{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "import ipywidgets as widgets\n",
    "from IPython.display import display\n",
    "\n",
    "# add parent directory to path\n",
    "library_path = os.path.abspath('..')\n",
    "if library_path not in sys.path:\n",
    "    sys.path.append(library_path)\n",
    "\n",
    "from ideal_genom.zoom_heatmap import filter_sumstats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0b2f73ce225b44f189770ac5426ce105",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Text(value='/home/luis/data/gwasResult/', description='Path to project folder:', style=TextStyle(description_w…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0e1f73cae35443259c0a05fd41588299",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Text(value='annotated_normalized_combined_R2_0.3.dose_step2_sex_pheno-glm.PHENO1.glm.logistic.hybrid', descrip…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f8cc31bfdde44ba193e985bfb8490ef0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Text(value='table_lead_SNPS_GWAS_glm_logistic_final_paper', description='Name of file with SNPs to highlight:'…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Create interactive widgets for input\n",
    "input_path = widgets.Text(\n",
    "    value='/home/luis/data/gwasResult/',\n",
    "    description='Path to project folder:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "input_name = widgets.Text(\n",
    "    value='annotated_normalized_combined_R2_0.3.dose_step2_sex_pheno-glm.PHENO1.glm.logistic.hybrid',\n",
    "    description='Name of GWAS summary file:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "top_snp = widgets.Text(\n",
    "    value='table_lead_SNPS_GWAS_glm_logistic_final_paper',\n",
    "    description='Name of file with SNPs to highlight:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "# Display the widgets\n",
    "display(input_path, input_name, top_snp)\n",
    "\n",
    "# Function to get the text parameter values\n",
    "def get_params():\n",
    "    return input_path.value, input_name.value, top_snp.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Parameter 1: /home/luis/data/gwasResult/\n",
      "Parameter 2: annotated_normalized_combined_R2_0.3.dose_step2_sex_pheno-glm.PHENO1.glm.logistic.hybrid\n",
      "Parameter 3: table_lead_SNPS_GWAS_glm_logistic_final_paper\n"
     ]
    }
   ],
   "source": [
    "# Use the parameter values\n",
    "path_params = get_params()\n",
    "print(f\"Parameter 1: {path_params[0]}\")\n",
    "print(f\"Parameter 2: {path_params[1]}\")\n",
    "print(f\"Parameter 3: {path_params[2]}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "4ca8942cf6994d16ae4c95c4305605c5",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Textarea(value='#CHROM, POS, ID, P', description='Columns to use on the Mannhattan plot (comma-separated):', l…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cols_touse = widgets.Textarea(\n",
    "    value=\"#CHROM, POS, ID, P\",\n",
    "    description='Columns to use on the Mannhattan plot (comma-separated):',\n",
    "    style={'description_width': 'initial'},\n",
    "    layout=widgets.Layout(width='50%')\n",
    ")\n",
    "display(cols_touse)\n",
    "\n",
    "def get_cols():\n",
    "    return cols_touse.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>727242</td>\n",
       "      <td>rs61769339</td>\n",
       "      <td>0.355696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>727717</td>\n",
       "      <td>rs61769340</td>\n",
       "      <td>0.517232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>730869</td>\n",
       "      <td>rs200188737</td>\n",
       "      <td>0.510707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>732369</td>\n",
       "      <td>rs1315713498</td>\n",
       "      <td>0.741994</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>740738</td>\n",
       "      <td>rs146067153</td>\n",
       "      <td>0.816063</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #CHROM     POS            ID         P\n",
       "0       1  727242    rs61769339  0.355696\n",
       "1       1  727717    rs61769340  0.517232\n",
       "2       1  730869   rs200188737  0.510707\n",
       "3       1  732369  rs1315713498  0.741994\n",
       "4       1  740738   rs146067153  0.816063"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = get_cols()\n",
    "\n",
    "df_gwas = pd.read_csv(\n",
    "    os.path.join(path_params[0], path_params[1]), sep='\\t', usecols=[col.strip() for col in cols.split(',')]\n",
    ")\n",
    "df_gwas.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "32c32f412ca84a24b7e0cf05a7374475",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Text(value='SNP', description='Column name with the rsID:', style=TextStyle(description_width='initial'))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "rsID = widgets.Text(\n",
    "    value='SNP',\n",
    "    description='Column name with the rsID:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "display(rsID)\n",
    "\n",
    "def get_rsID():\n",
    "    return rsID.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['rs3747973', 'rs74990530', 'rs34311866', 'rs356182', 'rs7681440', 'rs1846190', 'rs2517680', 'rs4909940', 'rs528813377', 'rs56328224']\n"
     ]
    }
   ],
   "source": [
    "gwas_rsID = get_rsID()\n",
    "\n",
    "if path_params[2] is not None or path_params[2] != '':\n",
    "    highlit_path = os.path.join(path_params[0], path_params[2])\n",
    "    if os.path.exists(highlit_path):\n",
    "        df_high = pd.read_csv(\n",
    "            highlit_path, sep='\\t'\n",
    "        )\n",
    "        to_highlight = df_high[gwas_rsID].to_list()\n",
    "        print(to_highlight[:10])\n",
    "        del df_high\n",
    "    else:\n",
    "        print('Path to file with SNPs to highlight does not exist')\n",
    "        to_highlight = []\n",
    "else:\n",
    "    print('No file with SNPs to highlight')\n",
    "    to_highlight = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0b37615c7e674a7482d1561d389be2b4",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Text(value='ID', description='Column with rsID:', style=TextStyle(description_width='initial'))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "67479b32f08d42689a61482a4323d59f",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Text(value='#CHROM', description='Column with chromosome:', style=TextStyle(description_width='initial'))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "fd943b417f9a490aaa2e8f7691fb0e7c",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Text(value='POS', description='Column with base-pair position:', style=TextStyle(description_width='initial'))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "9f798eed82a74cedb1d58f232058f1f6",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Text(value='P', description='Column with p-values:', style=TextStyle(description_width='initial'))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "SNP_col = widgets.Text(\n",
    "    value='ID',\n",
    "    description='Column with rsID:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "CHR_col = widgets.Text(\n",
    "    value='#CHROM',\n",
    "    description='Column with chromosome:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "POS_col = widgets.Text(\n",
    "    value='POS',\n",
    "    description='Column with base-pair position:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "P_col = widgets.Text(\n",
    "    value='P',\n",
    "    description='Column with p-values:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "display(SNP_col, CHR_col, POS_col, P_col)\n",
    "\n",
    "def get_col_names():\n",
    "    return SNP_col.value, CHR_col.value, POS_col.value, P_col.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_col, chr_col, pos_col, p_col = get_col_names()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024/12/11 16:51:46 Start to annotate variants with nearest gene name(s)...\n",
      "2024/12/11 16:51:46  -Assigning Gene name using NCBI refseq latest GRCh38 for protein coding genes\n",
      "2024/12/11 16:51:46  -Using user-provided gtf:/home/luis/CGE/ideal-genom/GCF_000001405.40_GRCh38.p14_genomic.gtf\n",
      "2024/12/11 16:51:49 Finished annotating variants with nearest gene name(s) successfully!\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>P</th>\n",
       "      <th>log10p</th>\n",
       "      <th>LOCATION</th>\n",
       "      <th>GENENAME</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>204840546</td>\n",
       "      <td>rs11801418</td>\n",
       "      <td>2.044970e-06</td>\n",
       "      <td>5.689313</td>\n",
       "      <td>0</td>\n",
       "      <td>NFASC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>204842990</td>\n",
       "      <td>rs869841</td>\n",
       "      <td>2.770490e-06</td>\n",
       "      <td>5.557443</td>\n",
       "      <td>0</td>\n",
       "      <td>NFASC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>204843348</td>\n",
       "      <td>rs12563611</td>\n",
       "      <td>3.140220e-06</td>\n",
       "      <td>5.503040</td>\n",
       "      <td>0</td>\n",
       "      <td>NFASC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>204843570</td>\n",
       "      <td>rs12407417</td>\n",
       "      <td>2.770490e-06</td>\n",
       "      <td>5.557443</td>\n",
       "      <td>0</td>\n",
       "      <td>NFASC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>204843922</td>\n",
       "      <td>rs12217091</td>\n",
       "      <td>9.794050e-07</td>\n",
       "      <td>6.009038</td>\n",
       "      <td>0</td>\n",
       "      <td>NFASC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #CHROM        POS          ID             P    log10p  LOCATION GENENAME\n",
       "0       1  204840546  rs11801418  2.044970e-06  5.689313         0    NFASC\n",
       "1       1  204842990    rs869841  2.770490e-06  5.557443         0    NFASC\n",
       "2       1  204843348  rs12563611  3.140220e-06  5.503040         0    NFASC\n",
       "3       1  204843570  rs12407417  2.770490e-06  5.557443         0    NFASC\n",
       "4       1  204843922  rs12217091  9.794050e-07  6.009038         0    NFASC"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered = filter_sumstats(\n",
    "    data_df       =df_gwas, \n",
    "    lead_snp      =to_highlight[0], \n",
    "    snp_col       =snp_col, \n",
    "    p_col         =p_col, \n",
    "    pos_col       =pos_col, \n",
    "    chr_col       =chr_col, \n",
    "    pval_threshold=5e-6, \n",
    "    radius        =10e6\n",
    ")\n",
    "filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>510224</th>\n",
       "      <td>1</td>\n",
       "      <td>205708020</td>\n",
       "      <td>rs3747973</td>\n",
       "      <td>6.094640e-11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        #CHROM        POS         ID             P\n",
       "510224       1  205708020  rs3747973  6.094640e-11"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gwas[df_gwas[snp_col] == to_highlight[0]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tables in the database: [('start_codon',), ('stop_codon',), ('transcript',), ('gene',), ('exon',), ('CDS',), ('_datacache_metadata',)]\n"
     ]
    }
   ],
   "source": [
    "import sqlite3\n",
    "\n",
    "# Path to your .db file\n",
    "db_file = \"/home/luis/CGE/ideal-genom/GCF_000001405.40_GRCh38.p14_genomall_genes.gtf.db\"\n",
    "\n",
    "# Connect to the database\n",
    "connection = sqlite3.connect(db_file)\n",
    "\n",
    "# Create a cursor object to execute SQL queries\n",
    "cursor = connection.cursor()\n",
    "\n",
    "# Get the list of all tables\n",
    "cursor.execute(\"SELECT name FROM sqlite_master WHERE type='table';\")\n",
    "tables = cursor.fetchall()\n",
    "print(\"Tables in the database:\", tables)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>protein_id</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>end</th>\n",
       "      <th>transcript_biotype</th>\n",
       "      <th>source</th>\n",
       "      <th>exon_number</th>\n",
       "      <th>seqname</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>feature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11874</td>\n",
       "      <td></td>\n",
       "      <td>+</td>\n",
       "      <td></td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>12227</td>\n",
       "      <td>transcript</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td>NR_046018.2</td>\n",
       "      <td>exon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12613</td>\n",
       "      <td></td>\n",
       "      <td>+</td>\n",
       "      <td></td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>12721</td>\n",
       "      <td>transcript</td>\n",
       "      <td></td>\n",
       "      <td>2</td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td>NR_046018.2</td>\n",
       "      <td>exon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>13221</td>\n",
       "      <td></td>\n",
       "      <td>+</td>\n",
       "      <td></td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>14409</td>\n",
       "      <td>transcript</td>\n",
       "      <td></td>\n",
       "      <td>3</td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td>NR_046018.2</td>\n",
       "      <td>exon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>29321</td>\n",
       "      <td></td>\n",
       "      <td>-</td>\n",
       "      <td></td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>29370</td>\n",
       "      <td>transcript</td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>exon</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>24738</td>\n",
       "      <td></td>\n",
       "      <td>-</td>\n",
       "      <td></td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>24891</td>\n",
       "      <td>transcript</td>\n",
       "      <td></td>\n",
       "      <td>2</td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>exon</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   start protein_id strand gene_biotype  gene_id    end transcript_biotype  \\\n",
       "0  11874                 +               DDX11L1  12227         transcript   \n",
       "1  12613                 +               DDX11L1  12721         transcript   \n",
       "2  13221                 +               DDX11L1  14409         transcript   \n",
       "3  29321                 -                WASH7P  29370         transcript   \n",
       "4  24738                 -                WASH7P  24891         transcript   \n",
       "\n",
       "  source exon_number       seqname transcript_id feature  \n",
       "0                  1  NC_000001.11   NR_046018.2    exon  \n",
       "1                  2  NC_000001.11   NR_046018.2    exon  \n",
       "2                  3  NC_000001.11   NR_046018.2    exon  \n",
       "3                  1  NC_000001.11   NR_024540.1    exon  \n",
       "4                  2  NC_000001.11   NR_024540.1    exon  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_name = 'exon'\n",
    "df_exon = pd.read_sql_query(f\"SELECT * FROM {table_name};\", connection)\n",
    "df_exon.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>protein_id</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>end</th>\n",
       "      <th>transcript_biotype</th>\n",
       "      <th>source</th>\n",
       "      <th>exon_number</th>\n",
       "      <th>seqname</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>feature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11874</td>\n",
       "      <td></td>\n",
       "      <td>+</td>\n",
       "      <td>transcribed_pseudogene</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>14409</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td></td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>14362</td>\n",
       "      <td></td>\n",
       "      <td>-</td>\n",
       "      <td>transcribed_pseudogene</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>29370</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td></td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>17369</td>\n",
       "      <td></td>\n",
       "      <td>-</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>17436</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td></td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>29774</td>\n",
       "      <td></td>\n",
       "      <td>+</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>MIR1302-2HG</td>\n",
       "      <td>35418</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td></td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>30366</td>\n",
       "      <td></td>\n",
       "      <td>+</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>MIR1302-2</td>\n",
       "      <td>30503</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td></td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   start protein_id strand            gene_biotype      gene_id    end  \\\n",
       "0  11874                 +  transcribed_pseudogene      DDX11L1  14409   \n",
       "1  14362                 -  transcribed_pseudogene       WASH7P  29370   \n",
       "2  17369                 -                   miRNA    MIR6859-1  17436   \n",
       "3  29774                 +                  lncRNA  MIR1302-2HG  35418   \n",
       "4  30366                 +                   miRNA    MIR1302-2  30503   \n",
       "\n",
       "  transcript_biotype source exon_number       seqname transcript_id feature  \n",
       "0                                        NC_000001.11                  gene  \n",
       "1                                        NC_000001.11                  gene  \n",
       "2                                        NC_000001.11                  gene  \n",
       "3                                        NC_000001.11                  gene  \n",
       "4                                        NC_000001.11                  gene  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "table_name = 'gene'\n",
    "df_gene = pd.read_sql_query(f\"SELECT * FROM {table_name};\", connection)\n",
    "df_gene.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>start</th>\n",
       "      <th>protein_id</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_biotype</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>end</th>\n",
       "      <th>transcript_biotype</th>\n",
       "      <th>source</th>\n",
       "      <th>exon_number</th>\n",
       "      <th>seqname</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>feature</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4517</th>\n",
       "      <td>204828652</td>\n",
       "      <td></td>\n",
       "      <td>+</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NFASC</td>\n",
       "      <td>205022822</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td></td>\n",
       "      <td>gene</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          start protein_id strand    gene_biotype gene_id        end  \\\n",
       "4517  204828652                 +  protein_coding   NFASC  205022822   \n",
       "\n",
       "     transcript_biotype source exon_number       seqname transcript_id feature  \n",
       "4517                                        NC_000001.11                  gene  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gene[df_gene['gene_id'] == 'NFASC']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pyensembl import Genome\n",
    "\n",
    "data = Genome(\n",
    "                reference_name='GRCh38',\n",
    "                annotation_name='Refseq',\n",
    "                gtf_path_or_url='/home/luis/CGE/ideal-genom/GCF_000001405.40_GRCh38.p14_genomall_genes.gtf.db')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Transcript(transcript_id='NM_001005388.3', transcript_name='None', gene_id='NFASC', biotype='mRNA', contig='NC_000001.11', start=204828652, end=205022822, strand='+', genome='GRCh38')"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transcripts = data.transcripts_at_locus(contig='NC_000001.11', position=204840546)\n",
    "test = transcripts[0]\n",
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NM_001005388.3'"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test.transcript_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[{'assembly_name': 'GRCh38', 'start': 204840546, 'allele_string': 'T/A/C', 'strand': 1, 'input': 'rs11801418', 'end': 204840546, 'most_severe_consequence': 'intron_variant', 'colocated_variants': [{'end': 204840546, 'start': 204840546, 'seq_region_name': '1', 'id': 'rs11801418', 'allele_string': 'T/A/C', 'frequencies': {'A': {'gnomadg_asj': 0.09054, 'afr': 0.2421, 'gnomadg_nfe': 0.08936, 'gnomadg_remaining': 0.1569, 'sas': 0.0879, 'eas': 0.4821, 'gnomadg_ami': 0.1195, 'eur': 0.0915, 'gnomadg': 0.1562, 'gnomadg_sas': 0.1122, 'af': 0.2328, 'gnomadg_eas': 0.4538, 'gnomadg_amr': 0.2252, 'gnomadg_afr': 0.2124, 'amr': 0.2622, 'gnomadg_mid': 0.119, 'gnomadg_fin': 0.1655}}, 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080'}, {'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543'}, {'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}], 'id': 'rs11801418'}, {'assembly_name': 'GRCh38', 'start': 204842990, 'colocated_variants': [{'phenotype_or_disease': 1, 'strand': 1, 'start': 204842990, 'pubmed': [29875488], 'end': 204842990, 'frequencies': {'C': {'gnomadg_afr': 0.4079, 'amr': 0.2853, 'gnomadg_mid': 0.1361, 'gnomadg_amr': 0.2504, 'gnomadg_fin': 0.1653, 'gnomadg_eas': 0.5268, 'af': 0.3083, 'gnomadg_sas': 0.1164, 'gnomadg': 0.2162, 'eur': 0.0944, 'gnomadg_ami': 0.1195, 'eas': 0.5565, 'sas': 0.089, 'afr': 0.4561, 'gnomadg_asj': 0.1165, 'gnomadg_remaining': 0.1933, 'gnomadg_nfe': 0.09033}, 'T': {'gnomadg_fin': 0, 'gnomadg_amr': 0, 'gnomadg_afr': 0, 'gnomadg_mid': 0, 'gnomadg_eas': 0, 'gnomadg_remaining': 0, 'gnomadg_nfe': 1.47e-05, 'gnomadg': 6.575e-06, 'gnomadg_sas': 0, 'gnomadg_asj': 0, 'gnomadg_ami': 0}}, 'id': 'rs869841', 'seq_region_name': '1', 'allele_string': 'G/A/C/T'}], 'transcript_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant']}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080'}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080', 'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC'}, {'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC'}, {'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC'}, {'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC'}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531'}, {'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'strand': 1}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'protein_coding', 'strand': 1}], 'id': 'rs869841', 'seq_region_name': '1', 'allele_string': 'G/A/C/T', 'strand': 1, 'input': 'rs869841', 'regulatory_feature_consequences': [{'regulatory_feature_id': 'ENSR1_952B8P', 'biotype': 'enhancer', 'consequence_terms': ['regulatory_region_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'enhancer', 'consequence_terms': ['regulatory_region_variant'], 'regulatory_feature_id': 'ENSR1_952B8P'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'regulatory_feature_id': 'ENSR1_952B8P', 'biotype': 'enhancer', 'consequence_terms': ['regulatory_region_variant']}], 'most_severe_consequence': 'intron_variant', 'end': 204842990}, {'assembly_name': 'GRCh38', 'start': 204843348, 'colocated_variants': [{'strand': 1, 'id': 'rs12563611', 'seq_region_name': '1', 'allele_string': 'A/G', 'frequencies': {'G': {'sas': 0.091, 'eas': 0.5595, 'afr': 0.4569, 'gnomadg_asj': 0.1223, 'gnomadg_remaining': 0.1959, 'gnomadg_nfe': 0.09084, 'gnomadg_eas': 0.5294, 'af': 0.3101, 'amr': 0.2896, 'gnomadg_afr': 0.4086, 'gnomadg_mid': 0.1336, 'gnomadg_amr': 0.252, 'gnomadg_fin': 0.1663, 'eur': 0.0944, 'gnomadg_ami': 0.1195, 'gnomadg_sas': 0.1213, 'gnomadg': 0.2172}}, 'end': 204843348, 'start': 204843348}], 'id': 'rs12563611', 'transcript_consequences': [{'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}], 'seq_region_name': '1', 'input': 'rs12563611', 'allele_string': 'A/G', 'strand': 1, 'most_severe_consequence': 'intron_variant', 'end': 204843348}]\n"
     ]
    }
   ],
   "source": [
    "import requests, sys\n",
    " \n",
    "server = \"https://rest.ensembl.org\"\n",
    "ext = \"/vep/human/id\"\n",
    "headers={ \"Content-Type\" : \"application/json\", \"Accept\" : \"application/json\"}\n",
    "r = requests.post(server+ext, headers=headers, data='{ \"ids\" : [\"rs11801418\", \"rs869841\", \"rs12563611\" ] }')\n",
    " \n",
    "if not r.ok:\n",
    "  r.raise_for_status()\n",
    "  sys.exit()\n",
    " \n",
    "decoded = r.json()\n",
    "print(repr(decoded))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['assembly_name', 'start', 'allele_string', 'strand', 'input', 'end', 'most_severe_consequence', 'colocated_variants', 'seq_region_name', 'transcript_consequences', 'id'])"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "decoded[0].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'intron_variant'"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "decoded[0]['most_severe_consequence']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rs11801418 intron_variant\n",
      "rs869841 intron_variant\n",
      "rs12563611 intron_variant\n"
     ]
    }
   ],
   "source": [
    "for res in decoded:\n",
    "    print(res['id'], res['most_severe_consequence'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Response: [{'start': 204840546, 'assembly_name': 'GRCh38', 'most_severe_consequence': 'intron_variant', 'end': 204840546, 'allele_string': 'T/A/C', 'strand': 1, 'input': 'rs11801418', 'colocated_variants': [{'allele_string': 'T/A/C', 'seq_region_name': '1', 'id': 'rs11801418', 'frequencies': {'A': {'gnomadg_asj': 0.09054, 'afr': 0.2421, 'gnomadg_remaining': 0.1569, 'gnomadg_nfe': 0.08936, 'sas': 0.0879, 'eas': 0.4821, 'gnomadg_ami': 0.1195, 'eur': 0.0915, 'gnomadg': 0.1562, 'gnomadg_sas': 0.1122, 'af': 0.2328, 'gnomadg_eas': 0.4538, 'gnomadg_amr': 0.2252, 'gnomadg_afr': 0.2124, 'amr': 0.2622, 'gnomadg_mid': 0.119, 'gnomadg_fin': 0.1655}}, 'end': 204840546, 'start': 204840546, 'strand': 1}], 'transcript_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant']}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC'}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined'}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant']}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC'}], 'seq_region_name': '1', 'id': 'rs11801418'}, {'start': 204842990, 'assembly_name': 'GRCh38', 'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080', 'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977'}, {'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC'}, {'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543', 'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC'}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531'}, {'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}], 'colocated_variants': [{'phenotype_or_disease': 1, 'strand': 1, 'start': 204842990, 'end': 204842990, 'pubmed': [29875488], 'frequencies': {'C': {'af': 0.3083, 'gnomadg_eas': 0.5268, 'gnomadg_amr': 0.2504, 'gnomadg_afr': 0.4079, 'gnomadg_mid': 0.1361, 'amr': 0.2853, 'gnomadg_fin': 0.1653, 'gnomadg_ami': 0.1195, 'eur': 0.0944, 'gnomadg': 0.2162, 'gnomadg_sas': 0.1164, 'sas': 0.089, 'eas': 0.5565, 'afr': 0.4561, 'gnomadg_asj': 0.1165, 'gnomadg_remaining': 0.1933, 'gnomadg_nfe': 0.09033}, 'T': {'gnomadg_ami': 0, 'gnomadg_asj': 0, 'gnomadg_sas': 0, 'gnomadg_remaining': 0, 'gnomadg': 6.575e-06, 'gnomadg_nfe': 1.47e-05, 'gnomadg_eas': 0, 'gnomadg_afr': 0, 'gnomadg_mid': 0, 'gnomadg_amr': 0, 'gnomadg_fin': 0}}, 'allele_string': 'G/A/C/T', 'seq_region_name': '1', 'id': 'rs869841'}], 'id': 'rs869841', 'seq_region_name': '1', 'end': 204842990, 'most_severe_consequence': 'intron_variant', 'regulatory_feature_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'regulatory_feature_id': 'ENSR1_952B8P', 'biotype': 'enhancer', 'consequence_terms': ['regulatory_region_variant']}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['regulatory_region_variant'], 'biotype': 'enhancer', 'regulatory_feature_id': 'ENSR1_952B8P'}, {'consequence_terms': ['regulatory_region_variant'], 'biotype': 'enhancer', 'regulatory_feature_id': 'ENSR1_952B8P', 'impact': 'MODIFIER', 'variant_allele': 'T'}], 'allele_string': 'G/A/C/T', 'strand': 1, 'input': 'rs869841'}, {'start': 204843348, 'assembly_name': 'GRCh38', 'most_severe_consequence': 'intron_variant', 'end': 204843348, 'strand': 1, 'allele_string': 'A/G', 'input': 'rs12563611', 'colocated_variants': [{'start': 204843348, 'end': 204843348, 'frequencies': {'G': {'gnomadg_ami': 0.1195, 'eur': 0.0944, 'gnomadg': 0.2172, 'gnomadg_sas': 0.1213, 'af': 0.3101, 'gnomadg_eas': 0.5294, 'gnomadg_fin': 0.1663, 'gnomadg_amr': 0.252, 'gnomadg_afr': 0.4086, 'gnomadg_mid': 0.1336, 'amr': 0.2896, 'gnomadg_remaining': 0.1959, 'gnomadg_nfe': 0.09084, 'gnomadg_asj': 0.1223, 'afr': 0.4569, 'sas': 0.091, 'eas': 0.5595}}, 'id': 'rs12563611', 'seq_region_name': '1', 'allele_string': 'A/G', 'strand': 1}], 'id': 'rs12563611', 'transcript_consequences': [{'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876'}, {'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}], 'seq_region_name': '1'}, {'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531'}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC'}], 'colocated_variants': [{'somatic': 1, 'strand': 1, 'phenotype_or_disease': 1, 'seq_region_name': '1', 'id': 'COSV58381597', 'allele_string': 'COSMIC_MUTATION', 'end': 204843570, 'start': 204843570}, {'strand': 1, 'frequencies': {'T': {'gnomadg': 0.2062, 'gnomadg_sas': 0.1156, 'gnomadg_ami': 0.1186, 'eur': 0.0944, 'gnomadg_fin': 0.1585, 'gnomadg_amr': 0.2466, 'gnomadg_afr': 0.381, 'amr': 0.2853, 'gnomadg_mid': 0.137, 'af': 0.3013, 'gnomadg_eas': 0.5268, 'gnomadg_remaining': 0.1865, 'gnomadg_nfe': 0.09, 'afr': 0.4297, 'gnomadg_asj': 0.1157, 'eas': 0.5565, 'sas': 0.089}}, 'id': 'rs12407417', 'seq_region_name': '1', 'allele_string': 'C/T', 'start': 204843570, 'end': 204843570}], 'id': 'rs12407417', 'seq_region_name': '1', 'strand': 1, 'allele_string': 'C/T', 'input': 'rs12407417', 'most_severe_consequence': 'intron_variant', 'end': 204843570, 'assembly_name': 'GRCh38', 'start': 204843570}, {'allele_string': 'C/G/T', 'strand': 1, 'input': 'rs12217091', 'end': 204843922, 'most_severe_consequence': 'intron_variant', 'colocated_variants': [{'end': 204843922, 'start': 204843922, 'allele_string': 'C/G/T', 'seq_region_name': '1', 'id': 'rs12217091', 'frequencies': {'T': {'gnomadg_nfe': 0.08885, 'gnomadg_remaining': 0.1229, 'gnomadg_asj': 0.09002, 'afr': 0.0961, 'sas': 0.0879, 'eas': 0.4831, 'eur': 0.0915, 'gnomadg_ami': 0.1195, 'gnomadg_sas': 0.1121, 'gnomadg': 0.1184, 'gnomadg_eas': 0.4518, 'af': 0.1925, 'gnomadg_fin': 0.166, 'gnomadg_mid': 0.1088, 'gnomadg_afr': 0.08206, 'amr': 0.2478, 'gnomadg_amr': 0.2116}}, 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}], 'id': 'rs12217091', 'assembly_name': 'GRCh38', 'start': 204843922}, {'assembly_name': 'GRCh38', 'start': 204844325, 'input': 'rs6699817', 'strand': 1, 'allele_string': 'A/G', 'most_severe_consequence': 'intron_variant', 'end': 204844325, 'seq_region_name': '1', 'colocated_variants': [{'start': 204844325, 'end': 204844325, 'frequencies': {'G': {'afr': 0.4569, 'gnomadg_asj': 0.1225, 'gnomadg_remaining': 0.1968, 'gnomadg_nfe': 0.09088, 'eas': 0.5595, 'sas': 0.091, 'gnomadg_sas': 0.1213, 'gnomadg': 0.2174, 'eur': 0.0944, 'gnomadg_ami': 0.1176, 'amr': 0.2911, 'gnomadg_afr': 0.4092, 'gnomadg_mid': 0.1361, 'gnomadg_amr': 0.2528, 'gnomadg_fin': 0.1656, 'gnomadg_eas': 0.5293, 'af': 0.3103}}, 'id': 'rs6699817', 'seq_region_name': '1', 'allele_string': 'A/G', 'strand': 1}], 'transcript_consequences': [{'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'NFASC'}, {'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant']}], 'id': 'rs6699817'}, {'input': 'rs6702731', 'strand': 1, 'allele_string': 'T/A/C', 'end': 204844602, 'most_severe_consequence': 'intron_variant', 'id': 'rs6702731', 'colocated_variants': [{'end': 204844602, 'start': 204844602, 'id': 'rs6702731', 'seq_region_name': '1', 'allele_string': 'T/A/C', 'frequencies': {'C': {'eas': 0.5595, 'sas': 0.091, 'gnomadg_remaining': 0.1965, 'gnomadg_nfe': 0.0908, 'afr': 0.4569, 'gnomadg_asj': 0.1225, 'gnomadg_fin': 0.1661, 'gnomadg_amr': 0.2517, 'gnomadg_mid': 0.1361, 'amr': 0.2896, 'gnomadg_afr': 0.4082, 'af': 0.3101, 'gnomadg_eas': 0.5306, 'gnomadg': 0.217, 'gnomadg_sas': 0.1213, 'gnomadg_ami': 0.1176, 'eur': 0.0944}}, 'strand': 1}], 'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080'}, {'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977'}, {'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977'}, {'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543'}, {'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}], 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 204844602}, {'assembly_name': 'GRCh38', 'start': 204844772, 'id': 'rs57128821', 'colocated_variants': [{'strand': 1, 'seq_region_name': '1', 'id': 'rs57128821', 'allele_string': 'G/A', 'frequencies': {'A': {'gnomadg': 0.2171, 'gnomadg_sas': 0.1206, 'gnomadg_ami': 0.1195, 'eur': 0.0944, 'gnomadg_fin': 0.1662, 'gnomadg_amr': 0.2521, 'amr': 0.2896, 'gnomadg_afr': 0.4085, 'gnomadg_mid': 0.1361, 'af': 0.3101, 'gnomadg_eas': 0.53, 'gnomadg_remaining': 0.1951, 'gnomadg_nfe': 0.09082, 'gnomadg_asj': 0.1225, 'afr': 0.4569, 'eas': 0.5595, 'sas': 0.091}}, 'end': 204844772, 'start': 204844772}], 'transcript_consequences': [{'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393'}, {'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543', 'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC'}], 'seq_region_name': '1', 'strand': 1, 'allele_string': 'G/A', 'input': 'rs57128821', 'most_severe_consequence': 'intron_variant', 'end': 204844772}, {'seq_region_name': '1', 'colocated_variants': [{'strand': 1, 'end': 204844794, 'start': 204844794, 'allele_string': 'C/T', 'seq_region_name': '1', 'id': 'rs57482207', 'frequencies': {'T': {'eur': 0.0944, 'gnomadg_ami': 0.1195, 'gnomadg_sas': 0.1214, 'gnomadg': 0.2172, 'gnomadg_eas': 0.5291, 'af': 0.3101, 'amr': 0.2896, 'gnomadg_afr': 0.4084, 'gnomadg_mid': 0.1463, 'gnomadg_amr': 0.2525, 'gnomadg_fin': 0.1668, 'gnomadg_asj': 0.1223, 'afr': 0.4569, 'gnomadg_remaining': 0.1958, 'gnomadg_nfe': 0.09086, 'sas': 0.091, 'eas': 0.5595}}}], 'transcript_consequences': [{'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'T', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC'}], 'id': 'rs57482207', 'end': 204844794, 'most_severe_consequence': 'intron_variant', 'input': 'rs57482207', 'allele_string': 'C/T', 'strand': 1, 'start': 204844794, 'assembly_name': 'GRCh38'}, {'start': 204846450, 'assembly_name': 'GRCh38', 'end': 204846450, 'most_severe_consequence': 'intron_variant', 'allele_string': 'G/A', 'strand': 1, 'input': 'rs4951134', 'seq_region_name': '1', 'colocated_variants': [{'allele_string': 'G/A', 'id': 'rs4951134', 'seq_region_name': '1', 'frequencies': {'A': {'af': 0.2979, 'gnomadg_eas': 0.5295, 'gnomadg_amr': 0.2471, 'amr': 0.2896, 'gnomadg_mid': 0.1327, 'gnomadg_afr': 0.3728, 'gnomadg_fin': 0.1658, 'gnomadg_ami': 0.1195, 'eur': 0.0944, 'gnomadg': 0.2066, 'gnomadg_sas': 0.1221, 'sas': 0.091, 'eas': 0.5565, 'gnomadg_asj': 0.1224, 'afr': 0.413, 'gnomadg_nfe': 0.09057, 'gnomadg_remaining': 0.1885}}, 'end': 204846450, 'start': 204846450, 'strand': 1}], 'transcript_consequences': [{'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC'}, {'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543'}, {'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}], 'id': 'rs4951134'}, {'assembly_name': 'GRCh38', 'start': 204846484, 'input': 'rs4951135', 'strand': 1, 'allele_string': 'T/C', 'most_severe_consequence': 'intron_variant', 'end': 204846484, 'colocated_variants': [{'strand': 1, 'end': 204846484, 'start': 204846484, 'allele_string': 'T/C', 'seq_region_name': '1', 'id': 'rs4951135', 'frequencies': {'C': {'gnomadg_remaining': 0.1868, 'gnomadg_nfe': 0.09054, 'afr': 0.413, 'gnomadg_asj': 0.1225, 'sas': 0.091, 'eas': 0.5565, 'gnomadg_ami': 0.1198, 'eur': 0.0944, 'gnomadg': 0.2066, 'gnomadg_sas': 0.1218, 'af': 0.2979, 'gnomadg_eas': 0.5296, 'gnomadg_fin': 0.1664, 'gnomadg_amr': 0.2473, 'amr': 0.2896, 'gnomadg_afr': 0.3726, 'gnomadg_mid': 0.1327}}}], 'id': 'rs4951135', 'transcript_consequences': [{'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531'}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}], 'seq_region_name': '1'}, {'start': 204846601, 'assembly_name': 'GRCh38', 'transcript_consequences': [{'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}], 'colocated_variants': [{'frequencies': {'T': {'gnomadg_remaining': 0.1864, 'gnomadg_nfe': 0.09047, 'afr': 0.413, 'gnomadg_asj': 0.1225, 'sas': 0.091, 'eas': 0.5565, 'eur': 0.0944, 'gnomadg_ami': 0.1195, 'gnomadg_sas': 0.1211, 'gnomadg': 0.2061, 'gnomadg_eas': 0.5279, 'af': 0.2979, 'gnomadg_fin': 0.1657, 'gnomadg_afr': 0.3719, 'gnomadg_mid': 0.1336, 'amr': 0.2896, 'gnomadg_amr': 0.2472}}, 'seq_region_name': '1', 'id': 'rs4951136', 'allele_string': 'G/C/T', 'start': 204846601, 'end': 204846601, 'strand': 1}], 'id': 'rs4951136', 'seq_region_name': '1', 'end': 204846601, 'most_severe_consequence': 'intron_variant', 'allele_string': 'G/C/T', 'strand': 1, 'input': 'rs4951136'}, {'seq_region_name': '1', 'colocated_variants': [{'start': 204847545, 'end': 204847545, 'frequencies': {'T': {'gnomadg_remaining': 0.1868, 'gnomadg_nfe': 0.0904, 'gnomadg_asj': 0.1225, 'afr': 0.4115, 'sas': 0.091, 'eas': 0.5565, 'eur': 0.0944, 'gnomadg_ami': 0.1195, 'gnomadg_sas': 0.1215, 'gnomadg': 0.2066, 'gnomadg_eas': 0.529, 'af': 0.2975, 'gnomadg_fin': 0.1666, 'gnomadg_afr': 0.3726, 'gnomadg_mid': 0.1301, 'amr': 0.2896, 'gnomadg_amr': 0.2474}}, 'seq_region_name': '1', 'id': 'rs67020185', 'allele_string': 'C/T', 'strand': 1}], 'transcript_consequences': [{'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876'}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1}, {'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC'}], 'id': 'rs67020185', 'input': 'rs67020185', 'strand': 1, 'allele_string': 'C/T', 'most_severe_consequence': 'intron_variant', 'end': 204847545, 'assembly_name': 'GRCh38', 'start': 204847545}, {'end': 204849935, 'most_severe_consequence': 'intron_variant', 'input': 'rs1107714', 'allele_string': 'A/G/T', 'strand': 1, 'colocated_variants': [{'strand': 1, 'allele_string': 'A/G/T', 'id': 'rs1107714', 'seq_region_name': '1', 'frequencies': {'G': {'eas': 0.5556, 'sas': 0.091, 'gnomadg_asj': 0.1222, 'afr': 0.4115, 'gnomadg_nfe': 0.09041, 'gnomadg_remaining': 0.1875, 'amr': 0.2896, 'gnomadg_afr': 0.3731, 'gnomadg_mid': 0.1327, 'gnomadg_amr': 0.2477, 'gnomadg_fin': 0.1663, 'gnomadg_eas': 0.5275, 'af': 0.2973, 'gnomadg_sas': 0.1213, 'gnomadg': 0.2067, 'eur': 0.0944, 'gnomadg_ami': 0.1167}}, 'end': 204849935, 'start': 204849935}], 'transcript_consequences': [{'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant']}, {'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC'}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC'}], 'seq_region_name': '1', 'id': 'rs1107714', 'start': 204849935, 'assembly_name': 'GRCh38'}, {'allele_string': 'T/A/C/G', 'strand': 1, 'input': 'rs7529113', 'end': 204850633, 'most_severe_consequence': 'intron_variant', 'colocated_variants': [{'strand': 1, 'frequencies': {'C': {'amr': 0.2896, 'gnomadg_afr': 0.3733, 'gnomadg_mid': 0.1327, 'gnomadg_amr': 0.2473, 'gnomadg_fin': 0.1658, 'gnomadg_eas': 0.5286, 'af': 0.2971, 'gnomadg_sas': 0.1208, 'gnomadg': 0.2067, 'eur': 0.0944, 'gnomadg_ami': 0.1195, 'eas': 0.5556, 'sas': 0.091, 'gnomadg_asj': 0.1225, 'afr': 0.4107, 'gnomadg_nfe': 0.09035, 'gnomadg_remaining': 0.1879}}, 'id': 'rs7529113', 'seq_region_name': '1', 'allele_string': 'T/A/C/G', 'start': 204850633, 'end': 204850633}], 'id': 'rs7529113', 'transcript_consequences': [{'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'NFASC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543', 'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC'}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427'}, {'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427'}, {'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}], 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 204850633}, {'start': 204850789, 'assembly_name': 'GRCh38', 'colocated_variants': [{'strand': 1, 'seq_region_name': '1', 'id': 'rs7515134', 'allele_string': 'C/A/G', 'frequencies': {'G': {'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_amr': 6.539e-05, 'gnomadg_fin': 0, 'gnomadg_eas': 0, 'gnomadg_asj': 0, 'gnomadg_sas': 0, 'gnomadg_remaining': 0, 'gnomadg_nfe': 0, 'gnomadg': 6.571e-06, 'gnomadg_ami': 0}, 'A': {'gnomadg_asj': 0.1164, 'afr': 0.2806, 'gnomadg_remaining': 0.1575, 'gnomadg_nfe': 0.08937, 'eas': 0.5536, 'sas': 0.089, 'gnomadg_sas': 0.1159, 'gnomadg': 0.1691, 'eur': 0.0944, 'gnomadg_ami': 0.1195, 'gnomadg_afr': 0.2459, 'amr': 0.2637, 'gnomadg_mid': 0.1164, 'gnomadg_amr': 0.2326, 'gnomadg_fin': 0.1663, 'gnomadg_eas': 0.5229, 'af': 0.2584}}, 'end': 204850789, 'start': 204850789}], 'transcript_consequences': [{'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC'}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined'}, {'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined'}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393'}, {'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'gene_symbol': 'NFASC', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC'}], 'id': 'rs7515134', 'seq_region_name': '1', 'end': 204850789, 'most_severe_consequence': 'intron_variant', 'allele_string': 'C/A/G', 'strand': 1, 'input': 'rs7515134'}, {'start': 204850829, 'assembly_name': 'GRCh38', 'colocated_variants': [{'strand': 1, 'start': 204850829, 'end': 204850829, 'frequencies': {'C': {'eur': 0.0944, 'gnomadg_ami': 0.1181, 'gnomadg_sas': 0.1216, 'gnomadg': 0.2066, 'gnomadg_eas': 0.5267, 'af': 0.2973, 'gnomadg_fin': 0.1664, 'amr': 0.2896, 'gnomadg_afr': 0.3729, 'gnomadg_mid': 0.1276, 'gnomadg_amr': 0.2473, 'gnomadg_nfe': 0.09034, 'gnomadg_remaining': 0.1871, 'gnomadg_asj': 0.1225, 'afr': 0.4107, 'sas': 0.091, 'eas': 0.5565}}, 'id': 'rs7529301', 'seq_region_name': '1', 'allele_string': 'T/C/G'}], 'transcript_consequences': [{'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'protein_coding', 'strand': 1}, {'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393'}, {'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC'}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427'}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC'}], 'id': 'rs7529301', 'seq_region_name': '1', 'end': 204850829, 'most_severe_consequence': 'intron_variant', 'input': 'rs7529301', 'strand': 1, 'allele_string': 'T/C/G'}, {'strand': 1, 'allele_string': 'C/T', 'input': 'rs4951139', 'end': 204851418, 'most_severe_consequence': 'intron_variant', 'id': 'rs4951139', 'colocated_variants': [{'end': 204851418, 'start': 204851418, 'allele_string': 'C/T', 'id': 'rs4951139', 'seq_region_name': '1', 'frequencies': {'T': {'gnomadg_eas': 0.5235, 'af': 0.259, 'gnomadg_mid': 0.1233, 'gnomadg_afr': 0.2457, 'amr': 0.2637, 'gnomadg_amr': 0.2325, 'gnomadg_fin': 0.1651, 'eur': 0.0944, 'gnomadg_ami': 0.1195, 'gnomadg_sas': 0.1153, 'gnomadg': 0.1689, 'sas': 0.089, 'eas': 0.5526, 'afr': 0.2837, 'gnomadg_asj': 0.1164, 'gnomadg_remaining': 0.1565, 'gnomadg_nfe': 0.08939}}, 'strand': 1}], 'transcript_consequences': [{'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}], 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 204851418}, {'start': 204852036, 'assembly_name': 'GRCh38', 'seq_region_name': '1', 'colocated_variants': [{'strand': 1, 'end': 204852036, 'start': 204852036, 'allele_string': 'G/A', 'seq_region_name': '1', 'id': 'rs16854533', 'frequencies': {'A': {'eas': 0.5526, 'sas': 0.089, 'gnomadg_asj': 0.1164, 'afr': 0.357, 'gnomadg_remaining': 0.1706, 'gnomadg_nfe': 0.08978, 'gnomadg_amr': 0.2402, 'gnomadg_afr': 0.3116, 'gnomadg_mid': 0.1293, 'amr': 0.2767, 'gnomadg_fin': 0.1658, 'af': 0.2802, 'gnomadg_eas': 0.5242, 'gnomadg': 0.1882, 'gnomadg_sas': 0.116, 'gnomadg_ami': 0.1195, 'eur': 0.0944}}}], 'transcript_consequences': [{'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1}, {'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'strand': 1}], 'id': 'rs16854533', 'most_severe_consequence': 'intron_variant', 'end': 204852036, 'allele_string': 'G/A', 'strand': 1, 'input': 'rs16854533'}, {'assembly_name': 'GRCh38', 'start': 204852085, 'input': 'rs16854537', 'strand': 1, 'allele_string': 'T/C', 'most_severe_consequence': 'intron_variant', 'end': 204852085, 'transcript_consequences': [{'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393'}, {'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531'}, {'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}], 'colocated_variants': [{'strand': 1, 'frequencies': {'C': {'eas': 0.5556, 'sas': 0.091, 'gnomadg_remaining': 0.1761, 'gnomadg_nfe': 0.09036, 'afr': 0.3676, 'gnomadg_asj': 0.1225, 'gnomadg_fin': 0.1662, 'gnomadg_amr': 0.2423, 'gnomadg_mid': 0.1293, 'gnomadg_afr': 0.3224, 'amr': 0.281, 'af': 0.2845, 'gnomadg_eas': 0.5275, 'gnomadg': 0.1921, 'gnomadg_sas': 0.1209, 'gnomadg_ami': 0.1187, 'eur': 0.0944}}, 'allele_string': 'T/C', 'seq_region_name': '1', 'id': 'rs16854537', 'start': 204852085, 'end': 204852085}], 'seq_region_name': '1', 'id': 'rs16854537'}, {'input': 'rs12405886', 'allele_string': 'G/A/T', 'strand': 1, 'end': 204852168, 'most_severe_consequence': 'intron_variant', 'transcript_consequences': [{'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC'}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'strand': 1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC'}, {'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC'}, {'strand': 1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543'}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000513543'}, {'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC'}, {'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}], 'colocated_variants': [{'strand': 1, 'start': 204852168, 'end': 204852168, 'frequencies': {'A': {'gnomadg_asj': 0.1221, 'afr': 0.3676, 'gnomadg_nfe': 0.09031, 'gnomadg_remaining': 0.1766, 'eas': 0.5556, 'sas': 0.091, 'gnomadg': 0.1921, 'gnomadg_sas': 0.1205, 'gnomadg_ami': 0.1198, 'eur': 0.0944, 'gnomadg_amr': 0.2421, 'gnomadg_mid': 0.1293, 'amr': 0.281, 'gnomadg_afr': 0.3224, 'gnomadg_fin': 0.1663, 'af': 0.2845, 'gnomadg_eas': 0.5271}}, 'seq_region_name': '1', 'id': 'rs12405886', 'allele_string': 'G/A/T'}], 'seq_region_name': '1', 'id': 'rs12405886', 'assembly_name': 'GRCh38', 'start': 204852168}, {'input': 'rs12408680', 'allele_string': 'C/T', 'strand': 1, 'end': 204852360, 'most_severe_consequence': 'intron_variant', 'colocated_variants': [{'start': 204852360, 'end': 204852360, 'frequencies': {'T': {'gnomadg_remaining': 0.1758, 'gnomadg_nfe': 0.09034, 'afr': 0.3676, 'gnomadg_asj': 0.1226, 'sas': 0.091, 'eas': 0.5556, 'eur': 0.0944, 'gnomadg_ami': 0.1176, 'gnomadg_sas': 0.1206, 'gnomadg': 0.192, 'gnomadg_eas': 0.527, 'af': 0.2845, 'gnomadg_fin': 0.1665, 'gnomadg_afr': 0.3222, 'gnomadg_mid': 0.1293, 'amr': 0.281, 'gnomadg_amr': 0.2424}}, 'seq_region_name': '1', 'id': 'rs12408680', 'allele_string': 'C/T', 'strand': 1}], 'transcript_consequences': [{'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000403080', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000430393', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531'}, {'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}, {'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427'}], 'id': 'rs12408680', 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 204852360}, {'assembly_name': 'GRCh38', 'start': 204852424, 'allele_string': 'T/C', 'strand': 1, 'input': 'rs12405298', 'end': 204852424, 'most_severe_consequence': 'intron_variant', 'colocated_variants': [{'start': 204852424, 'end': 204852424, 'frequencies': {'C': {'eur': 0.0944, 'gnomadg_ami': 0.1195, 'gnomadg_sas': 0.1202, 'gnomadg': 0.192, 'gnomadg_eas': 0.528, 'af': 0.2847, 'gnomadg_mid': 0.1293, 'gnomadg_afr': 0.3221, 'amr': 0.281, 'gnomadg_amr': 0.2422, 'gnomadg_fin': 0.1658, 'afr': 0.3676, 'gnomadg_asj': 0.1224, 'gnomadg_remaining': 0.1766, 'gnomadg_nfe': 0.0903, 'sas': 0.092, 'eas': 0.5556}}, 'seq_region_name': '1', 'id': 'rs12405298', 'allele_string': 'T/C', 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000339876', 'gene_symbol_source': 'HGNC'}, {'strand': 1, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'transcript_id': 'ENST00000404977', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1, 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866'}, {'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000680427', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'C', 'impact': 'MODIFIER'}], 'id': 'rs12405298'}, {'start': 204852559, 'assembly_name': 'GRCh38', 'most_severe_consequence': 'intron_variant', 'end': 204852559, 'input': 'rs11240292', 'strand': 1, 'allele_string': 'A/G', 'seq_region_name': '1', 'colocated_variants': [{'strand': 1, 'end': 204852559, 'start': 204852559, 'seq_region_name': '1', 'id': 'rs11240292', 'allele_string': 'A/G', 'frequencies': {'G': {'gnomadg_eas': 0.5277, 'af': 0.2971, 'gnomadg_mid': 0.1327, 'gnomadg_afr': 0.3731, 'amr': 0.2896, 'gnomadg_amr': 0.2471, 'gnomadg_fin': 0.1662, 'eur': 0.0944, 'gnomadg_ami': 0.1195, 'gnomadg_sas': 0.1214, 'gnomadg': 0.2066, 'sas': 0.091, 'eas': 0.5556, 'afr': 0.4107, 'gnomadg_asj': 0.122, 'gnomadg_remaining': 0.1864, 'gnomadg_nfe': 0.09038}}}], 'transcript_consequences': [{'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000339876', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000403080', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': 1}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': 1, 'gene_symbol': 'NFASC', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000404977'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000430393', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant']}, {'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29866', 'gene_id': 'ENSG00000163531', 'transcript_id': 'ENST00000513543', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': 1, 'gene_symbol': 'NFASC', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'transcript_id': 'ENST00000680427', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000163531', 'hgnc_id': 'HGNC:29866', 'consequence_terms': ['intron_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NFASC', 'strand': 1, 'biotype': 'protein_coding'}], 'id': 'rs11240292'}, {'start': 205680319, 'assembly_name': 'GRCh38', 'colocated_variants': [{'strand': 1, 'start': 205680319, 'end': 205680319, 'frequencies': {'A': {'gnomadg_fin': 0.03119, 'gnomadg_mid': 0.07192, 'amr': 0.111, 'gnomadg_afr': 0.01332, 'gnomadg_amr': 0.07722, 'gnomadg_eas': 0.1542, 'af': 0.0725, 'gnomadg_sas': 0.08987, 'gnomadg': 0.03891, 'gnomade': 0.05102, 'eur': 0.0278, 'gnomadg_ami': 0.01978, 'eas': 0.1538, 'gnomade_eas': 0.5, 'gnomade_nfe': 0.04444, 'sas': 0.0992, 'gnomadg_remaining': 0.04309, 'gnomadg_nfe': 0.03295, 'gnomade_amr': 0, 'afr': 0.0045, 'gnomadg_asj': 0.07604, 'gnomade_remaining': 0}}, 'id': 'rs61824662', 'seq_region_name': '1', 'allele_string': 'G/A'}], 'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367145', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000158715', 'hgnc_id': 'HGNC:8642', 'gene_symbol': 'SLC45A3', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429', 'gene_id': 'ENSG00000275213', 'consequence_terms': ['upstream_gene_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'Metazoa_SRP', 'strand': 1, 'biotype': 'misc_RNA', 'distance': 3699}, {'biotype': 'lncRNA', 'distance': 2958, 'strand': -1, 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767938'}, {'transcript_id': 'ENST00000767939', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 2964, 'strand': -1}, {'strand': -1, 'distance': 3004, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000767940'}, {'biotype': 'lncRNA', 'distance': 3205, 'strand': 1, 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768286'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768287', 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'cdna_start': 48, 'transcript_id': 'ENST00000768288', 'consequence_terms': ['non_coding_transcript_exon_variant'], 'cdna_end': 48, 'gene_id': 'ENSG00000300035', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}], 'seq_region_name': '1', 'id': 'rs61824662', 'most_severe_consequence': 'non_coding_transcript_exon_variant', 'regulatory_feature_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['regulatory_region_variant'], 'biotype': 'enhancer', 'regulatory_feature_id': 'ENSR1_CQF3G'}], 'end': 205680319, 'input': 'rs61824662', 'strand': 1, 'allele_string': 'G/A'}, {'start': 205681143, 'assembly_name': 'GRCh38', 'most_severe_consequence': 'intron_variant', 'end': 205681143, 'strand': 1, 'allele_string': 'T/C/G', 'input': 'rs1775153', 'colocated_variants': [{'strand': 1, 'start': 205681143, 'end': 205681143, 'frequencies': {'G': {'sas': 0.4755, 'eas': 0.5089, 'gnomadg_remaining': 0.6204, 'gnomadg_nfe': 0.7513, 'afr': 0.2118, 'gnomadg_asj': 0.6897, 'af': 0.5012, 'gnomadg_eas': 0.4959, 'gnomadg_fin': 0.7189, 'gnomadg_amr': 0.6989, 'gnomadg_afr': 0.2963, 'amr': 0.7421, 'gnomadg_mid': 0.585, 'gnomadg_ami': 0.7203, 'eur': 0.7326, 'gnomadg': 0.6008, 'gnomadg_sas': 0.542}}, 'id': 'rs1775153', 'seq_region_name': '1', 'allele_string': 'T/C/G'}], 'transcript_consequences': [{'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC45A3', 'strand': -1, 'biotype': 'protein_coding', 'distance': 634, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367145', 'gene_id': 'ENSG00000158715', 'hgnc_id': 'HGNC:8642', 'consequence_terms': ['upstream_gene_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367145', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000158715', 'hgnc_id': 'HGNC:8642', 'gene_symbol': 'SLC45A3', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 634, 'biotype': 'protein_coding', 'strand': -1}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'Metazoa_SRP', 'strand': 1, 'distance': 2875, 'biotype': 'misc_RNA', 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429', 'gene_id': 'ENSG00000275213', 'consequence_terms': ['upstream_gene_variant']}, {'gene_id': 'ENSG00000275213', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429', 'strand': 1, 'distance': 2875, 'biotype': 'misc_RNA', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'Metazoa_SRP'}, {'transcript_id': 'ENST00000767938', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 2134, 'biotype': 'lncRNA', 'strand': -1}, {'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'strand': -1, 'distance': 2134, 'biotype': 'lncRNA'}, {'strand': -1, 'distance': 2140, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000767939'}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'strand': -1, 'biotype': 'lncRNA', 'distance': 2140, 'transcript_id': 'ENST00000767939', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant']}, {'distance': 2180, 'biotype': 'lncRNA', 'strand': -1, 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767940'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 2180, 'strand': -1, 'transcript_id': 'ENST00000767940', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300008'}, {'biotype': 'lncRNA', 'distance': 2381, 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768286'}, {'biotype': 'lncRNA', 'distance': 2381, 'strand': 1, 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768286'}, {'transcript_id': 'ENST00000768287', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000768287', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035'}, {'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768288'}, {'transcript_id': 'ENST00000768288', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}], 'seq_region_name': '1', 'id': 'rs1775153'}, {'input': 'rs34606094', 'strand': 1, 'allele_string': 'C/A/T', 'end': 205682206, 'most_severe_consequence': 'intron_variant', 'id': 'rs34606094', 'colocated_variants': [{'strand': 1, 'id': 'rs34606094', 'seq_region_name': '1', 'allele_string': 'C/A/T', 'frequencies': {'T': {'gnomadg_sas': 0.09768, 'gnomadg': 0.06485, 'eur': 0.0775, 'gnomadg_ami': 0.06828, 'gnomadg_fin': 0.09776, 'amr': 0.1239, 'gnomadg_afr': 0.01947, 'gnomadg_mid': 0.102, 'gnomadg_amr': 0.09549, 'gnomadg_eas': 0.1546, 'af': 0.0869, 'gnomadg_remaining': 0.06708, 'gnomadg_nfe': 0.06969, 'afr': 0.0053, 'gnomadg_asj': 0.09359, 'eas': 0.1528, 'sas': 0.1125}}, 'end': 205682206, 'start': 205682206}], 'transcript_consequences': [{'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:8642', 'gene_id': 'ENSG00000158715', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367145', 'distance': 1697, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC45A3', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 1697, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'SLC45A3', 'gene_id': 'ENSG00000158715', 'hgnc_id': 'HGNC:8642', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367145'}, {'strand': 1, 'distance': 1812, 'biotype': 'misc_RNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'Metazoa_SRP', 'gene_id': 'ENSG00000275213', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429'}, {'gene_id': 'ENSG00000275213', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429', 'strand': 1, 'biotype': 'misc_RNA', 'distance': 1812, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'Metazoa_SRP'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 1071, 'strand': -1, 'transcript_id': 'ENST00000767938', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300008'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': -1, 'biotype': 'lncRNA', 'distance': 1071, 'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant']}, {'distance': 1077, 'biotype': 'lncRNA', 'strand': -1, 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767939'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 1077, 'strand': -1, 'transcript_id': 'ENST00000767939', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300008'}, {'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000767940', 'strand': -1, 'distance': 1117, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000767940', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': -1, 'distance': 1117, 'biotype': 'lncRNA'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 1318, 'strand': 1, 'transcript_id': 'ENST00000768286', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300035'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768286', 'biotype': 'lncRNA', 'distance': 1318, 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000768287', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035'}, {'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768287'}, {'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000768288'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000768288', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}], 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 205682206}, {'assembly_name': 'GRCh38', 'start': 205682349, 'seq_region_name': '1', 'colocated_variants': [{'id': 'rs1772136', 'seq_region_name': '1', 'allele_string': 'C/G/T', 'frequencies': {'T': {'eas': 0.5079, 'sas': 0.4755, 'gnomadg_asj': 0.6894, 'afr': 0.2126, 'gnomadg_remaining': 0.6195, 'gnomadg_nfe': 0.7518, 'gnomadg_amr': 0.6976, 'gnomadg_afr': 0.2969, 'amr': 0.7378, 'gnomadg_mid': 0.5822, 'gnomadg_fin': 0.7179, 'af': 0.5004, 'gnomadg_eas': 0.4936, 'gnomadg': 0.6007, 'gnomadg_sas': 0.5406, 'gnomadg_ami': 0.7192, 'eur': 0.7316}}, 'end': 205682349, 'start': 205682349, 'strand': 1}], 'transcript_consequences': [{'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'SLC45A3', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1840, 'transcript_id': 'ENST00000367145', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:8642', 'gene_id': 'ENSG00000158715', 'consequence_terms': ['upstream_gene_variant']}, {'hgnc_id': 'HGNC:8642', 'gene_id': 'ENSG00000158715', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000367145', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 1840, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'SLC45A3'}, {'transcript_id': 'ENST00000619429', 'gene_symbol_source': 'RFAM', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000275213', 'gene_symbol': 'Metazoa_SRP', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'misc_RNA', 'distance': 1669, 'strand': 1}, {'gene_symbol': 'Metazoa_SRP', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 1669, 'biotype': 'misc_RNA', 'strand': 1, 'transcript_id': 'ENST00000619429', 'gene_symbol_source': 'RFAM', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000275213'}, {'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'strand': -1, 'distance': 928, 'biotype': 'lncRNA'}, {'transcript_id': 'ENST00000767938', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 928, 'biotype': 'lncRNA', 'strand': -1}, {'transcript_id': 'ENST00000767939', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'strand': -1, 'biotype': 'lncRNA', 'distance': 934}, {'strand': -1, 'biotype': 'lncRNA', 'distance': 934, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000767939'}, {'transcript_id': 'ENST00000767940', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 974, 'biotype': 'lncRNA', 'strand': -1}, {'transcript_id': 'ENST00000767940', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 974, 'strand': -1}, {'distance': 1175, 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'G', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768286'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 1175, 'strand': 1, 'transcript_id': 'ENST00000768286', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300035'}, {'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768287'}, {'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768287'}, {'transcript_id': 'ENST00000768288', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000768288', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035'}], 'id': 'rs1772136', 'input': 'rs1772136', 'strand': 1, 'allele_string': 'C/G/T', 'end': 205682349, 'most_severe_consequence': 'intron_variant'}, {'colocated_variants': [{'id': 'rs61824663', 'seq_region_name': '1', 'allele_string': 'G/A/T', 'frequencies': {'T': {'gnomadg_eas': 0, 'af': 0.0002, 'amr': 0.0014, 'gnomadg_afr': 2.415e-05, 'gnomadg_mid': 0, 'gnomadg_amr': 0, 'gnomadg_fin': 0, 'eur': 0, 'gnomadg_ami': 0, 'gnomadg_sas': 0, 'gnomadg': 6.583e-06, 'sas': 0, 'eas': 0, 'afr': 0, 'gnomadg_asj': 0, 'gnomadg_remaining': 0, 'gnomadg_nfe': 0}, 'A': {'gnomadg_sas': 0.08968, 'gnomadg': 0.03907, 'eur': 0.0278, 'gnomadg_ami': 0.01978, 'gnomadg_mid': 0.07192, 'amr': 0.111, 'gnomadg_afr': 0.01328, 'gnomadg_amr': 0.0776, 'gnomadg_fin': 0.03129, 'gnomadg_eas': 0.155, 'af': 0.0721, 'afr': 0.0045, 'gnomadg_asj': 0.07612, 'gnomadg_remaining': 0.04373, 'gnomadg_nfe': 0.03308, 'eas': 0.1528, 'sas': 0.0982}}, 'end': 205682722, 'start': 205682722, 'strand': 1}], 'id': 'rs61824663', 'transcript_consequences': [{'hgnc_id': 'HGNC:8642', 'gene_id': 'ENSG00000158715', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000367145', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'distance': 2213, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC45A3'}, {'gene_symbol': 'SLC45A3', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'protein_coding', 'distance': 2213, 'strand': -1, 'transcript_id': 'ENST00000367145', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000158715', 'hgnc_id': 'HGNC:8642'}, {'transcript_id': 'ENST00000619429', 'gene_symbol_source': 'RFAM', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000275213', 'gene_symbol': 'Metazoa_SRP', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 1296, 'biotype': 'misc_RNA', 'strand': 1}, {'strand': 1, 'biotype': 'misc_RNA', 'distance': 1296, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'Metazoa_SRP', 'gene_id': 'ENSG00000275213', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429'}, {'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000767938', 'strand': -1, 'distance': 555, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000767938', 'strand': -1, 'distance': 555, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'strand': -1, 'biotype': 'lncRNA', 'distance': 561, 'transcript_id': 'ENST00000767939', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant']}, {'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000767939', 'strand': -1, 'biotype': 'lncRNA', 'distance': 561, 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'strand': -1, 'distance': 601, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767940', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['downstream_gene_variant']}, {'transcript_id': 'ENST00000767940', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 601, 'strand': -1}, {'distance': 802, 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768286'}, {'strand': 1, 'distance': 802, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000768286'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000768287', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'T', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768287'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768288', 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'transcript_id': 'ENST00000768288', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': 1, 'biotype': 'lncRNA'}], 'seq_region_name': '1', 'input': 'rs61824663', 'strand': 1, 'allele_string': 'G/A/T', 'end': 205682722, 'most_severe_consequence': 'intron_variant', 'assembly_name': 'GRCh38', 'start': 205682722}, {'start': 205684628, 'assembly_name': 'GRCh38', 'colocated_variants': [{'frequencies': {'C': {'gnomadg_asj': 0.6898, 'afr': 0.2474, 'gnomadg_remaining': 0.6257, 'gnomadg_nfe': 0.7507, 'eas': 0.5089, 'sas': 0.4755, 'gnomadg_sas': 0.541, 'gnomadg': 0.6062, 'eur': 0.7326, 'gnomadg_ami': 0.7192, 'gnomadg_mid': 0.5897, 'amr': 0.7421, 'gnomadg_afr': 0.3165, 'gnomadg_amr': 0.7015, 'gnomadg_fin': 0.7165, 'gnomadg_eas': 0.4927, 'af': 0.5106}}, 'seq_region_name': '1', 'id': 'rs6685188', 'allele_string': 'T/A/C', 'start': 205684628, 'pubmed': [36206743, 27863252], 'end': 205684628, 'phenotype_or_disease': 1, 'strand': 1}], 'transcript_consequences': [{'gene_symbol': 'SLC45A3', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'distance': 4119, 'strand': -1, 'transcript_id': 'ENST00000367145', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000158715', 'hgnc_id': 'HGNC:8642'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367145', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000158715', 'hgnc_id': 'HGNC:8642', 'gene_symbol': 'SLC45A3', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding', 'distance': 4119, 'strand': -1}, {'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429', 'gene_id': 'ENSG00000275213', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'Metazoa_SRP', 'strand': 1, 'biotype': 'misc_RNA', 'distance': 321}, {'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429', 'gene_id': 'ENSG00000275213', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'Metazoa_SRP', 'strand': 1, 'distance': 321, 'biotype': 'misc_RNA'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': -1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'strand': -1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'biotype': 'lncRNA', 'strand': -1, 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767939'}, {'biotype': 'lncRNA', 'strand': -1, 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767939'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': -1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767940', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'transcript_id': 'ENST00000767940', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'strand': -1, 'biotype': 'lncRNA'}, {'gene_id': 'ENSG00000300035', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000768286', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000768286', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 904, 'strand': 1, 'transcript_id': 'ENST00000768287', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'strand': 1, 'distance': 904, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000768287', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['downstream_gene_variant']}, {'strand': 1, 'distance': 907, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000768288'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'strand': 1, 'distance': 907, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000768288', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['downstream_gene_variant']}], 'seq_region_name': '1', 'id': 'rs6685188', 'end': 205684628, 'most_severe_consequence': 'intron_variant', 'allele_string': 'T/A/C', 'strand': 1, 'input': 'rs6685188'}, {'assembly_name': 'GRCh38', 'start': 205686421, 'colocated_variants': [{'strand': 1, 'end': 205686421, 'start': 205686421, 'seq_region_name': '1', 'id': 'rs12402185', 'allele_string': 'G/T', 'frequencies': {'T': {'afr': 0.0045, 'gnomadg_asj': 0.07575, 'gnomadg_remaining': 0.04352, 'gnomadg_nfe': 0.03303, 'eas': 0.1528, 'sas': 0.0992, 'gnomadg': 0.03904, 'gnomadg_sas': 0.09031, 'gnomadg_ami': 0.01978, 'eur': 0.0278, 'gnomadg_amr': 0.07729, 'gnomadg_mid': 0.07143, 'gnomadg_afr': 0.01327, 'amr': 0.1124, 'gnomadg_fin': 0.0313, 'af': 0.0725, 'gnomadg_eas': 0.1548}}}], 'transcript_consequences': [{'gene_id': 'ENSG00000275213', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429', 'strand': 1, 'biotype': 'misc_RNA', 'distance': 2114, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'Metazoa_SRP'}, {'strand': -1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000767938'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'strand': -1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767939', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'transcript_id': 'ENST00000767940', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': -1}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768286', 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'strand': 1, 'biotype': 'lncRNA', 'distance': 2697, 'transcript_id': 'ENST00000768287', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['downstream_gene_variant']}, {'strand': 1, 'distance': 2700, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000768288'}], 'id': 'rs12402185', 'seq_region_name': '1', 'strand': 1, 'allele_string': 'G/T', 'input': 'rs12402185', 'most_severe_consequence': 'intron_variant', 'end': 205686421}, {'input': 'rs12404824', 'allele_string': 'C/G', 'strand': 1, 'end': 205686692, 'most_severe_consequence': 'intron_variant', 'colocated_variants': [{'strand': 1, 'allele_string': 'C/G', 'id': 'rs12404824', 'seq_region_name': '1', 'frequencies': {'G': {'sas': 0.0992, 'eas': 0.1528, 'gnomadg_remaining': 0.04356, 'gnomadg_nfe': 0.03304, 'gnomadg_asj': 0.0746, 'afr': 0.0045, 'gnomadg_eas': 0.1548, 'af': 0.0725, 'gnomadg_fin': 0.03133, 'gnomadg_afr': 0.01335, 'amr': 0.1124, 'gnomadg_mid': 0.07143, 'gnomadg_amr': 0.07707, 'eur': 0.0278, 'gnomadg_ami': 0.01974, 'gnomadg_sas': 0.09012, 'gnomadg': 0.03898}}, 'end': 205686692, 'start': 205686692}], 'transcript_consequences': [{'gene_symbol': 'Metazoa_SRP', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'misc_RNA', 'distance': 2385, 'strand': 1, 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000275213'}, {'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'strand': -1, 'biotype': 'lncRNA'}, {'biotype': 'lncRNA', 'strand': -1, 'impact': 'MODIFIER', 'variant_allele': 'G', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767939'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'strand': -1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767940', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768286', 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'gene_id': 'ENSG00000300035', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000768287', 'strand': 1, 'biotype': 'lncRNA', 'distance': 2968, 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000768288', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'lncRNA', 'distance': 2971, 'strand': 1}], 'id': 'rs12404824', 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 205686692}, {'start': 205686896, 'assembly_name': 'GRCh38', 'colocated_variants': [{'start': 205686896, 'end': 205686896, 'frequencies': {'C': {'eur': 0.0278, 'gnomadg_ami': 0.01978, 'gnomadg_sas': 0.08989, 'gnomadg': 0.03786, 'gnomadg_eas': 0.1544, 'af': 0.0725, 'gnomadg_fin': 0.03126, 'gnomadg_mid': 0.06803, 'amr': 0.1124, 'gnomadg_afr': 0.008633, 'gnomadg_amr': 0.07744, 'gnomadg_remaining': 0.04269, 'gnomadg_nfe': 0.03334, 'afr': 0.0045, 'gnomadg_asj': 0.07555, 'sas': 0.0992, 'eas': 0.1528}, 'A': {'gnomadg_eas': 0, 'gnomadg_fin': 0, 'gnomadg_amr': 0, 'gnomadg_afr': 0, 'gnomadg_mid': 0, 'gnomadg_ami': 0, 'gnomadg_nfe': 5.882e-05, 'gnomadg_remaining': 0, 'gnomadg': 2.632e-05, 'gnomadg_sas': 0, 'gnomadg_asj': 0}}, 'allele_string': 'G/A/C', 'id': 'rs12737668', 'seq_region_name': '1', 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'gene_id': 'ENSG00000275213', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429', 'strand': 1, 'distance': 2589, 'biotype': 'misc_RNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'Metazoa_SRP'}, {'gene_id': 'ENSG00000275213', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429', 'strand': 1, 'biotype': 'misc_RNA', 'distance': 2589, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'Metazoa_SRP'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': -1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': -1, 'transcript_id': 'ENST00000767938', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008'}, {'transcript_id': 'ENST00000767939', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': -1}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767939', 'biotype': 'lncRNA', 'strand': -1, 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'biotype': 'lncRNA', 'strand': -1, 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767940'}, {'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000767940', 'strand': -1, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000768286', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': 1, 'biotype': 'lncRNA'}, {'gene_id': 'ENSG00000300035', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000768286', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000768287', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 3172, 'strand': 1}, {'gene_id': 'ENSG00000300035', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000768287', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3172, 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000768288', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 3175, 'strand': 1}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768288', 'distance': 3175, 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'C', 'impact': 'MODIFIER'}], 'id': 'rs12737668', 'end': 205686896, 'most_severe_consequence': 'intron_variant', 'strand': 1, 'allele_string': 'G/A/C', 'input': 'rs12737668'}, {'assembly_name': 'GRCh38', 'start': 205687325, 'colocated_variants': [{'end': 205687325, 'pubmed': [23921447, 37993489, 38064665], 'start': 205687325, 'allele_string': 'G/A/C', 'id': 'rs823106', 'seq_region_name': '1', 'frequencies': {'C': {'afr': 0.5439, 'gnomadg_asj': 0.762, 'gnomadg_nfe': 0.8642, 'gnomadg_remaining': 0.7486, 'sas': 0.5685, 'eas': 0.5169, 'eur': 0.8549, 'gnomadg_ami': 0.8004, 'gnomadg_sas': 0.6274, 'gnomadg': 0.7521, 'gnomadg_eas': 0.4986, 'af': 0.643, 'gnomadg_mid': 0.6871, 'gnomadg_afr': 0.5763, 'amr': 0.8127, 'gnomadg_amr': 0.7889, 'gnomadg_fin': 0.8428}}, 'strand': 1, 'phenotype_or_disease': 1}], 'transcript_consequences': [{'gene_id': 'ENSG00000275213', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429', 'strand': 1, 'biotype': 'misc_RNA', 'distance': 3018, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'Metazoa_SRP'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'Metazoa_SRP', 'strand': 1, 'biotype': 'misc_RNA', 'distance': 3018, 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000619429', 'gene_id': 'ENSG00000275213', 'consequence_terms': ['downstream_gene_variant']}, {'transcript_id': 'ENST00000767938', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': -1}, {'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000767938', 'strand': -1, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': -1, 'transcript_id': 'ENST00000767939', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008'}, {'transcript_id': 'ENST00000767939', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'strand': -1, 'biotype': 'lncRNA'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767940', 'biotype': 'lncRNA', 'strand': -1, 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'strand': -1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767940', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000768286', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035'}, {'transcript_id': 'ENST00000768286', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'distance': 3601, 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768287'}, {'transcript_id': 'ENST00000768287', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 3601, 'strand': 1}, {'biotype': 'lncRNA', 'distance': 3604, 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'transcript_id': 'ENST00000768288'}, {'strand': 1, 'distance': 3604, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000768288'}], 'seq_region_name': '1', 'id': 'rs823106', 'strand': 1, 'allele_string': 'G/A/C', 'input': 'rs823106', 'end': 205687325, 'most_severe_consequence': 'intron_variant'}, {'start': 205691812, 'assembly_name': 'GRCh38', 'most_severe_consequence': 'intron_variant', 'end': 205691812, 'input': 'rs12132270', 'strand': 1, 'allele_string': 'C/G/T', 'transcript_consequences': [{'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': -1, 'transcript_id': 'ENST00000767938', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': -1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'transcript_id': 'ENST00000767939', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': -1}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': -1, 'transcript_id': 'ENST00000767939', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767940', 'biotype': 'lncRNA', 'strand': -1, 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': -1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767940', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'transcript_id': 'ENST00000768286', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 1074, 'strand': 1}, {'transcript_id': 'ENST00000768286', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 1074, 'biotype': 'lncRNA', 'strand': 1}], 'colocated_variants': [{'id': 'rs12132270', 'seq_region_name': '1', 'allele_string': 'C/G/T', 'frequencies': {'T': {'amr': 0.1455, 'gnomadg_mid': 0.2226, 'gnomadg_afr': 0.06038, 'gnomadg_amr': 0.1641, 'gnomadg_fin': 0.1557, 'gnomadg_eas': 0.5177, 'af': 0.2474, 'gnomadg_sas': 0.3829, 'gnomadg': 0.1416, 'eur': 0.1342, 'gnomadg_ami': 0.1974, 'eas': 0.498, 'sas': 0.4387, 'gnomadg_asj': 0.187, 'afr': 0.0545, 'gnomadg_nfe': 0.1344, 'gnomadg_remaining': 0.1588}}, 'end': 205691812, 'start': 205691812, 'strand': 1}], 'seq_region_name': '1', 'id': 'rs12132270'}, {'start': 205692290, 'assembly_name': 'GRCh38', 'id': 'rs1891091', 'colocated_variants': [{'start': 205692290, 'end': 205692290, 'frequencies': {'T': {'gnomadg_eas': 0.5174, 'af': 0.2264, 'gnomadg_fin': 0.1511, 'gnomadg_mid': 0.1701, 'gnomadg_afr': 0.02304, 'amr': 0.1196, 'gnomadg_amr': 0.1363, 'eur': 0.1193, 'gnomadg_ami': 0.06798, 'gnomadg_sas': 0.3763, 'gnomadg': 0.1173, 'sas': 0.4335, 'eas': 0.499, 'gnomadg_nfe': 0.116, 'gnomadg_remaining': 0.132, 'afr': 0.003, 'gnomadg_asj': 0.1293}, 'A': {'gnomadg_eas': 0, 'gnomadg_fin': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_amr': 0, 'gnomadg_ami': 0, 'gnomadg_sas': 0, 'gnomadg_remaining': 0, 'gnomadg_nfe': 2.941e-05, 'gnomadg': 1.316e-05, 'gnomadg_asj': 0}}, 'id': 'rs1891091', 'seq_region_name': '1', 'allele_string': 'C/A/T', 'strand': 1}], 'transcript_consequences': [{'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': -1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'strand': -1, 'biotype': 'lncRNA'}, {'transcript_id': 'ENST00000767939', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': -1}, {'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000767939', 'strand': -1, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000767940', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': -1}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': -1, 'transcript_id': 'ENST00000767940', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008'}, {'transcript_id': 'ENST00000768286', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 1552, 'strand': 1}, {'transcript_id': 'ENST00000768286', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 1552, 'biotype': 'lncRNA', 'strand': 1}], 'seq_region_name': '1', 'end': 205692290, 'most_severe_consequence': 'intron_variant', 'strand': 1, 'allele_string': 'C/A/T', 'input': 'rs1891091'}, {'assembly_name': 'GRCh38', 'start': 205693016, 'allele_string': 'T/A/C', 'strand': 1, 'input': 'rs10751444', 'end': 205693016, 'most_severe_consequence': 'intron_variant', 'colocated_variants': [{'end': 205693016, 'start': 205693016, 'allele_string': 'T/A/C', 'id': 'rs10751444', 'seq_region_name': '1', 'frequencies': {'C': {'af': 0.3842, 'gnomadg_eas': 0.5214, 'gnomadg_amr': 0.2537, 'gnomadg_afr': 0.3925, 'gnomadg_mid': 0.3129, 'amr': 0.2075, 'gnomadg_fin': 0.2782, 'gnomadg_ami': 0.273, 'eur': 0.2475, 'gnomadg': 0.3036, 'gnomadg_sas': 0.4535, 'sas': 0.5133, 'eas': 0.502, 'gnomadg_asj': 0.2323, 'afr': 0.3956, 'gnomadg_nfe': 0.2422, 'gnomadg_remaining': 0.2827}, 'A': {'gnomadg_amr': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_fin': 0, 'gnomadg_eas': 0.000387, 'gnomadg_asj': 0, 'gnomadg': 1.316e-05, 'gnomadg_remaining': 0, 'gnomadg_nfe': 0, 'gnomadg_sas': 0, 'gnomadg_ami': 0}}, 'strand': 1}], 'transcript_consequences': [{'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': -1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767938', 'biotype': 'lncRNA', 'strand': -1, 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767939', 'biotype': 'lncRNA', 'distance': 412, 'strand': -1, 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767939', 'biotype': 'lncRNA', 'distance': 412, 'strand': -1, 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767940', 'biotype': 'lncRNA', 'distance': 465, 'strand': -1, 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000767940', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 465, 'biotype': 'lncRNA', 'strand': -1}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'lncRNA', 'distance': 2278, 'strand': 1, 'transcript_id': 'ENST00000768286', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 2278, 'strand': 1, 'transcript_id': 'ENST00000768286', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035'}], 'seq_region_name': '1', 'id': 'rs10751444'}, {'end': 205693209, 'most_severe_consequence': 'intron_variant', 'input': 'rs6593964', 'strand': 1, 'allele_string': 'A/C/T', 'colocated_variants': [{'strand': 1, 'start': 205693209, 'end': 205693209, 'frequencies': {'C': {'gnomadg_remaining': 0.2829, 'gnomadg_nfe': 0.2418, 'afr': 0.3964, 'gnomadg_asj': 0.2321, 'sas': 0.5133, 'eas': 0.502, 'eur': 0.2475, 'gnomadg_ami': 0.2714, 'gnomadg_sas': 0.4542, 'gnomadg': 0.3035, 'gnomadg_eas': 0.5195, 'af': 0.3844, 'gnomadg_fin': 0.2781, 'gnomadg_mid': 0.3129, 'gnomadg_afr': 0.3927, 'amr': 0.2075, 'gnomadg_amr': 0.2535}}, 'seq_region_name': '1', 'id': 'rs6593964', 'allele_string': 'A/C/T'}], 'seq_region_name': '1', 'transcript_consequences': [{'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000767938', 'strand': -1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'strand': -1, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000767938'}, {'strand': -1, 'biotype': 'lncRNA', 'distance': 605, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000767939'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767939', 'biotype': 'lncRNA', 'distance': 605, 'strand': -1, 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'strand': -1, 'biotype': 'lncRNA', 'distance': 658, 'transcript_id': 'ENST00000767940', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['upstream_gene_variant']}, {'transcript_id': 'ENST00000767940', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['upstream_gene_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'strand': -1, 'biotype': 'lncRNA', 'distance': 658}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 2471, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000768286', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035'}, {'transcript_id': 'ENST00000768286', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 2471, 'strand': 1}], 'id': 'rs6593964', 'start': 205693209, 'assembly_name': 'GRCh38'}, {'assembly_name': 'GRCh38', 'start': 205693849, 'id': 'rs6676110', 'colocated_variants': [{'start': 205693849, 'end': 205693849, 'frequencies': {'A': {'gnomadg': 0.2874, 'gnomadg_sas': 0.4533, 'gnomadg_ami': 0.273, 'eur': 0.2475, 'gnomadg_fin': 0.2789, 'gnomadg_amr': 0.2479, 'gnomadg_mid': 0.3095, 'amr': 0.2017, 'gnomadg_afr': 0.3363, 'af': 0.368, 'gnomadg_eas': 0.5221, 'gnomadg_remaining': 0.2696, 'gnomadg_nfe': 0.2416, 'afr': 0.3374, 'gnomadg_asj': 0.2321, 'eas': 0.502, 'sas': 0.5133}}, 'seq_region_name': '1', 'id': 'rs6676110', 'allele_string': 'G/A', 'strand': 1}], 'transcript_consequences': [{'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767938', 'biotype': 'lncRNA', 'strand': -1, 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'strand': -1, 'distance': 1245, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767939', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['upstream_gene_variant']}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': -1, 'distance': 1298, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767940', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['upstream_gene_variant']}, {'transcript_id': 'ENST00000768286', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3111}], 'seq_region_name': '1', 'strand': 1, 'allele_string': 'G/A', 'input': 'rs6676110', 'end': 205693849, 'most_severe_consequence': 'intron_variant'}, {'start': 205694350, 'assembly_name': 'GRCh38', 'transcript_consequences': [{'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767938', 'biotype': 'lncRNA', 'strand': -1, 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'strand': -1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767939', 'distance': 1746, 'biotype': 'lncRNA', 'strand': -1, 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000767939', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': -1, 'distance': 1746, 'biotype': 'lncRNA'}, {'transcript_id': 'ENST00000767940', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 1799, 'strand': -1}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'distance': 1799, 'biotype': 'lncRNA', 'strand': -1, 'transcript_id': 'ENST00000767940', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300008'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 3612, 'strand': 1, 'transcript_id': 'ENST00000768286', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000300035'}, {'strand': 1, 'distance': 3612, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_id': 'ENSG00000300035', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000768286'}], 'colocated_variants': [{'strand': 1, 'phenotype_or_disease': 1, 'pubmed': [38155330], 'end': 205694350, 'start': 205694350, 'seq_region_name': '1', 'id': 'rs12118655', 'allele_string': 'A/G/T', 'frequencies': {'G': {'eas': 0.495, 'sas': 0.4387, 'afr': 0.0053, 'gnomadg_asj': 0.187, 'gnomadg_remaining': 0.152, 'gnomadg_nfe': 0.1344, 'gnomadg_amr': 0.1601, 'amr': 0.1412, 'gnomadg_afr': 0.02605, 'gnomadg_mid': 0.2158, 'gnomadg_fin': 0.1555, 'af': 0.2332, 'gnomadg_eas': 0.5193, 'gnomadg': 0.1318, 'gnomadg_sas': 0.3832, 'gnomadg_ami': 0.1989, 'eur': 0.1342}}}], 'id': 'rs12118655', 'seq_region_name': '1', 'end': 205694350, 'most_severe_consequence': 'intron_variant', 'input': 'rs12118655', 'strand': 1, 'allele_string': 'A/G/T'}, {'start': 205701598, 'assembly_name': 'GRCh38', 'most_severe_consequence': 'intron_variant', 'regulatory_feature_consequences': [{'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'enhancer', 'consequence_terms': ['regulatory_region_variant'], 'regulatory_feature_id': 'ENSR1_B3HS82'}, {'consequence_terms': ['regulatory_region_variant'], 'biotype': 'enhancer', 'regulatory_feature_id': 'ENSR1_B3HS82', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'regulatory_feature_id': 'ENSR1_B3HS82', 'consequence_terms': ['regulatory_region_variant'], 'biotype': 'enhancer', 'variant_allele': 'T', 'impact': 'MODIFIER'}], 'end': 205701598, 'input': 'rs823098', 'strand': 1, 'allele_string': 'A/C/G/T', 'seq_region_name': '1', 'colocated_variants': [{'start': 205701598, 'end': 205701598, 'frequencies': {'G': {'eas': 0.8512, 'sas': 0.8834, 'gnomadg_asj': 0.9073, 'afr': 0.7315, 'gnomadg_remaining': 0.9021, 'gnomadg_nfe': 0.9654, 'amr': 0.8646, 'gnomadg_mid': 0.8946, 'gnomadg_afr': 0.7738, 'gnomadg_amr': 0.8956, 'gnomadg_fin': 0.9691, 'gnomadg_eas': 0.851, 'af': 0.8508, 'gnomadg_sas': 0.8993, 'gnomadg': 0.8982, 'eur': 0.9662, 'gnomadg_ami': 0.9803}}, 'allele_string': 'A/C/G/T', 'id': 'rs823098', 'seq_region_name': '1', 'strand': 1}], 'transcript_consequences': [{'transcript_id': 'ENST00000767938', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': -1}, {'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000767938', 'strand': -1, 'biotype': 'lncRNA', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': -1, 'biotype': 'lncRNA'}], 'id': 'rs823098'}, {'allele_string': 'A/G/T', 'strand': 1, 'input': 'rs1772140', 'regulatory_feature_consequences': [{'impact': 'MODIFIER', 'variant_allele': 'G', 'regulatory_feature_id': 'ENSR1_8839X9', 'biotype': 'enhancer', 'consequence_terms': ['regulatory_region_variant']}, {'regulatory_feature_id': 'ENSR1_8839X9', 'biotype': 'enhancer', 'consequence_terms': ['regulatory_region_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER'}], 'most_severe_consequence': 'intron_variant', 'end': 205703354, 'colocated_variants': [{'seq_region_name': '1', 'id': 'rs1772140', 'allele_string': 'A/G/T', 'frequencies': {'G': {'afr': 0.7292, 'gnomadg_asj': 0.9075, 'gnomadg_remaining': 0.9019, 'gnomadg_nfe': 0.9652, 'sas': 0.8845, 'eas': 0.8433, 'eur': 0.9662, 'gnomadg_ami': 0.9803, 'gnomadg_sas': 0.8983, 'gnomadg': 0.8975, 'gnomadg_eas': 0.8469, 'af': 0.8486, 'amr': 0.8631, 'gnomadg_afr': 0.7727, 'gnomadg_mid': 0.8844, 'gnomadg_amr': 0.8947, 'gnomadg_fin': 0.9692}}, 'end': 205703354, 'start': 205703354, 'strand': 1}], 'id': 'rs1772140', 'transcript_consequences': [{'biotype': 'lncRNA', 'strand': -1, 'impact': 'MODIFIER', 'variant_allele': 'G', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767938'}, {'biotype': 'lncRNA', 'strand': -1, 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767938'}], 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 205703354}, {'colocated_variants': [{'strand': 1, 'phenotype_or_disease': 1, 'pubmed': [21738478, 37805635], 'end': 205707135, 'start': 205707135, 'id': 'rs12748961', 'seq_region_name': '1', 'allele_string': 'T/A/C', 'frequencies': {'C': {'gnomadg_eas': 0.5276, 'af': 0.233, 'gnomadg_mid': 0.2007, 'gnomadg_afr': 0.03253, 'amr': 0.1268, 'gnomadg_amr': 0.1503, 'gnomadg_fin': 0.1537, 'eur': 0.1292, 'gnomadg_ami': 0.2007, 'gnomadg_sas': 0.3788, 'gnomadg': 0.1309, 'sas': 0.4397, 'eas': 0.495, 'afr': 0.0151, 'gnomadg_asj': 0.1856, 'gnomadg_remaining': 0.147, 'gnomadg_nfe': 0.1309}}}], 'seq_region_name': '1', 'transcript_consequences': [{'gene_id': 'ENSG00000300008', 'consequence_terms': ['non_coding_transcript_exon_variant'], 'cdna_end': 30, 'transcript_id': 'ENST00000767938', 'cdna_start': 30, 'strand': -1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'consequence_terms': ['non_coding_transcript_exon_variant'], 'cdna_end': 30, 'gene_id': 'ENSG00000300008', 'cdna_start': 30, 'transcript_id': 'ENST00000767938', 'biotype': 'lncRNA', 'strand': -1, 'impact': 'MODIFIER', 'variant_allele': 'C'}], 'id': 'rs12748961', 'most_severe_consequence': 'non_coding_transcript_exon_variant', 'end': 205707135, 'regulatory_feature_consequences': [{'biotype': 'enhancer', 'consequence_terms': ['regulatory_region_variant'], 'regulatory_feature_id': 'ENSR1_B3HS8S', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['regulatory_region_variant'], 'biotype': 'enhancer', 'regulatory_feature_id': 'ENSR1_B3HS8S'}], 'input': 'rs12748961', 'strand': 1, 'allele_string': 'T/A/C', 'start': 205707135, 'assembly_name': 'GRCh38'}, {'input': 'rs3747973', 'strand': 1, 'allele_string': 'A/C/G/T', 'end': 205708020, 'most_severe_consequence': 'upstream_gene_variant', 'id': 'rs3747973', 'colocated_variants': [{'phenotype_or_disease': 1, 'strand': 1, 'frequencies': {'T': {'gnomadg_ami': 0, 'gnomadg_asj': 0, 'gnomadg_sas': 0, 'gnomadg_nfe': 0, 'gnomadg_remaining': 0, 'gnomadg': 5.295e-05, 'gnomadg_eas': 0.001556, 'gnomadg_afr': 0, 'gnomadg_mid': 0, 'gnomadg_amr': 0, 'gnomadg_fin': 0}, 'G': {'afr': 0.1868, 'gnomadg_asj': 0.6431, 'gnomadg_remaining': 0.5342, 'gnomadg_nfe': 0.5992, 'sas': 0.5869, 'eas': 0.499, 'eur': 0.5895, 'gnomadg_ami': 0.6148, 'gnomadg_sas': 0.5691, 'gnomadg': 0.4855, 'gnomadg_eas': 0.5309, 'af': 0.4397, 'amr': 0.4107, 'gnomadg_mid': 0.6327, 'gnomadg_afr': 0.2359, 'gnomadg_amr': 0.4628, 'gnomadg_fin': 0.6246}}, 'allele_string': 'A/C/G/T', 'seq_region_name': '1', 'id': 'rs3747973', 'start': 205708020, 'end': 205708020, 'pubmed': [36914875, 33462484, 34887591, 30595370, 32203549, 38538606, 37277458]}], 'transcript_consequences': [{'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'distance': 4802, 'strand': -1, 'gene_symbol': 'NUCKS1', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'distance': 4802, 'strand': -1, 'gene_symbol': 'NUCKS1', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NUCKS1', 'strand': -1, 'distance': 4802, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['downstream_gene_variant']}, {'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'strand': -1, 'distance': 856, 'biotype': 'lncRNA'}, {'transcript_id': 'ENST00000767938', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 856, 'biotype': 'lncRNA', 'strand': -1}, {'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': -1, 'biotype': 'lncRNA', 'distance': 856}], 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 205708020}, {'colocated_variants': [{'strand': 1, 'allele_string': 'C/G/T', 'id': 'rs3747972', 'seq_region_name': '1', 'frequencies': {'T': {'gnomadg_asj': 0.1854, 'afr': 0.0053, 'gnomadg_remaining': 0.1437, 'gnomadg_nfe': 0.129, 'sas': 0.4274, 'eas': 0.4931, 'gnomadg_ami': 0.2007, 'eur': 0.1292, 'gnomadg': 0.1267, 'gnomadg_sas': 0.364, 'af': 0.2272, 'gnomadg_eas': 0.5216, 'gnomadg_amr': 0.146, 'gnomadg_afr': 0.02596, 'gnomadg_mid': 0.2041, 'amr': 0.1239, 'gnomadg_fin': 0.1492}}, 'end': 205708207, 'start': 205708207}], 'seq_region_name': '1', 'transcript_consequences': [{'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4615}, {'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4615, 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NUCKS1'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'strand': -1, 'distance': 1043, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000767938', 'gene_id': 'ENSG00000300008', 'consequence_terms': ['upstream_gene_variant']}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000300008', 'transcript_id': 'ENST00000767938', 'distance': 1043, 'biotype': 'lncRNA', 'strand': -1, 'variant_allele': 'T', 'impact': 'MODIFIER'}], 'id': 'rs3747972', 'input': 'rs3747972', 'allele_string': 'C/G/T', 'strand': 1, 'end': 205708207, 'most_severe_consequence': 'upstream_gene_variant', 'assembly_name': 'GRCh38', 'start': 205708207}, {'assembly_name': 'GRCh38', 'start': 205714227, 'input': 'rs3805', 'allele_string': 'T/A/G', 'strand': 1, 'most_severe_consequence': '3_prime_UTR_variant', 'end': 205714227, 'transcript_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'strand': -1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'cdna_start': 4994, 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'cdna_end': 4994, 'consequence_terms': ['3_prime_UTR_variant']}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'NUCKS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'cdna_end': 4994, 'consequence_terms': ['3_prime_UTR_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'cdna_start': 4994, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'transcript_id': 'ENST00000464938', 'gene_symbol_source': 'HGNC', 'distance': 4225, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'NUCKS1', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'strand': -1, 'distance': 4225, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'NUCKS1', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000464938', 'gene_symbol_source': 'HGNC'}], 'colocated_variants': [{'strand': 1, 'phenotype_or_disease': 1, 'id': 'rs3805', 'seq_region_name': '1', 'allele_string': 'T/A/G', 'frequencies': {'G': {'gnomadg_fin': 0.1543, 'gnomadg_amr': 0.15, 'gnomadg_mid': 0.2021, 'amr': 0.1254, 'gnomadg_afr': 0.03729, 'af': 0.2336, 'gnomadg_eas': 0.534, 'gnomadg': 0.1322, 'gnomadg_sas': 0.3699, 'gnomade': 0.125, 'gnomadg_ami': 0.2011, 'eur': 0.1292, 'eas': 0.497, 'gnomade_eas': 0.5, 'gnomade_nfe': 0, 'sas': 0.4315, 'gnomadg_nfe': 0.1307, 'gnomadg_remaining': 0.1478, 'afr': 0.0227, 'gnomadg_asj': 0.1855}}, 'pubmed': [32201043], 'end': 205714227, 'start': 205714227}], 'seq_region_name': '1', 'id': 'rs3805'}, {'colocated_variants': [{'strand': 1, 'start': 205718745, 'end': 205718745, 'pubmed': [29619377], 'frequencies': {'T': {'gnomadg_eas': 0.8546, 'af': 0.8223, 'gnomadg_fin': 0.9589, 'gnomadg_mid': 0.8503, 'amr': 0.83, 'gnomadg_afr': 0.7271, 'gnomadg_amr': 0.8686, 'eur': 0.9394, 'gnomadg_ami': 0.9803, 'gnomadg_sas': 0.8849, 'gnomadg': 0.873, 'sas': 0.8732, 'eas': 0.8542, 'gnomadg_remaining': 0.87, 'gnomadg_nfe': 0.949, 'afr': 0.6672, 'gnomadg_asj': 0.8723}}, 'allele_string': 'A/C/G/T', 'seq_region_name': '1', 'id': 'rs823092'}], 'transcript_consequences': [{'gene_symbol': 'NUCKS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1, 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'NUCKS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142'}, {'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000464938', 'gene_symbol_source': 'HGNC'}, {'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000464938', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000464938', 'gene_symbol_source': 'HGNC'}], 'id': 'rs823092', 'seq_region_name': '1', 'input': 'rs823092', 'allele_string': 'A/C/G/T', 'strand': 1, 'end': 205718745, 'most_severe_consequence': 'intron_variant', 'assembly_name': 'GRCh38', 'start': 205718745}, {'most_severe_consequence': 'intron_variant', 'end': 205720099, 'input': 'rs823093', 'strand': 1, 'allele_string': 'G/A/T', 'colocated_variants': [{'start': 205720099, 'pubmed': [31719764, 35695515], 'end': 205720099, 'frequencies': {'T': {'gnomadg_eas': 0, 'gnomadg_fin': 0, 'gnomadg_amr': 0, 'gnomadg_afr': 0, 'gnomadg_mid': 0, 'gnomadg_ami': 0, 'gnomadg_remaining': 0, 'gnomadg': 0, 'gnomadg_nfe': 0, 'gnomadg_sas': 0, 'gnomadg_asj': 0}, 'A': {'gnomadg_fin': 0.9688, 'gnomadg_amr': 0.8822, 'gnomadg_mid': 0.8912, 'gnomadg_afr': 0.7096, 'amr': 0.8429, 'af': 0.8231, 'gnomadg_eas': 0.8545, 'gnomadg': 0.8793, 'gnomadg_sas': 0.892, 'gnomadg_ami': 0.9803, 'eur': 0.9662, 'eas': 0.8522, 'sas': 0.8783, 'gnomadg_remaining': 0.8861, 'gnomadg_nfe': 0.9657, 'afr': 0.6407, 'gnomadg_asj': 0.9119}}, 'id': 'rs823093', 'seq_region_name': '1', 'allele_string': 'G/A/T', 'strand': 1}], 'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NUCKS1', 'strand': -1, 'biotype': 'protein_coding'}, {'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 373, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000464938'}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 373, 'strand': -1, 'gene_symbol': 'NUCKS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'transcript_id': 'ENST00000464938', 'gene_symbol_source': 'HGNC'}], 'seq_region_name': '1', 'id': 'rs823093', 'start': 205720099, 'assembly_name': 'GRCh38'}, {'allele_string': 'T/A/C/G', 'strand': 1, 'input': 'rs823108', 'end': 205724473, 'most_severe_consequence': 'intron_variant', 'colocated_variants': [{'frequencies': {'G': {'gnomadg_ami': 0, 'gnomadg_remaining': 0, 'gnomadg': 6.569e-06, 'gnomadg_nfe': 0, 'gnomadg_sas': 0.0002071, 'gnomadg_asj': 0, 'gnomadg_eas': 0, 'gnomadg_fin': 0, 'gnomadg_amr': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0}, 'C': {'eur': 0.9662, 'gnomadg_ami': 0.9803, 'gnomadg_sas': 0.892, 'gnomadg': 0.8788, 'gnomadg_eas': 0.8558, 'af': 0.8225, 'gnomadg_fin': 0.9689, 'amr': 0.8444, 'gnomadg_mid': 0.8912, 'gnomadg_afr': 0.7076, 'gnomadg_amr': 0.8826, 'gnomadg_remaining': 0.8866, 'gnomadg_nfe': 0.9657, 'afr': 0.6392, 'gnomadg_asj': 0.9119, 'sas': 0.8783, 'eas': 0.8502}}, 'allele_string': 'T/A/C/G', 'id': 'rs823108', 'seq_region_name': '1', 'start': 205724473, 'end': 205724473, 'strand': 1}], 'transcript_consequences': [{'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'strand': -1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['intron_variant']}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'NUCKS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'NUCKS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4747, 'strand': -1, 'transcript_id': 'ENST00000464938', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275'}, {'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000464938', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 4747, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NUCKS1'}, {'gene_symbol': 'NUCKS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 4747, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000464938', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275'}], 'id': 'rs823108', 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 205724473}, {'seq_region_name': '1', 'colocated_variants': [{'strand': 1, 'start': 205727393, 'end': 205727393, 'frequencies': {'T': {'afr': 0.6415, 'gnomadg_asj': 0.9119, 'gnomadg_remaining': 0.8883, 'gnomadg_nfe': 0.9658, 'eas': 0.8522, 'sas': 0.8783, 'gnomadg': 0.8807, 'gnomadg_sas': 0.8919, 'gnomadg_ami': 0.9803, 'eur': 0.9662, 'gnomadg_amr': 0.8827, 'gnomadg_afr': 0.7143, 'amr': 0.8444, 'gnomadg_mid': 0.8912, 'gnomadg_fin': 0.9688, 'af': 0.8235, 'gnomadg_eas': 0.8558}}, 'allele_string': 'G/C/T', 'id': 'rs1772146', 'seq_region_name': '1'}], 'transcript_consequences': [{'biotype': 'snoRNA', 'distance': 3828, 'strand': -1, 'gene_symbol': 'SNORA72', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000201944', 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000365074'}, {'transcript_id': 'ENST00000365074', 'gene_symbol_source': 'RFAM', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000201944', 'gene_symbol': 'SNORA72', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'snoRNA', 'distance': 3828, 'strand': -1}, {'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1'}, {'gene_symbol': 'NUCKS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1, 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923'}], 'id': 'rs1772146', 'allele_string': 'G/C/T', 'strand': 1, 'input': 'rs1772146', 'most_severe_consequence': 'intron_variant', 'end': 205727393, 'assembly_name': 'GRCh38', 'start': 205727393}, {'start': 205727543, 'assembly_name': 'GRCh38', 'colocated_variants': [{'allele_string': 'G/A', 'seq_region_name': '1', 'id': 'rs1772147', 'frequencies': {'A': {'gnomadg_remaining': 0.8868, 'gnomadg_nfe': 0.9657, 'gnomadg_asj': 0.9119, 'afr': 0.6384, 'sas': 0.8783, 'eas': 0.8522, 'gnomadg_ami': 0.9803, 'eur': 0.9662, 'gnomadg': 0.8788, 'gnomadg_sas': 0.8917, 'af': 0.8227, 'gnomadg_eas': 0.8554, 'gnomadg_fin': 0.9686, 'gnomadg_amr': 0.8822, 'amr': 0.8444, 'gnomadg_afr': 0.7079, 'gnomadg_mid': 0.8912}}, 'end': 205727543, 'start': 205727543, 'strand': 1}], 'id': 'rs1772147', 'transcript_consequences': [{'gene_symbol': 'SNORA72', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'snoRNA', 'distance': 3678, 'strand': -1, 'transcript_id': 'ENST00000365074', 'gene_symbol_source': 'RFAM', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000201944'}, {'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NUCKS1'}], 'seq_region_name': '1', 'end': 205727543, 'most_severe_consequence': 'intron_variant', 'strand': 1, 'allele_string': 'G/A', 'input': 'rs1772147'}, {'colocated_variants': [{'frequencies': {'T': {'gnomadg_ami': 0.9803, 'eur': 0.9662, 'gnomadg': 0.8792, 'gnomadg_sas': 0.8921, 'af': 0.8233, 'gnomadg_eas': 0.8558, 'gnomadg_amr': 0.8823, 'gnomadg_afr': 0.7093, 'gnomadg_mid': 0.8912, 'amr': 0.8444, 'gnomadg_fin': 0.9687, 'afr': 0.6407, 'gnomadg_asj': 0.9119, 'gnomadg_remaining': 0.8864, 'gnomadg_nfe': 0.9657, 'sas': 0.8783, 'eas': 0.8522}, 'G': {'gnomadg_nfe': 0, 'gnomadg_remaining': 0, 'gnomadg': 0, 'gnomadg_sas': 0, 'gnomadg_asj': 0, 'gnomadg_ami': 0, 'gnomadg_fin': 0, 'gnomadg_amr': 0, 'gnomadg_afr': 0, 'gnomadg_mid': 0, 'gnomadg_eas': 0}}, 'id': 'rs1620334', 'seq_region_name': '1', 'allele_string': 'C/G/T', 'start': 205728717, 'pubmed': [29619377], 'end': 205728717, 'strand': 1}], 'id': 'rs1620334', 'transcript_consequences': [{'strand': -1, 'biotype': 'snoRNA', 'distance': 2504, 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'SNORA72', 'gene_id': 'ENSG00000201944', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000365074', 'gene_symbol_source': 'RFAM'}, {'gene_symbol': 'SNORA72', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 2504, 'biotype': 'snoRNA', 'strand': -1, 'transcript_id': 'ENST00000365074', 'gene_symbol_source': 'RFAM', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000201944'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'NUCKS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142'}, {'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'NUCKS1'}], 'seq_region_name': '1', 'end': 205728717, 'most_severe_consequence': 'intron_variant', 'allele_string': 'C/G/T', 'strand': 1, 'input': 'rs1620334', 'start': 205728717, 'assembly_name': 'GRCh38'}, {'most_severe_consequence': 'intron_variant', 'end': 205729947, 'input': 'rs71633561', 'strand': 1, 'allele_string': 'G/A/C/T', 'seq_region_name': '1', 'colocated_variants': [{'strand': 1, 'frequencies': {'T': {'gnomadg_ami': 0, 'gnomadg_sas': 0, 'gnomadg_remaining': 0, 'gnomadg': 6.596e-06, 'gnomadg_nfe': 1.472e-05, 'gnomadg_asj': 0, 'gnomadg_eas': 0, 'gnomadg_fin': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_amr': 0}, 'C': {'gnomadg_fin': 0.1529, 'gnomadg_mid': 0.2041, 'amr': 0.1268, 'gnomadg_afr': 0.02588, 'gnomadg_amr': 0.148, 'gnomadg_eas': 0.5333, 'af': 0.229, 'gnomadg_sas': 0.3689, 'gnomadg': 0.1284, 'eur': 0.1282, 'gnomadg_ami': 0.1993, 'eas': 0.497, 'sas': 0.4315, 'gnomadg_remaining': 0.1452, 'gnomadg_nfe': 0.1304, 'gnomadg_asj': 0.1856, 'afr': 0.0053}}, 'allele_string': 'G/A/C/T', 'id': 'rs71633561', 'seq_region_name': '1', 'start': 205729947, 'end': 205729947}], 'transcript_consequences': [{'gene_id': 'ENSG00000201944', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000365074', 'gene_symbol_source': 'RFAM', 'strand': -1, 'biotype': 'snoRNA', 'distance': 1274, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SNORA72'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000201944', 'transcript_id': 'ENST00000365074', 'gene_symbol_source': 'RFAM', 'distance': 1274, 'biotype': 'snoRNA', 'strand': -1, 'gene_symbol': 'SNORA72', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'strand': -1, 'biotype': 'snoRNA', 'distance': 1274, 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'SNORA72', 'gene_id': 'ENSG00000201944', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000365074', 'gene_symbol_source': 'RFAM'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'NUCKS1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'gene_symbol': 'NUCKS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1'}], 'id': 'rs71633561', 'start': 205729947, 'assembly_name': 'GRCh38'}, {'end': 205731634, 'most_severe_consequence': 'intron_variant', 'input': 'rs823109', 'allele_string': 'A/G', 'strand': 1, 'colocated_variants': [{'strand': 1, 'id': 'rs823109', 'seq_region_name': '1', 'allele_string': 'A/G', 'frequencies': {'G': {'afr': 0.6399, 'gnomadg_asj': 0.9119, 'gnomadg_nfe': 0.9657, 'gnomadg_remaining': 0.8866, 'sas': 0.8783, 'eas': 0.8522, 'eur': 0.9662, 'gnomadg_ami': 0.9803, 'gnomadg_sas': 0.8922, 'gnomadg': 0.8792, 'gnomadg_eas': 0.8559, 'af': 0.8231, 'gnomadg_mid': 0.8912, 'amr': 0.8444, 'gnomadg_afr': 0.7089, 'gnomadg_amr': 0.883, 'gnomadg_fin': 0.9687}}, 'end': 205731634, 'start': 205731634}], 'id': 'rs823109', 'transcript_consequences': [{'biotype': 'snoRNA', 'distance': 282, 'strand': -1, 'gene_symbol': 'SNORA72', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000201944', 'gene_symbol_source': 'RFAM', 'transcript_id': 'ENST00000365074'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'NUCKS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC'}], 'seq_region_name': '1', 'start': 205731634, 'assembly_name': 'GRCh38'}, {'start': 205739496, 'assembly_name': 'GRCh38', 'end': 205739496, 'most_severe_consequence': 'intron_variant', 'allele_string': 'G/A/C/T', 'strand': 1, 'input': 'rs823112', 'colocated_variants': [{'allele_string': 'G/A/C/T', 'seq_region_name': '1', 'id': 'rs823112', 'frequencies': {'A': {'gnomadg_ami': 0.9803, 'eur': 0.9662, 'gnomadg': 0.881, 'gnomadg_sas': 0.8919, 'af': 0.8227, 'gnomadg_eas': 0.8553, 'gnomadg_amr': 0.8825, 'gnomadg_afr': 0.7155, 'amr': 0.8429, 'gnomadg_mid': 0.8912, 'gnomadg_fin': 0.9687, 'gnomadg_asj': 0.9119, 'afr': 0.6399, 'gnomadg_remaining': 0.8879, 'gnomadg_nfe': 0.9658, 'sas': 0.8773, 'eas': 0.8522}}, 'end': 205739496, 'start': 205739496, 'strand': 1}], 'transcript_consequences': [{'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NUCKS1'}, {'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['intron_variant'], 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'strand': -1, 'biotype': 'protein_coding'}, {'gene_symbol': 'NUCKS1', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'protein_coding', 'strand': -1, 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923'}], 'id': 'rs823112', 'seq_region_name': '1'}, {'most_severe_consequence': 'intron_variant', 'end': 205742109, 'input': 'rs823113', 'strand': 1, 'allele_string': 'C/A/G/T', 'id': 'rs823113', 'colocated_variants': [{'id': 'rs823113', 'seq_region_name': '1', 'allele_string': 'C/A/G/T', 'frequencies': {'G': {'eur': 0.9662, 'gnomadg_ami': 0.9803, 'gnomadg_sas': 0.8918, 'gnomadg': 0.8791, 'gnomadg_eas': 0.8547, 'af': 0.8227, 'amr': 0.8429, 'gnomadg_afr': 0.7092, 'gnomadg_mid': 0.8912, 'gnomadg_amr': 0.8822, 'gnomadg_fin': 0.9688, 'gnomadg_asj': 0.9119, 'afr': 0.6399, 'gnomadg_remaining': 0.8858, 'gnomadg_nfe': 0.9657, 'sas': 0.8773, 'eas': 0.8522}}, 'end': 205742109, 'start': 205742109, 'strand': 1}], 'transcript_consequences': [{'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'NUCKS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'NUCKS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1, 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923'}, {'gene_symbol': 'NUCKS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923'}], 'seq_region_name': '1', 'start': 205742109, 'assembly_name': 'GRCh38'}, {'assembly_name': 'GRCh38', 'start': 205744250, 'colocated_variants': [{'allele_string': 'G/A/C/T', 'seq_region_name': '1', 'id': 'rs823128', 'frequencies': {'A': {'gnomadg_remaining': 0.8865, 'gnomadg_nfe': 0.9658, 'gnomadg_asj': 0.9119, 'afr': 0.6415, 'sas': 0.8793, 'eas': 0.8522, 'eur': 0.9662, 'gnomadg_ami': 0.9803, 'gnomadg_sas': 0.8913, 'gnomadg': 0.88, 'gnomadg_eas': 0.8549, 'af': 0.8235, 'gnomadg_fin': 0.9687, 'amr': 0.8429, 'gnomadg_mid': 0.8912, 'gnomadg_afr': 0.7124, 'gnomadg_amr': 0.8823}}, 'end': 205744250, 'pubmed': [21738487, 21121051, 29745862, 27299523, 22086882, 20683486, 21248740, 21425343, 21840748, 25758099, 28927418, 19915575, 20697102, 21044948, 21268244, 21858135, 23747047, 24373818, 26944116, 27174169, 28749816, 29192552, 35695987, 33574311, 24842889, 21172849, 32655481, 31366155, 28239348, 22878430, 37398170, 37396535], 'start': 205744250, 'strand': 1, 'phenotype_or_disease': 1}], 'id': 'rs823128', 'transcript_consequences': [{'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NUCKS1'}, {'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'gene_symbol': 'NUCKS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'strand': -1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['intron_variant']}], 'seq_region_name': '1', 'allele_string': 'G/A/C/T', 'strand': 1, 'input': 'rs823128', 'end': 205744250, 'most_severe_consequence': 'intron_variant'}, {'start': 205744671, 'assembly_name': 'GRCh38', 'colocated_variants': [{'strand': 1, 'frequencies': {'C': {'af': 0.8059, 'gnomadg_eas': 0.8613, 'gnomadg_fin': 0.9717, 'gnomadg_amr': 0.883, 'gnomadg_afr': 0.6762, 'amr': 0.8386, 'gnomadg_mid': 0.8957, 'gnomadg_ami': 0.9798, 'eur': 0.9662, 'gnomadg': 0.875, 'gnomadg_sas': 0.8982, 'sas': 0.8793, 'eas': 0.8522, 'gnomadg_remaining': 0.8798, 'gnomadg_nfe': 0.966, 'afr': 0.5772, 'gnomadg_asj': 0.9127}}, 'id': 'rs823129', 'seq_region_name': '1', 'allele_string': 'T/A/C', 'start': 205744671, 'end': 205744671}], 'id': 'rs823129', 'transcript_consequences': [{'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NUCKS1', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NUCKS1', 'strand': -1, 'biotype': 'protein_coding'}], 'seq_region_name': '1', 'end': 205744671, 'most_severe_consequence': 'intron_variant', 'input': 'rs823129', 'strand': 1, 'allele_string': 'T/A/C'}, {'assembly_name': 'GRCh38', 'start': 205745309, 'input': 'rs823131', 'strand': 1, 'allele_string': 'A/C/G/T', 'end': 205745309, 'most_severe_consequence': 'intron_variant', 'id': 'rs823131', 'colocated_variants': [{'strand': 1, 'start': 205745309, 'end': 205745309, 'frequencies': {'T': {'gnomadg_eas': 0, 'gnomadg_fin': 0, 'gnomadg_amr': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_ami': 0, 'gnomadg_remaining': 0, 'gnomadg_nfe': 0, 'gnomadg': 0, 'gnomadg_sas': 0, 'gnomadg_asj': 0}, 'G': {'gnomadg_remaining': 0.8754, 'gnomadg_nfe': 0.9656, 'gnomadg_asj': 0.9117, 'afr': 0.5772, 'eas': 0.8522, 'sas': 0.8793, 'gnomadg_sas': 0.8914, 'gnomadg': 0.8651, 'eur': 0.9662, 'gnomadg_ami': 0.9802, 'gnomadg_fin': 0.9687, 'amr': 0.8386, 'gnomadg_mid': 0.8912, 'gnomadg_afr': 0.6598, 'gnomadg_amr': 0.8769, 'gnomadg_eas': 0.8553, 'af': 0.8059}}, 'allele_string': 'A/C/G/T', 'seq_region_name': '1', 'id': 'rs823131'}], 'transcript_consequences': [{'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'gene_symbol': 'NUCKS1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding', 'strand': -1}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'NUCKS1', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC'}], 'seq_region_name': '1'}, {'id': 'rs823114', 'colocated_variants': [{'frequencies': {'A': {'af': 0.4073, 'gnomadg_eas': 0.5384, 'gnomadg_amr': 0.4347, 'amr': 0.3746, 'gnomadg_mid': 0.5646, 'gnomadg_afr': 0.2287, 'gnomadg_fin': 0.5913, 'gnomadg_ami': 0.5549, 'gnomade_fin': 0.5498, 'eur': 0.5358, 'gnomade': 0.494, 'gnomadg': 0.4542, 'gnomadg_sas': 0.5262, 'gnomade_asj': 0.5452, 'gnomade_nfe': 0.5129, 'sas': 0.5552, 'gnomade_eas': 0.4342, 'gnomade_sas': 0.5014, 'eas': 0.5, 'gnomade_remaining': 0.462, 'gnomade_afr': 0.1667, 'gnomadg_asj': 0.5844, 'afr': 0.1467, 'gnomade_amr': 0.3617, 'gnomadg_remaining': 0.4872, 'gnomadg_nfe': 0.5538, 'gnomade_mid': 0.6301}}, 'id': 'rs823114', 'seq_region_name': '1', 'allele_string': 'G/A', 'start': 205750404, 'pubmed': [29745862, 21812969, 28535700, 33736926, 34030174, 23395371, 36325427, 30643251, 28867149, 31384434, 24842889, 26207393, 28239348, 29607885, 33820434, 30044860, 29966015, 36224396, 28892062, 30275531, 38443977, 38254936, 37996473], 'end': 205750404, 'phenotype_or_disease': 1, 'strand': 1}], 'transcript_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'strand': -1, 'biotype': 'protein_coding', 'distance': 222, 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['upstream_gene_variant']}], 'seq_region_name': '1', 'most_severe_consequence': 'upstream_gene_variant', 'end': 205750404, 'regulatory_feature_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'regulatory_feature_id': 'ENSR1_952MZ9', 'biotype': 'promoter', 'consequence_terms': ['regulatory_region_variant']}], 'input': 'rs823114', 'strand': 1, 'allele_string': 'G/A', 'start': 205750404, 'assembly_name': 'GRCh38'}, {'most_severe_consequence': 'upstream_gene_variant', 'end': 205750729, 'input': 'rs865402', 'allele_string': 'A/C/G/T', 'strand': 1, 'transcript_consequences': [{'gene_symbol': 'NUCKS1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'distance': 547, 'biotype': 'protein_coding', 'strand': -1, 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275'}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 547, 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142'}, {'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'gene_symbol': 'NUCKS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 547, 'strand': -1}], 'colocated_variants': [{'strand': 1, 'allele_string': 'A/C/G/T', 'id': 'rs865402', 'seq_region_name': '1', 'frequencies': {'T': {'sas': 0.8783, 'eas': 0.8522, 'gnomadg_remaining': 0.8852, 'gnomadg_nfe': 0.9658, 'gnomadg_asj': 0.9119, 'afr': 0.6301, 'gnomadg_eas': 0.8546, 'af': 0.8201, 'gnomadg_fin': 0.9689, 'gnomadg_mid': 0.8946, 'amr': 0.8415, 'gnomadg_afr': 0.7027, 'gnomadg_amr': 0.8815, 'eur': 0.9662, 'gnomadg_ami': 0.9803, 'gnomadg_sas': 0.8922, 'gnomadg': 0.8774}}, 'end': 205750729, 'start': 205750729}], 'id': 'rs865402', 'seq_region_name': '1', 'start': 205750729, 'assembly_name': 'GRCh38'}, {'most_severe_consequence': 'upstream_gene_variant', 'end': 205750950, 'strand': 1, 'allele_string': 'G/A/C/T', 'input': 'rs823115', 'colocated_variants': [{'seq_region_name': '1', 'id': 'rs823115', 'allele_string': 'G/A/C/T', 'frequencies': {'A': {'eas': 0.8522, 'sas': 0.8783, 'gnomadg_remaining': 0.8867, 'gnomadg_nfe': 0.9657, 'gnomadg_asj': 0.9119, 'afr': 0.6301, 'gnomadg_fin': 0.9688, 'gnomadg_amr': 0.8815, 'gnomadg_afr': 0.7029, 'amr': 0.8415, 'gnomadg_mid': 0.8946, 'af': 0.8201, 'gnomadg_eas': 0.8551, 'gnomadg': 0.8774, 'gnomadg_sas': 0.8919, 'gnomadg_ami': 0.9802, 'eur': 0.9662}}, 'end': 205750950, 'start': 205750950, 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'strand': -1, 'biotype': 'protein_coding', 'distance': 768, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142'}, {'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'gene_symbol': 'NUCKS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 768, 'biotype': 'protein_coding', 'strand': -1}, {'strand': -1, 'distance': 768, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142'}], 'id': 'rs823115', 'start': 205750950, 'assembly_name': 'GRCh38'}, {'most_severe_consequence': 'upstream_gene_variant', 'end': 205751084, 'allele_string': 'C/A/G/T', 'strand': 1, 'input': 'rs12564209', 'colocated_variants': [{'end': 205751084, 'start': 205751084, 'allele_string': 'C/A/G/T', 'id': 'rs12564209', 'seq_region_name': '1', 'frequencies': {'G': {'gnomadg_amr': 0.1484, 'gnomadg_mid': 0.2041, 'gnomadg_afr': 0.03107, 'amr': 0.1239, 'gnomadg_fin': 0.1527, 'af': 0.2304, 'gnomadg_eas': 0.533, 'gnomadg': 0.1291, 'gnomadg_sas': 0.3684, 'gnomadg_ami': 0.2004, 'eur': 0.1272, 'eas': 0.498, 'sas': 0.4294, 'afr': 0.0136, 'gnomadg_asj': 0.1856, 'gnomadg_remaining': 0.1446, 'gnomadg_nfe': 0.1287}}, 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'strand': -1, 'biotype': 'protein_coding', 'distance': 902, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'NUCKS1'}, {'gene_symbol': 'NUCKS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 902, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275'}, {'gene_symbol': 'NUCKS1', 'impact': 'MODIFIER', 'variant_allele': 'T', 'distance': 902, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923'}], 'id': 'rs12564209', 'start': 205751084, 'assembly_name': 'GRCh38'}, {'start': 205751355, 'assembly_name': 'GRCh38', 'colocated_variants': [{'end': 205751355, 'pubmed': [36477530, 35661827], 'start': 205751355, 'allele_string': 'G/A/C', 'id': 'rs823116', 'seq_region_name': '1', 'frequencies': {'A': {'gnomadg': 0.4547, 'gnomadg_sas': 0.5263, 'gnomadg_ami': 0.5559, 'eur': 0.5348, 'gnomadg_amr': 0.4355, 'gnomadg_afr': 0.2289, 'amr': 0.3761, 'gnomadg_mid': 0.568, 'gnomadg_fin': 0.5944, 'af': 0.4071, 'gnomadg_eas': 0.5366, 'afr': 0.1467, 'gnomadg_asj': 0.5832, 'gnomadg_remaining': 0.4886, 'gnomadg_nfe': 0.5541, 'eas': 0.501, 'sas': 0.5532}}, 'strand': 1, 'phenotype_or_disease': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1173, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['upstream_gene_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NUCKS1', 'strand': -1, 'distance': 1173, 'biotype': 'protein_coding'}], 'id': 'rs823116', 'end': 205751355, 'most_severe_consequence': 'upstream_gene_variant', 'strand': 1, 'allele_string': 'G/A/C', 'input': 'rs823116'}, {'start': 205754385, 'assembly_name': 'GRCh38', 'colocated_variants': [{'start': 205754385, 'end': 205754385, 'frequencies': {'A': {'gnomadg_eas': 0.8544, 'af': 0.8191, 'gnomadg_fin': 0.9687, 'amr': 0.8429, 'gnomadg_afr': 0.7011, 'gnomadg_mid': 0.898, 'gnomadg_amr': 0.8806, 'eur': 0.9662, 'gnomadg_ami': 0.9803, 'gnomadg_sas': 0.8926, 'gnomadg': 0.8769, 'sas': 0.8783, 'eas': 0.8522, 'gnomadg_remaining': 0.886, 'gnomadg_nfe': 0.9658, 'afr': 0.6256, 'gnomadg_asj': 0.9119}}, 'allele_string': 'T/A/C', 'seq_region_name': '1', 'id': 'rs823117', 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'gene_symbol': 'NUCKS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 4203, 'strand': -1}, {'gene_symbol': 'NUCKS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 4203, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275'}], 'id': 'rs823117', 'end': 205754385, 'most_severe_consequence': 'upstream_gene_variant', 'input': 'rs823117', 'strand': 1, 'allele_string': 'T/A/C'}, {'assembly_name': 'GRCh38', 'start': 205754444, 'transcript_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'strand': -1, 'distance': 4262, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['upstream_gene_variant']}, {'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'gene_symbol': 'NUCKS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 4262, 'strand': -1}], 'colocated_variants': [{'seq_region_name': '1', 'id': 'rs823118', 'allele_string': 'C/A/T', 'frequencies': {'T': {'eas': 0.5099, 'sas': 0.5532, 'gnomadg_remaining': 0.4919, 'gnomadg_nfe': 0.5542, 'afr': 0.1551, 'gnomadg_asj': 0.584, 'gnomadg_fin': 0.5944, 'amr': 0.3761, 'gnomadg_afr': 0.2353, 'gnomadg_mid': 0.568, 'gnomadg_amr': 0.4386, 'gnomadg_eas': 0.5424, 'af': 0.4111, 'gnomadg_sas': 0.5257, 'gnomadg': 0.4569, 'eur': 0.5348, 'gnomadg_ami': 0.5549}}, 'pubmed': [31681140, 28927418, 26793951, 26914237, 27287230, 31818509, 30618224, 28078311, 30598082, 27182965, 35094046, 34712594, 34454617, 28887905, 28867149, 25064009, 29141588, 36477530, 26101198, 27090875, 29499164, 30071902, 33436766, 37292720, 26738859, 33935957, 35695515, 31366155, 32669683, 28239348, 33820434, 28892059, 31767922, 34322689, 34534445, 34633332, 28917824, 30957308, 27393345, 25737818, 31701892, 28949048, 31462381, 36759515, 34440451, 30595370, 34594039, 37693475, 37396535, 37671566, 38538606], 'end': 205754444, 'start': 205754444, 'strand': 1, 'phenotype_or_disease': 1}], 'id': 'rs823118', 'seq_region_name': '1', 'input': 'rs823118', 'strand': 1, 'allele_string': 'C/A/T', 'end': 205754444, 'most_severe_consequence': 'upstream_gene_variant'}, {'start': 205754491, 'assembly_name': 'GRCh38', 'end': 205754491, 'most_severe_consequence': 'upstream_gene_variant', 'input': 'rs860311', 'strand': 1, 'allele_string': 'A/C/T', 'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'NUCKS1', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4309}, {'biotype': 'protein_coding', 'distance': 4309, 'strand': -1, 'gene_symbol': 'NUCKS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'transcript_id': 'ENST00000367142', 'gene_symbol_source': 'HGNC'}], 'colocated_variants': [{'end': 205754491, 'start': 205754491, 'allele_string': 'A/C/T', 'id': 'rs860311', 'seq_region_name': '1', 'frequencies': {'C': {'gnomadg_nfe': 0.952, 'gnomadg_remaining': 0.8592, 'afr': 0.5477, 'gnomadg_asj': 0.9066, 'eas': 0.8522, 'sas': 0.8558, 'gnomadg_sas': 0.8682, 'gnomadg': 0.8482, 'eur': 0.9553, 'gnomadg_ami': 0.9682, 'gnomadg_fin': 0.9583, 'gnomadg_afr': 0.6309, 'gnomadg_mid': 0.881, 'amr': 0.8271, 'gnomadg_amr': 0.8681, 'gnomadg_eas': 0.8545, 'af': 0.7897}}, 'strand': 1}], 'id': 'rs860311', 'seq_region_name': '1'}, {'colocated_variants': [{'strand': 1, 'allele_string': 'A/G/T', 'seq_region_name': '1', 'id': 'rs823119', 'frequencies': {'T': {'gnomadg_asj': 0.9096, 'afr': 0.6362, 'gnomadg_remaining': 0.8872, 'gnomadg_nfe': 0.9654, 'eas': 0.8512, 'sas': 0.8783, 'gnomadg_sas': 0.8918, 'gnomadg': 0.8786, 'eur': 0.9662, 'gnomadg_ami': 0.9803, 'gnomadg_afr': 0.7078, 'amr': 0.8429, 'gnomadg_mid': 0.898, 'gnomadg_amr': 0.8817, 'gnomadg_fin': 0.9681, 'gnomadg_eas': 0.8544, 'af': 0.8217}}, 'end': 205754510, 'start': 205754510}], 'id': 'rs823119', 'transcript_consequences': [{'gene_symbol': 'NUCKS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 4328, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923'}, {'gene_symbol': 'NUCKS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 4328, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923'}], 'seq_region_name': '1', 'most_severe_consequence': 'upstream_gene_variant', 'end': 205754510, 'input': 'rs823119', 'strand': 1, 'allele_string': 'A/G/T', 'start': 205754510, 'assembly_name': 'GRCh38'}, {'assembly_name': 'GRCh38', 'start': 205754922, 'colocated_variants': [{'strand': 1, 'start': 205754922, 'end': 205754922, 'frequencies': {'G': {'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_amr': 0, 'gnomadg_fin': 0, 'gnomadg_eas': 0, 'gnomadg_asj': 0, 'gnomadg_sas': 0, 'gnomadg': 0, 'gnomadg_remaining': 0, 'gnomadg_nfe': 0, 'gnomadg_ami': 0}, 'T': {'af': 0.8043, 'gnomadg_eas': 0.8546, 'gnomadg_fin': 0.9686, 'gnomadg_amr': 0.8763, 'gnomadg_afr': 0.659, 'amr': 0.8386, 'gnomadg_mid': 0.898, 'gnomadg_ami': 0.9803, 'eur': 0.9662, 'gnomadg': 0.8647, 'gnomadg_sas': 0.8911, 'sas': 0.8783, 'eas': 0.8522, 'gnomadg_remaining': 0.8767, 'gnomadg_nfe': 0.9657, 'afr': 0.5719, 'gnomadg_asj': 0.9121}}, 'seq_region_name': '1', 'id': 'rs823120', 'allele_string': 'C/G/T'}], 'seq_region_name': '1', 'transcript_consequences': [{'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'NUCKS1', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4740, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'gene_id': 'ENSG00000069275', 'hgnc_id': 'HGNC:29923', 'consequence_terms': ['upstream_gene_variant']}, {'gene_symbol': 'NUCKS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 4740, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367142', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:29923', 'gene_id': 'ENSG00000069275'}], 'id': 'rs823120', 'input': 'rs823120', 'allele_string': 'C/G/T', 'strand': 1, 'most_severe_consequence': 'upstream_gene_variant', 'end': 205754922}, {'colocated_variants': [{'strand': 1, 'frequencies': {'T': {'af': 0.7706, 'gnomadg_eas': 0.8554, 'gnomadg_fin': 0.957, 'gnomadg_amr': 0.8588, 'gnomadg_afr': 0.5879, 'amr': 0.817, 'gnomadg_mid': 0.8776, 'gnomadg_ami': 0.955, 'eur': 0.9473, 'gnomadg': 0.8318, 'gnomadg_sas': 0.867, 'sas': 0.8558, 'eas': 0.8512, 'gnomadg_nfe': 0.9458, 'gnomadg_remaining': 0.8447, 'afr': 0.4871, 'gnomadg_asj': 0.8796}, 'G': {'gnomadg_asj': 0, 'gnomadg_sas': 0.0002074, 'gnomadg_nfe': 0, 'gnomadg_remaining': 0, 'gnomadg': 6.575e-06, 'gnomadg_ami': 0, 'gnomadg_afr': 0, 'gnomadg_mid': 0, 'gnomadg_amr': 0, 'gnomadg_fin': 0, 'gnomadg_eas': 0}}, 'allele_string': 'C/G/T', 'id': 'rs823122', 'seq_region_name': '1', 'start': 205755900, 'end': 205755900, 'pubmed': [22086882, 20683486, 21840748, 28927418, 35695515, 31366155]}], 'id': 'rs823122', 'seq_region_name': '1', 'most_severe_consequence': 'intergenic_variant', 'end': 205755900, 'strand': 1, 'allele_string': 'C/G/T', 'input': 'rs823122', 'intergenic_consequences': [{'impact': 'MODIFIER', 'variant_allele': 'G', 'consequence_terms': ['intergenic_variant']}, {'consequence_terms': ['intergenic_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER'}], 'start': 205755900, 'assembly_name': 'GRCh38'}, {'intergenic_consequences': [{'consequence_terms': ['intergenic_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'consequence_terms': ['intergenic_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intergenic_variant']}], 'assembly_name': 'GRCh38', 'start': 205756218, 'colocated_variants': [{'start': 205756218, 'pubmed': [28139693, 25434496, 29619377], 'end': 205756218, 'frequencies': {'T': {'af': 0.8047, 'gnomadg_eas': 0.8553, 'gnomadg_fin': 0.9686, 'gnomadg_amr': 0.8767, 'gnomadg_mid': 0.898, 'gnomadg_afr': 0.6606, 'amr': 0.8401, 'gnomadg_ami': 0.9803, 'eur': 0.9662, 'gnomadg': 0.8652, 'gnomadg_sas': 0.8916, 'sas': 0.8783, 'eas': 0.8522, 'gnomadg_nfe': 0.9657, 'gnomadg_remaining': 0.8767, 'afr': 0.5726, 'gnomadg_asj': 0.9118}}, 'id': 'rs823123', 'seq_region_name': '1', 'allele_string': 'C/A/G/T', 'phenotype_or_disease': 1, 'strand': 1}], 'id': 'rs823123', 'seq_region_name': '1', 'input': 'rs823123', 'allele_string': 'C/A/G/T', 'strand': 1, 'end': 205756218, 'most_severe_consequence': 'intergenic_variant'}, {'seq_region_name': '1', 'colocated_variants': [{'strand': 1, 'allele_string': 'C/A/G/T', 'seq_region_name': '1', 'id': 'rs823124', 'frequencies': {'T': {'eas': 0.8522, 'sas': 0.8783, 'afr': 0.5726, 'gnomadg_asj': 0.9121, 'gnomadg_remaining': 0.8775, 'gnomadg_nfe': 0.9657, 'gnomadg_mid': 0.898, 'amr': 0.8401, 'gnomadg_afr': 0.6614, 'gnomadg_amr': 0.8775, 'gnomadg_fin': 0.9692, 'gnomadg_eas': 0.8564, 'af': 0.8047, 'gnomadg_sas': 0.8925, 'gnomadg': 0.8662, 'eur': 0.9662, 'gnomadg_ami': 0.9792}}, 'end': 205757208, 'start': 205757208}], 'id': 'rs823124', 'most_severe_consequence': 'intergenic_variant', 'end': 205757208, 'input': 'rs823124', 'allele_string': 'C/A/G/T', 'strand': 1, 'intergenic_consequences': [{'consequence_terms': ['intergenic_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intergenic_variant']}, {'consequence_terms': ['intergenic_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T'}], 'start': 205757208, 'assembly_name': 'GRCh38'}, {'start': 205758338, 'assembly_name': 'GRCh38', 'intergenic_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intergenic_variant']}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intergenic_variant']}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intergenic_variant']}], 'end': 205758338, 'most_severe_consequence': 'intergenic_variant', 'allele_string': 'C/A/G/T', 'strand': 1, 'input': 'rs1626710', 'colocated_variants': [{'strand': 1, 'allele_string': 'C/A/G/T', 'id': 'rs1626710', 'seq_region_name': '1', 'frequencies': {'A': {'afr': 0.5484, 'gnomadg_asj': 0.9066, 'gnomadg_remaining': 0.8589, 'gnomadg_nfe': 0.9519, 'eas': 0.8522, 'sas': 0.8569, 'gnomadg_sas': 0.8669, 'gnomadg': 0.8484, 'eur': 0.9553, 'gnomadg_ami': 0.9682, 'gnomadg_mid': 0.881, 'gnomadg_afr': 0.6316, 'amr': 0.8285, 'gnomadg_amr': 0.8682, 'gnomadg_fin': 0.9581, 'gnomadg_eas': 0.8545, 'af': 0.7903}}, 'end': 205758338, 'start': 205758338}], 'seq_region_name': '1', 'id': 'rs1626710'}, {'intergenic_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intergenic_variant']}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['intergenic_variant']}], 'start': 205758340, 'assembly_name': 'GRCh38', 'colocated_variants': [{'strand': 1, 'allele_string': 'G/A/C', 'id': 'rs1772150', 'seq_region_name': '1', 'frequencies': {'A': {'gnomadg_ami': 0.9682, 'eur': 0.9563, 'gnomadg': 0.8485, 'gnomadg_sas': 0.8669, 'af': 0.7903, 'gnomadg_eas': 0.8541, 'gnomadg_amr': 0.8684, 'gnomadg_afr': 0.6319, 'gnomadg_mid': 0.881, 'amr': 0.8271, 'gnomadg_fin': 0.958, 'afr': 0.5484, 'gnomadg_asj': 0.9066, 'gnomadg_remaining': 0.8589, 'gnomadg_nfe': 0.9519, 'sas': 0.8569, 'eas': 0.8522}}, 'end': 205758340, 'start': 205758340}], 'seq_region_name': '1', 'id': 'rs1772150', 'most_severe_consequence': 'intergenic_variant', 'end': 205758340, 'input': 'rs1772150', 'strand': 1, 'allele_string': 'G/A/C'}, {'start': 205758693, 'assembly_name': 'GRCh38', 'intergenic_consequences': [{'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intergenic_variant']}, {'consequence_terms': ['intergenic_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER'}], 'most_severe_consequence': 'intergenic_variant', 'end': 205758693, 'strand': 1, 'allele_string': 'G/C/T', 'input': 'rs823125', 'seq_region_name': '1', 'colocated_variants': [{'frequencies': {'C': {'gnomadg_mid': 0.881, 'gnomadg_afr': 0.6332, 'amr': 0.8285, 'gnomadg_amr': 0.8685, 'gnomadg_fin': 0.9587, 'gnomadg_eas': 0.8549, 'af': 0.7903, 'gnomadg_sas': 0.8674, 'gnomadg': 0.8489, 'eur': 0.9553, 'gnomadg_ami': 0.9681, 'eas': 0.8522, 'sas': 0.8558, 'afr': 0.5492, 'gnomadg_asj': 0.9077, 'gnomadg_remaining': 0.8596, 'gnomadg_nfe': 0.952}}, 'allele_string': 'G/C/T', 'seq_region_name': '1', 'id': 'rs823125', 'start': 205758693, 'end': 205758693, 'strand': 1}], 'id': 'rs823125'}, {'input': 'rs823126', 'allele_string': 'C/A/G/T', 'strand': 1, 'most_severe_consequence': 'intergenic_variant', 'end': 205759241, 'colocated_variants': [{'strand': 1, 'seq_region_name': '1', 'id': 'rs823126', 'allele_string': 'C/A/G/T', 'frequencies': {'A': {'gnomadg_ami': 0, 'gnomadg_sas': 0.0002073, 'gnomadg_nfe': 0, 'gnomadg_remaining': 0, 'gnomadg': 6.574e-06, 'gnomadg_asj': 0, 'gnomadg_eas': 0, 'gnomadg_fin': 0, 'gnomadg_afr': 0, 'gnomadg_mid': 0, 'gnomadg_amr': 0}, 'G': {'gnomadg_sas': 0.8673, 'gnomadg': 0.8484, 'eur': 0.9553, 'gnomadg_ami': 0.9681, 'gnomadg_fin': 0.9581, 'gnomadg_afr': 0.6325, 'amr': 0.8285, 'gnomadg_mid': 0.881, 'gnomadg_amr': 0.8684, 'gnomadg_eas': 0.8544, 'af': 0.7903, 'gnomadg_remaining': 0.8589, 'gnomadg_nfe': 0.9519, 'afr': 0.5492, 'gnomadg_asj': 0.9066, 'eas': 0.8522, 'sas': 0.8558}}, 'end': 205759241, 'start': 205759241}], 'id': 'rs823126', 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 205759241, 'intergenic_consequences': [{'consequence_terms': ['intergenic_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'consequence_terms': ['intergenic_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intergenic_variant']}]}, {'colocated_variants': [{'strand': 1, 'start': 205760574, 'end': 205760574, 'frequencies': {'G': {'af': 0.7704, 'gnomadg_eas': 0.8558, 'gnomadg_amr': 0.8591, 'amr': 0.817, 'gnomadg_afr': 0.5875, 'gnomadg_mid': 0.8776, 'gnomadg_fin': 0.9568, 'gnomadg_ami': 0.955, 'eur': 0.9473, 'gnomadg': 0.8321, 'gnomadg_sas': 0.8666, 'sas': 0.8558, 'eas': 0.8512, 'afr': 0.4864, 'gnomadg_asj': 0.8795, 'gnomadg_nfe': 0.9459, 'gnomadg_remaining': 0.8438}}, 'allele_string': 'A/C/G/T', 'seq_region_name': '1', 'id': 'rs4951024'}], 'id': 'rs4951024', 'seq_region_name': '1', 'allele_string': 'A/C/G/T', 'strand': 1, 'input': 'rs4951024', 'most_severe_consequence': 'intergenic_variant', 'end': 205760574, 'intergenic_consequences': [{'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['intergenic_variant']}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intergenic_variant']}, {'consequence_terms': ['intergenic_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER'}], 'assembly_name': 'GRCh38', 'start': 205760574}, {'colocated_variants': [{'frequencies': {'T': {'gnomadg_fin': 0, 'gnomadg_amr': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_eas': 0, 'gnomadg': 0, 'gnomadg_remaining': 0, 'gnomadg_nfe': 0, 'gnomadg_sas': 0, 'gnomadg_asj': 0, 'gnomadg_ami': 0}, 'C': {'gnomadg_eas': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_amr': 0, 'gnomadg_fin': 9.808e-05, 'gnomadg_ami': 0, 'gnomadg_asj': 0, 'gnomadg_sas': 0, 'gnomadg': 6.633e-06, 'gnomadg_remaining': 0, 'gnomadg_nfe': 0}, 'A': {'sas': 0.8599, 'eas': 0.8542, 'gnomadg_nfe': 0.9465, 'gnomadg_remaining': 0.8311, 'afr': 0.4115, 'gnomadg_asj': 0.88, 'gnomadg_eas': 0.8527, 'af': 0.7518, 'gnomadg_fin': 0.9564, 'amr': 0.8156, 'gnomadg_afr': 0.5181, 'gnomadg_mid': 0.881, 'gnomadg_amr': 0.8521, 'eur': 0.9473, 'gnomadg_ami': 0.9548, 'gnomadg_sas': 0.8668, 'gnomadg': 0.8125}}, 'allele_string': 'G/A/C/T', 'id': 'rs4338418', 'seq_region_name': '1', 'start': 205763255, 'end': 205763255, 'strand': 1}], 'transcript_consequences': [{'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'distance': 4731, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4731, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'distance': 4731, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC'}], 'seq_region_name': '1', 'id': 'rs4338418', 'allele_string': 'G/A/C/T', 'strand': 1, 'input': 'rs4338418', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205763255, 'assembly_name': 'GRCh38', 'start': 205763255}, {'assembly_name': 'GRCh38', 'start': 205763384, 'transcript_consequences': [{'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4602}, {'distance': 4602, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4602}], 'colocated_variants': [{'start': 205763384, 'end': 205763384, 'frequencies': {'A': {'gnomadg_amr': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_fin': 0, 'gnomadg_eas': 0, 'gnomadg_asj': 0, 'gnomadg': 0, 'gnomadg_remaining': 0, 'gnomadg_nfe': 0, 'gnomadg_sas': 0, 'gnomadg_ami': 0}, 'C': {'af': 0.8057, 'gnomadg_eas': 0.8567, 'gnomadg_fin': 0.9687, 'gnomadg_amr': 0.8762, 'amr': 0.8401, 'gnomadg_mid': 0.9007, 'gnomadg_afr': 0.655, 'gnomadg_ami': 0.9803, 'eur': 0.9662, 'gnomadg': 0.8643, 'gnomadg_sas': 0.8916, 'sas': 0.8824, 'eas': 0.8552, 'gnomadg_nfe': 0.9666, 'gnomadg_remaining': 0.8774, 'afr': 0.5711, 'gnomadg_asj': 0.912}}, 'allele_string': 'T/A/C/G', 'seq_region_name': '1', 'id': 'rs823145', 'strand': 1}], 'seq_region_name': '1', 'id': 'rs823145', 'strand': 1, 'allele_string': 'T/A/C/G', 'input': 'rs823145', 'end': 205763384, 'most_severe_consequence': 'downstream_gene_variant'}, {'most_severe_consequence': 'downstream_gene_variant', 'end': 205763686, 'strand': 1, 'allele_string': 'C/A/G/T', 'input': 'rs823146', 'colocated_variants': [{'seq_region_name': '1', 'id': 'rs823146', 'allele_string': 'C/A/G/T', 'frequencies': {'G': {'gnomadg_amr': 0.88, 'amr': 0.8429, 'gnomadg_afr': 0.6903, 'gnomadg_mid': 0.9014, 'gnomadg_fin': 0.969, 'af': 0.8183, 'gnomadg_eas': 0.8576, 'gnomadg': 0.8744, 'gnomadg_sas': 0.8922, 'gnomadg_ami': 0.9803, 'eur': 0.9662, 'eas': 0.8552, 'sas': 0.8824, 'afr': 0.6172, 'gnomadg_asj': 0.9119, 'gnomadg_nfe': 0.9668, 'gnomadg_remaining': 0.8833}}, 'end': 205763686, 'start': 205763686, 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'strand': -1, 'biotype': 'protein_coding', 'distance': 4300, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 4300, 'biotype': 'protein_coding', 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 4300, 'biotype': 'protein_coding'}], 'id': 'rs823146', 'start': 205763686, 'assembly_name': 'GRCh38'}, {'start': 205764010, 'assembly_name': 'GRCh38', 'end': 205764010, 'most_severe_consequence': 'downstream_gene_variant', 'input': 'rs823147', 'allele_string': 'A/C/G/T', 'strand': 1, 'colocated_variants': [{'strand': 1, 'start': 205764010, 'end': 205764010, 'frequencies': {'G': {'gnomadg_ami': 0.9803, 'eur': 0.9662, 'gnomadg': 0.8742, 'gnomadg_sas': 0.8918, 'af': 0.8183, 'gnomadg_eas': 0.8567, 'gnomadg_amr': 0.8803, 'amr': 0.8429, 'gnomadg_afr': 0.6902, 'gnomadg_mid': 0.9014, 'gnomadg_fin': 0.9689, 'afr': 0.6172, 'gnomadg_asj': 0.9119, 'gnomadg_nfe': 0.9667, 'gnomadg_remaining': 0.8835, 'sas': 0.8824, 'eas': 0.8552}}, 'allele_string': 'A/C/G/T', 'id': 'rs823147', 'seq_region_name': '1'}], 'transcript_consequences': [{'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding', 'distance': 3976, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 3976, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'distance': 3976, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T'}], 'seq_region_name': '1', 'id': 'rs823147'}, {'start': 205765993, 'assembly_name': 'GRCh38', 'end': 205765993, 'most_severe_consequence': 'downstream_gene_variant', 'strand': 1, 'allele_string': 'A/C/G/T', 'input': 'rs823148', 'colocated_variants': [{'strand': 1, 'end': 205765993, 'start': 205765993, 'id': 'rs823148', 'seq_region_name': '1', 'allele_string': 'A/C/G/T', 'frequencies': {'T': {'gnomadg_eas': 0, 'gnomadg_amr': 0, 'gnomadg_afr': 0, 'gnomadg_mid': 0, 'gnomadg_fin': 0, 'gnomadg_ami': 0, 'gnomadg_asj': 0, 'gnomadg': 0, 'gnomadg_remaining': 0, 'gnomadg_nfe': 0, 'gnomadg_sas': 0}, 'G': {'gnomadg_amr': 0.8803, 'gnomadg_afr': 0.6903, 'amr': 0.8429, 'gnomadg_mid': 0.9014, 'gnomadg_fin': 0.969, 'af': 0.8183, 'gnomadg_eas': 0.8559, 'gnomadg': 0.8744, 'gnomadg_sas': 0.8927, 'gnomadg_ami': 0.9802, 'eur': 0.9662, 'eas': 0.8552, 'sas': 0.8824, 'afr': 0.6172, 'gnomadg_asj': 0.9119, 'gnomadg_remaining': 0.8832, 'gnomadg_nfe': 0.9668}}}], 'transcript_consequences': [{'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 3444, 'biotype': 'protein_coding', 'strand': -1, 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 3444, 'strand': -1}, {'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 3444}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'distance': 1993, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'distance': 1993, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 1993, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139'}, {'strand': -1, 'distance': 4307, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4307, 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4307, 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29'}, {'biotype': 'protein_coding', 'distance': 3710, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'protein_coding', 'distance': 3710, 'strand': -1, 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'distance': 3710, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'strand': -1, 'distance': 3725, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 3725, 'biotype': 'protein_coding'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 3725}, {'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 4168, 'biotype': 'protein_coding_CDS_not_defined'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4168, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 4168, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'nonsense_mediated_decay', 'distance': 3454, 'strand': -1, 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'strand': -1, 'distance': 3454, 'biotype': 'nonsense_mediated_decay', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'strand': -1, 'biotype': 'nonsense_mediated_decay', 'distance': 3454, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}], 'id': 'rs823148', 'seq_region_name': '1'}, {'strand': 1, 'allele_string': 'A/C/G', 'input': 'rs823149', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205766101, 'seq_region_name': '1', 'colocated_variants': [{'end': 205766101, 'start': 205766101, 'id': 'rs823149', 'seq_region_name': '1', 'allele_string': 'A/C/G', 'frequencies': {'C': {'gnomadg_sas': 0.842, 'gnomadg': 0.8377, 'eur': 0.8976, 'gnomadg_ami': 0.9276, 'amr': 0.8199, 'gnomadg_afr': 0.6822, 'gnomadg_mid': 0.8571, 'gnomadg_amr': 0.8523, 'gnomadg_fin': 0.8985, 'gnomadg_eas': 0.8524, 'af': 0.7893, 'afr': 0.6172, 'gnomadg_asj': 0.8874, 'gnomadg_remaining': 0.8511, 'gnomadg_nfe': 0.914, 'eas': 0.8462, 'sas': 0.8303}}, 'strand': 1}], 'transcript_consequences': [{'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'distance': 3336, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'distance': 3336, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'biotype': 'protein_coding', 'distance': 1885, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139'}, {'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 1885, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4199, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'strand': -1, 'distance': 4199, 'biotype': 'protein_coding', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 3602, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'biotype': 'protein_coding', 'distance': 3602, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 3617, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'distance': 3617, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4060, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G', 'distance': 4060, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'biotype': 'retained_intron', 'distance': 4961, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'biotype': 'retained_intron', 'distance': 4961, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'nonsense_mediated_decay', 'distance': 3346}, {'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G', 'distance': 3346, 'biotype': 'nonsense_mediated_decay', 'strand': -1}], 'id': 'rs823149', 'assembly_name': 'GRCh38', 'start': 205766101}, {'strand': 1, 'allele_string': 'C/A/G/T', 'input': 'rs823150', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205766358, 'id': 'rs823150', 'colocated_variants': [{'end': 205766358, 'start': 205766358, 'seq_region_name': '1', 'id': 'rs823150', 'allele_string': 'C/A/G/T', 'frequencies': {'T': {'sas': 0.8824, 'eas': 0.8562, 'afr': 0.6256, 'gnomadg_asj': 0.9118, 'gnomadg_remaining': 0.8851, 'gnomadg_nfe': 0.9667, 'af': 0.8209, 'gnomadg_eas': 0.8588, 'gnomadg_amr': 0.8805, 'amr': 0.8444, 'gnomadg_afr': 0.6969, 'gnomadg_mid': 0.9014, 'gnomadg_fin': 0.9689, 'gnomadg_ami': 0.9803, 'eur': 0.9662, 'gnomadg': 0.8763, 'gnomadg_sas': 0.8925}}, 'strand': 1}], 'transcript_consequences': [{'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 3079, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'biotype': 'protein_coding', 'distance': 3079, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 3079, 'biotype': 'protein_coding'}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 1628, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1628, 'biotype': 'protein_coding'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'strand': -1, 'distance': 1628, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'distance': 3942, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'strand': -1, 'biotype': 'protein_coding', 'distance': 3942, 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 3942, 'biotype': 'protein_coding', 'strand': -1}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 3345, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 3345, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 3345, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29'}, {'strand': -1, 'distance': 3360, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390'}, {'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 3360, 'biotype': 'protein_coding', 'strand': -1}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'protein_coding', 'distance': 3360, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 3803, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 3803, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 3803, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC'}, {'biotype': 'retained_intron', 'distance': 4704, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534'}, {'distance': 4704, 'biotype': 'retained_intron', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534'}, {'strand': -1, 'biotype': 'retained_intron', 'distance': 4704, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'biotype': 'nonsense_mediated_decay', 'distance': 3089, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'distance': 3089, 'biotype': 'nonsense_mediated_decay', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'distance': 3089, 'biotype': 'nonsense_mediated_decay', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T'}], 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 205766358}, {'assembly_name': 'GRCh38', 'start': 205766389, 'colocated_variants': [{'strand': 1, 'frequencies': {'C': {'gnomadg_remaining': 0.8713, 'gnomadg_nfe': 0.9666, 'afr': 0.5514, 'gnomadg_asj': 0.9118, 'sas': 0.8824, 'eas': 0.8552, 'gnomadg_ami': 0.9803, 'eur': 0.9662, 'gnomadg': 0.8581, 'gnomadg_sas': 0.8923, 'af': 0.8007, 'gnomadg_eas': 0.8573, 'gnomadg_fin': 0.9689, 'gnomadg_amr': 0.8745, 'gnomadg_mid': 0.9014, 'amr': 0.8415, 'gnomadg_afr': 0.6333}}, 'allele_string': 'T/C', 'id': 'rs823151', 'seq_region_name': '1', 'start': 205766389, 'end': 205766389}], 'transcript_consequences': [{'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'strand': -1, 'distance': 3048, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1597}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 3911, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 3314, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'strand': -1, 'distance': 3329, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC'}, {'distance': 3772, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'retained_intron', 'distance': 4673, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'nonsense_mediated_decay', 'distance': 3058, 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}], 'seq_region_name': '1', 'id': 'rs823151', 'strand': 1, 'allele_string': 'T/C', 'input': 'rs823151', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205766389}, {'id': 'rs71633563', 'colocated_variants': [{'strand': 1, 'start': 205766726, 'end': 205766726, 'frequencies': {'T': {'eur': 0.1272, 'gnomadg_ami': 0.2066, 'gnomadg_sas': 0.3686, 'gnomadg': 0.1296, 'gnomadg_eas': 0.5378, 'af': 0.233, 'gnomadg_fin': 0.1538, 'amr': 0.1225, 'gnomadg_afr': 0.03133, 'gnomadg_mid': 0.1986, 'gnomadg_amr': 0.1466, 'gnomadg_nfe': 0.1294, 'gnomadg_remaining': 0.1433, 'gnomadg_asj': 0.1858, 'afr': 0.0151, 'sas': 0.4335, 'eas': 0.506}}, 'allele_string': 'C/G/T', 'seq_region_name': '1', 'id': 'rs71633563'}], 'transcript_consequences': [{'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 2711, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'distance': 2711, 'biotype': 'protein_coding', 'strand': -1}, {'distance': 1260, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC'}, {'distance': 1260, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 3574}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 3574, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 2977, 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 2977, 'biotype': 'protein_coding'}, {'distance': 2992, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'strand': -1, 'distance': 2992, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 3435, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887'}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 3435, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'strand': -1, 'distance': 4336, 'biotype': 'retained_intron', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'strand': -1, 'biotype': 'retained_intron', 'distance': 4336, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'distance': 2721, 'biotype': 'nonsense_mediated_decay', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078'}, {'strand': -1, 'distance': 2721, 'biotype': 'nonsense_mediated_decay', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G', 'distance': 4872, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4872, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T'}], 'seq_region_name': '1', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205766726, 'strand': 1, 'allele_string': 'C/G/T', 'input': 'rs71633563', 'start': 205766726, 'assembly_name': 'GRCh38'}, {'assembly_name': 'GRCh38', 'start': 205767435, 'seq_region_name': '1', 'colocated_variants': [{'strand': 1, 'frequencies': {'G': {'eas': 0.8552, 'sas': 0.8824, 'gnomadg_asj': 0.9119, 'afr': 0.6263, 'gnomadg_remaining': 0.8855, 'gnomadg_nfe': 0.9667, 'gnomadg_amr': 0.8803, 'gnomadg_afr': 0.6979, 'gnomadg_mid': 0.9014, 'amr': 0.8444, 'gnomadg_fin': 0.9689, 'af': 0.8209, 'gnomadg_eas': 0.8571, 'gnomadg': 0.8765, 'gnomadg_sas': 0.8924, 'gnomadg_ami': 0.9803, 'eur': 0.9662}}, 'seq_region_name': '1', 'id': 'rs823153', 'allele_string': 'A/C/G/T', 'start': 205767435, 'end': 205767435}], 'transcript_consequences': [{'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 2002, 'strand': -1}, {'distance': 2002, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 2002, 'strand': -1}, {'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 551}, {'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 551, 'strand': -1}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'strand': -1, 'distance': 551, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 2865, 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 2865, 'biotype': 'protein_coding'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'distance': 2865, 'biotype': 'protein_coding', 'strand': -1}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 2268, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant']}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'protein_coding', 'distance': 2268, 'strand': -1, 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'distance': 2268, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding', 'distance': 2283, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'biotype': 'protein_coding', 'distance': 2283, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 2283, 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant']}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 2726, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887'}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 2726, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC'}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 2726, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 3627, 'biotype': 'retained_intron', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'retained_intron', 'distance': 3627}, {'distance': 3627, 'biotype': 'retained_intron', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 2012, 'biotype': 'nonsense_mediated_decay', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 2012, 'biotype': 'nonsense_mediated_decay', 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 2012, 'biotype': 'nonsense_mediated_decay', 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4163, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4163, 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111'}, {'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 4163, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}], 'id': 'rs823153', 'input': 'rs823153', 'strand': 1, 'allele_string': 'A/C/G/T', 'end': 205767435, 'most_severe_consequence': 'downstream_gene_variant'}, {'assembly_name': 'GRCh38', 'start': 205767534, 'strand': 1, 'allele_string': 'T/A/G', 'input': 'rs1772152', 'end': 205767534, 'most_severe_consequence': 'downstream_gene_variant', 'colocated_variants': [{'strand': 1, 'start': 205767534, 'end': 205767534, 'frequencies': {'A': {'afr': 0.6263, 'gnomadg_asj': 0.9118, 'gnomadg_remaining': 0.8847, 'gnomadg_nfe': 0.9667, 'eas': 0.8552, 'sas': 0.8824, 'gnomadg_sas': 0.8924, 'gnomadg': 0.876, 'eur': 0.9672, 'gnomadg_ami': 0.9803, 'amr': 0.8444, 'gnomadg_afr': 0.6962, 'gnomadg_mid': 0.9014, 'gnomadg_amr': 0.8802, 'gnomadg_fin': 0.9688, 'gnomadg_eas': 0.8565, 'af': 0.8211}}, 'allele_string': 'T/A/G', 'seq_region_name': '1', 'id': 'rs1772152'}], 'id': 'rs1772152', 'transcript_consequences': [{'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1903, 'biotype': 'protein_coding'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'strand': -1, 'distance': 1903, 'biotype': 'protein_coding', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'distance': 452, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29'}, {'distance': 452, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'distance': 2766, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 2766, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'strand': -1, 'biotype': 'protein_coding', 'distance': 2169, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 2169, 'strand': -1}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 2184, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'distance': 2184, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'distance': 2627, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G', 'distance': 2627, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'strand': -1, 'biotype': 'retained_intron', 'distance': 3528, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 3528, 'biotype': 'retained_intron', 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 1913, 'biotype': 'nonsense_mediated_decay', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29'}, {'biotype': 'nonsense_mediated_decay', 'distance': 1913, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078'}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'distance': 4064, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 4064, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}], 'seq_region_name': '1'}, {'start': 205767535, 'assembly_name': 'GRCh38', 'colocated_variants': [{'frequencies': {'A': {'gnomadg_sas': 0.8925, 'gnomadg': 0.8761, 'eur': 0.9672, 'gnomadg_ami': 0.9803, 'gnomadg_fin': 0.9688, 'gnomadg_afr': 0.6964, 'gnomadg_mid': 0.9014, 'amr': 0.8444, 'gnomadg_amr': 0.8802, 'gnomadg_eas': 0.8568, 'af': 0.8211, 'gnomadg_nfe': 0.9667, 'gnomadg_remaining': 0.8847, 'gnomadg_asj': 0.9118, 'afr': 0.6263, 'eas': 0.8552, 'sas': 0.8824}}, 'allele_string': 'C/A', 'seq_region_name': '1', 'id': 'rs1618936', 'start': 205767535, 'end': 205767535, 'strand': 1}], 'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 1902, 'biotype': 'protein_coding', 'strand': -1}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'distance': 451, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 2765, 'biotype': 'protein_coding', 'strand': -1}, {'biotype': 'protein_coding', 'distance': 2168, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324'}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 2183, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 2626, 'biotype': 'protein_coding_CDS_not_defined', 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'retained_intron', 'distance': 3527, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant']}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1912, 'biotype': 'nonsense_mediated_decay', 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4063}], 'seq_region_name': '1', 'id': 'rs1618936', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205767535, 'strand': 1, 'allele_string': 'C/A', 'input': 'rs1618936'}, {'start': 205768322, 'assembly_name': 'GRCh38', 'most_severe_consequence': '3_prime_UTR_variant', 'end': 205768322, 'strand': 1, 'allele_string': 'T/A/C', 'input': 'rs844005', 'seq_region_name': '1', 'colocated_variants': [{'strand': 1, 'start': 205768322, 'end': 205768322, 'frequencies': {'C': {'gnomadg_asj': 0.9119, 'afr': 0.6263, 'gnomadg_remaining': 0.8853, 'gnomadg_nfe': 0.9667, 'sas': 0.8824, 'eas': 0.8552, 'gnomadg_ami': 0.9803, 'eur': 0.9662, 'gnomadg': 0.8763, 'gnomadg_sas': 0.8925, 'af': 0.8209, 'gnomadg_eas': 0.8568, 'gnomadg_amr': 0.8806, 'gnomadg_afr': 0.6973, 'amr': 0.8444, 'gnomadg_mid': 0.9007, 'gnomadg_fin': 0.9688}}, 'allele_string': 'T/A/C', 'id': 'rs844005', 'seq_region_name': '1'}], 'transcript_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1115, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 1115, 'biotype': 'protein_coding', 'strand': -1}, {'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 2972, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_start': 2972, 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'cdna_start': 2972, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_end': 2972, 'consequence_terms': ['3_prime_UTR_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 1978, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding', 'distance': 1978, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'strand': -1, 'distance': 1381, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1381, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'biotype': 'protein_coding', 'distance': 1396, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 1396, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390'}, {'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 1839, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1839, 'biotype': 'protein_coding_CDS_not_defined'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'retained_intron', 'distance': 2740, 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 2740, 'biotype': 'retained_intron'}, {'strand': -1, 'distance': 1125, 'biotype': 'nonsense_mediated_decay', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'nonsense_mediated_decay', 'distance': 1125, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'distance': 3276, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 3276, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'distance': 4670, 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'distance': 4670, 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'C', 'impact': 'MODIFIER'}], 'id': 'rs844005'}, {'most_severe_consequence': '3_prime_UTR_variant', 'end': 205768611, 'input': 'rs11557080', 'allele_string': 'G/A', 'strand': 1, 'colocated_variants': [{'phenotype_or_disease': 1, 'strand': 1, 'frequencies': {'A': {'gnomade_remaining': 0, 'afr': 0.0136, 'gnomadg_asj': 0.1859, 'gnomadg_nfe': 0.1294, 'gnomadg_remaining': 0.1442, 'sas': 0.4335, 'gnomade_nfe': 0, 'eas': 0.506, 'eur': 0.1272, 'gnomadg_ami': 0.2083, 'gnomade': 0, 'gnomadg_sas': 0.3702, 'gnomadg': 0.1299, 'gnomadg_eas': 0.5377, 'af': 0.2328, 'gnomadg_mid': 0.2, 'gnomadg_afr': 0.03082, 'amr': 0.1239, 'gnomadg_amr': 0.1474, 'gnomadg_fin': 0.1557}}, 'allele_string': 'G/A', 'seq_region_name': '1', 'id': 'rs11557080', 'start': 205768611, 'end': 205768611, 'pubmed': [31701892, 36759515]}], 'id': 'rs11557080', 'transcript_consequences': [{'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'distance': 826, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 2683, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_start': 2683, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 1689, 'biotype': 'protein_coding', 'strand': -1}, {'biotype': 'protein_coding', 'distance': 1092, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'distance': 1107, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1550, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'retained_intron', 'distance': 2451, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'strand': -1, 'distance': 836, 'biotype': 'nonsense_mediated_decay', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2987, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 4381, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839048', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521'}], 'seq_region_name': '1', 'start': 205768611, 'assembly_name': 'GRCh38'}, {'colocated_variants': [{'end': 205768636, 'start': 205768636, 'allele_string': 'C/A/T', 'id': 'rs823132', 'seq_region_name': '1', 'frequencies': {'T': {'gnomade': 0.9286, 'gnomadg_sas': 0.8931, 'gnomadg': 0.8763, 'eur': 0.9662, 'gnomadg_ami': 0.9802, 'gnomadg_mid': 0.9007, 'amr': 0.8444, 'gnomadg_afr': 0.6969, 'gnomadg_amr': 0.8804, 'gnomadg_fin': 0.9687, 'gnomadg_eas': 0.856, 'af': 0.8209, 'afr': 0.6263, 'gnomadg_asj': 0.9118, 'gnomadg_remaining': 0.8848, 'gnomadg_nfe': 0.9667, 'gnomade_remaining': 1, 'gnomade_sas': 1, 'eas': 0.8552, 'sas': 0.8824, 'gnomade_nfe': 0.9545, 'gnomade_asj': 0.5}}, 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 801, 'strand': -1, 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'distance': 801, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'cdna_start': 2658, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_end': 2658, 'consequence_terms': ['3_prime_UTR_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding'}, {'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_end': 2658, 'consequence_terms': ['3_prime_UTR_variant'], 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'cdna_start': 2658}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 1664, 'biotype': 'protein_coding', 'strand': -1}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'distance': 1664, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'distance': 1067, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'strand': -1, 'distance': 1067, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1082, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'distance': 1082, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 1525, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'distance': 1525, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887'}, {'strand': -1, 'distance': 2426, 'biotype': 'retained_intron', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'biotype': 'retained_intron', 'distance': 2426, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 811, 'biotype': 'nonsense_mediated_decay'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'distance': 811, 'biotype': 'nonsense_mediated_decay', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2962, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2962, 'strand': -1}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 4356, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839048', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': 1, 'distance': 4356, 'biotype': 'lncRNA'}], 'id': 'rs823132', 'input': 'rs823132', 'strand': 1, 'allele_string': 'C/A/T', 'most_severe_consequence': '3_prime_UTR_variant', 'end': 205768636, 'assembly_name': 'GRCh38', 'start': 205768636}, {'strand': 1, 'allele_string': 'T/A/C', 'input': 'rs823133', 'most_severe_consequence': '3_prime_UTR_variant', 'end': 205768653, 'id': 'rs823133', 'colocated_variants': [{'strand': 1, 'start': 205768653, 'end': 205768653, 'frequencies': {'C': {'gnomade_remaining': 1, 'gnomade_afr': 0.5, 'gnomade_amr': 1, 'afr': 0.6263, 'gnomadg_asj': 0.9119, 'gnomadg_nfe': 0.9667, 'gnomadg_remaining': 0.8849, 'sas': 0.8824, 'gnomade_nfe': 0.9773, 'gnomade_asj': 0.5, 'gnomade_eas': 0.75, 'gnomade_sas': 1, 'eas': 0.8552, 'eur': 0.9662, 'gnomadg_ami': 0.9802, 'gnomade': 0.931, 'gnomadg_sas': 0.893, 'gnomadg': 0.8762, 'gnomadg_eas': 0.8562, 'af': 0.8209, 'amr': 0.8444, 'gnomadg_mid': 0.9007, 'gnomadg_afr': 0.6966, 'gnomadg_amr': 0.8807, 'gnomadg_fin': 0.9688}}, 'id': 'rs823133', 'seq_region_name': '1', 'allele_string': 'T/A/C'}], 'transcript_consequences': [{'biotype': 'protein_coding', 'distance': 784, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 784, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 2641, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'cdna_start': 2641, 'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'cdna_start': 2641, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 2641, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding'}, {'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1647}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1647, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'biotype': 'protein_coding', 'distance': 1050, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'strand': -1, 'distance': 1050, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'distance': 1065, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1065}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1508, 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'strand': -1, 'distance': 1508, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'strand': -1, 'biotype': 'retained_intron', 'distance': 2409, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'retained_intron', 'distance': 2409, 'strand': -1, 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 794, 'biotype': 'nonsense_mediated_decay', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'strand': -1, 'distance': 794, 'biotype': 'nonsense_mediated_decay', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 2945, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111'}, {'strand': -1, 'distance': 2945, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC'}, {'biotype': 'lncRNA', 'distance': 4339, 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'strand': 1, 'biotype': 'lncRNA', 'distance': 4339, 'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant']}], 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 205768653}, {'assembly_name': 'GRCh38', 'start': 205768828, 'input': 'rs823134', 'allele_string': 'G/A/C', 'strand': 1, 'end': 205768828, 'most_severe_consequence': '3_prime_UTR_variant', 'colocated_variants': [{'seq_region_name': '1', 'id': 'rs823134', 'allele_string': 'G/A/C', 'frequencies': {'A': {'gnomadg_ami': 0, 'gnomadg_asj': 0, 'gnomadg_sas': 0, 'gnomadg_nfe': 0, 'gnomadg_remaining': 0, 'gnomadg': 0, 'gnomadg_eas': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_amr': 0, 'gnomadg_fin': 0}, 'C': {'eas': 0.8552, 'gnomade_nfe': 1, 'sas': 0.8824, 'afr': 0.6263, 'gnomadg_asj': 0.9119, 'gnomadg_nfe': 0.9667, 'gnomadg_remaining': 0.8854, 'gnomade_remaining': 1, 'gnomade_afr': 1, 'gnomadg_mid': 0.9007, 'amr': 0.8444, 'gnomadg_afr': 0.6974, 'gnomadg_amr': 0.8806, 'gnomadg_fin': 0.9689, 'gnomadg_eas': 0.857, 'af': 0.8209, 'gnomade': 1, 'gnomadg_sas': 0.8931, 'gnomadg': 0.8764, 'eur': 0.9662, 'gnomadg_ami': 0.9803}}, 'end': 205768828, 'start': 205768828, 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'biotype': 'protein_coding', 'distance': 609, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'strand': -1, 'distance': 609, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'cdna_end': 2466, 'consequence_terms': ['3_prime_UTR_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_start': 2466, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 2466, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_start': 2466, 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1472}, {'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 1472, 'strand': -1}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'biotype': 'protein_coding', 'distance': 875, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'strand': -1, 'distance': 875, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324'}, {'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'distance': 890, 'strand': -1}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 890, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1333}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 1333, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 2234, 'biotype': 'retained_intron', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 2234, 'biotype': 'retained_intron', 'strand': -1, 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'strand': -1, 'distance': 619, 'biotype': 'nonsense_mediated_decay', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'strand': -1, 'biotype': 'nonsense_mediated_decay', 'distance': 619, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2770, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'strand': -1, 'distance': 2770, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'distance': 4164, 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'strand': 1, 'distance': 4164, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant']}], 'id': 'rs823134'}, {'most_severe_consequence': '3_prime_UTR_variant', 'end': 205769038, 'allele_string': 'T/A/C', 'strand': 1, 'input': 'rs823135', 'seq_region_name': '1', 'colocated_variants': [{'allele_string': 'T/A/C', 'id': 'rs823135', 'seq_region_name': '1', 'frequencies': {'C': {'gnomadg_eas': 0.8568, 'af': 0.8209, 'gnomadg_afr': 0.6973, 'amr': 0.8444, 'gnomadg_mid': 0.9014, 'gnomadg_amr': 0.8803, 'gnomadg_fin': 0.9689, 'eur': 0.9662, 'gnomadg_ami': 0.9803, 'gnomadg_sas': 0.8925, 'gnomadg': 0.8763, 'sas': 0.8824, 'eas': 0.8552, 'gnomadg_asj': 0.9119, 'afr': 0.6263, 'gnomadg_remaining': 0.8868, 'gnomadg_nfe': 0.9667}}, 'end': 205769038, 'pubmed': [29619377], 'start': 205769038, 'strand': 1}], 'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 399, 'biotype': 'protein_coding', 'strand': -1}, {'strand': -1, 'distance': 399, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC'}, {'cdna_start': 2256, 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 2256, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 2256, 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'cdna_start': 2256}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1262, 'biotype': 'protein_coding'}, {'biotype': 'protein_coding', 'distance': 1262, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 665}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 665, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 680, 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant']}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'biotype': 'protein_coding', 'distance': 680, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1123, 'strand': -1, 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1123, 'biotype': 'protein_coding_CDS_not_defined'}, {'distance': 2024, 'biotype': 'retained_intron', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'retained_intron', 'distance': 2024, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'biotype': 'nonsense_mediated_decay', 'distance': 409, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 409, 'biotype': 'nonsense_mediated_decay', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2560, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111'}, {'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2560}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3954, 'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant']}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3954, 'variant_allele': 'C', 'impact': 'MODIFIER'}], 'id': 'rs823135', 'start': 205769038, 'assembly_name': 'GRCh38'}, {'id': 'rs823136', 'colocated_variants': [{'strand': 1, 'start': 205769123, 'end': 205769123, 'pubmed': [22606281, 29619377], 'frequencies': {'A': {'gnomadg_fin': 0.9691, 'gnomadg_afr': 0.6974, 'gnomadg_mid': 0.9014, 'amr': 0.8444, 'gnomadg_amr': 0.8802, 'gnomadg_eas': 0.8569, 'af': 0.8209, 'gnomadg_sas': 0.8925, 'gnomadg': 0.8763, 'gnomade': 1, 'eur': 0.9662, 'gnomadg_ami': 0.9803, 'eas': 0.8552, 'sas': 0.8824, 'gnomade_nfe': 1, 'gnomadg_remaining': 0.8859, 'gnomadg_nfe': 0.9667, 'gnomadg_asj': 0.9119, 'afr': 0.6263, 'gnomade_afr': 1, 'gnomade_remaining': 1}, 'T': {'gnomadg_fin': 0, 'gnomadg_amr': 6.548e-05, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_eas': 0, 'gnomadg_nfe': 0, 'gnomadg_remaining': 0, 'gnomadg': 6.575e-06, 'gnomadg_sas': 0, 'gnomadg_asj': 0, 'gnomadg_ami': 0}}, 'allele_string': 'C/A/G/T', 'id': 'rs823136', 'seq_region_name': '1'}], 'transcript_consequences': [{'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'distance': 314, 'biotype': 'protein_coding', 'strand': -1}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'distance': 314, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'biotype': 'protein_coding', 'distance': 314, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'cdna_start': 2171, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_end': 2171, 'consequence_terms': ['3_prime_UTR_variant']}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'cdna_start': 2171, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 2171}, {'cdna_start': 2171, 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 2171, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1177, 'biotype': 'protein_coding'}, {'distance': 1177, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729'}, {'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1177}, {'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 580, 'strand': -1}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'strand': -1, 'biotype': 'protein_coding', 'distance': 580, 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 580, 'strand': -1}, {'strand': -1, 'distance': 595, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'distance': 595, 'biotype': 'protein_coding', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390'}, {'strand': -1, 'distance': 595, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390'}, {'distance': 1038, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC'}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 1038, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887'}, {'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1038, 'strand': -1}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'strand': -1, 'distance': 1939, 'biotype': 'retained_intron', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29'}, {'distance': 1939, 'biotype': 'retained_intron', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 1939, 'biotype': 'retained_intron', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'nonsense_mediated_decay', 'distance': 324, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'strand': -1, 'distance': 324, 'biotype': 'nonsense_mediated_decay', 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'biotype': 'nonsense_mediated_decay', 'distance': 324, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 2475, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'distance': 2475, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 2475, 'biotype': 'protein_coding_CDS_not_defined', 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant']}, {'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'strand': 1, 'distance': 3869, 'biotype': 'lncRNA'}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'strand': 1, 'distance': 3869, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant']}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'distance': 3869, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839048', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521'}], 'seq_region_name': '1', 'end': 205769123, 'most_severe_consequence': '3_prime_UTR_variant', 'strand': 1, 'allele_string': 'C/A/G/T', 'input': 'rs823136', 'start': 205769123, 'assembly_name': 'GRCh38'}, {'assembly_name': 'GRCh38', 'start': 205770064, 'input': 'rs823138', 'strand': 1, 'allele_string': 'T/A/C', 'end': 205770064, 'most_severe_consequence': '3_prime_UTR_variant', 'seq_region_name': '1', 'colocated_variants': [{'strand': 1, 'frequencies': {'C': {'gnomadg_ami': 0.9803, 'eur': 0.9662, 'gnomade_fin': 0.9617, 'gnomade': 0.9375, 'gnomadg': 0.8761, 'gnomadg_sas': 0.8924, 'af': 0.8211, 'gnomadg_eas': 0.8563, 'gnomadg_amr': 0.8804, 'gnomadg_afr': 0.6971, 'gnomadg_mid': 0.9014, 'amr': 0.8444, 'gnomadg_fin': 0.9688, 'gnomade_remaining': 0.9263, 'gnomade_afr': 0.6947, 'afr': 0.6263, 'gnomadg_asj': 0.9119, 'gnomade_amr': 0.8658, 'gnomadg_remaining': 0.8853, 'gnomadg_nfe': 0.9665, 'gnomade_mid': 0.9294, 'gnomade_asj': 0.9161, 'gnomade_nfe': 0.9672, 'sas': 0.8834, 'gnomade_eas': 0.8734, 'gnomade_sas': 0.9042, 'eas': 0.8552}, 'A': {'gnomade_fin': 0, 'gnomade_afr': 0, 'gnomade_remaining': 0, 'gnomade_mid': 0, 'gnomade': 0, 'gnomade_amr': 0, 'gnomade_eas': 0, 'gnomade_nfe': 0, 'gnomade_asj': 0, 'gnomade_sas': 0}}, 'id': 'rs823138', 'seq_region_name': '1', 'allele_string': 'T/A/C', 'start': 205770064, 'end': 205770064}], 'transcript_consequences': [{'cdna_start': 1096, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'cdna_end': 1096, 'consequence_terms': ['3_prime_UTR_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'strand': -1}, {'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'cdna_start': 1096, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_end': 1096, 'consequence_terms': ['3_prime_UTR_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'cdna_end': 1230, 'consequence_terms': ['3_prime_UTR_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_start': 1230, 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC'}, {'cdna_start': 1230, 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'cdna_end': 1230, 'consequence_terms': ['3_prime_UTR_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'distance': 236, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29'}, {'biotype': 'protein_coding', 'distance': 236, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_end': 945, 'consequence_terms': ['3_prime_UTR_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'cdna_start': 945, 'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding', 'strand': -1, 'cdna_start': 945, 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 945, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'cdna_start': 1095, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_end': 1095, 'consequence_terms': ['3_prime_UTR_variant']}, {'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_end': 1095, 'consequence_terms': ['3_prime_UTR_variant'], 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'cdna_start': 1095}, {'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 97, 'biotype': 'protein_coding_CDS_not_defined'}, {'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 97, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 998, 'biotype': 'retained_intron', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29'}, {'biotype': 'retained_intron', 'distance': 998, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534'}, {'consequence_terms': ['3_prime_UTR_variant', 'NMD_transcript_variant'], 'cdna_end': 954, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_start': 954, 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'biotype': 'nonsense_mediated_decay', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'strand': -1, 'biotype': 'nonsense_mediated_decay', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['3_prime_UTR_variant', 'NMD_transcript_variant'], 'cdna_end': 954, 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'cdna_start': 954}, {'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1534, 'strand': -1}, {'strand': -1, 'distance': 1534, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111'}, {'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': 1, 'distance': 2928, 'biotype': 'lncRNA'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'strand': 1, 'distance': 2928, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant']}], 'id': 'rs823138'}, {'start': 205770138, 'assembly_name': 'GRCh38', 'end': 205770138, 'most_severe_consequence': '3_prime_UTR_variant', 'input': 'rs708723', 'strand': 1, 'allele_string': 'C/T', 'seq_region_name': '1', 'colocated_variants': [{'allele_string': 'C/T', 'seq_region_name': '1', 'id': 'rs708723', 'frequencies': {'T': {'gnomadg_eas': 0.5418, 'af': 0.4147, 'gnomadg_fin': 0.5912, 'gnomadg_mid': 0.568, 'gnomadg_afr': 0.247, 'amr': 0.379, 'gnomadg_amr': 0.4412, 'eur': 0.5318, 'gnomade_fin': 0.5838, 'gnomadg_ami': 0.554, 'gnomadg_sas': 0.5272, 'gnomadg': 0.4606, 'gnomade': 0.5374, 'gnomade_eas': 0.5277, 'sas': 0.5583, 'gnomade_nfe': 0.5589, 'gnomade_asj': 0.5902, 'eas': 0.5079, 'gnomade_sas': 0.5188, 'gnomade_afr': 0.2448, 'gnomade_remaining': 0.5213, 'gnomade_mid': 0.6219, 'gnomadg_nfe': 0.5552, 'gnomadg_remaining': 0.4901, 'gnomade_amr': 0.3945, 'afr': 0.1672, 'gnomadg_asj': 0.5838}}, 'end': 205770138, 'pubmed': [21738488, 22086882, 22892372, 22911860, 25528405, 33494262, 28386217, 24303760, 33398198, 33436766, 24510904, 27814993, 36823318, 33820434, 34534445, 24319646, 36687396, 37805635, 37671566, 37945903], 'start': 205770138, 'strand': 1, 'phenotype_or_disease': 1}], 'transcript_consequences': [{'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_end': 1022, 'consequence_terms': ['3_prime_UTR_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'cdna_start': 1022}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'cdna_end': 1156, 'consequence_terms': ['3_prime_UTR_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_start': 1156, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139'}, {'distance': 162, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1, 'cdna_start': 871, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'cdna_end': 871, 'consequence_terms': ['3_prime_UTR_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 1021, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_start': 1021, 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 23, 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'retained_intron', 'distance': 924, 'strand': -1}, {'consequence_terms': ['3_prime_UTR_variant', 'NMD_transcript_variant'], 'cdna_end': 880, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_start': 880, 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'biotype': 'nonsense_mediated_decay', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1460, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant']}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'distance': 2854, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER'}], 'id': 'rs708723'}, {'colocated_variants': [{'somatic': 1, 'strand': 1, 'phenotype_or_disease': 1, 'id': 'COSV52521096', 'seq_region_name': '1', 'allele_string': 'COSMIC_MUTATION', 'end': 205770268, 'start': 205770268}, {'frequencies': {'G': {'gnomadg_eas': 0.8545, 'af': 0.721, 'gnomadg_fin': 0.8781, 'gnomadg_mid': 0.7789, 'amr': 0.7767, 'gnomadg_afr': 0.5263, 'gnomadg_amr': 0.8122, 'gnomade_fin': 0.8762, 'eur': 0.8579, 'gnomadg_ami': 0.9046, 'gnomadg_sas': 0.8103, 'gnomadg': 0.7723, 'gnomade': 0.8588, 'gnomade_eas': 0.8654, 'sas': 0.7904, 'gnomade_nfe': 0.881, 'gnomade_asj': 0.8221, 'eas': 0.8462, 'gnomade_sas': 0.8106, 'gnomade_afr': 0.5108, 'gnomade_remaining': 0.8346, 'gnomade_mid': 0.7957, 'gnomadg_remaining': 0.7758, 'gnomadg_nfe': 0.8833, 'gnomade_amr': 0.79, 'afr': 0.441, 'gnomadg_asj': 0.8183}}, 'allele_string': 'T/C/G', 'seq_region_name': '1', 'id': 'rs823139', 'start': 205770268, 'end': 205770268, 'pubmed': [33494262, 27814993], 'strand': 1}], 'transcript_consequences': [{'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'cdna_start': 892, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_end': 892, 'consequence_terms': ['3_prime_UTR_variant']}, {'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 892, 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'cdna_start': 892}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_end': 1026, 'consequence_terms': ['3_prime_UTR_variant'], 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'cdna_start': 1026, 'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_end': 1026, 'consequence_terms': ['3_prime_UTR_variant'], 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'cdna_start': 1026, 'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding', 'distance': 32, 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 32, 'biotype': 'protein_coding', 'strand': -1}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_end': 741, 'consequence_terms': ['3_prime_UTR_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'cdna_start': 741, 'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'cdna_end': 741, 'consequence_terms': ['3_prime_UTR_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_start': 741, 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 891, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'cdna_start': 891}, {'cdna_start': 891, 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 891, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'cdna_start': 476, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'consequence_terms': ['non_coding_transcript_exon_variant'], 'cdna_end': 476, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_end': 476, 'consequence_terms': ['non_coding_transcript_exon_variant'], 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'cdna_start': 476, 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 794, 'biotype': 'retained_intron', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'strand': -1, 'biotype': 'retained_intron', 'distance': 794, 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'strand': -1, 'biotype': 'nonsense_mediated_decay', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_end': 750, 'consequence_terms': ['3_prime_UTR_variant', 'NMD_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'cdna_start': 750}, {'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'cdna_start': 750, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['3_prime_UTR_variant', 'NMD_transcript_variant'], 'cdna_end': 750, 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'nonsense_mediated_decay'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'distance': 1330, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'distance': 1330, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'strand': 1, 'biotype': 'lncRNA', 'distance': 2724, 'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant']}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'biotype': 'lncRNA', 'distance': 2724, 'strand': 1, 'variant_allele': 'G', 'impact': 'MODIFIER'}], 'id': 'rs823139', 'seq_region_name': '1', 'input': 'rs823139', 'allele_string': 'T/C/G', 'strand': 1, 'most_severe_consequence': '3_prime_UTR_variant', 'end': 205770268, 'assembly_name': 'GRCh38', 'start': 205770268}, {'start': 205772688, 'assembly_name': 'GRCh38', 'end': 205772688, 'most_severe_consequence': 'intron_variant', 'regulatory_feature_consequences': [{'consequence_terms': ['regulatory_region_variant'], 'biotype': 'enhancer', 'regulatory_feature_id': 'ENSR1_952N7Q', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'regulatory_feature_id': 'ENSR1_952N7Q', 'biotype': 'enhancer', 'consequence_terms': ['regulatory_region_variant']}], 'input': 'rs823142', 'allele_string': 'A/C/T', 'strand': 1, 'id': 'rs823142', 'colocated_variants': [{'start': 205772688, 'end': 205772688, 'pubmed': [29619377], 'frequencies': {'T': {'gnomadg_amr': 0.8817, 'amr': 0.8458, 'gnomadg_mid': 0.9014, 'gnomadg_afr': 0.7057, 'gnomadg_fin': 0.9687, 'af': 0.8239, 'gnomadg_eas': 0.8561, 'gnomade': 0.9393, 'gnomadg': 0.8786, 'gnomadg_sas': 0.8926, 'gnomadg_ami': 0.9803, 'eur': 0.9672, 'gnomade_fin': 0.9619, 'gnomade_sas': 0.9031, 'eas': 0.8542, 'gnomade_asj': 0.9121, 'sas': 0.8824, 'gnomade_nfe': 0.9673, 'gnomade_eas': 0.8673, 'afr': 0.6369, 'gnomadg_asj': 0.909, 'gnomade_amr': 0.8399, 'gnomadg_remaining': 0.8903, 'gnomadg_nfe': 0.9665, 'gnomade_mid': 0.9225, 'gnomade_remaining': 0.924, 'gnomade_afr': 0.6966}}, 'allele_string': 'A/C/T', 'id': 'rs823142', 'seq_region_name': '1', 'strand': 1}], 'transcript_consequences': [{'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1, 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29'}, {'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['intron_variant']}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'strand': -1, 'biotype': 'retained_intron', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'biotype': 'retained_intron', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['intron_variant', 'NMD_transcript_variant'], 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'nonsense_mediated_decay', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'nonsense_mediated_decay', 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['intron_variant', 'NMD_transcript_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'biotype': 'lncRNA', 'distance': 2871, 'strand': 1, 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000648961', 'strand': 1, 'distance': 2871, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'strand': 1, 'distance': 304, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant']}, {'biotype': 'lncRNA', 'distance': 304, 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048'}, {'transcript_id': 'ENST00000839049', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 2585, 'biotype': 'lncRNA', 'strand': 1}, {'distance': 2585, 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'T', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049'}], 'seq_region_name': '1'}, {'most_severe_consequence': '5_prime_UTR_variant', 'regulatory_feature_consequences': [{'regulatory_feature_id': 'ENSR1_952N8H', 'biotype': 'promoter', 'consequence_terms': ['regulatory_region_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'regulatory_feature_id': 'ENSR1_952N8H', 'consequence_terms': ['regulatory_region_variant'], 'biotype': 'promoter', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'regulatory_feature_id': 'ENSR1_952N8H', 'biotype': 'promoter', 'consequence_terms': ['regulatory_region_variant']}], 'end': 205775418, 'strand': 1, 'allele_string': 'C/A/G/T', 'input': 'rs823144', 'colocated_variants': [{'somatic': 1, 'strand': 1, 'phenotype_or_disease': 1, 'id': 'CR120223', 'seq_region_name': '1', 'allele_string': 'HGMD_MUTATION', 'end': 205775418, 'start': 205775418}, {'allele_string': 'C/A/G/T', 'id': 'rs823144', 'seq_region_name': '1', 'frequencies': {'G': {'gnomade_mid': 0, 'gnomade_amr': 0, 'gnomade': 0, 'gnomade_fin': 0, 'gnomade_afr': 0, 'gnomade_remaining': 0, 'gnomade_sas': 0, 'gnomade_eas': 0, 'gnomade_asj': 0, 'gnomade_nfe': 0}, 'T': {'gnomade_sas': 0, 'gnomade_asj': 0, 'gnomade_nfe': 0, 'gnomade_eas': 0, 'gnomade_amr': 0, 'gnomade': 0, 'gnomade_mid': 0, 'gnomade_remaining': 0, 'gnomade_fin': 0, 'gnomade_afr': 0}, 'A': {'gnomadg_asj': 0.7376, 'afr': 0.3563, 'gnomade_amr': 0.6111, 'gnomadg_nfe': 0.7589, 'gnomadg_remaining': 0.6603, 'gnomade_mid': 1, 'gnomade_remaining': 0.8333, 'gnomade_afr': 0.4, 'gnomade_sas': 0.6515, 'eas': 0.5089, 'gnomade_asj': 0.8929, 'sas': 0.6288, 'gnomade_nfe': 0.7941, 'gnomade_eas': 0.5, 'gnomade': 0.749, 'gnomadg': 0.6368, 'gnomadg_sas': 0.6026, 'gnomadg_ami': 0.7434, 'gnomade_fin': 0.9, 'eur': 0.7515, 'gnomadg_amr': 0.5628, 'gnomadg_afr': 0.431, 'gnomadg_mid': 0.6599, 'amr': 0.4798, 'gnomadg_fin': 0.784, 'af': 0.5367, 'gnomadg_eas': 0.5334}}, 'end': 205775418, 'pubmed': [27299523, 28245721, 33494262, 33938021, 25966061, 32014019, 33820434, 36116551], 'start': 205775418, 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1, 'cdna_start': 32, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'consequence_terms': ['5_prime_UTR_variant'], 'cdna_end': 32, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['5_prime_UTR_variant'], 'cdna_end': 32, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_start': 32, 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'protein_coding', 'strand': -1, 'cdna_start': 32, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'consequence_terms': ['5_prime_UTR_variant'], 'cdna_end': 32, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'cdna_end': 65, 'consequence_terms': ['5_prime_UTR_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_start': 65, 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'cdna_start': 65, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_end': 65, 'consequence_terms': ['5_prime_UTR_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['5_prime_UTR_variant'], 'cdna_end': 65, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_start': 65, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139'}, {'cdna_end': 42, 'consequence_terms': ['5_prime_UTR_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_start': 42, 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'cdna_start': 42, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_end': 42, 'consequence_terms': ['5_prime_UTR_variant']}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['5_prime_UTR_variant'], 'cdna_end': 42, 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'cdna_start': 42, 'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29'}, {'consequence_terms': ['5_prime_UTR_variant'], 'cdna_end': 34, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_start': 34, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'cdna_end': 34, 'consequence_terms': ['5_prime_UTR_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_start': 34, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['5_prime_UTR_variant'], 'cdna_end': 34, 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'cdna_start': 34, 'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'distance': 185, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'distance': 185, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 185, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'cdna_end': 23, 'consequence_terms': ['non_coding_transcript_exon_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'cdna_start': 23, 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'cdna_start': 23, 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['non_coding_transcript_exon_variant'], 'cdna_end': 23, 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'cdna_end': 23, 'consequence_terms': ['non_coding_transcript_exon_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_start': 23, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887'}, {'distance': 267, 'biotype': 'retained_intron', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 267, 'biotype': 'retained_intron', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'retained_intron', 'distance': 267}, {'strand': -1, 'biotype': 'nonsense_mediated_decay', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'cdna_end': 43, 'consequence_terms': ['5_prime_UTR_variant', 'NMD_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'cdna_start': 43}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'nonsense_mediated_decay', 'strand': -1, 'cdna_start': 43, 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['5_prime_UTR_variant', 'NMD_transcript_variant'], 'cdna_end': 43, 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'cdna_start': 43, 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'cdna_end': 43, 'consequence_terms': ['5_prime_UTR_variant', 'NMD_transcript_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'nonsense_mediated_decay', 'strand': -1}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 209, 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 209, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 209, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant']}, {'transcript_id': 'ENST00000648961', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 141, 'strand': 1}, {'transcript_id': 'ENST00000648961', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 141, 'biotype': 'lncRNA', 'strand': 1}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000648961', 'strand': 1, 'distance': 141, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000839048'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'lncRNA', 'strand': 1, 'cdna_start': 146, 'transcript_id': 'ENST00000839049', 'consequence_terms': ['non_coding_transcript_exon_variant'], 'cdna_end': 146, 'gene_id': 'ENSG00000285521'}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839049', 'cdna_start': 146, 'gene_id': 'ENSG00000285521', 'cdna_end': 146, 'consequence_terms': ['non_coding_transcript_exon_variant']}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839049', 'cdna_start': 146, 'gene_id': 'ENSG00000285521', 'cdna_end': 146, 'consequence_terms': ['non_coding_transcript_exon_variant']}], 'id': 'rs823144', 'start': 205775418, 'assembly_name': 'GRCh38'}, {'assembly_name': 'GRCh38', 'start': 205776556, 'seq_region_name': '1', 'colocated_variants': [{'strand': 1, 'frequencies': {'T': {'eur': 0.7515, 'gnomadg_ami': 0.7434, 'gnomadg_sas': 0.6036, 'gnomadg': 0.6222, 'gnomadg_eas': 0.5374, 'af': 0.5194, 'gnomadg_mid': 0.6575, 'amr': 0.4784, 'gnomadg_afr': 0.3781, 'gnomadg_amr': 0.5586, 'gnomadg_fin': 0.7847, 'afr': 0.289, 'gnomadg_asj': 0.7376, 'gnomadg_remaining': 0.6523, 'gnomadg_nfe': 0.7592, 'sas': 0.6309, 'eas': 0.5099}}, 'allele_string': 'C/T', 'id': 'rs1775151', 'seq_region_name': '1', 'start': 205776556, 'end': 205776556}], 'transcript_consequences': [{'strand': -1, 'biotype': 'protein_coding', 'distance': 1107, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 1074, 'strand': -1}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 1097, 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729'}, {'strand': -1, 'distance': 1105, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324'}, {'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 1323, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1116, 'biotype': 'protein_coding_CDS_not_defined'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1405, 'biotype': 'retained_intron', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'biotype': 'nonsense_mediated_decay', 'distance': 1096, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1347, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961', 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000839048'}, {'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'T', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049'}], 'id': 'rs1775151', 'allele_string': 'C/T', 'strand': 1, 'input': 'rs1775151', 'most_severe_consequence': 'intron_variant', 'end': 205776556}, {'strand': 1, 'allele_string': 'T/A/C/G', 'input': 'rs1772153', 'most_severe_consequence': 'intron_variant', 'end': 205776654, 'transcript_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1205, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'strand': -1, 'distance': 1205, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC'}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1205, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'biotype': 'protein_coding', 'distance': 1172, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139'}, {'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 1172, 'strand': -1}, {'strand': -1, 'distance': 1172, 'biotype': 'protein_coding', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'distance': 1195, 'strand': -1}, {'distance': 1195, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1195, 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29'}, {'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 1203, 'strand': -1}, {'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C', 'distance': 1203, 'biotype': 'protein_coding', 'strand': -1}, {'distance': 1203, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 1421, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 1421, 'strand': -1, 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'distance': 1421, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC'}, {'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'distance': 1214, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1214, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1214, 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'distance': 1503, 'biotype': 'retained_intron', 'strand': -1, 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'biotype': 'retained_intron', 'distance': 1503, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'biotype': 'retained_intron', 'distance': 1503, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'distance': 1194, 'biotype': 'nonsense_mediated_decay', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'C', 'distance': 1194, 'biotype': 'nonsense_mediated_decay', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'nonsense_mediated_decay', 'distance': 1194, 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 1445, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'distance': 1445, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1445, 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111'}, {'transcript_id': 'ENST00000648961', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000648961', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961'}, {'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000839049', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049'}, {'transcript_id': 'ENST00000839049', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}], 'colocated_variants': [{'strand': 1, 'frequencies': {'C': {'sas': 0.6667, 'eas': 0.5099, 'gnomadg_remaining': 0.6901, 'gnomadg_nfe': 0.7736, 'afr': 0.4395, 'gnomadg_asj': 0.7496, 'af': 0.5715, 'gnomadg_eas': 0.5368, 'gnomadg_fin': 0.7977, 'gnomadg_amr': 0.5768, 'gnomadg_mid': 0.6769, 'gnomadg_afr': 0.5054, 'amr': 0.4986, 'gnomadg_ami': 0.7549, 'eur': 0.7644, 'gnomadg': 0.6677, 'gnomadg_sas': 0.6288}}, 'allele_string': 'T/A/C/G', 'seq_region_name': '1', 'id': 'rs1772153', 'start': 205776654, 'end': 205776654}], 'id': 'rs1772153', 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 205776654}, {'most_severe_consequence': 'intron_variant', 'end': 205776804, 'input': 'rs1772154', 'strand': 1, 'allele_string': 'C/T', 'id': 'rs1772154', 'colocated_variants': [{'strand': 1, 'allele_string': 'C/T', 'seq_region_name': '1', 'id': 'rs1772154', 'frequencies': {'T': {'gnomadg_remaining': 0.6907, 'gnomadg_nfe': 0.7739, 'gnomadg_asj': 0.7497, 'afr': 0.4395, 'eas': 0.5099, 'sas': 0.6524, 'gnomadg_sas': 0.6297, 'gnomadg': 0.6683, 'eur': 0.7644, 'gnomadg_ami': 0.7555, 'gnomadg_fin': 0.798, 'gnomadg_mid': 0.6769, 'amr': 0.4971, 'gnomadg_afr': 0.5062, 'gnomadg_amr': 0.5771, 'gnomadg_eas': 0.5388, 'af': 0.5685}}, 'end': 205776804, 'start': 205776804}], 'transcript_consequences': [{'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1355, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'protein_coding', 'distance': 1322, 'strand': -1}, {'distance': 1345, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1353, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1571}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1364, 'biotype': 'protein_coding_CDS_not_defined', 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'retained_intron', 'distance': 1653, 'strand': -1}, {'strand': -1, 'biotype': 'nonsense_mediated_decay', 'distance': 1344, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078'}, {'distance': 1595, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961', 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000839048', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049', 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'T'}], 'seq_region_name': '1', 'start': 205776804, 'assembly_name': 'GRCh38'}, {'most_severe_consequence': 'intron_variant', 'end': 205776832, 'input': 'rs6683517', 'allele_string': 'G/C/T', 'strand': 1, 'transcript_consequences': [{'strand': -1, 'biotype': 'protein_coding', 'distance': 1383, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1383, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'transcript_id': 'ENST00000367139', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 1350, 'strand': -1}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'biotype': 'protein_coding', 'distance': 1350, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'strand': -1, 'distance': 1373, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1373, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'strand': -1, 'distance': 1381, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1381, 'biotype': 'protein_coding'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1599, 'biotype': 'protein_coding'}, {'transcript_id': 'ENST00000446390', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 1599, 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1392, 'biotype': 'protein_coding_CDS_not_defined'}, {'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1392, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'strand': -1, 'distance': 1681, 'biotype': 'retained_intron', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'distance': 1681, 'biotype': 'retained_intron', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1372, 'biotype': 'nonsense_mediated_decay'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'nonsense_mediated_decay', 'distance': 1372, 'strand': -1, 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1623, 'biotype': 'protein_coding_CDS_not_defined'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 1623, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961', 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000648961', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839048', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000839049', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000839049', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}], 'colocated_variants': [{'strand': 1, 'start': 205776832, 'end': 205776832, 'frequencies': {'C': {'gnomadg_remaining': 0.2237, 'gnomadg_nfe': 0.266, 'afr': 0.0439, 'gnomadg_asj': 0.2526, 'sas': 0.4509, 'eas': 0.498, 'gnomadg_ami': 0.3322, 'eur': 0.2505, 'gnomadg': 0.2212, 'gnomadg_sas': 0.394, 'af': 0.2742, 'gnomadg_eas': 0.5304, 'gnomadg_fin': 0.2691, 'gnomadg_amr': 0.2091, 'gnomadg_afr': 0.07662, 'amr': 0.1729, 'gnomadg_mid': 0.2143}}, 'seq_region_name': '1', 'id': 'rs6683517', 'allele_string': 'G/C/T'}], 'seq_region_name': '1', 'id': 'rs6683517', 'start': 205776832, 'assembly_name': 'GRCh38'}, {'assembly_name': 'GRCh38', 'start': 205777861, 'colocated_variants': [{'seq_region_name': '1', 'id': 'rs12031215', 'allele_string': 'A/C/G/T', 'frequencies': {'G': {'gnomadg_remaining': 0.2166, 'gnomadg_nfe': 0.2658, 'gnomadg_asj': 0.2526, 'afr': 0.0113, 'eas': 0.498, 'sas': 0.4509, 'gnomadg_sas': 0.3927, 'gnomadg': 0.2129, 'eur': 0.2505, 'gnomadg_ami': 0.3322, 'gnomadg_fin': 0.2688, 'gnomadg_afr': 0.04815, 'amr': 0.1729, 'gnomadg_mid': 0.2143, 'gnomadg_amr': 0.2067, 'gnomadg_eas': 0.5288, 'af': 0.2656}}, 'end': 205777861, 'start': 205777861, 'strand': 1}], 'transcript_consequences': [{'strand': -1, 'distance': 2412, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'distance': 2412, 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 2412, 'biotype': 'protein_coding', 'strand': -1}, {'distance': 2379, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139'}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 2379, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 2379, 'strand': -1}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 2402, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant']}, {'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 2402}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000414729', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 2402}, {'strand': -1, 'distance': 2410, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324'}, {'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 2410, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 2410, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'strand': -1, 'distance': 2628, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390'}, {'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 2628, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280'}, {'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'biotype': 'protein_coding', 'distance': 2628, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2421, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 2421, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'RAB29'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'distance': 2421, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'distance': 2710, 'biotype': 'retained_intron', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'retained_intron', 'distance': 2710, 'strand': -1}, {'biotype': 'retained_intron', 'distance': 2710, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'biotype': 'nonsense_mediated_decay', 'distance': 2401, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'nonsense_mediated_decay', 'distance': 2401, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000528078', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'nonsense_mediated_decay', 'distance': 2401, 'strand': -1}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 2652, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 2652, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant']}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2652, 'transcript_id': 'ENST00000533111', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000648961', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521'}, {'transcript_id': 'ENST00000648961', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961', 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'strand': 1, 'biotype': 'lncRNA'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049', 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049', 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000839049', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}], 'id': 'rs12031215', 'seq_region_name': '1', 'strand': 1, 'allele_string': 'A/C/G/T', 'input': 'rs12031215', 'end': 205777861, 'most_severe_consequence': 'intron_variant'}, {'colocated_variants': [{'start': 205779181, 'end': 205779181, 'frequencies': {'A': {'afr': 0.4395, 'gnomadg_asj': 0.7497, 'gnomadg_remaining': 0.6906, 'gnomadg_nfe': 0.7734, 'eas': 0.5099, 'sas': 0.6524, 'gnomadg_sas': 0.6294, 'gnomadg': 0.6678, 'eur': 0.7644, 'gnomadg_ami': 0.7549, 'amr': 0.4957, 'gnomadg_afr': 0.5057, 'gnomadg_mid': 0.6769, 'gnomadg_amr': 0.5769, 'gnomadg_fin': 0.7974, 'gnomadg_eas': 0.5376, 'af': 0.5683}, 'T': {'gnomadg_fin': 0, 'gnomadg_amr': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_eas': 0, 'gnomadg_nfe': 0, 'gnomadg_remaining': 0, 'gnomadg': 0, 'gnomadg_sas': 0, 'gnomadg_asj': 0, 'gnomadg_ami': 0}}, 'seq_region_name': '1', 'id': 'rs10900524', 'allele_string': 'G/A/T', 'strand': 1}], 'transcript_consequences': [{'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'transcript_id': 'ENST00000235932', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'distance': 3732, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'protein_coding', 'distance': 3732, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000235932', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'strand': -1, 'biotype': 'protein_coding', 'distance': 3699, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367139', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 3699, 'biotype': 'protein_coding', 'strand': -1}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 3722, 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant']}, {'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000414729', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 3722, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29'}, {'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000437324', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'distance': 3730, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000437324', 'strand': -1, 'biotype': 'protein_coding', 'distance': 3730, 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 3948}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'protein_coding', 'distance': 3948, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000446390', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant']}, {'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000468887', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 3741, 'strand': -1, 'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'distance': 3741, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468887'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000492534', 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'consequence_terms': ['upstream_gene_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 4030, 'biotype': 'retained_intron'}, {'distance': 4030, 'biotype': 'retained_intron', 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'transcript_id': 'ENST00000492534', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'RAB29', 'impact': 'MODIFIER', 'variant_allele': 'A', 'distance': 3721, 'biotype': 'nonsense_mediated_decay', 'strand': -1, 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'biotype': 'nonsense_mediated_decay', 'distance': 3721, 'transcript_id': 'ENST00000528078', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant']}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 3972, 'strand': -1, 'gene_symbol': 'RAB29', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000117280', 'hgnc_id': 'HGNC:9789', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000533111', 'hgnc_id': 'HGNC:9789', 'gene_id': 'ENSG00000117280', 'consequence_terms': ['upstream_gene_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'RAB29', 'strand': -1, 'distance': 3972, 'biotype': 'protein_coding_CDS_not_defined'}, {'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961'}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000648961'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839048', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521'}, {'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000839049'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049', 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'T'}], 'seq_region_name': '1', 'id': 'rs10900524', 'most_severe_consequence': 'intron_variant', 'end': 205779181, 'strand': 1, 'allele_string': 'G/A/T', 'input': 'rs10900524', 'start': 205779181, 'assembly_name': 'GRCh38'}, {'start': 205780658, 'assembly_name': 'GRCh38', 'transcript_consequences': [{'transcript_id': 'ENST00000648961', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': 1, 'biotype': 'lncRNA'}, {'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961'}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000839048'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839048', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839049', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049', 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'T'}], 'colocated_variants': [{'start': 205780658, 'end': 205780658, 'frequencies': {'A': {'gnomadg_asj': 0.08701, 'afr': 0.0053, 'gnomadg_nfe': 0.03722, 'gnomadg_remaining': 0.05714, 'sas': 0.1166, 'eas': 0.1488, 'eur': 0.0298, 'gnomadg_ami': 0.02514, 'gnomadg_sas': 0.1351, 'gnomadg': 0.04903, 'gnomadg_eas': 0.1796, 'af': 0.0767, 'amr': 0.1196, 'gnomadg_afr': 0.01028, 'gnomadg_mid': 0.1071, 'gnomadg_amr': 0.1229, 'gnomadg_fin': 0.0625}, 'T': {'gnomadg_fin': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 3.495e-05, 'gnomadg_amr': 0, 'gnomadg_eas': 0, 'gnomadg_sas': 0, 'gnomadg_remaining': 0, 'gnomadg_nfe': 0, 'gnomadg': 8.842e-06, 'gnomadg_asj': 0, 'gnomadg_ami': 0}}, 'allele_string': 'G/A/T', 'seq_region_name': '1', 'id': 'rs11394622', 'strand': 1}], 'id': 'rs11394622', 'seq_region_name': '1', 'end': 205780658, 'most_severe_consequence': 'intron_variant', 'allele_string': 'G/A/T', 'strand': 1, 'input': 'rs11394622'}, {'start': 205782068, 'assembly_name': 'GRCh38', 'most_severe_consequence': 'intron_variant', 'end': 205782068, 'strand': 1, 'allele_string': 'C/T', 'input': 'rs863725', 'colocated_variants': [{'end': 205782068, 'start': 205782068, 'allele_string': 'C/T', 'id': 'rs863725', 'seq_region_name': '1', 'frequencies': {'T': {'gnomadg_sas': 0.6014, 'gnomadg': 0.6439, 'eur': 0.7525, 'gnomadg_ami': 0.7429, 'gnomadg_mid': 0.6803, 'gnomadg_afr': 0.4526, 'amr': 0.487, 'gnomadg_amr': 0.5674, 'gnomadg_fin': 0.7839, 'gnomadg_eas': 0.5365, 'af': 0.5423, 'gnomadg_asj': 0.7467, 'afr': 0.3737, 'gnomadg_nfe': 0.7594, 'gnomadg_remaining': 0.6742, 'eas': 0.5099, 'sas': 0.6268}}, 'strand': 1}], 'transcript_consequences': [{'transcript_id': 'ENST00000648961', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': 1, 'biotype': 'lncRNA'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049', 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER'}], 'seq_region_name': '1', 'id': 'rs863725'}, {'input': 'rs947211', 'allele_string': 'A/G', 'strand': 1, 'end': 205783537, 'most_severe_consequence': 'non_coding_transcript_exon_variant', 'seq_region_name': '1', 'colocated_variants': [{'frequencies': {'G': {'afr': 0.3623, 'gnomadg_asj': 0.7465, 'gnomadg_nfe': 0.7594, 'gnomadg_remaining': 0.6687, 'sas': 0.6258, 'eas': 0.5099, 'eur': 0.7525, 'gnomadg_ami': 0.7451, 'gnomadg_sas': 0.6005, 'gnomadg': 0.6396, 'gnomadg_eas': 0.5363, 'af': 0.5387, 'amr': 0.4841, 'gnomadg_afr': 0.4374, 'gnomadg_mid': 0.6769, 'gnomadg_amr': 0.5658, 'gnomadg_fin': 0.7854}}, 'allele_string': 'A/G', 'id': 'rs947211', 'seq_region_name': '1', 'start': 205783537, 'end': 205783537, 'pubmed': [27299523, 22086882, 22892372, 20683486, 21248740, 21425343, 21840748, 25758099, 28927418, 27287230, 20697102, 23747047, 24373818, 26944116, 27174169, 35695987, 22438815, 20413655, 21419001, 28807727, 33574311, 23395371, 31384434, 23792949, 24842889, 25966061, 23504498, 35695515, 31366155, 28239348, 36100329, 31187092, 25606459, 31217489, 19915576, 37805635], 'phenotype_or_disease': 1, 'strand': 1}], 'transcript_consequences': [{'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'lncRNA', 'strand': 1, 'cdna_start': 1069, 'transcript_id': 'ENST00000648961', 'consequence_terms': ['non_coding_transcript_exon_variant'], 'cdna_end': 1069, 'gene_id': 'ENSG00000285521'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['non_coding_transcript_exon_variant'], 'cdna_end': 1009, 'transcript_id': 'ENST00000839048', 'cdna_start': 1009, 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000839049', 'cdna_start': 1187, 'gene_id': 'ENSG00000285521', 'cdna_end': 1187, 'consequence_terms': ['non_coding_transcript_exon_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'strand': 1, 'biotype': 'lncRNA'}], 'id': 'rs947211', 'assembly_name': 'GRCh38', 'start': 205783537}, {'assembly_name': 'GRCh38', 'start': 205783766, 'input': 'rs1772156', 'allele_string': 'T/A/C', 'strand': 1, 'most_severe_consequence': 'downstream_gene_variant', 'end': 205783766, 'colocated_variants': [{'strand': 1, 'frequencies': {'A': {'gnomadg_ami': 0, 'gnomadg_sas': 0, 'gnomadg_remaining': 0, 'gnomadg': 0, 'gnomadg_nfe': 0, 'gnomadg_asj': 0, 'gnomadg_eas': 0, 'gnomadg_fin': 0, 'gnomadg_afr': 0, 'gnomadg_mid': 0, 'gnomadg_amr': 0}, 'C': {'gnomadg_eas': 0.536, 'af': 0.5699, 'amr': 0.5014, 'gnomadg_mid': 0.6939, 'gnomadg_afr': 0.5113, 'gnomadg_amr': 0.5792, 'gnomadg_fin': 0.7975, 'eur': 0.7654, 'gnomadg_ami': 0.7555, 'gnomadg_sas': 0.6266, 'gnomadg': 0.6699, 'sas': 0.6483, 'eas': 0.5099, 'afr': 0.4448, 'gnomadg_asj': 0.7581, 'gnomadg_nfe': 0.7739, 'gnomadg_remaining': 0.6968}}, 'allele_string': 'T/A/C', 'id': 'rs1772156', 'seq_region_name': '1', 'start': 205783766, 'end': 205783766}], 'seq_region_name': '1', 'transcript_consequences': [{'biotype': 'lncRNA', 'distance': 143, 'strand': 1, 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 143, 'strand': 1, 'transcript_id': 'ENST00000648961', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 145, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'biotype': 'lncRNA', 'distance': 145, 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 145, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839049', 'strand': 1, 'biotype': 'lncRNA', 'distance': 145, 'variant_allele': 'C', 'impact': 'MODIFIER'}], 'id': 'rs1772156'}, {'start': 205783969, 'assembly_name': 'GRCh38', 'colocated_variants': [{'strand': 1, 'start': 205783969, 'end': 205783969, 'pubmed': [28564610], 'frequencies': {'G': {'gnomadg_asj': 0.7465, 'afr': 0.3737, 'gnomadg_remaining': 0.673, 'gnomadg_nfe': 0.7594, 'eas': 0.5099, 'sas': 0.6258, 'gnomadg': 0.6436, 'gnomadg_sas': 0.6016, 'gnomadg_ami': 0.7434, 'eur': 0.7525, 'gnomadg_amr': 0.5673, 'gnomadg_afr': 0.4513, 'gnomadg_mid': 0.6803, 'amr': 0.4856, 'gnomadg_fin': 0.785, 'af': 0.5419, 'gnomadg_eas': 0.5365}}, 'allele_string': 'A/G', 'seq_region_name': '1', 'id': 'rs885224'}], 'id': 'rs885224', 'transcript_consequences': [{'strand': 1, 'biotype': 'lncRNA', 'distance': 346, 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000648961'}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 348, 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048'}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 348, 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839049'}], 'seq_region_name': '1', 'end': 205783969, 'most_severe_consequence': 'downstream_gene_variant', 'input': 'rs885224', 'strand': 1, 'allele_string': 'A/G'}, {'colocated_variants': [{'frequencies': {'C': {'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_amr': 0, 'gnomadg_fin': 0, 'gnomadg_eas': 0, 'gnomadg_asj': 0, 'gnomadg_sas': 0, 'gnomadg_remaining': 0, 'gnomadg': 0, 'gnomadg_nfe': 0, 'gnomadg_ami': 0}, 'A': {'gnomadg': 0.6556, 'gnomadg_sas': 0.5988, 'gnomadg_ami': 0.7434, 'eur': 0.7515, 'gnomadg_fin': 0.7834, 'gnomadg_amr': 0.5715, 'amr': 0.4914, 'gnomadg_mid': 0.6803, 'gnomadg_afr': 0.4938, 'af': 0.5565, 'gnomadg_eas': 0.5371, 'gnomadg_remaining': 0.6823, 'gnomadg_nfe': 0.7592, 'afr': 0.4281, 'gnomadg_asj': 0.7462, 'eas': 0.5089, 'sas': 0.6247}}, 'allele_string': 'G/A/C', 'id': 'rs1775158', 'seq_region_name': '1', 'start': 205784547, 'end': 205784547, 'strand': 1}], 'transcript_consequences': [{'strand': -1, 'distance': 4548, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137'}, {'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 4548, 'strand': -1}, {'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'strand': -1, 'distance': 4547, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1'}, {'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4547, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000648961', 'strand': 1, 'biotype': 'lncRNA', 'distance': 924, 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 924, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000648961', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 926, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'biotype': 'lncRNA', 'distance': 926, 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000839049', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'strand': 1, 'biotype': 'lncRNA', 'distance': 926}, {'transcript_id': 'ENST00000839049', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'strand': 1, 'biotype': 'lncRNA', 'distance': 926}], 'id': 'rs1775158', 'seq_region_name': '1', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205784547, 'allele_string': 'G/A/C', 'strand': 1, 'input': 'rs1775158', 'start': 205784547, 'assembly_name': 'GRCh38'}, {'assembly_name': 'GRCh38', 'start': 205784680, 'input': 'rs1772157', 'allele_string': 'T/A/C', 'strand': 1, 'most_severe_consequence': 'downstream_gene_variant', 'end': 205784680, 'transcript_consequences': [{'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'distance': 4415, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4415}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 4414, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant']}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 4414, 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': 1, 'biotype': 'lncRNA', 'distance': 1057, 'transcript_id': 'ENST00000648961', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant']}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 1057, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000648961', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'distance': 1059, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'distance': 1059, 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049', 'distance': 1059, 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'distance': 1059, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}], 'colocated_variants': [{'allele_string': 'T/A/C', 'id': 'rs1772157', 'seq_region_name': '1', 'frequencies': {'C': {'gnomadg_sas': 0.6212, 'gnomadg': 0.667, 'eur': 0.7644, 'gnomadg_ami': 0.7522, 'gnomadg_fin': 0.7925, 'gnomadg_afr': 0.5048, 'amr': 0.4986, 'gnomadg_mid': 0.6862, 'gnomadg_amr': 0.5755, 'gnomadg_eas': 0.5285, 'af': 0.5675, 'gnomadg_remaining': 0.6935, 'gnomadg_nfe': 0.7727, 'afr': 0.444, 'gnomadg_asj': 0.758, 'eas': 0.5069, 'sas': 0.6431}}, 'end': 205784680, 'pubmed': [33603002], 'start': 205784680, 'strand': 1, 'phenotype_or_disease': 1}], 'id': 'rs1772157', 'seq_region_name': '1'}, {'most_severe_consequence': 'downstream_gene_variant', 'end': 205784684, 'allele_string': 'A/G/T', 'strand': 1, 'input': 'rs1775159', 'colocated_variants': [{'allele_string': 'A/G/T', 'id': 'rs1775159', 'seq_region_name': '1', 'frequencies': {'T': {'gnomadg_amr': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 2.471e-05, 'gnomadg_fin': 0, 'gnomadg_eas': 0, 'gnomadg_asj': 0, 'gnomadg_remaining': 0, 'gnomadg': 6.706e-06, 'gnomadg_nfe': 0, 'gnomadg_sas': 0, 'gnomadg_ami': 0}, 'G': {'gnomadg': 0.6519, 'gnomadg_sas': 0.5935, 'gnomadg_ami': 0.7404, 'eur': 0.7515, 'gnomadg_fin': 0.7778, 'gnomadg_amr': 0.5669, 'amr': 0.4885, 'gnomadg_afr': 0.4862, 'gnomadg_mid': 0.6701, 'af': 0.5547, 'gnomadg_eas': 0.5274, 'gnomadg_remaining': 0.6784, 'gnomadg_nfe': 0.7581, 'afr': 0.4274, 'gnomadg_asj': 0.7468, 'eas': 0.5069, 'sas': 0.6207}}, 'end': 205784684, 'start': 205784684, 'strand': 1}], 'transcript_consequences': [{'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol': 'SLC41A1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 4411, 'biotype': 'protein_coding', 'strand': -1}, {'distance': 4411, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4410, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant']}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4410, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961', 'biotype': 'lncRNA', 'distance': 1061, 'strand': 1, 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000648961', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': 1, 'biotype': 'lncRNA', 'distance': 1061}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'distance': 1063, 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': 1, 'distance': 1063, 'biotype': 'lncRNA'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 1063, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'strand': 1, 'distance': 1063, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839049'}], 'seq_region_name': '1', 'id': 'rs1775159', 'start': 205784684, 'assembly_name': 'GRCh38'}, {'start': 205784692, 'assembly_name': 'GRCh38', 'colocated_variants': [{'frequencies': {'A': {'gnomadg_eas': 0, 'gnomadg_fin': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_amr': 0, 'gnomadg_ami': 0, 'gnomadg_sas': 0, 'gnomadg_nfe': 0, 'gnomadg_remaining': 0, 'gnomadg': 0, 'gnomadg_asj': 0}, 'C': {'gnomadg_asj': 0.7469, 'afr': 0.4274, 'gnomadg_remaining': 0.6784, 'gnomadg_nfe': 0.7581, 'sas': 0.6207, 'eas': 0.5069, 'gnomadg_ami': 0.7415, 'eur': 0.7515, 'gnomadg': 0.6519, 'gnomadg_sas': 0.591, 'af': 0.5547, 'gnomadg_eas': 0.5291, 'gnomadg_amr': 0.5667, 'amr': 0.4885, 'gnomadg_mid': 0.6748, 'gnomadg_afr': 0.4864, 'gnomadg_fin': 0.7759}}, 'allele_string': 'T/A/C', 'id': 'rs1772158', 'seq_region_name': '1', 'start': 205784692, 'end': 205784692, 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4403, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1'}, {'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'strand': -1, 'biotype': 'protein_coding', 'distance': 4403, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'SLC41A1'}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'distance': 4402, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4402, 'strand': -1}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 1069, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000648961'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961', 'distance': 1069, 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'distance': 1071, 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'biotype': 'lncRNA', 'distance': 1071, 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049', 'distance': 1071, 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 1071, 'strand': 1, 'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}], 'id': 'rs1772158', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205784692, 'allele_string': 'T/A/C', 'strand': 1, 'input': 'rs1772158'}, {'input': 'rs5014256', 'strand': 1, 'allele_string': 'C/A/G', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205784749, 'colocated_variants': [{'strand': 1, 'allele_string': 'C/A/G', 'id': 'rs5014256', 'seq_region_name': '1', 'frequencies': {'A': {'gnomadg_eas': 0.5247, 'af': 0.5393, 'gnomadg_fin': 0.7488, 'gnomadg_mid': 0.6808, 'gnomadg_afr': 0.4322, 'amr': 0.4885, 'gnomadg_amr': 0.5419, 'eur': 0.7515, 'gnomadg_ami': 0.7414, 'gnomadg_sas': 0.5983, 'gnomadg': 0.6342, 'sas': 0.6207, 'eas': 0.5079, 'gnomadg_nfe': 0.7599, 'gnomadg_remaining': 0.664, 'afr': 0.3684, 'gnomadg_asj': 0.7497}}, 'end': 205784749, 'start': 205784749}], 'transcript_consequences': [{'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 4346, 'biotype': 'protein_coding', 'strand': -1}, {'distance': 4346, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4345, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant']}, {'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 4345, 'biotype': 'protein_coding_CDS_not_defined'}, {'transcript_id': 'ENST00000648961', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 1126, 'biotype': 'lncRNA', 'strand': 1}, {'transcript_id': 'ENST00000648961', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'strand': 1, 'biotype': 'lncRNA', 'distance': 1126}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 1128, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839048', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'distance': 1128, 'biotype': 'lncRNA', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 1128, 'strand': 1}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839049', 'strand': 1, 'distance': 1128, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'G'}], 'seq_region_name': '1', 'id': 'rs5014256', 'assembly_name': 'GRCh38', 'start': 205784749}, {'start': 205785560, 'assembly_name': 'GRCh38', 'transcript_consequences': [{'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 3535, 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'distance': 3535}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'distance': 3535}, {'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 3534, 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 3534, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC'}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 3534, 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961', 'distance': 1937, 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961', 'biotype': 'lncRNA', 'distance': 1937, 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'distance': 1937, 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961'}, {'transcript_id': 'ENST00000839048', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 1939, 'biotype': 'lncRNA', 'strand': 1}, {'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'strand': 1, 'distance': 1939, 'biotype': 'lncRNA'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'lncRNA', 'distance': 1939, 'strand': 1, 'transcript_id': 'ENST00000839048', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'strand': 1, 'distance': 1939, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839049', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant']}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'strand': 1, 'distance': 1939, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839049', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant']}, {'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 1939, 'biotype': 'lncRNA', 'strand': 1}], 'colocated_variants': [{'strand': 1, 'start': 205785560, 'end': 205785560, 'frequencies': {'T': {'gnomadg_fin': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_amr': 0, 'gnomadg_eas': 0, 'gnomadg_sas': 0, 'gnomadg_nfe': 1.47e-05, 'gnomadg_remaining': 0, 'gnomadg': 6.579e-06, 'gnomadg_asj': 0, 'gnomadg_ami': 0}, 'C': {'sas': 0.4479, 'eas': 0.497, 'gnomadg_remaining': 0.2227, 'gnomadg_nfe': 0.2656, 'afr': 0.0446, 'gnomadg_asj': 0.2514, 'af': 0.2732, 'gnomadg_eas': 0.5298, 'gnomadg_fin': 0.2686, 'gnomadg_amr': 0.2074, 'gnomadg_afr': 0.07677, 'gnomadg_mid': 0.2109, 'amr': 0.1715, 'gnomadg_ami': 0.3322, 'eur': 0.2495, 'gnomadg': 0.2208, 'gnomadg_sas': 0.391}}, 'allele_string': 'G/A/C/T', 'id': 'rs10494874', 'seq_region_name': '1'}], 'id': 'rs10494874', 'seq_region_name': '1', 'end': 205785560, 'most_severe_consequence': 'downstream_gene_variant', 'input': 'rs10494874', 'strand': 1, 'allele_string': 'G/A/C/T'}, {'assembly_name': 'GRCh38', 'start': 205785918, 'input': 'rs12565968', 'allele_string': 'G/A/T', 'strand': 1, 'most_severe_consequence': 'downstream_gene_variant', 'end': 205785918, 'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 3177, 'biotype': 'protein_coding', 'strand': -1}, {'biotype': 'protein_coding', 'distance': 3177, 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137'}, {'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 3176, 'strand': -1}, {'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 3176, 'biotype': 'protein_coding_CDS_not_defined'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'strand': 1, 'distance': 2295, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000648961', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant']}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': 1, 'biotype': 'lncRNA', 'distance': 2295, 'transcript_id': 'ENST00000648961', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant']}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'distance': 2297, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 2297, 'strand': 1, 'transcript_id': 'ENST00000839048', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 2297, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049', 'biotype': 'lncRNA', 'distance': 2297, 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER'}], 'colocated_variants': [{'strand': 1, 'end': 205785918, 'start': 205785918, 'allele_string': 'G/A/T', 'seq_region_name': '1', 'id': 'rs12565968', 'frequencies': {'A': {'gnomadg_eas': 0.5363, 'af': 0.5126, 'gnomadg_fin': 0.7964, 'amr': 0.4784, 'gnomadg_mid': 0.6712, 'gnomadg_afr': 0.354, 'gnomadg_amr': 0.5573, 'eur': 0.7565, 'gnomadg_ami': 0.7429, 'gnomadg_sas': 0.6247, 'gnomadg': 0.6205, 'sas': 0.6493, 'eas': 0.5069, 'gnomadg_remaining': 0.6534, 'gnomadg_nfe': 0.7672, 'gnomadg_asj': 0.7226, 'afr': 0.2481}}}], 'seq_region_name': '1', 'id': 'rs12565968'}, {'start': 205786016, 'assembly_name': 'GRCh38', 'colocated_variants': [{'start': 205786016, 'end': 205786016, 'frequencies': {'C': {'gnomadg_fin': 0.7958, 'amr': 0.4784, 'gnomadg_mid': 0.6701, 'gnomadg_afr': 0.3515, 'gnomadg_amr': 0.5552, 'gnomadg_eas': 0.5352, 'af': 0.5128, 'gnomadg_sas': 0.623, 'gnomadg': 0.6188, 'eur': 0.7565, 'gnomadg_ami': 0.7411, 'eas': 0.5069, 'sas': 0.6493, 'gnomadg_remaining': 0.6532, 'gnomadg_nfe': 0.7662, 'gnomadg_asj': 0.7208, 'afr': 0.2489}}, 'seq_region_name': '1', 'id': 'rs12565321', 'allele_string': 'T/C', 'strand': 1}], 'id': 'rs12565321', 'transcript_consequences': [{'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding', 'distance': 3079, 'strand': -1, 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429'}, {'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'distance': 3078, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065'}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'transcript_id': 'ENST00000484228', 'gene_symbol_source': 'HGNC', 'distance': 4904, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000648961', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 2393, 'strand': 1}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 2395, 'strand': 1, 'transcript_id': 'ENST00000839048', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'distance': 2395, 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049'}], 'seq_region_name': '1', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205786016, 'input': 'rs12565321', 'allele_string': 'T/C', 'strand': 1}, {'transcript_consequences': [{'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'distance': 3072}, {'biotype': 'protein_coding', 'distance': 3072, 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'G', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 3072, 'biotype': 'protein_coding'}, {'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 3071, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 3071, 'biotype': 'protein_coding_CDS_not_defined'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 3071, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant']}, {'distance': 4897, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000484228', 'gene_symbol_source': 'HGNC'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 4897, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant']}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 4897, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant']}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961', 'distance': 2400, 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'transcript_id': 'ENST00000648961', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'strand': 1, 'distance': 2400, 'biotype': 'lncRNA'}, {'biotype': 'lncRNA', 'distance': 2400, 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961'}, {'biotype': 'lncRNA', 'distance': 2402, 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048'}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 2402, 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'biotype': 'lncRNA', 'distance': 2402, 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'distance': 2402, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049', 'distance': 2402, 'biotype': 'lncRNA', 'strand': 1, 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': 1, 'biotype': 'lncRNA', 'distance': 2402, 'transcript_id': 'ENST00000839049', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant']}], 'colocated_variants': [{'start': 205786023, 'end': 205786023, 'frequencies': {'T': {'gnomadg_sas': 0.6233, 'gnomadg': 0.6192, 'eur': 0.7565, 'gnomadg_ami': 0.7411, 'gnomadg_fin': 0.7959, 'gnomadg_mid': 0.6678, 'gnomadg_afr': 0.3524, 'amr': 0.4769, 'gnomadg_amr': 0.5552, 'gnomadg_eas': 0.5334, 'af': 0.5126, 'gnomadg_remaining': 0.6526, 'gnomadg_nfe': 0.7664, 'afr': 0.2489, 'gnomadg_asj': 0.7209, 'eas': 0.5069, 'sas': 0.6493}, 'A': {'gnomadg_ami': 0, 'gnomadg_sas': 0, 'gnomadg_remaining': 0, 'gnomadg': 6.63e-06, 'gnomadg_nfe': 0, 'gnomadg_asj': 0, 'gnomadg_eas': 0.0001955, 'gnomadg_fin': 0, 'gnomadg_mid': 0, 'gnomadg_afr': 0, 'gnomadg_amr': 0}}, 'allele_string': 'C/A/G/T', 'seq_region_name': '1', 'id': 'rs12567894', 'strand': 1}], 'id': 'rs12567894', 'seq_region_name': '1', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205786023, 'allele_string': 'C/A/G/T', 'strand': 1, 'input': 'rs12567894', 'start': 205786023, 'assembly_name': 'GRCh38'}, {'assembly_name': 'GRCh38', 'start': 205786137, 'allele_string': 'T/A/C', 'strand': 1, 'input': 'rs11240567', 'end': 205786137, 'most_severe_consequence': 'downstream_gene_variant', 'id': 'rs11240567', 'colocated_variants': [{'strand': 1, 'start': 205786137, 'end': 205786137, 'frequencies': {'C': {'sas': 0.6493, 'eas': 0.5069, 'gnomadg_asj': 0.7228, 'afr': 0.3169, 'gnomadg_remaining': 0.6654, 'gnomadg_nfe': 0.7669, 'af': 0.5312, 'gnomadg_eas': 0.5347, 'gnomadg_amr': 0.5617, 'gnomadg_mid': 0.6701, 'gnomadg_afr': 0.4067, 'amr': 0.4813, 'gnomadg_fin': 0.7961, 'gnomadg_ami': 0.7418, 'eur': 0.7565, 'gnomadg': 0.6351, 'gnomadg_sas': 0.6257}}, 'allele_string': 'T/A/C', 'id': 'rs11240567', 'seq_region_name': '1'}], 'transcript_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'distance': 2958, 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant']}, {'biotype': 'protein_coding', 'distance': 2958, 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC'}, {'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2957, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1'}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 2957, 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 4783, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4783, 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': 1, 'distance': 2514, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000648961', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant']}, {'strand': 1, 'distance': 2514, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000648961'}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': 1, 'biotype': 'lncRNA', 'distance': 2516, 'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant']}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 2516, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 2516, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 2516, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}], 'seq_region_name': '1'}, {'colocated_variants': [{'allele_string': 'C/T', 'seq_region_name': '1', 'id': 'rs1775143', 'frequencies': {'T': {'gnomadg_fin': 0.796, 'gnomadg_amr': 0.5596, 'amr': 0.4798, 'gnomadg_mid': 0.6667, 'gnomadg_afr': 0.3947, 'af': 0.5274, 'gnomadg_eas': 0.5369, 'gnomadg': 0.6314, 'gnomadg_sas': 0.624, 'gnomadg_ami': 0.7423, 'eur': 0.7565, 'eas': 0.5069, 'sas': 0.6493, 'gnomadg_remaining': 0.6632, 'gnomadg_nfe': 0.7667, 'afr': 0.3033, 'gnomadg_asj': 0.7225}}, 'end': 205786422, 'pubmed': [28535700, 25855069], 'start': 205786422, 'strand': 1}], 'transcript_consequences': [{'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'distance': 2673, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2672, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant']}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4498, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000484228', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000648961', 'strand': 1, 'biotype': 'lncRNA', 'distance': 2799, 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'biotype': 'lncRNA', 'distance': 2801, 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'lncRNA', 'distance': 2801, 'strand': 1, 'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}], 'seq_region_name': '1', 'id': 'rs1775143', 'end': 205786422, 'most_severe_consequence': 'downstream_gene_variant', 'allele_string': 'C/T', 'strand': 1, 'input': 'rs1775143', 'start': 205786422, 'assembly_name': 'GRCh38'}, {'colocated_variants': [{'strand': 1, 'end': 205786884, 'start': 205786884, 'allele_string': 'G/A/C', 'seq_region_name': '1', 'id': 'rs1775144', 'frequencies': {'A': {'gnomadg_remaining': 0.8935, 'gnomadg_nfe': 0.9671, 'afr': 0.6407, 'gnomadg_asj': 0.9093, 'sas': 0.8824, 'eas': 0.8492, 'eur': 0.9672, 'gnomadg_ami': 0.9803, 'gnomadg_sas': 0.8913, 'gnomadg': 0.8799, 'gnomadg_eas': 0.8516, 'af': 0.8241, 'gnomadg_fin': 0.9687, 'gnomadg_mid': 0.898, 'gnomadg_afr': 0.7098, 'amr': 0.8473, 'gnomadg_amr': 0.8836}}}], 'id': 'rs1775144', 'transcript_consequences': [{'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 2211, 'biotype': 'protein_coding'}, {'distance': 2211, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137'}, {'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2210, 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2210, 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 4036, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 4036, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000648961', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3261, 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 3261, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000648961'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'biotype': 'lncRNA', 'distance': 3263, 'strand': 1, 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3263, 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 3263, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839049'}, {'strand': 1, 'distance': 3263, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839049'}], 'seq_region_name': '1', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205786884, 'strand': 1, 'allele_string': 'G/A/C', 'input': 'rs1775144', 'start': 205786884, 'assembly_name': 'GRCh38'}, {'start': 205787015, 'assembly_name': 'GRCh38', 'colocated_variants': [{'id': 'rs1775145', 'seq_region_name': '1', 'allele_string': 'C/A', 'frequencies': {'A': {'gnomadg_remaining': 0.6406, 'gnomadg_nfe': 0.7527, 'gnomadg_asj': 0.7105, 'afr': 0.2322, 'eas': 0.5079, 'sas': 0.6268, 'gnomadg_sas': 0.5983, 'gnomadg': 0.6059, 'eur': 0.7435, 'gnomadg_ami': 0.7297, 'gnomadg_fin': 0.7841, 'gnomadg_mid': 0.6565, 'gnomadg_afr': 0.3366, 'amr': 0.4669, 'gnomadg_amr': 0.5486, 'gnomadg_eas': 0.5369, 'af': 0.5}}, 'pubmed': [37805635], 'end': 205787015, 'start': 205787015, 'strand': 1}], 'transcript_consequences': [{'biotype': 'protein_coding', 'distance': 2080, 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137'}, {'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2079, 'strand': -1}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 3905, 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228'}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 3392, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000648961'}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 3394, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839049', 'distance': 3394, 'biotype': 'lncRNA', 'strand': 1, 'impact': 'MODIFIER', 'variant_allele': 'A'}], 'seq_region_name': '1', 'id': 'rs1775145', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205787015, 'strand': 1, 'allele_string': 'C/A', 'input': 'rs1775145'}, {'id': 'rs1775146', 'colocated_variants': [{'frequencies': {'G': {'gnomadg_ami': 0.7418, 'eur': 0.7565, 'gnomadg': 0.62, 'gnomadg_sas': 0.6247, 'af': 0.513, 'gnomadg_eas': 0.5382, 'gnomadg_amr': 0.5563, 'gnomadg_afr': 0.3537, 'amr': 0.4784, 'gnomadg_mid': 0.6701, 'gnomadg_fin': 0.7963, 'afr': 0.2489, 'gnomadg_asj': 0.7219, 'gnomadg_nfe': 0.7671, 'gnomadg_remaining': 0.6541, 'sas': 0.6493, 'eas': 0.5079}}, 'allele_string': 'A/G/T', 'id': 'rs1775146', 'seq_region_name': '1', 'start': 205787040, 'end': 205787040, 'strand': 1}], 'transcript_consequences': [{'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'strand': -1, 'distance': 2055, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'SLC41A1'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 2055, 'strand': -1}, {'gene_symbol': 'SLC41A1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'distance': 2054, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429'}, {'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2054, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 3880, 'biotype': 'protein_coding_CDS_not_defined'}, {'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 3880, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'transcript_id': 'ENST00000484228', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 3417, 'strand': 1, 'transcript_id': 'ENST00000648961', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000648961', 'biotype': 'lncRNA', 'distance': 3417, 'strand': 1, 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'biotype': 'lncRNA', 'distance': 3419, 'strand': 1, 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'distance': 3419, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'impact': 'MODIFIER', 'variant_allele': 'G', 'strand': 1, 'distance': 3419, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000839049', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant']}, {'transcript_id': 'ENST00000839049', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'strand': 1, 'distance': 3419, 'biotype': 'lncRNA'}], 'seq_region_name': '1', 'strand': 1, 'allele_string': 'A/G/T', 'input': 'rs1775146', 'end': 205787040, 'most_severe_consequence': 'downstream_gene_variant', 'assembly_name': 'GRCh38', 'start': 205787040}, {'start': 205787356, 'assembly_name': 'GRCh38', 'colocated_variants': [{'phenotype_or_disease': 1, 'strand': 1, 'start': 205787356, 'end': 205787356, 'pubmed': [28139693, 25434496, 36038597, 32310270, 34454617, 31007957, 37227164, 38203667], 'frequencies': {'A': {'gnomadg_remaining': 0.2219, 'gnomadg_nfe': 0.266, 'gnomadg_asj': 0.2523, 'afr': 0.0348, 'sas': 0.4479, 'eas': 0.497, 'gnomadg_ami': 0.333, 'eur': 0.2495, 'gnomadg': 0.2195, 'gnomadg_sas': 0.3923, 'af': 0.2706, 'gnomadg_eas': 0.5323, 'gnomadg_fin': 0.27, 'gnomadg_amr': 0.2068, 'gnomadg_afr': 0.07088, 'amr': 0.1715, 'gnomadg_mid': 0.2109}}, 'allele_string': 'C/A', 'id': 'rs6679073', 'seq_region_name': '1'}], 'id': 'rs6679073', 'transcript_consequences': [{'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'strand': -1, 'distance': 1739, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1'}, {'strand': -1, 'distance': 1738, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 3564}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': 1, 'distance': 3733, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000648961', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant']}, {'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'strand': 1, 'distance': 3735, 'biotype': 'lncRNA'}, {'transcript_id': 'ENST00000839049', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3735}], 'seq_region_name': '1', 'end': 205787356, 'most_severe_consequence': 'downstream_gene_variant', 'strand': 1, 'allele_string': 'C/A', 'input': 'rs6679073'}, {'colocated_variants': [{'strand': 1, 'frequencies': {'A': {'af': 0.0773, 'gnomadg_eas': 0.1482, 'gnomadg_fin': 0.03127, 'gnomadg_amr': 0.07909, 'gnomadg_afr': 0.007271, 'gnomadg_mid': 0.06122, 'amr': 0.1196, 'gnomadg_ami': 0.01974, 'eur': 0.0298, 'gnomadg': 0.03693, 'gnomadg_sas': 0.1069, 'sas': 0.1176, 'eas': 0.1508, 'gnomadg_remaining': 0.04119, 'gnomadg_nfe': 0.03096, 'afr': 0.0053, 'gnomadg_asj': 0.07719}}, 'id': 'rs12740609', 'seq_region_name': '1', 'allele_string': 'G/A', 'start': 205787407, 'end': 205787407}], 'transcript_consequences': [{'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'distance': 1688, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1'}, {'strand': -1, 'distance': 1687, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 3513, 'biotype': 'protein_coding_CDS_not_defined'}, {'strand': 1, 'distance': 3784, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000648961'}, {'transcript_id': 'ENST00000839048', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3786}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 3786, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839049'}], 'seq_region_name': '1', 'id': 'rs12740609', 'input': 'rs12740609', 'strand': 1, 'allele_string': 'G/A', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205787407, 'assembly_name': 'GRCh38', 'start': 205787407}, {'start': 205788441, 'assembly_name': 'GRCh38', 'end': 205788441, 'most_severe_consequence': 'downstream_gene_variant', 'input': 'rs7536285', 'allele_string': 'C/T', 'strand': 1, 'transcript_consequences': [{'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'distance': 654, 'strand': -1}, {'distance': 653, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 2479, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'transcript_id': 'ENST00000648961', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 4818, 'strand': 1}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 4820, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839048', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839049', 'strand': 1, 'distance': 4820, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER'}], 'colocated_variants': [{'strand': 1, 'allele_string': 'C/T', 'id': 'rs7536285', 'seq_region_name': '1', 'frequencies': {'T': {'afr': 0.32, 'gnomadg_asj': 0.7226, 'gnomadg_remaining': 0.6689, 'gnomadg_nfe': 0.767, 'sas': 0.6493, 'eas': 0.5069, 'eur': 0.7565, 'gnomadg_ami': 0.7423, 'gnomadg_sas': 0.6252, 'gnomadg': 0.6384, 'gnomadg_eas': 0.5398, 'af': 0.5321, 'amr': 0.4827, 'gnomadg_afr': 0.4144, 'gnomadg_mid': 0.6735, 'gnomadg_amr': 0.5645, 'gnomadg_fin': 0.7986}}, 'end': 205788441, 'start': 205788441}], 'id': 'rs7536285', 'seq_region_name': '1'}, {'colocated_variants': [{'allele_string': 'G/A/C', 'seq_region_name': '1', 'id': 'rs7513115', 'frequencies': {'A': {'sas': 0.6421, 'eas': 0.5079, 'gnomadg_remaining': 0.6538, 'gnomadg_nfe': 0.7666, 'afr': 0.2474, 'gnomadg_asj': 0.7224, 'gnomadg_eas': 0.5351, 'af': 0.5112, 'gnomadg_fin': 0.7979, 'amr': 0.4784, 'gnomadg_afr': 0.3556, 'gnomadg_mid': 0.6678, 'gnomadg_amr': 0.5573, 'eur': 0.7565, 'gnomadg_ami': 0.7445, 'gnomadg_sas': 0.6192, 'gnomadg': 0.6205}}, 'end': 205788564, 'start': 205788564, 'strand': 1}], 'transcript_consequences': [{'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'distance': 531, 'strand': -1, 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'distance': 531}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 530, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 530}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 2356, 'biotype': 'protein_coding_CDS_not_defined', 'transcript_id': 'ENST00000484228', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant']}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'distance': 2356, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'transcript_id': 'ENST00000648961', 'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'strand': 1, 'distance': 4941, 'biotype': 'lncRNA'}, {'transcript_id': 'ENST00000648961', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'distance': 4941, 'strand': 1}, {'gene_id': 'ENSG00000285521', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000839048', 'strand': 1, 'biotype': 'lncRNA', 'distance': 4943, 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521', 'transcript_id': 'ENST00000839048', 'biotype': 'lncRNA', 'distance': 4943, 'strand': 1, 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 4943, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'distance': 4943, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000839049', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000285521'}], 'id': 'rs7513115', 'seq_region_name': '1', 'strand': 1, 'allele_string': 'G/A/C', 'input': 'rs7513115', 'most_severe_consequence': 'downstream_gene_variant', 'end': 205788564, 'assembly_name': 'GRCh38', 'start': 205788564}, {'transcript_consequences': [{'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'distance': 423, 'biotype': 'protein_coding', 'strand': -1, 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 422, 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057'}, {'transcript_id': 'ENST00000484228', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2248}], 'colocated_variants': [{'strand': 1, 'end': 205788672, 'start': 205788672, 'id': 'rs12747475', 'seq_region_name': '1', 'allele_string': 'C/T', 'frequencies': {'T': {'gnomadg_remaining': 0.5969, 'gnomadg_nfe': 0.6374, 'gnomadg_asj': 0.6447, 'afr': 0.2784, 'sas': 0.4939, 'eas': 0.4117, 'eur': 0.5467, 'gnomadg_ami': 0.6189, 'gnomadg_sas': 0.579, 'gnomadg': 0.5532, 'gnomadg_eas': 0.5369, 'af': 0.4165, 'gnomadg_fin': 0.6621, 'amr': 0.389, 'gnomadg_afr': 0.3918, 'gnomadg_mid': 0.6727, 'gnomadg_amr': 0.5031}}}], 'seq_region_name': '1', 'id': 'rs12747475', 'input': 'rs12747475', 'allele_string': 'C/T', 'strand': 1, 'end': 205788672, 'most_severe_consequence': 'downstream_gene_variant', 'assembly_name': 'GRCh38', 'start': 205788672}, {'start': 205790748, 'assembly_name': 'GRCh38', 'colocated_variants': [{'start': 205790748, 'end': 205790748, 'frequencies': {'C': {'gnomade_remaining': 0.75, 'gnomade_amr': 0, 'afr': 0.4198, 'gnomadg_asj': 0.7958, 'gnomadg_remaining': 0.7287, 'gnomadg_nfe': 0.8303, 'sas': 0.6513, 'gnomade_nfe': 0.9375, 'gnomade_asj': 0.5, 'eas': 0.5119, 'eur': 0.8191, 'gnomadg_ami': 0.7774, 'gnomade': 0.8387, 'gnomadg_sas': 0.6335, 'gnomadg': 0.7045, 'gnomadg_eas': 0.5414, 'af': 0.5765, 'gnomadg_afr': 0.5103, 'gnomadg_mid': 0.7483, 'amr': 0.5115, 'gnomadg_amr': 0.6088, 'gnomadg_fin': 0.8638}}, 'allele_string': 'T/A/C', 'id': 'rs1341249', 'seq_region_name': '1', 'strand': 1}], 'transcript_consequences': [{'cdna_start': 3364, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'cdna_end': 3364, 'consequence_terms': ['3_prime_UTR_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'consequence_terms': ['3_prime_UTR_variant'], 'cdna_end': 3364, 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'cdna_start': 3364, 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['non_coding_transcript_exon_variant'], 'cdna_end': 2123, 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'cdna_start': 2123, 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC'}, {'consequence_terms': ['non_coding_transcript_exon_variant'], 'cdna_end': 2123, 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'cdna_start': 2123, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 172}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 172, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}], 'seq_region_name': '1', 'id': 'rs1341249', 'most_severe_consequence': '3_prime_UTR_variant', 'end': 205790748, 'allele_string': 'T/A/C', 'strand': 1, 'input': 'rs1341249'}, {'assembly_name': 'GRCh38', 'start': 205793011, 'input': 'rs708726', 'strand': 1, 'allele_string': 'T/A/C/G', 'most_severe_consequence': 'intron_variant', 'end': 205793011, 'colocated_variants': [{'pubmed': [23889843, 37805635, 38116116], 'end': 205793011, 'start': 205793011, 'id': 'rs708726', 'seq_region_name': '1', 'allele_string': 'T/A/C/G', 'frequencies': {'G': {'afr': 0.2852, 'gnomadg_asj': 0.7471, 'gnomadg_remaining': 0.6572, 'gnomadg_nfe': 0.7579, 'sas': 0.6278, 'eas': 0.5198, 'gnomadg_ami': 0.7549, 'eur': 0.7545, 'gnomadg': 0.6219, 'gnomadg_sas': 0.6103, 'af': 0.5184, 'gnomadg_eas': 0.5427, 'gnomadg_amr': 0.5512, 'amr': 0.464, 'gnomadg_mid': 0.6939, 'gnomadg_afr': 0.3766, 'gnomadg_fin': 0.7952}}, 'strand': 1, 'phenotype_or_disease': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['intron_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['intron_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'transcript_id': 'ENST00000484228', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}], 'id': 'rs708726'}, {'start': 205793818, 'assembly_name': 'GRCh38', 'id': 'rs823155', 'colocated_variants': [{'frequencies': {'C': {'af': 0.5164, 'gnomadg_eas': 0.5427, 'gnomadg_fin': 0.7954, 'gnomadg_amr': 0.5515, 'amr': 0.4611, 'gnomadg_mid': 0.6939, 'gnomadg_afr': 0.3767, 'gnomadg_ami': 0.7533, 'eur': 0.7535, 'gnomadg': 0.6221, 'gnomadg_sas': 0.61, 'sas': 0.6278, 'eas': 0.5129, 'gnomadg_remaining': 0.6575, 'gnomadg_nfe': 0.7573, 'gnomadg_asj': 0.7471, 'afr': 0.2852}, 'A': {'gnomadg_ami': 0, 'gnomadg_sas': 0, 'gnomadg_remaining': 0, 'gnomadg': 0, 'gnomadg_nfe': 0, 'gnomadg_asj': 0, 'gnomadg_eas': 0, 'gnomadg_fin': 0, 'gnomadg_afr': 0, 'gnomadg_mid': 0, 'gnomadg_amr': 0}}, 'allele_string': 'T/A/C', 'id': 'rs823155', 'seq_region_name': '1', 'start': 205793818, 'end': 205793818, 'strand': 1}], 'transcript_consequences': [{'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['intron_variant']}, {'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000484228', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228'}], 'seq_region_name': '1', 'end': 205793818, 'most_severe_consequence': 'intron_variant', 'input': 'rs823155', 'allele_string': 'T/A/C', 'strand': 1}, {'start': 205795512, 'assembly_name': 'GRCh38', 'end': 205795512, 'most_severe_consequence': 'non_coding_transcript_exon_variant', 'allele_string': 'G/A/C/T', 'strand': 1, 'input': 'rs823156', 'colocated_variants': [{'end': 205795512, 'start': 205795512, 'id': 'COSV65650550', 'seq_region_name': '1', 'allele_string': 'COSMIC_MUTATION', 'strand': 1, 'phenotype_or_disease': 1, 'somatic': 1}, {'start': 205795512, 'pubmed': [21738487, 27299523, 22086882, 22892372, 20683486, 31546642, 35163527, 21248740, 21425343, 21840748, 25758099, 28927418, 19915575, 21268244, 23747047, 26944116, 27174169, 28749816, 35695987, 22438815, 33574311, 28604798, 33494262, 34712594, 31384434, 30013184, 34025389, 35695515, 31366155, 27814993], 'end': 205795512, 'frequencies': {'A': {'eas': 0.755, 'gnomade_sas': 0.7418, 'gnomade_eas': 0.8199, 'gnomade_asj': 0.7933, 'sas': 0.726, 'gnomade_nfe': 0.8248, 'gnomadg_nfe': 0.8273, 'gnomadg_remaining': 0.7275, 'gnomade_mid': 0.7537, 'afr': 0.3381, 'gnomadg_asj': 0.7762, 'gnomade_amr': 0.7452, 'gnomade_afr': 0.4002, 'gnomade_remaining': 0.7787, 'gnomadg_fin': 0.8331, 'gnomadg_amr': 0.7605, 'amr': 0.7464, 'gnomadg_mid': 0.7295, 'gnomadg_afr': 0.4207, 'af': 0.6492, 'gnomadg_eas': 0.7916, 'gnomadg': 0.7042, 'gnomadg_sas': 0.7451, 'gnomade': 0.8049, 'gnomadg_ami': 0.9145, 'eur': 0.8101, 'gnomade_fin': 0.8278}}, 'var_synonyms': {'ClinVar': ['RCV001540981', 'VCV001183197']}, 'id': 'rs823156', 'seq_region_name': '1', 'allele_string': 'G/A/C/T', 'phenotype_or_disease': 1, 'clin_sig': ['benign'], 'strand': 1, 'clin_sig_allele': 'A:benign'}], 'transcript_consequences': [{'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065'}, {'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'SLC41A1'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'cdna_end': 1105, 'consequence_terms': ['non_coding_transcript_exon_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'cdna_start': 1105}, {'transcript_id': 'ENST00000484228', 'gene_symbol_source': 'HGNC', 'cdna_start': 1105, 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['non_coding_transcript_exon_variant'], 'cdna_end': 1105, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined'}, {'cdna_end': 1105, 'consequence_terms': ['non_coding_transcript_exon_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'cdna_start': 1105, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'T'}], 'id': 'rs823156', 'seq_region_name': '1'}, {'colocated_variants': [{'phenotype_or_disease': 1, 'clin_sig': ['benign'], 'strand': 1, 'clin_sig_allele': 'A:benign', 'start': 205796667, 'end': 205796667, 'frequencies': {'A': {'eur': 0.8757, 'gnomade_fin': 0.8971, 'gnomadg_ami': 0.9386, 'gnomadg_sas': 0.7706, 'gnomadg': 0.7935, 'gnomade': 0.86, 'gnomadg_eas': 0.7928, 'af': 0.7175, 'gnomadg_fin': 0.9024, 'amr': 0.8012, 'gnomadg_mid': 0.7891, 'gnomadg_afr': 0.5756, 'gnomadg_amr': 0.8213, 'gnomade_afr': 0.5714, 'gnomade_remaining': 0.8444, 'gnomade_mid': 0.8191, 'gnomadg_remaining': 0.8046, 'gnomadg_nfe': 0.9001, 'gnomade_amr': 0.8036, 'gnomadg_asj': 0.8341, 'afr': 0.4992, 'gnomade_eas': 0.8247, 'sas': 0.7515, 'gnomade_nfe': 0.897, 'gnomade_asj': 0.8378, 'eas': 0.755, 'gnomade_sas': 0.7707}, 'G': {'gnomade_mid': 0, 'gnomade': 0, 'gnomade_amr': 0, 'gnomade_fin': 0, 'gnomade_afr': 0, 'gnomade_remaining': 0, 'gnomade_sas': 0, 'gnomade_eas': 0, 'gnomade_nfe': 0, 'gnomade_asj': 0}}, 'var_synonyms': {'ClinVar': ['RCV001656738', 'VCV001249217']}, 'id': 'rs823157', 'seq_region_name': '1', 'allele_string': 'T/A/C/G'}], 'seq_region_name': '1', 'transcript_consequences': [{'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['intron_variant']}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137'}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000484228', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 51}, {'distance': 51, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 51, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484228', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['upstream_gene_variant']}], 'id': 'rs823157', 'input': 'rs823157', 'allele_string': 'T/A/C/G', 'strand': 1, 'end': 205796667, 'most_severe_consequence': 'intron_variant', 'assembly_name': 'GRCh38', 'start': 205796667}, {'colocated_variants': [{'clin_sig_allele': 'T:benign', 'clin_sig': ['benign'], 'strand': 1, 'phenotype_or_disease': 1, 'allele_string': 'C/G/T', 'var_synonyms': {'ClinVar': ['RCV001615549', 'VCV001224700']}, 'id': 'rs4396169', 'seq_region_name': '1', 'frequencies': {'T': {'gnomadg': 0.2453, 'gnomadg_sas': 0.3842, 'gnomade': 0.2925, 'gnomadg_ami': 0.3531, 'gnomade_fin': 0.2792, 'eur': 0.2893, 'gnomadg_fin': 0.288, 'gnomadg_amr': 0.2276, 'gnomadg_mid': 0.2857, 'amr': 0.1931, 'gnomadg_afr': 0.1017, 'af': 0.2877, 'gnomadg_eas': 0.5309, 'gnomadg_nfe': 0.2945, 'gnomadg_remaining': 0.2555, 'gnomade_mid': 0.3364, 'afr': 0.0711, 'gnomadg_asj': 0.2939, 'gnomade_amr': 0.2028, 'gnomade_afr': 0.09261, 'gnomade_remaining': 0.2941, 'eas': 0.499, 'gnomade_sas': 0.375, 'gnomade_eas': 0.5253, 'gnomade_asj': 0.2936, 'sas': 0.4284, 'gnomade_nfe': 0.2878}, 'G': {'gnomade': 6.956e-07, 'gnomade_amr': 0, 'gnomade_mid': 0, 'gnomade_remaining': 0, 'gnomade_fin': 0, 'gnomade_afr': 0, 'gnomade_sas': 0, 'gnomade_nfe': 9.099e-07, 'gnomade_asj': 0, 'gnomade_eas': 0}}, 'end': 205799181, 'pubmed': [33462482], 'start': 205799181}], 'seq_region_name': '1', 'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429'}, {'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1'}, {'strand': -1, 'distance': 2565, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000484228', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2565, 'strand': -1, 'transcript_id': 'ENST00000484228', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065'}], 'id': 'rs4396169', 'strand': 1, 'allele_string': 'C/G/T', 'input': 'rs4396169', 'end': 205799181, 'most_severe_consequence': 'intron_variant', 'assembly_name': 'GRCh38', 'start': 205799181}, {'colocated_variants': [{'end': 205801933, 'start': 205801933, 'id': 'rs823065', 'seq_region_name': '1', 'allele_string': 'T/A/C/G', 'frequencies': {'G': {'eas': 0.7619, 'sas': 0.8088, 'gnomadg_nfe': 0.9421, 'gnomadg_remaining': 0.8642, 'afr': 0.6112, 'gnomadg_asj': 0.8701, 'gnomadg_fin': 0.9548, 'gnomadg_afr': 0.6795, 'amr': 0.8242, 'gnomadg_mid': 0.8401, 'gnomadg_amr': 0.8534, 'gnomadg_eas': 0.7886, 'af': 0.7742, 'gnomadg_sas': 0.8189, 'gnomadg': 0.8508, 'eur': 0.9324, 'gnomadg_ami': 0.989}}, 'strand': 1}], 'id': 'rs823065', 'transcript_consequences': [{'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1'}, {'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1, 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429'}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 445}, {'distance': 445, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 445, 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['upstream_gene_variant']}], 'seq_region_name': '1', 'input': 'rs823065', 'strand': 1, 'allele_string': 'T/A/C/G', 'end': 205801933, 'most_severe_consequence': 'intron_variant', 'assembly_name': 'GRCh38', 'start': 205801933}, {'start': 205802045, 'assembly_name': 'GRCh38', 'id': 'rs823066', 'colocated_variants': [{'frequencies': {'G': {'gnomadg_remaining': 0.8629, 'gnomadg_nfe': 0.9422, 'afr': 0.6067, 'gnomadg_asj': 0.8701, 'eas': 0.7619, 'sas': 0.8078, 'gnomadg_sas': 0.8191, 'gnomadg': 0.8496, 'eur': 0.9324, 'gnomadg_ami': 0.989, 'gnomadg_fin': 0.9548, 'gnomadg_afr': 0.6755, 'gnomadg_mid': 0.8401, 'amr': 0.8242, 'gnomadg_amr': 0.8527, 'gnomadg_eas': 0.788, 'af': 0.7728}}, 'seq_region_name': '1', 'id': 'rs823066', 'allele_string': 'A/C/G/T', 'start': 205802045, 'end': 205802045, 'strand': 1}], 'transcript_consequences': [{'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1'}, {'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 557, 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057'}, {'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 557, 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'SLC41A1'}, {'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'T', 'distance': 557, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065'}], 'seq_region_name': '1', 'most_severe_consequence': 'intron_variant', 'end': 205802045, 'input': 'rs823066', 'allele_string': 'A/C/G/T', 'strand': 1}, {'assembly_name': 'GRCh38', 'start': 205802576, 'seq_region_name': '1', 'colocated_variants': [{'end': 205802576, 'start': 205802576, 'allele_string': 'T/A/C/G', 'id': 'rs823067', 'seq_region_name': '1', 'frequencies': {'C': {'eas': 0.7619, 'sas': 0.8078, 'gnomadg_asj': 0.8704, 'afr': 0.612, 'gnomadg_remaining': 0.8661, 'gnomadg_nfe': 0.9422, 'gnomadg_afr': 0.6798, 'amr': 0.8242, 'gnomadg_mid': 0.8401, 'gnomadg_amr': 0.8528, 'gnomadg_fin': 0.9552, 'gnomadg_eas': 0.7887, 'af': 0.774, 'gnomadg_sas': 0.8204, 'gnomadg': 0.8513, 'eur': 0.9314, 'gnomadg_ami': 0.9879}}, 'strand': 1}], 'transcript_consequences': [{'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1'}, {'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'protein_coding', 'strand': -1, 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429'}, {'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 1088, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 1088, 'biotype': 'protein_coding_CDS_not_defined', 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['upstream_gene_variant']}, {'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'G', 'distance': 1088, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065'}], 'id': 'rs823067', 'allele_string': 'T/A/C/G', 'strand': 1, 'input': 'rs823067', 'end': 205802576, 'most_severe_consequence': 'intron_variant'}, {'assembly_name': 'GRCh38', 'start': 205802592, 'allele_string': 'T/A/C', 'strand': 1, 'input': 'rs823068', 'most_severe_consequence': 'intron_variant', 'end': 205802592, 'colocated_variants': [{'end': 205802592, 'start': 205802592, 'allele_string': 'T/A/C', 'id': 'rs823068', 'seq_region_name': '1', 'frequencies': {'C': {'gnomadg_ami': 0.989, 'eur': 0.9314, 'gnomadg': 0.8534, 'gnomadg_sas': 0.8205, 'af': 0.7762, 'gnomadg_eas': 0.79, 'gnomadg_amr': 0.8541, 'amr': 0.8242, 'gnomadg_afr': 0.6868, 'gnomadg_mid': 0.8401, 'gnomadg_fin': 0.9556, 'gnomadg_asj': 0.8707, 'afr': 0.6203, 'gnomadg_remaining': 0.8682, 'gnomadg_nfe': 0.9424, 'sas': 0.8078, 'eas': 0.7619}}, 'strand': 1}], 'transcript_consequences': [{'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['intron_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1104, 'strand': -1}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1104, 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C'}], 'id': 'rs823068', 'seq_region_name': '1'}, {'start': 205802654, 'assembly_name': 'GRCh38', 'end': 205802654, 'most_severe_consequence': 'intron_variant', 'allele_string': 'G/A/C', 'strand': 1, 'input': 'rs7540981', 'id': 'rs7540981', 'colocated_variants': [{'seq_region_name': '1', 'id': 'rs7540981', 'allele_string': 'G/A/C', 'frequencies': {'A': {'gnomadg_ami': 0.3531, 'eur': 0.2883, 'gnomadg': 0.2344, 'gnomadg_sas': 0.3847, 'af': 0.2764, 'gnomadg_eas': 0.554, 'gnomadg_fin': 0.29, 'gnomadg_amr': 0.2247, 'gnomadg_afr': 0.05749, 'gnomadg_mid': 0.2862, 'amr': 0.1873, 'gnomadg_remaining': 0.2457, 'gnomadg_nfe': 0.2948, 'afr': 0.0174, 'gnomadg_asj': 0.2989, 'sas': 0.4274, 'eas': 0.5188}}, 'end': 205802654, 'start': 205802654, 'strand': 1}], 'transcript_consequences': [{'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429'}, {'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1, 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065'}, {'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1166, 'strand': -1}, {'strand': -1, 'distance': 1166, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC'}], 'seq_region_name': '1'}, {'most_severe_consequence': 'intron_variant', 'end': 205803242, 'input': 'rs823070', 'allele_string': 'T/A/C/G', 'strand': 1, 'colocated_variants': [{'strand': 1, 'end': 205803242, 'start': 205803242, 'seq_region_name': '1', 'id': 'rs823070', 'allele_string': 'T/A/C/G', 'frequencies': {'C': {'gnomadg_sas': 0.8203, 'gnomadg': 0.848, 'eur': 0.9314, 'gnomadg_ami': 0.989, 'gnomadg_fin': 0.9549, 'gnomadg_mid': 0.8401, 'gnomadg_afr': 0.6696, 'amr': 0.8242, 'gnomadg_amr': 0.8519, 'gnomadg_eas': 0.789, 'af': 0.772, 'gnomadg_remaining': 0.861, 'gnomadg_nfe': 0.9421, 'gnomadg_asj': 0.8705, 'afr': 0.6044, 'eas': 0.7619, 'sas': 0.8078}, 'A': {'gnomadg_afr': 0, 'gnomadg_mid': 0, 'gnomadg_amr': 0, 'gnomadg_fin': 0, 'gnomadg_eas': 0, 'gnomadg_asj': 0, 'gnomadg_sas': 0, 'gnomadg_remaining': 0, 'gnomadg': 0, 'gnomadg_nfe': 0, 'gnomadg_ami': 0}}}], 'seq_region_name': '1', 'transcript_consequences': [{'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137'}, {'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065'}, {'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 1754, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'distance': 1754, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'distance': 1754, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'G', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057'}], 'id': 'rs823070', 'start': 205803242, 'assembly_name': 'GRCh38'}, {'colocated_variants': [{'strand': 1, 'start': 205804251, 'end': 205804251, 'frequencies': {'T': {'gnomadg_ami': 0.9901, 'eur': 0.9304, 'gnomadg': 0.8476, 'gnomadg_sas': 0.8191, 'af': 0.7716, 'gnomadg_eas': 0.7883, 'gnomadg_amr': 0.8515, 'gnomadg_afr': 0.6688, 'amr': 0.8242, 'gnomadg_mid': 0.8401, 'gnomadg_fin': 0.9547, 'gnomadg_asj': 0.87, 'afr': 0.6044, 'gnomadg_nfe': 0.942, 'gnomadg_remaining': 0.8609, 'sas': 0.8067, 'eas': 0.7619}}, 'id': 'rs823071', 'seq_region_name': '1', 'allele_string': 'G/C/T'}], 'seq_region_name': '1', 'transcript_consequences': [{'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant']}, {'strand': -1, 'biotype': 'protein_coding', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'SLC41A1', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant'], 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC'}, {'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'distance': 2763, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'biotype': 'protein_coding_CDS_not_defined', 'distance': 2763, 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'transcript_id': 'ENST00000468057', 'gene_symbol_source': 'HGNC'}], 'id': 'rs823071', 'most_severe_consequence': 'intron_variant', 'end': 205804251, 'input': 'rs823071', 'allele_string': 'G/C/T', 'strand': 1, 'start': 205804251, 'assembly_name': 'GRCh38'}, {'start': 205805028, 'assembly_name': 'GRCh38', 'transcript_consequences': [{'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 3540, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000468057', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429'}], 'colocated_variants': [{'phenotype_or_disease': 1, 'strand': 1, 'frequencies': {'C': {'afr': 0.5492, 'gnomadg_asj': 0.8697, 'gnomadg_remaining': 0.8483, 'gnomadg_nfe': 0.942, 'eas': 0.7609, 'sas': 0.8078, 'gnomadg': 0.8361, 'gnomadg_sas': 0.8188, 'gnomadg_ami': 0.989, 'eur': 0.9314, 'gnomadg_amr': 0.8454, 'gnomadg_mid': 0.8401, 'amr': 0.8213, 'gnomadg_afr': 0.6298, 'gnomadg_fin': 0.9547, 'af': 0.7568, 'gnomadg_eas': 0.7871}}, 'id': 'rs823073', 'seq_region_name': '1', 'allele_string': 'T/C', 'start': 205805028, 'pubmed': [32888494], 'end': 205805028}], 'seq_region_name': '1', 'id': 'rs823073', 'most_severe_consequence': 'intron_variant', 'end': 205805028, 'input': 'rs823073', 'strand': 1, 'allele_string': 'T/C'}, {'end': 205806682, 'most_severe_consequence': 'intron_variant', 'allele_string': 'T/C', 'strand': 1, 'input': 'rs823076', 'colocated_variants': [{'start': 205806682, 'pubmed': [24303760], 'end': 205806682, 'frequencies': {'C': {'gnomadg_remaining': 0.8525, 'gnomadg_nfe': 0.942, 'gnomadg_asj': 0.87, 'afr': 0.553, 'eas': 0.7609, 'sas': 0.8078, 'gnomadg_sas': 0.8187, 'gnomadg': 0.8373, 'eur': 0.9314, 'gnomadg_ami': 0.989, 'gnomadg_fin': 0.955, 'gnomadg_mid': 0.8401, 'amr': 0.8213, 'gnomadg_afr': 0.6334, 'gnomadg_amr': 0.8456, 'gnomadg_eas': 0.7876, 'af': 0.7578}}, 'seq_region_name': '1', 'id': 'rs823076', 'allele_string': 'T/C', 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000484000', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 3617}], 'id': 'rs823076', 'start': 205806682, 'assembly_name': 'GRCh38'}, {'assembly_name': 'GRCh38', 'start': 205806981, 'id': 'rs823077', 'colocated_variants': [{'end': 205806981, 'start': 205806981, 'allele_string': 'T/C', 'seq_region_name': '1', 'id': 'rs823077', 'frequencies': {'C': {'sas': 0.8078, 'eas': 0.7609, 'gnomadg_remaining': 0.8507, 'gnomadg_nfe': 0.9421, 'gnomadg_asj': 0.8697, 'afr': 0.553, 'gnomadg_eas': 0.7871, 'af': 0.7578, 'gnomadg_fin': 0.9546, 'amr': 0.8213, 'gnomadg_afr': 0.6323, 'gnomadg_mid': 0.8401, 'gnomadg_amr': 0.8457, 'eur': 0.9314, 'gnomadg_ami': 0.989, 'gnomadg_sas': 0.8189, 'gnomadg': 0.8369}}, 'strand': 1}], 'transcript_consequences': [{'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant']}, {'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000484000', 'strand': -1, 'distance': 3318, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'SLC41A1'}], 'seq_region_name': '1', 'input': 'rs823077', 'allele_string': 'T/C', 'strand': 1, 'most_severe_consequence': 'intron_variant', 'end': 205806981}, {'assembly_name': 'GRCh38', 'start': 205807330, 'seq_region_name': '1', 'colocated_variants': [{'end': 205807330, 'start': 205807330, 'id': 'rs823078', 'seq_region_name': '1', 'allele_string': 'T/A/C/G', 'frequencies': {'C': {'gnomadg_sas': 0.8198, 'gnomadg': 0.8502, 'eur': 0.9324, 'gnomadg_ami': 0.989, 'gnomadg_mid': 0.8401, 'amr': 0.8242, 'gnomadg_afr': 0.6775, 'gnomadg_amr': 0.8523, 'gnomadg_fin': 0.9547, 'gnomadg_eas': 0.7886, 'af': 0.7738, 'gnomadg_asj': 0.8703, 'afr': 0.6104, 'gnomadg_nfe': 0.9422, 'gnomadg_remaining': 0.864, 'eas': 0.7619, 'sas': 0.8078}}, 'strand': 1}], 'transcript_consequences': [{'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['intron_variant'], 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding'}, {'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'transcript_id': 'ENST00000484000', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 2969, 'biotype': 'protein_coding_CDS_not_defined'}, {'strand': -1, 'distance': 2969, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000484000', 'gene_symbol_source': 'HGNC'}, {'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000484000', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2969, 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'SLC41A1'}], 'id': 'rs823078', 'input': 'rs823078', 'allele_string': 'T/A/C/G', 'strand': 1, 'end': 205807330, 'most_severe_consequence': 'intron_variant'}, {'assembly_name': 'GRCh38', 'start': 205808031, 'allele_string': 'T/A/C', 'strand': 1, 'input': 'rs823079', 'most_severe_consequence': 'intron_variant', 'end': 205808031, 'seq_region_name': '1', 'colocated_variants': [{'frequencies': {'C': {'gnomadg_sas': 0.8199, 'gnomadg': 0.839, 'eur': 0.9314, 'gnomadg_ami': 0.989, 'gnomadg_afr': 0.6371, 'amr': 0.817, 'gnomadg_mid': 0.8401, 'gnomadg_amr': 0.8487, 'gnomadg_fin': 0.9548, 'gnomadg_eas': 0.7895, 'af': 0.7586, 'afr': 0.5582, 'gnomadg_asj': 0.8697, 'gnomadg_remaining': 0.8525, 'gnomadg_nfe': 0.9421, 'eas': 0.7609, 'sas': 0.8078}}, 'seq_region_name': '1', 'id': 'rs823079', 'allele_string': 'T/A/C', 'start': 205808031, 'end': 205808031, 'strand': 1}], 'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367137', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding', 'strand': -1}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000367137', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['intron_variant']}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000484000', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 2268, 'strand': -1, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'transcript_id': 'ENST00000484000', 'gene_symbol_source': 'HGNC', 'distance': 2268, 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'SLC41A1', 'impact': 'MODIFIER', 'variant_allele': 'C'}], 'id': 'rs823079'}, {'strand': 1, 'allele_string': 'G/A/C', 'input': 'rs11240569', 'most_severe_consequence': 'synonymous_variant', 'end': 205810103, 'transcript_consequences': [{'transcript_id': 'ENST00000367137', 'cdna_end': 1376, 'gene_id': 'ENSG00000133065', 'cds_start': 339, 'biotype': 'protein_coding', 'protein_start': 113, 'amino_acids': 'T', 'cdna_start': 1376, 'gene_symbol_source': 'HGNC', 'protein_end': 113, 'consequence_terms': ['synonymous_variant'], 'hgnc_id': 'HGNC:19429', 'cds_end': 339, 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'LOW', 'codons': 'acC/acT', 'strand': -1}, {'codons': 'acC/acG', 'variant_allele': 'C', 'impact': 'LOW', 'gene_symbol': 'SLC41A1', 'strand': -1, 'gene_symbol_source': 'HGNC', 'cdna_start': 1376, 'amino_acids': 'T', 'hgnc_id': 'HGNC:19429', 'cds_end': 339, 'protein_end': 113, 'consequence_terms': ['synonymous_variant'], 'protein_start': 113, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000367137', 'cds_start': 339, 'gene_id': 'ENSG00000133065', 'cdna_end': 1376}, {'transcript_id': 'ENST00000484000', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:19429', 'gene_id': 'ENSG00000133065', 'gene_symbol': 'SLC41A1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 196, 'strand': -1}, {'transcript_id': 'ENST00000484000', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000133065', 'hgnc_id': 'HGNC:19429', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'SLC41A1', 'strand': -1, 'distance': 196, 'biotype': 'protein_coding_CDS_not_defined'}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 3250, 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653632', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'distance': 3250, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000653632', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['upstream_gene_variant']}, {'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'distance': 3306, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653838', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['upstream_gene_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653838', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['upstream_gene_variant'], 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'distance': 3306, 'biotype': 'lncRNA'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000656162', 'biotype': 'lncRNA', 'distance': 3245, 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000656162', 'gene_symbol_source': 'HGNC', 'strand': 1, 'distance': 3245, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000656763', 'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 3219, 'biotype': 'lncRNA', 'strand': 1}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 3219, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000656763'}, {'biotype': 'lncRNA', 'distance': 3259, 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000658744', 'gene_symbol_source': 'HGNC'}, {'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['upstream_gene_variant'], 'transcript_id': 'ENST00000658744', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3259, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1'}, {'transcript_id': 'ENST00000659357', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'distance': 3244, 'biotype': 'lncRNA', 'strand': 1}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659357', 'biotype': 'lncRNA', 'distance': 3244, 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'distance': 3234, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659611', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['upstream_gene_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659611', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['upstream_gene_variant'], 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'distance': 3234, 'biotype': 'lncRNA'}, {'transcript_id': 'ENST00000659861', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['upstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3234}, {'strand': 1, 'distance': 3234, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659861'}, {'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662563', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3242, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'PM20D1-AS1'}, {'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'distance': 3242, 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000662563', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 3433, 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662659'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000662659', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'distance': 3433, 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'transcript_id': 'ENST00000664013', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'distance': 3234, 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'distance': 3234, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000664013', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['upstream_gene_variant']}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000664067', 'gene_symbol_source': 'HGNC', 'distance': 3242, 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664067', 'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'distance': 3242, 'biotype': 'lncRNA', 'strand': 1}, {'strand': 1, 'distance': 3234, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664074'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'distance': 3234, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664074', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['upstream_gene_variant']}, {'strand': 1, 'biotype': 'lncRNA', 'distance': 3226, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664917'}, {'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664917', 'biotype': 'lncRNA', 'distance': 3226, 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'consequence_terms': ['upstream_gene_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669196', 'biotype': 'lncRNA', 'distance': 3244, 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3244, 'transcript_id': 'ENST00000669196', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['upstream_gene_variant']}, {'strand': 1, 'distance': 3244, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669230'}, {'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['upstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669230', 'strand': 1, 'biotype': 'lncRNA', 'distance': 3244, 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'PM20D1-AS1'}, {'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000670417', 'distance': 3244, 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'consequence_terms': ['upstream_gene_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'transcript_id': 'ENST00000670417', 'gene_symbol_source': 'HGNC', 'distance': 3244, 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'C'}], 'colocated_variants': [{'allele_string': 'COSMIC_MUTATION', 'seq_region_name': '1', 'id': 'COSV65650164', 'var_synonyms': {'COSMIC': ['COSM146744']}, 'end': 205810103, 'start': 205810103, 'somatic': 1, 'strand': 1, 'phenotype_or_disease': 1}, {'pubmed': [20683486, 31546642, 35163527, 26308152, 27612022, 31384434, 34025389], 'end': 205810103, 'start': 205810103, 'seq_region_name': '1', 'id': 'rs11240569', 'var_synonyms': {'ClinVar': ['RCV001612573', 'VCV001231809']}, 'allele_string': 'G/A/C', 'frequencies': {'A': {'gnomade': 0.2909, 'gnomadg': 0.2329, 'gnomadg_sas': 0.3844, 'gnomadg_ami': 0.3516, 'gnomade_fin': 0.2785, 'eur': 0.2883, 'gnomadg_amr': 0.2203, 'amr': 0.1873, 'gnomadg_mid': 0.2842, 'gnomadg_afr': 0.05715, 'gnomadg_fin': 0.2879, 'af': 0.2766, 'gnomadg_eas': 0.553, 'gnomadg_asj': 0.2986, 'afr': 0.0174, 'gnomade_amr': 0.1964, 'gnomadg_remaining': 0.2443, 'gnomadg_nfe': 0.294, 'gnomade_mid': 0.3384, 'gnomade_remaining': 0.292, 'gnomade_afr': 0.04806, 'gnomade_sas': 0.3722, 'eas': 0.5198, 'gnomade_asj': 0.2977, 'gnomade_nfe': 0.2866, 'sas': 0.4274, 'gnomade_eas': 0.5501}}, 'strand': 1, 'clin_sig': ['benign'], 'phenotype_or_disease': 1, 'clin_sig_allele': 'A:benign'}], 'id': 'rs11240569', 'seq_region_name': '1', 'assembly_name': 'GRCh38', 'start': 205810103}, {'end': 205823552, 'most_severe_consequence': 'intron_variant', 'allele_string': 'T/A/C/G', 'strand': 1, 'input': 'rs844004', 'colocated_variants': [{'allele_string': 'T/A/C/G', 'seq_region_name': '1', 'id': 'rs844004', 'frequencies': {'C': {'gnomadg_asj': 0.8655, 'afr': 0.649, 'gnomadg_nfe': 0.9416, 'gnomadg_remaining': 0.871, 'eas': 0.7659, 'sas': 0.8037, 'gnomadg_sas': 0.8105, 'gnomadg': 0.8607, 'eur': 0.9294, 'gnomadg_ami': 0.989, 'gnomadg_afr': 0.7183, 'amr': 0.8271, 'gnomadg_mid': 0.8367, 'gnomadg_amr': 0.852, 'gnomadg_fin': 0.955, 'gnomadg_eas': 0.7881, 'af': 0.7837}}, 'end': 205823552, 'start': 205823552, 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'distance': 4473, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'PM20D1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367136'}, {'distance': 4473, 'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'PM20D1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['downstream_gene_variant'], 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'transcript_id': 'ENST00000367136', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'distance': 4473, 'biotype': 'protein_coding', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1', 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367136'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4474, 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'PM20D1', 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000460624', 'gene_symbol_source': 'HGNC'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4474, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1', 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000460624'}, {'transcript_id': 'ENST00000460624', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'consequence_terms': ['downstream_gene_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1', 'strand': -1, 'distance': 4474, 'biotype': 'protein_coding_CDS_not_defined'}, {'strand': -1, 'distance': 4801, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1', 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'consequence_terms': ['downstream_gene_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000461807'}, {'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000461807', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4801, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1'}, {'transcript_id': 'ENST00000461807', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'gene_symbol': 'PM20D1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4801, 'strand': -1}, {'gene_symbol': 'PM20D1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4781, 'strand': -1, 'transcript_id': 'ENST00000469861', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['downstream_gene_variant'], 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'distance': 4781, 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1', 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'consequence_terms': ['downstream_gene_variant'], 'transcript_id': 'ENST00000469861', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000469861', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'consequence_terms': ['downstream_gene_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'PM20D1', 'strand': -1, 'distance': 4781, 'biotype': 'protein_coding_CDS_not_defined'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000653632', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653632', 'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'PM20D1-AS1'}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000653632', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653838', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653838'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653838'}, {'transcript_id': 'ENST00000656162', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000656162', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000656162', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000656763'}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000656763', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'transcript_id': 'ENST00000656763', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'transcript_id': 'ENST00000658744', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000658744', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER'}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000658744', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'transcript_id': 'ENST00000659357', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659357'}, {'transcript_id': 'ENST00000659357', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000659611', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'transcript_id': 'ENST00000659611', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659611', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000659861', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659861', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'PM20D1-AS1', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000659861', 'gene_symbol_source': 'HGNC'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000662563', 'gene_symbol_source': 'HGNC'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000662563', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662563', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'G', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000662659', 'gene_symbol_source': 'HGNC'}, {'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000662659', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'PM20D1-AS1'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000662659', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664013', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664013', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000664013', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000664067', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000664067', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000664067', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'transcript_id': 'ENST00000664074', 'gene_symbol_source': 'HGNC'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664074'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664074'}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664917'}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664917', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664917', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669196', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669196', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000669196', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669230', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669230'}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000669230', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000670417', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'lncRNA', 'strand': 1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000670417', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000670417', 'gene_symbol_source': 'HGNC'}], 'id': 'rs844004', 'start': 205823552, 'assembly_name': 'GRCh38'}, {'colocated_variants': [{'phenotype_or_disease': 1, 'strand': 1, 'somatic': 1, 'start': 205828905, 'end': 205828905, 'id': 'COSV65649363', 'seq_region_name': '1', 'allele_string': 'COSMIC_MUTATION'}, {'allele_string': 'G/A/C/T', 'seq_region_name': '1', 'id': 'rs823085', 'frequencies': {'A': {'eas': 0.7649, 'sas': 0.8037, 'gnomadg_remaining': 0.871, 'gnomadg_nfe': 0.9415, 'afr': 0.6483, 'gnomadg_asj': 0.866, 'gnomadg_fin': 0.9548, 'gnomadg_amr': 0.8521, 'amr': 0.8271, 'gnomadg_afr': 0.7181, 'gnomadg_mid': 0.8367, 'af': 0.7833, 'gnomadg_eas': 0.7876, 'gnomadg': 0.8605, 'gnomadg_sas': 0.8099, 'gnomadg_ami': 0.989, 'eur': 0.9294}}, 'end': 205828905, 'start': 205828905, 'strand': 1}], 'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367136', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'gene_symbol': 'PM20D1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'biotype': 'protein_coding', 'strand': -1, 'gene_symbol': 'PM20D1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant'], 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367136'}, {'transcript_id': 'ENST00000367136', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'consequence_terms': ['intron_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1', 'strand': -1, 'biotype': 'protein_coding'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'PM20D1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'transcript_id': 'ENST00000460624', 'gene_symbol_source': 'HGNC'}, {'gene_symbol': 'PM20D1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000460624', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518'}, {'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000460624', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000461807', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'PM20D1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000461807', 'gene_symbol_source': 'HGNC', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1'}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1', 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000461807'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000469861', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'gene_symbol': 'PM20D1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1}, {'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000469861', 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'strand': -1, 'biotype': 'protein_coding_CDS_not_defined', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1', 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000469861'}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653632', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000653632', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'transcript_id': 'ENST00000653632', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000653838', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619'}, {'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653838', 'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'PM20D1-AS1'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653838', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000656162', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619'}, {'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000656162', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1'}, {'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000656162', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1'}, {'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000656763', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1'}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000656763', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000656763', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000658744', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'transcript_id': 'ENST00000658744', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000658744', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'transcript_id': 'ENST00000659357', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659357', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'lncRNA', 'strand': 1}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659357'}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659611'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659611', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'C'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659611', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000659861', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659861'}, {'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000659861', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1'}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000662563', 'gene_symbol_source': 'HGNC'}, {'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662563', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662563', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662659'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662659'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'transcript_id': 'ENST00000662659', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664013', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664013'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000664013', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000664067', 'gene_symbol_source': 'HGNC'}, {'transcript_id': 'ENST00000664067', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664067', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664074', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000664074', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000664074', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664917', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'C', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664917'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664917', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669196'}, {'transcript_id': 'ENST00000669196', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'C', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669196', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000669230', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'C', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669230', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669230', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000670417', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1'}, {'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000670417', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'C', 'gene_symbol': 'PM20D1-AS1'}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000670417'}], 'id': 'rs823085', 'seq_region_name': '1', 'end': 205828905, 'most_severe_consequence': 'intron_variant', 'strand': 1, 'allele_string': 'G/A/C/T', 'input': 'rs823085', 'start': 205828905, 'assembly_name': 'GRCh38'}, {'assembly_name': 'GRCh38', 'start': 205838376, 'input': 'rs954206', 'allele_string': 'C/G/T', 'strand': 1, 'end': 205838376, 'most_severe_consequence': 'intron_variant', 'id': 'rs954206', 'colocated_variants': [{'strand': 1, 'start': 205838376, 'end': 205838376, 'pubmed': [23597238], 'frequencies': {'G': {'gnomadg_ami': 0, 'gnomadg_nfe': 1.47e-05, 'gnomadg_remaining': 0, 'gnomadg': 6.575e-06, 'gnomadg_sas': 0, 'gnomadg_asj': 0, 'gnomadg_eas': 0, 'gnomadg_fin': 0, 'gnomadg_amr': 0, 'gnomadg_afr': 0, 'gnomadg_mid': 0}, 'T': {'eas': 0.4296, 'sas': 0.4059, 'gnomadg_remaining': 0.2498, 'gnomadg_nfe': 0.2917, 'gnomadg_asj': 0.3008, 'afr': 0.0635, 'gnomadg_fin': 0.2891, 'gnomadg_mid': 0.2755, 'gnomadg_afr': 0.094, 'amr': 0.1902, 'gnomadg_amr': 0.2216, 'gnomadg_eas': 0.4756, 'af': 0.2658, 'gnomadg_sas': 0.3683, 'gnomadg': 0.2391, 'eur': 0.2833, 'gnomadg_ami': 0.3516}}, 'allele_string': 'C/G/T', 'id': 'rs954206', 'seq_region_name': '1'}], 'transcript_consequences': [{'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1', 'strand': -1, 'biotype': 'protein_coding', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367136', 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'consequence_terms': ['intron_variant']}, {'strand': -1, 'biotype': 'protein_coding', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1', 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'consequence_terms': ['intron_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367136'}, {'gene_symbol': 'PM20D1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000460624', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'transcript_id': 'ENST00000460624', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'PM20D1', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'gene_symbol': 'PM20D1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000461807', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'PM20D1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000461807'}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000653632', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619'}, {'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'T', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000653632', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653838', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1'}, {'transcript_id': 'ENST00000653838', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000656162', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000656162', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000656763', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'transcript_id': 'ENST00000656763', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000658744'}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000658744', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659357'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659357', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000659611', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'transcript_id': 'ENST00000659611', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'G', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659861'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000659861', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662563', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'transcript_id': 'ENST00000662563', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662659'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'transcript_id': 'ENST00000662659', 'gene_symbol_source': 'HGNC'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000664013', 'gene_symbol_source': 'HGNC'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000664013', 'gene_symbol_source': 'HGNC'}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664067'}, {'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664067', 'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1-AS1'}, {'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'G', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664074', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664074', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000664917', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664917', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669196', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'G', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669196'}, {'variant_allele': 'G', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000669230', 'gene_symbol_source': 'HGNC', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669230', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'transcript_id': 'ENST00000670417', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'G'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000670417', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}], 'seq_region_name': '1'}, {'most_severe_consequence': 'intron_variant', 'end': 205838885, 'strand': 1, 'allele_string': 'C/A/T', 'input': 'rs11240572', 'colocated_variants': [{'start': 205838885, 'pubmed': [20683486, 21248740, 21425343, 21840748, 19915575, 23747047, 26944116, 27174169, 32682840, 30814270, 34373950, 24514572, 33494262, 25061481, 31366155, 27814993], 'end': 205838885, 'frequencies': {'A': {'eas': 0.1835, 'sas': 0.1339, 'gnomadg_nfe': 0.03242, 'gnomadg_remaining': 0.04214, 'afr': 0.0038, 'gnomadg_asj': 0.07905, 'gnomadg_fin': 0.03172, 'gnomadg_afr': 0.007482, 'amr': 0.1196, 'gnomadg_mid': 0.07143, 'gnomadg_amr': 0.08197, 'gnomadg_eas': 0.1773, 'af': 0.0873, 'gnomadg_sas': 0.1275, 'gnomadg': 0.03961, 'eur': 0.0328, 'gnomadg_ami': 0.009868}}, 'seq_region_name': '1', 'id': 'rs11240572', 'allele_string': 'C/A/T', 'strand': 1}], 'seq_region_name': '1', 'transcript_consequences': [{'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000367136', 'consequence_terms': ['intron_variant'], 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'gene_symbol': 'PM20D1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding', 'strand': -1}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1', 'strand': -1, 'biotype': 'protein_coding', 'transcript_id': 'ENST00000367136', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'consequence_terms': ['intron_variant']}, {'gene_symbol': 'PM20D1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'transcript_id': 'ENST00000460624', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'transcript_id': 'ENST00000460624', 'gene_symbol_source': 'HGNC', 'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'PM20D1', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000461807', 'hgnc_id': 'HGNC:26518', 'gene_id': 'ENSG00000162877', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1', 'strand': -1, 'biotype': 'protein_coding_CDS_not_defined'}, {'biotype': 'protein_coding_CDS_not_defined', 'strand': -1, 'gene_symbol': 'PM20D1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000162877', 'hgnc_id': 'HGNC:26518', 'transcript_id': 'ENST00000461807', 'gene_symbol_source': 'HGNC'}, {'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653632', 'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'PM20D1-AS1'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000653632', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'T'}, {'transcript_id': 'ENST00000653838', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'transcript_id': 'ENST00000653838', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000656162', 'gene_symbol_source': 'HGNC'}, {'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000656162', 'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'A', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'transcript_id': 'ENST00000656763', 'gene_symbol_source': 'HGNC'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'transcript_id': 'ENST00000656763', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000658744', 'gene_symbol_source': 'HGNC', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000658744', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000659357', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000659357', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000659611', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000659611', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'transcript_id': 'ENST00000659861', 'gene_symbol_source': 'HGNC', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619'}, {'transcript_id': 'ENST00000659861', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662563'}, {'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662563', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662659', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000662659'}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000664013', 'gene_symbol_source': 'HGNC'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000664013', 'gene_symbol_source': 'HGNC'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664067', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'biotype': 'lncRNA', 'strand': 1}, {'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000664067', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1-AS1'}, {'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664074', 'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'impact': 'MODIFIER', 'variant_allele': 'A'}, {'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1-AS1', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664074'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664917', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'variant_allele': 'A', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000664917', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'A', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'transcript_id': 'ENST00000669196', 'gene_symbol_source': 'HGNC'}, {'impact': 'MODIFIER', 'variant_allele': 'T', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA', 'transcript_id': 'ENST00000669196', 'gene_symbol_source': 'HGNC', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant']}, {'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669230', 'gene_id': 'ENSG00000286619', 'hgnc_id': 'HGNC:27633', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'PM20D1-AS1', 'strand': 1, 'biotype': 'lncRNA'}, {'strand': 1, 'biotype': 'lncRNA', 'variant_allele': 'T', 'impact': 'MODIFIER', 'gene_symbol': 'PM20D1-AS1', 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'gene_symbol_source': 'HGNC', 'transcript_id': 'ENST00000669230'}, {'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'transcript_id': 'ENST00000670417', 'gene_symbol_source': 'HGNC', 'strand': 1, 'biotype': 'lncRNA', 'impact': 'MODIFIER', 'variant_allele': 'A', 'gene_symbol': 'PM20D1-AS1'}, {'biotype': 'lncRNA', 'strand': 1, 'gene_symbol': 'PM20D1-AS1', 'variant_allele': 'T', 'impact': 'MODIFIER', 'consequence_terms': ['intron_variant', 'non_coding_transcript_variant'], 'hgnc_id': 'HGNC:27633', 'gene_id': 'ENSG00000286619', 'transcript_id': 'ENST00000670417', 'gene_symbol_source': 'HGNC'}], 'id': 'rs11240572', 'start': 205838885, 'assembly_name': 'GRCh38'}]\n"
     ]
    }
   ],
   "source": [
    "from ideal_genom.api_client import EnsemblRestClient\n",
    "\n",
    "client = EnsemblRestClient()\n",
    "\n",
    "# Example list of IDs for the POST request\n",
    "ids = filtered['ID'].to_list()#[\"rs11801418\", \"rs869841\", \"rs12563611\"]\n",
    "\n",
    "response = client.post_vep_request(ids)\n",
    "\n",
    "if response:\n",
    "    print(\"Response:\", response)\n",
    "else:\n",
    "    print(\"Failed to get response.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "154"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(response)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rs11801418 intron_variant\n",
      "rs869841 intron_variant\n",
      "rs12563611 intron_variant\n",
      "rs12407417 intron_variant\n",
      "rs12217091 intron_variant\n",
      "rs6699817 intron_variant\n",
      "rs6702731 intron_variant\n",
      "rs57128821 intron_variant\n",
      "rs57482207 intron_variant\n",
      "rs4951134 intron_variant\n",
      "rs4951135 intron_variant\n",
      "rs4951136 intron_variant\n",
      "rs67020185 intron_variant\n",
      "rs1107714 intron_variant\n",
      "rs7529113 intron_variant\n",
      "rs7515134 intron_variant\n",
      "rs7529301 intron_variant\n",
      "rs4951139 intron_variant\n",
      "rs16854533 intron_variant\n",
      "rs16854537 intron_variant\n",
      "rs12405886 intron_variant\n",
      "rs12408680 intron_variant\n",
      "rs12405298 intron_variant\n",
      "rs11240292 intron_variant\n",
      "rs61824662 non_coding_transcript_exon_variant\n",
      "rs1775153 intron_variant\n",
      "rs34606094 intron_variant\n",
      "rs1772136 intron_variant\n",
      "rs61824663 intron_variant\n",
      "rs6685188 intron_variant\n",
      "rs12402185 intron_variant\n",
      "rs12404824 intron_variant\n",
      "rs12737668 intron_variant\n",
      "rs823106 intron_variant\n",
      "rs12132270 intron_variant\n",
      "rs1891091 intron_variant\n",
      "rs10751444 intron_variant\n",
      "rs6593964 intron_variant\n",
      "rs6676110 intron_variant\n",
      "rs12118655 intron_variant\n",
      "rs823098 intron_variant\n",
      "rs1772140 intron_variant\n",
      "rs12748961 non_coding_transcript_exon_variant\n",
      "rs3747973 upstream_gene_variant\n",
      "rs3747972 upstream_gene_variant\n",
      "rs3805 3_prime_UTR_variant\n",
      "rs823092 intron_variant\n",
      "rs823093 intron_variant\n",
      "rs823108 intron_variant\n",
      "rs1772146 intron_variant\n",
      "rs1772147 intron_variant\n",
      "rs1620334 intron_variant\n",
      "rs71633561 intron_variant\n",
      "rs823109 intron_variant\n",
      "rs823112 intron_variant\n",
      "rs823113 intron_variant\n",
      "rs823128 intron_variant\n",
      "rs823129 intron_variant\n",
      "rs823131 intron_variant\n",
      "rs823114 upstream_gene_variant\n",
      "rs865402 upstream_gene_variant\n",
      "rs823115 upstream_gene_variant\n",
      "rs12564209 upstream_gene_variant\n",
      "rs823116 upstream_gene_variant\n",
      "rs823117 upstream_gene_variant\n",
      "rs823118 upstream_gene_variant\n",
      "rs860311 upstream_gene_variant\n",
      "rs823119 upstream_gene_variant\n",
      "rs823120 upstream_gene_variant\n",
      "rs823122 intergenic_variant\n",
      "rs823123 intergenic_variant\n",
      "rs823124 intergenic_variant\n",
      "rs1626710 intergenic_variant\n",
      "rs1772150 intergenic_variant\n",
      "rs823125 intergenic_variant\n",
      "rs823126 intergenic_variant\n",
      "rs4951024 intergenic_variant\n",
      "rs4338418 downstream_gene_variant\n",
      "rs823145 downstream_gene_variant\n",
      "rs823146 downstream_gene_variant\n",
      "rs823147 downstream_gene_variant\n",
      "rs823148 downstream_gene_variant\n",
      "rs823149 downstream_gene_variant\n",
      "rs823150 downstream_gene_variant\n",
      "rs823151 downstream_gene_variant\n",
      "rs71633563 downstream_gene_variant\n",
      "rs823153 downstream_gene_variant\n",
      "rs1772152 downstream_gene_variant\n",
      "rs1618936 downstream_gene_variant\n",
      "rs844005 3_prime_UTR_variant\n",
      "rs11557080 3_prime_UTR_variant\n",
      "rs823132 3_prime_UTR_variant\n",
      "rs823133 3_prime_UTR_variant\n",
      "rs823134 3_prime_UTR_variant\n",
      "rs823135 3_prime_UTR_variant\n",
      "rs823136 3_prime_UTR_variant\n",
      "rs823138 3_prime_UTR_variant\n",
      "rs708723 3_prime_UTR_variant\n",
      "rs823139 3_prime_UTR_variant\n",
      "rs823142 intron_variant\n",
      "rs823144 5_prime_UTR_variant\n",
      "rs1775151 intron_variant\n",
      "rs1772153 intron_variant\n",
      "rs1772154 intron_variant\n",
      "rs6683517 intron_variant\n",
      "rs12031215 intron_variant\n",
      "rs10900524 intron_variant\n",
      "rs11394622 intron_variant\n",
      "rs863725 intron_variant\n",
      "rs947211 non_coding_transcript_exon_variant\n",
      "rs1772156 downstream_gene_variant\n",
      "rs885224 downstream_gene_variant\n",
      "rs1775158 downstream_gene_variant\n",
      "rs1772157 downstream_gene_variant\n",
      "rs1775159 downstream_gene_variant\n",
      "rs1772158 downstream_gene_variant\n",
      "rs5014256 downstream_gene_variant\n",
      "rs10494874 downstream_gene_variant\n",
      "rs12565968 downstream_gene_variant\n",
      "rs12565321 downstream_gene_variant\n",
      "rs12567894 downstream_gene_variant\n",
      "rs11240567 downstream_gene_variant\n",
      "rs1775143 downstream_gene_variant\n",
      "rs1775144 downstream_gene_variant\n",
      "rs1775145 downstream_gene_variant\n",
      "rs1775146 downstream_gene_variant\n",
      "rs6679073 downstream_gene_variant\n",
      "rs12740609 downstream_gene_variant\n",
      "rs7536285 downstream_gene_variant\n",
      "rs7513115 downstream_gene_variant\n",
      "rs12747475 downstream_gene_variant\n",
      "rs1341249 3_prime_UTR_variant\n",
      "rs708726 intron_variant\n",
      "rs823155 intron_variant\n",
      "rs823156 non_coding_transcript_exon_variant\n",
      "rs823157 intron_variant\n",
      "rs4396169 intron_variant\n",
      "rs823065 intron_variant\n",
      "rs823066 intron_variant\n",
      "rs823067 intron_variant\n",
      "rs823068 intron_variant\n",
      "rs7540981 intron_variant\n",
      "rs823070 intron_variant\n",
      "rs823071 intron_variant\n",
      "rs823073 intron_variant\n",
      "rs823076 intron_variant\n",
      "rs823077 intron_variant\n",
      "rs823078 intron_variant\n",
      "rs823079 intron_variant\n",
      "rs11240569 synonymous_variant\n",
      "rs844004 intron_variant\n",
      "rs823085 intron_variant\n",
      "rs954206 intron_variant\n",
      "rs11240572 intron_variant\n"
     ]
    }
   ],
   "source": [
    "for res in response:\n",
    "    print(res['id'], res['most_severe_consequence'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Ensembl Gene ID: ENSG00000139618\n",
      "Ensembl Gene ID: LRG_293\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "# Base API URL\n",
    "server = \"https://rest.ensembl.org\"\n",
    "ext = \"/xrefs/symbol/homo_sapiens/\"\n",
    "headers = {\"Content-Type\": \"application/json\", \"Accept\": \"application/json\"}\n",
    "\n",
    "# Specify the gene symbol\n",
    "gene_symbol = \"BRCA2\"\n",
    "url = f\"{server}{ext}{gene_symbol}\"\n",
    "\n",
    "# Send GET request\n",
    "response = requests.get(url, headers=headers)\n",
    "\n",
    "# Process the response\n",
    "if response.ok:\n",
    "    xrefs = response.json()\n",
    "    for entry in xrefs:\n",
    "        if entry[\"type\"] == \"gene\":\n",
    "            print(\"Ensembl Gene ID:\", entry[\"id\"])\n",
    "else:\n",
    "    print(f\"Error: {response.status_code}, {response.reason}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'type': 'gene', 'id': 'ENSG00000139618'}, {'id': 'LRG_293', 'type': 'gene'}]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "response.json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "luxgiant-dstream-dWyB7QIT-py3.10",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
