{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "import ipywidgets as widgets\n",
    "from IPython.display import display\n",
    "\n",
    "# add parent directory to path\n",
    "library_path = os.path.abspath('..')\n",
    "if library_path not in sys.path:\n",
    "    sys.path.append(library_path)\n",
    "\n",
    "from ideal_genom.zoom_heatmap import filter_sumstats, snp_annotations, get_gene_information\n",
    "from ideal_genom.zoom_heatmap import  get_ld_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create interactive widgets for input\n",
    "input_path = widgets.Text(\n",
    "    value='/home/luis/data/gwasResult/',\n",
    "    description='Path to project folder:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "input_name = widgets.Text(\n",
    "    value='annotated_normalized_combined_R2_0.3.dose_step2_sex_pheno-glm.PHENO1.glm.logistic.hybrid',\n",
    "    description='Name of GWAS summary file:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "top_snp = widgets.Text(\n",
    "    value='table_lead_SNPS_GWAS_glm_logistic_final_paper',\n",
    "    description='Name of file with SNPs to highlight:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "bfile_path = widgets.Text(\n",
    "    value='/home/luis/data/LuxGiantimputed/inputData/',\n",
    "    description='Path to project folder:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "bfile_name = widgets.Text(\n",
    "    value='luxgiant_imputed_noprobID',\n",
    "    description='Path to project folder:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "# Display the widgets\n",
    "display(input_path, input_name, top_snp, bfile_path, bfile_name)\n",
    "\n",
    "# Function to get the text parameter values\n",
    "def get_params():\n",
    "    return input_path.value, input_name.value, top_snp.value, bfile_path.value, bfile_name.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use the parameter values\n",
    "path_params = get_params()\n",
    "print(f\"Parameter 1: {path_params[0]}\")\n",
    "print(f\"Parameter 2: {path_params[1]}\")\n",
    "print(f\"Parameter 3: {path_params[2]}\")\n",
    "print(f\"Parameter 4: {path_params[3]}\")\n",
    "print(f\"Parameter 5: {path_params[4]}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols_touse = widgets.Textarea(\n",
    "    value=\"#CHROM, POS, ID, P\",\n",
    "    description='Columns to use on the Mannhattan plot (comma-separated):',\n",
    "    style={'description_width': 'initial'},\n",
    "    layout=widgets.Layout(width='50%')\n",
    ")\n",
    "display(cols_touse)\n",
    "\n",
    "def get_cols():\n",
    "    return cols_touse.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols = get_cols()\n",
    "\n",
    "df_gwas = pd.read_csv(\n",
    "    os.path.join(path_params[0], path_params[1]), sep='\\t', usecols=[col.strip() for col in cols.split(',')]\n",
    ")\n",
    "df_gwas.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rsID = widgets.Text(\n",
    "    value='SNP',\n",
    "    description='Column name with the rsID:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "display(rsID)\n",
    "\n",
    "def get_rsID():\n",
    "    return rsID.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas_rsID = get_rsID()\n",
    "\n",
    "if path_params[2] is not None or path_params[2] != '':\n",
    "    highlit_path = os.path.join(path_params[0], path_params[2])\n",
    "    if os.path.exists(highlit_path):\n",
    "        df_high = pd.read_csv(\n",
    "            highlit_path, sep='\\t'\n",
    "        )\n",
    "        to_highlight = df_high[gwas_rsID].to_list()\n",
    "        print(to_highlight[:10])\n",
    "        del df_high\n",
    "    else:\n",
    "        print('Path to file with SNPs to highlight does not exist')\n",
    "        to_highlight = []\n",
    "else:\n",
    "    print('No file with SNPs to highlight')\n",
    "    to_highlight = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "SNP_col = widgets.Text(\n",
    "    value='ID',\n",
    "    description='Column with rsID:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "CHR_col = widgets.Text(\n",
    "    value='#CHROM',\n",
    "    description='Column with chromosome:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "POS_col = widgets.Text(\n",
    "    value='POS',\n",
    "    description='Column with base-pair position:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "\n",
    "P_col = widgets.Text(\n",
    "    value='P',\n",
    "    description='Column with p-values:',\n",
    "    style={'description_width': 'initial'}\n",
    ")\n",
    "display(SNP_col, CHR_col, POS_col, P_col)\n",
    "\n",
    "def get_col_names():\n",
    "    return SNP_col.value, CHR_col.value, POS_col.value, P_col.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "snp_col, chr_col, pos_col, p_col = get_col_names()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered = filter_sumstats(\n",
    "    data_df       =df_gwas, \n",
    "    lead_snp      =to_highlight[0], \n",
    "    snp_col       =snp_col, \n",
    "    p_col         =p_col, \n",
    "    pos_col       =pos_col, \n",
    "    chr_col       =chr_col, \n",
    "    pval_threshold=5e-6, \n",
    "    radius        =1e6\n",
    ")\n",
    "filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotated = snp_annotations(\n",
    "    data_df=filtered, \n",
    "    snp_col=snp_col, \n",
    "    chr_col=chr_col, \n",
    "    pos_col=pos_col\n",
    ")\n",
    "\n",
    "annotated['Mbp'] = annotated['POS'] / 1e6\n",
    "annotated.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotated['consequence'].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = get_gene_information(\n",
    "    genes=annotated['GENENAME'].unique().tolist(),\n",
    "    gtf_path=None,\n",
    "    build='38'\n",
    ")\n",
    "genes['start_esc'] = genes['start']/1e6\n",
    "genes['end_esc'] = genes['end']/1e6\n",
    "genes['length_esc'] = genes['end_esc'] - genes['start_esc']\n",
    "genes['lenght_esc1'] = genes['length_esc']/1e6\n",
    "genes\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_ld_matrix(\n",
    "    data_df=annotated,\n",
    "    snp_col=snp_col,\n",
    "    pos_col=pos_col,\n",
    "    bfile_folder=path_params[3],\n",
    "    bfile_name=path_params[4],\n",
    "    output_path='/home/luis/data/gwasResult/',\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_LD = pd.read_csv(\n",
    "        '/home/luis/data/LuxGiantimputed/inputData/matrix-ld.ld', \n",
    "        sep=r'\\s+',\n",
    "        header=None,\n",
    "        index_col=None,\n",
    "        engine='python'\n",
    "    )\n",
    "df_LD.head()\n",
    "ld = df_LD.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ld.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "\n",
    "N=ld.shape[0]\n",
    "ld = np.tril(ld, k=0)\n",
    "ldm = np.ma.masked_where(ld==0, ld)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import matplotlib\n",
    "from itertools import cycle\n",
    "from matplotlib.patches import FancyArrow\n",
    "from matplotlib import transforms\n",
    "\n",
    "region = (annotated['Mbp'].min() - 0.05, annotated['Mbp'].max() + 0.05)\n",
    "\n",
    "lead_snp = to_highlight[0]\n",
    "\n",
    "plt.figure(figsize=(10, 10))\n",
    "\n",
    "# Define the overall grid size (9 rows, 1 column)\n",
    "ax1 = plt.subplot2grid((9, 1), (0, 0), rowspan=4)  # Top plot (4 rows)\n",
    "ax2 = plt.subplot2grid((9, 1), (4, 0), rowspan=1)  # Middle plot (1 row)\n",
    "ax3 = plt.subplot2grid((9, 1), (5, 0), rowspan=4)  # Bottom plot (4 rows)\n",
    "\n",
    "# Plot for ax1\n",
    "lead = annotated[annotated[snp_col] == lead_snp]\n",
    "missense = annotated[annotated['consequence'] == 'intron_variant']\n",
    "utr3 = annotated[annotated['consequence'] == '3_prime_UTR_variant']\n",
    "utr5 = annotated[annotated['consequence'] == '5_prime_UTR_variant']\n",
    "upstrm = annotated[annotated['consequence'] == 'upstream_gene_variant']\n",
    "\n",
    "ax1.scatter(annotated['Mbp'], annotated[p_col], s=15, color='grey', label='')\n",
    "ax1.scatter(missense['Mbp'], missense[p_col], s=30, color='orange', label='Intron variant')\n",
    "ax1.scatter(utr3['Mbp'], utr3[p_col], s=30, color='blue', label=\"3'-UTR variant\")\n",
    "ax1.scatter(utr5['Mbp'], utr5[p_col], s=30, color='green', label=\"5'-UTR variant\")\n",
    "ax1.scatter(upstrm['Mbp'], upstrm[p_col], s=30, color='black', label=\"Upstream variant\")\n",
    "ax1.scatter(lead['Mbp'], lead[p_col], s=30, color='red', label='Lead SNP')\n",
    "ax1.scatter\n",
    "ax1.set_xlim(region)\n",
    "ax1.xaxis.set_ticks_position('top')\n",
    "ax1.legend(loc='best')\n",
    "ax1.set_title(f\"Million basepairs on BTA{1}\", fontsize=12)\n",
    "ax1.set_ylabel('log10(P)', fontsize=12)\n",
    "\n",
    "# Plot for ax2\n",
    "ys = cycle([0.1, 0.4, 0.7, 1])\n",
    "\n",
    "for i in genes.index:\n",
    "    symbol, strand = genes.loc[i, 'gene'], genes.loc[i, 'strand']\n",
    "    start, end, length = genes.loc[i, 'start_esc'], genes.loc[i, 'end_esc'], genes.loc[i, 'length_esc']\n",
    "    y = next(ys)\n",
    "    \n",
    "    if symbol == lead['GENENAME'].values[0]:\n",
    "        color = 'red'\n",
    "    else:\n",
    "        color = 'black'\n",
    "    \n",
    "    if strand == '+':\n",
    "        arrow = FancyArrow(start, y, length, 0, width=0.001, head_width=0.03, head_length=0.01, color=color)\n",
    "        ax2.add_patch(arrow)\n",
    "        ax2.text(start + 0.5 * length, y + 0.05, symbol, ha='center', size=9)\n",
    "    elif strand == '-':\n",
    "        arrow_neg = FancyArrow(end, y, -length, 0, width=0.001, head_width=0.03, head_length=0.01, color=color)\n",
    "        ax2.add_patch(arrow_neg)\n",
    "        ax2.text(start + 0.5 * length, y + 0.05, symbol, ha='center', size=9)\n",
    "\n",
    "ax2.set_ylim(0, 1.2)\n",
    "ax2.set_xlim(region)\n",
    "ax2.axis('off')\n",
    "\n",
    "base = ax3.transData # to rotate triangle\n",
    "rotation = transforms.Affine2D().rotate_deg(180+90+45)\n",
    "cmap=matplotlib.cm.Reds\n",
    "im=ax3.imshow(ldm, cmap=cmap,transform=rotation+base,aspect='auto')\n",
    "ax3.set_xlim([2, 1.41*N])\n",
    "ax3.set_ylim([1*N, 2])\n",
    "ax3.axis('off')\n",
    "\n",
    "# Add a colorbar as the legend\n",
    "cbar = plt.colorbar(im, ax=ax3, orientation='horizontal', fraction=0.05, pad=0.2)\n",
    "cbar.set_label('LD Value', fontsize=10)  # Adjust label as needed\n",
    "\n",
    "plt.tight_layout()\n",
    "plt.show()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "luxgiant-dstream-dWyB7QIT-py3.10",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
