

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>PostImputation Module &mdash; ideal-genom 0.1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=e59714d7" />

  
      <script src="../_static/jquery.js?v=5d32c60e"></script>
      <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../_static/documentation_options.js?v=01f34227"></script>
      <script src="../_static/doctools.js?v=9bcbadda"></script>
      <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Preparatory Module" href="preparatory.html" />
    <link rel="prev" title="GWAS with Generalized Linear Mixed Models (GLMM)" href="../gwas/gwas_glmm.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            ideal-genom
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gwas/gwas_glm.html">GWAS with Generalized Linear Models (GLM)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../gwas/gwas_glmm.html">GWAS with Generalized Linear Mixed Models (GLMM)</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">PostImputation Module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.AnnotateVCF"><code class="docutils literal notranslate"><span class="pre">AnnotateVCF</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.AnnotateVCF.execute_task"><code class="docutils literal notranslate"><span class="pre">AnnotateVCF.execute_task()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.AnnotateVCF.annotate_vcf"><code class="docutils literal notranslate"><span class="pre">AnnotateVCF.annotate_vcf()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.AnnotateVCF.__init__"><code class="docutils literal notranslate"><span class="pre">AnnotateVCF.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id0"><code class="docutils literal notranslate"><span class="pre">AnnotateVCF.annotate_vcf()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id1"><code class="docutils literal notranslate"><span class="pre">AnnotateVCF.execute_task()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.FilterVariants"><code class="docutils literal notranslate"><span class="pre">FilterVariants</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.FilterVariants.execute_task"><code class="docutils literal notranslate"><span class="pre">FilterVariants.execute_task()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.FilterVariants.filter_variants"><code class="docutils literal notranslate"><span class="pre">FilterVariants.filter_variants()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.FilterVariants.Dependencies"><code class="docutils literal notranslate"><span class="pre">FilterVariants.Dependencies()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.FilterVariants.__init__"><code class="docutils literal notranslate"><span class="pre">FilterVariants.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id2"><code class="docutils literal notranslate"><span class="pre">FilterVariants.execute_task()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id3"><code class="docutils literal notranslate"><span class="pre">FilterVariants.filter_variants()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.GetPLINK"><code class="docutils literal notranslate"><span class="pre">GetPLINK</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.GetPLINK.__init__"><code class="docutils literal notranslate"><span class="pre">GetPLINK.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.GetPLINK.convert_vcf_to_plink"><code class="docutils literal notranslate"><span class="pre">GetPLINK.convert_vcf_to_plink()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.GetPLINK.update_fam"><code class="docutils literal notranslate"><span class="pre">GetPLINK.update_fam()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.IndexVCF"><code class="docutils literal notranslate"><span class="pre">IndexVCF</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.IndexVCF.execute_task"><code class="docutils literal notranslate"><span class="pre">IndexVCF.execute_task()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.IndexVCF.index_vcf"><code class="docutils literal notranslate"><span class="pre">IndexVCF.index_vcf()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.IndexVCF.__init__"><code class="docutils literal notranslate"><span class="pre">IndexVCF.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id4"><code class="docutils literal notranslate"><span class="pre">IndexVCF.execute_task()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id5"><code class="docutils literal notranslate"><span class="pre">IndexVCF.index_vcf()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.NormalizeVCF"><code class="docutils literal notranslate"><span class="pre">NormalizeVCF</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.NormalizeVCF.execute_task"><code class="docutils literal notranslate"><span class="pre">NormalizeVCF.execute_task()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.NormalizeVCF.normalize_vcf"><code class="docutils literal notranslate"><span class="pre">NormalizeVCF.normalize_vcf()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.NormalizeVCF.__init__"><code class="docutils literal notranslate"><span class="pre">NormalizeVCF.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id6"><code class="docutils literal notranslate"><span class="pre">NormalizeVCF.execute_task()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id7"><code class="docutils literal notranslate"><span class="pre">NormalizeVCF.normalize_vcf()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner"><code class="docutils literal notranslate"><span class="pre">ParallelTaskRunner</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.input_path"><code class="docutils literal notranslate"><span class="pre">ParallelTaskRunner.input_path</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.output_path"><code class="docutils literal notranslate"><span class="pre">ParallelTaskRunner.output_path</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.max_workers"><code class="docutils literal notranslate"><span class="pre">ParallelTaskRunner.max_workers</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.files"><code class="docutils literal notranslate"><span class="pre">ParallelTaskRunner.files</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.__init__"><code class="docutils literal notranslate"><span class="pre">ParallelTaskRunner.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.execute_task"><code class="docutils literal notranslate"><span class="pre">ParallelTaskRunner.execute_task()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF"><code class="docutils literal notranslate"><span class="pre">ProcessVCF</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.__init__"><code class="docutils literal notranslate"><span class="pre">ProcessVCF.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_annotate"><code class="docutils literal notranslate"><span class="pre">ProcessVCF.execute_annotate()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_concatenate"><code class="docutils literal notranslate"><span class="pre">ProcessVCF.execute_concatenate()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_filter"><code class="docutils literal notranslate"><span class="pre">ProcessVCF.execute_filter()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_index"><code class="docutils literal notranslate"><span class="pre">ProcessVCF.execute_index()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_normalize"><code class="docutils literal notranslate"><span class="pre">ProcessVCF.execute_normalize()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_reference_normalize"><code class="docutils literal notranslate"><span class="pre">ProcessVCF.execute_reference_normalize()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_unzip"><code class="docutils literal notranslate"><span class="pre">ProcessVCF.execute_unzip()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ReferenceNormalizeVCF"><code class="docutils literal notranslate"><span class="pre">ReferenceNormalizeVCF</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ReferenceNormalizeVCF.__init__"><code class="docutils literal notranslate"><span class="pre">ReferenceNormalizeVCF.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ReferenceNormalizeVCF.execute_task"><code class="docutils literal notranslate"><span class="pre">ReferenceNormalizeVCF.execute_task()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ReferenceNormalizeVCF.normalize_with_reference"><code class="docutils literal notranslate"><span class="pre">ReferenceNormalizeVCF.normalize_with_reference()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.UnzipVCF"><code class="docutils literal notranslate"><span class="pre">UnzipVCF</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.UnzipVCF.execute_task"><code class="docutils literal notranslate"><span class="pre">UnzipVCF.execute_task()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.UnzipVCF.unzip_files"><code class="docutils literal notranslate"><span class="pre">UnzipVCF.unzip_files()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.UnzipVCF.__init__"><code class="docutils literal notranslate"><span class="pre">UnzipVCF.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id20"><code class="docutils literal notranslate"><span class="pre">UnzipVCF.execute_task()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id21"><code class="docutils literal notranslate"><span class="pre">UnzipVCF.unzip_files()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="preparatory.html">Preparatory Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="../annotations.html">ideal_genom.annotations</a></li>
<li class="toctree-l1"><a class="reference internal" href="../api_client.html">ideal_genom.api_client</a></li>
<li class="toctree-l1"><a class="reference internal" href="../get_examples.html">ideal_genom.get_examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="../get_references.html">ideal_genom.get_references</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Helpers.html">ideal_genom.Helpers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../power_comp.html">ideal_genom.power_comp</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">ideal-genom</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">PostImputation Module</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/preprocessing/PostImputation.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-ideal_genom.preprocessing.PostImputation">
<span id="postimputation-module"></span><h1>PostImputation Module<a class="headerlink" href="#module-ideal_genom.preprocessing.PostImputation" title="Link to this heading"></a></h1>
<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.AnnotateVCF">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.preprocessing.PostImputation.</span></span><span class="sig-name descname"><span class="pre">AnnotateVCF</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.AnnotateVCF" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner" title="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner"><code class="xref py py-class docutils literal notranslate"><span class="pre">ParallelTaskRunner</span></code></a></p>
<p>A parallel task runner for annotating normalized VCF files using reference annotation.
This class provides functionality to annotate normalized VCF files with identifiers
from a reference annotation file using bcftools. It processes multiple VCF files in
parallel, making it efficient for large genomic datasets.</p>
<p>The class identifies all normalized VCF files matching a specified pattern and
annotates them using the provided reference annotation file. It adds identifiers
from the reference file to the VCF entries.</p>
<dl class="py attribute">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">Inherits</span> <span class="pre">all</span> <span class="pre">attributes</span> <span class="pre">from</span> <span class="pre">ParallelTaskRunner.</span></span></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.AnnotateVCF.execute_task">
<span class="sig-name descname"><span class="pre">execute_task</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ref_annotation</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'annotated-'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.AnnotateVCF.execute_task" title="Link to this definition"></a></dt>
<dd><p>Annotate all normalized VCF files using the reference annotation file.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.AnnotateVCF.annotate_vcf">
<span class="sig-name descname"><span class="pre">annotate_vcf</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_annotation</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'annotated-'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.AnnotateVCF.annotate_vcf" title="Link to this definition"></a></dt>
<dd><p>Annotate a single VCF file using bcftools.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.AnnotateVCF.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_annotation</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_workers</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'annotated-'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.AnnotateVCF.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a ParallelTask processor.</p>
<p>This constructor sets up paths for processing multiple VCF in parallel and validates
that both input and output directories exist and are properly formatted.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_path</strong> (<em>Path</em>) – Path to the directory containing input files to process.</p></li>
<li><p><strong>output_path</strong> (<em>Path</em>) – Path to the directory where processed files will be saved.</p></li>
<li><p><strong>max_workers</strong> (<em>Optional</em><em>[</em><em>int</em><em>]</em>) – Maximum number of concurrent workers for parallel processing. If None, defaults to min(8, os.cpu_count()).</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If input_path or output_path is not a Path object.:</p></li>
<li><p><strong>FileNotFoundError</strong> – If input_path or output_path does not exist.:</p></li>
<li><p><strong>NotADirectoryError</strong> – If input_path or output_path is not a directory.:</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id0">
<span class="sig-name descname"><span class="pre">annotate_vcf</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_annotation</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'annotated-'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id0" title="Link to this definition"></a></dt>
<dd><p>Annotates a VCF file with identifiers from a reference annotation file using bcftools.
This method takes an input VCF file and annotates it with IDs from a reference
annotation file. The annotated VCF is saved to a new file with the specified prefix.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_file</strong> (<em>Path</em>) – Path to the input VCF file to be annotated.</p></li>
<li><p><strong>ref_annotation</strong> (<em>Path</em>) – Path to the reference annotation file used for annotation.</p></li>
<li><p><strong>output_prefix</strong> (<em>str</em><em> (</em><em>optional</em><em>)</em>) – Prefix to add to the output filename. Defaults to ‘annotated-‘.</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>FileExistsError</strong> – If the input file does not exist.:</p></li>
<li><p><strong>IsADirectoryError</strong> – If the input file is a directory, not a file.:</p></li>
<li><p><strong>TypeError</strong> – If ref_annotation is not a Path object or output_prefix is not a string.:</p></li>
<li><p><strong>subprocess.CalledProcessError</strong> – If the bcftools command fails.:</p></li>
<li><p><strong>FileNotFoundError</strong> – If the input file is not found during execution.:</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id1">
<span class="sig-name descname"><span class="pre">execute_task</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id1" title="Link to this definition"></a></dt>
<dd><p>Annotates normalized VCF files using a reference annotation file.
This method collects all normalized VCF files matching the pattern <code class="docutils literal notranslate"><span class="pre">normalized-*dose.vcf.gz</span></code>
and annotates them using the provided reference annotation file. The annotated files
will be saved with the specified output prefix.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ref_annotation</strong> (<em>Path</em>) – Path to the reference annotation file.</p></li>
<li><p><strong>output_prefix</strong> (<em>str</em><em> (</em><em>optional</em><em>)</em>) – Prefix to add to the output files. Defaults to ‘annotated-‘.</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError:</strong> – If ref_annotation is not a Path object or output_prefix is not a string.</p></li>
<li><p><strong>FileNotFoundError:</strong> – If the reference annotation file does not exist.</p></li>
</ul>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>None</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.FilterVariants">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.preprocessing.PostImputation.</span></span><span class="sig-name descname"><span class="pre">FilterVariants</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.FilterVariants" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner" title="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner"><code class="xref py py-class docutils literal notranslate"><span class="pre">ParallelTaskRunner</span></code></a></p>
<p>A class for filtering genetic variants in VCF/BCF files based on imputation quality (R² statistic).
This class extends ParallelTaskRunner to provide parallel processing capabilities for filtering
variants across multiple VCF files. It identifies variants with imputation quality below a specified
R² threshold and removes them from the output files.</p>
<dl class="py attribute">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">Inherits</span> <span class="pre">all</span> <span class="pre">attributes</span> <span class="pre">from</span> <span class="pre">ParallelTaskRunner</span></span></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.FilterVariants.execute_task">
<span class="sig-name descname"><span class="pre">execute_task</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">r2_threshold</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'filtered-'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.FilterVariants.execute_task" title="Link to this definition"></a></dt>
<dd><p>Set up and execute parallel filtering of variants based on R² threshold</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.FilterVariants.filter_variants">
<span class="sig-name descname"><span class="pre">filter_variants</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">r2_threshold</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'filtered-'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.FilterVariants.filter_variants" title="Link to this definition"></a></dt>
<dd><p>Filter a single VCF/BCF file based on the R² imputation quality threshold</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.FilterVariants.Dependencies">
<span class="sig-name descname"><span class="pre">Dependencies</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.FilterVariants.Dependencies" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">------------</span></span></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">-</span> <span class="pre">bcftools</span> <span class="pre">must</span> <span class="pre">be</span> <span class="pre">installed</span> <span class="pre">and</span> <span class="pre">available</span> <span class="pre">in</span> <span class="pre">the</span> <span class="pre">system</span> <span class="pre">path</span></span></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">-</span> <span class="pre">ParallelTaskRunner</span> <span class="pre">parent</span> <span class="pre">class</span> <span class="pre">for</span> <span class="pre">handling</span> <span class="pre">parallel</span> <span class="pre">execution</span></span></dt>
<dd></dd></dl>

<p class="rubric">Notes</p>
<p>The class searches for files matching the pattern <code class="docutils literal notranslate"><span class="pre">*dose.vcf.gz</span></code> in the input directory
and processes them in parallel. The filtered output files will be saved in the output
directory with the specified prefix added to their original filenames.</p>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.FilterVariants.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_workers</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">r2_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.3</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'filtered-'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.FilterVariants.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a ParallelTask processor.</p>
<p>This constructor sets up paths for processing multiple VCF in parallel and validates
that both input and output directories exist and are properly formatted.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_path</strong> (<em>Path</em>) – Path to the directory containing input files to process.</p></li>
<li><p><strong>output_path</strong> (<em>Path</em>) – Path to the directory where processed files will be saved.</p></li>
<li><p><strong>max_workers</strong> (<em>Optional</em><em>[</em><em>int</em><em>]</em>) – Maximum number of concurrent workers for parallel processing. If None, defaults to min(8, os.cpu_count()).</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If input_path or output_path is not a Path object.:</p></li>
<li><p><strong>FileNotFoundError</strong> – If input_path or output_path does not exist.:</p></li>
<li><p><strong>NotADirectoryError</strong> – If input_path or output_path is not a directory.:</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id2">
<span class="sig-name descname"><span class="pre">execute_task</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id2" title="Link to this definition"></a></dt>
<dd><p>Execute the task of filtering variants based on an R² threshold.</p>
<p>This method collects the necessary files with the pattern <code class="docutils literal notranslate"><span class="pre">*dose.vcf.gz</span></code> and runs
the filtering task with the specified parameters.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>r2_threshold</strong> (<em>float</em>) – The threshold value for the R² statistic. Variants with an R² value below this threshold
will be filtered out.</p></li>
<li><p><strong>output_prefix</strong> (<em>str</em><em>, </em><em>optional</em>) – The prefix to be added to output filenames. Default is ‘filtered-‘.</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>TypeError</strong> – If r2_threshold is not a float or output_prefix is not a string.</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>The method uses internal methods _file_collector and _run_task to perform the filtering operation.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id3">
<span class="sig-name descname"><span class="pre">filter_variants</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">r2_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'filtered-'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id3" title="Link to this definition"></a></dt>
<dd><p>Filter variants from a VCF/BCF file based on R2 imputation quality threshold.
This method takes an imputed VCF/BCF file and filters out variants with
imputation quality (R2) below the specified threshold. The filtered output
is saved as a compressed VCF.</p>
<section id="parameters">
<h2>Parameters:<a class="headerlink" href="#parameters" title="Link to this heading"></a></h2>
<dl class="simple">
<dt>input_file (Path):</dt><dd><p>Path to the input VCF/BCF file to be filtered</p>
</dd>
<dt>r2_threshold (float):</dt><dd><p>Minimum R2 imputation quality threshold (variants with R2 &lt;= threshold will be removed)</p>
</dd>
<dt>output_prefix (str, optional):</dt><dd><p>Prefix to add to the output filename. Defaults to ‘filtered-‘.</p>
</dd>
</dl>
</section>
<section id="returns">
<h2>Returns:<a class="headerlink" href="#returns" title="Link to this heading"></a></h2>
<blockquote>
<div><p>None: The method outputs a filtered VCF file but doesn’t return a value.</p>
</div></blockquote>
</section>
<section id="raises">
<h2>Raises:<a class="headerlink" href="#raises" title="Link to this heading"></a></h2>
<blockquote>
<div><p>FileExistsError: If the input file does not exist
IsADirectoryError: If the input path is a directory, not a file
TypeError: If r2_threshold is not a float or output_prefix is not a string</p>
</div></blockquote>
</section>
<section id="notes">
<h2>Notes:<a class="headerlink" href="#notes" title="Link to this heading"></a></h2>
<blockquote>
<div><p>The output file will be saved in the instance’s output_path directory with
the name constructed as: output_prefix + input_file.name
This method requires bcftools to be installed and available in the system path.</p>
</div></blockquote>
</section>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.GetPLINK">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.preprocessing.PostImputation.</span></span><span class="sig-name descname"><span class="pre">GetPLINK</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.GetPLINK" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.GetPLINK.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">input_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.GetPLINK.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize the GetPLINK class.
This class handles conversion of a concatenated VCF file to a PLINK binary ready for further analysis.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_path</strong> (<em>Path</em>) – Directory path where the input VCF file is located.</p></li>
<li><p><strong>output_path</strong> (<em>Path</em>) – Directory path where the output files will be saved.</p></li>
<li><p><strong>input_name</strong> (<em>str</em>) – Name of the input VCF file (must end with .vcf or .vcf.gz).</p></li>
<li><p><strong>output_name</strong> (<em>str</em><em>, </em><em>optional</em>) – Name for the output file. If not provided, it will be derived from input_name.</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If input_path or output_path is not a Path object, or if input_name or output_name is not a string.</p></li>
<li><p><strong>FileNotFoundError</strong> – If input_path or output_path does not exist.</p></li>
<li><p><strong>NotADirectoryError</strong> – If input_path or output_path is not a directory.</p></li>
<li><p><strong>ValueError</strong> – If input_name is not provided or if it doesn’t end with .vcf or .vcf.gz.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.GetPLINK.convert_vcf_to_plink">
<span class="sig-name descname"><span class="pre">convert_vcf_to_plink</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">double_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">memory</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.GetPLINK.convert_vcf_to_plink" title="Link to this definition"></a></dt>
<dd><p>Convert a VCF file to PLINK binary format (.bed, .bim, .fam).
This method runs the plink2 command-line tool to convert the input VCF file to PLINK
binary format, filtering for SNPs with standard ACGT alleles only.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>double_id</strong> (<em>bool</em><em> (</em><em>optional</em><em>)</em>) – Whether to use the –double-id flag in plink2 command, which sets both FID and IID to the sample ID. Defaults to True.</p></li>
<li><p><strong>threads</strong> (<em>int</em><em> (</em><em>optional</em><em>)</em>) – Number of CPU threads to use. If None, defaults to (available CPU cores - 2) or 10 if CPU count can’t be determined.</p></li>
<li><p><strong>memory</strong> (<em>int</em><em> (</em><em>optional</em><em>)</em>) – Memory allocation in MB for plink2. If None, defaults to approximately 2/3 of available system memory.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><em>None</em></p></li>
<li><p><em>Side Effects</em></p></li>
<li><p><em>————</em></p></li>
<li><p><em>Creates PLINK binary files (.bed, .bim, .fam) in the self.analysis_ready directory with the prefix self.output_name + “-nosex”.</em></p></li>
</ul>
</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>subprocess.CalledProcessError</strong> – If the plink2 command execution fails.:</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.GetPLINK.update_fam">
<span class="sig-name descname"><span class="pre">update_fam</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">for_fam_update_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.GetPLINK.update_fam" title="Link to this definition"></a></dt>
<dd><p>Add family information to the PLINK .fam file.</p>
<p>This method reads a family information file and updates the PLINK .fam file
using the provided family information, via PLINK2.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>for_fam_update_file</strong> (<em>Path</em>) – Path to the family information file (.fam or without suffix).</p></li>
<li><p><strong>threads</strong> (<em>int</em><em>, </em><em>optional</em>) – Number of threads to use for PLINK2 (defaults to available CPUs - 2).</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.IndexVCF">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.preprocessing.PostImputation.</span></span><span class="sig-name descname"><span class="pre">IndexVCF</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.IndexVCF" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner" title="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner"><code class="xref py py-class docutils literal notranslate"><span class="pre">ParallelTaskRunner</span></code></a></p>
<p>A class for indexing VCF (Variant Call Format) files using bcftools in parallel.
This class extends ParallelTaskRunner to enable parallel processing of multiple VCF files.
It creates index files that facilitate quick random access to compressed VCF files.</p>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.IndexVCF.execute_task">
<span class="sig-name descname"><span class="pre">execute_task</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pattern</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.IndexVCF.execute_task" title="Link to this definition"></a></dt>
<dd><p>Collects VCF files based on the given pattern and runs indexing in parallel.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.IndexVCF.index_vcf">
<span class="sig-name descname"><span class="pre">index_vcf</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.IndexVCF.index_vcf" title="Link to this definition"></a></dt>
<dd><p>Indexes a single VCF file using bcftools.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.IndexVCF.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_workers</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pattern</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'normalized-*dose.vcf.gz'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.IndexVCF.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a ParallelTask processor.</p>
<p>This constructor sets up paths for processing multiple VCF in parallel and validates
that both input and output directories exist and are properly formatted.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_path</strong> (<em>Path</em>) – Path to the directory containing input files to process.</p></li>
<li><p><strong>output_path</strong> (<em>Path</em>) – Path to the directory where processed files will be saved.</p></li>
<li><p><strong>max_workers</strong> (<em>Optional</em><em>[</em><em>int</em><em>]</em>) – Maximum number of concurrent workers for parallel processing. If None, defaults to min(8, os.cpu_count()).</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If input_path or output_path is not a Path object.:</p></li>
<li><p><strong>FileNotFoundError</strong> – If input_path or output_path does not exist.:</p></li>
<li><p><strong>NotADirectoryError</strong> – If input_path or output_path is not a directory.:</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id4">
<span class="sig-name descname"><span class="pre">execute_task</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id4" title="Link to this definition"></a></dt>
<dd><p>Execute the task of indexing VCF files.</p>
<p>This method collects files based on the provided pattern and indexes
the VCF files.</p>
<section id="parameter">
<h2>Parameter<a class="headerlink" href="#parameter" title="Link to this heading"></a></h2>
<dl class="simple">
<dt>pattern<span class="classifier">str</span></dt><dd><p>The pattern used to filter/collect files for processing.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">rtype<span class="colon">:</span></dt>
<dd class="field-odd"><p>None</p>
</dd>
</dl>
</section>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id5">
<span class="sig-name descname"><span class="pre">index_vcf</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id5" title="Link to this definition"></a></dt>
<dd><p>Index a VCF file using bcftools.</p>
<p>This method creates an index for the specified VCF file using bcftools,
which is required for efficient querying and processing of VCF files.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>input_file</strong> (<em>Path</em>) – Path to the VCF file to be indexed. Must be an existing file.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>FileExistsError</strong> – If the input file does not exist.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.NormalizeVCF">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.preprocessing.PostImputation.</span></span><span class="sig-name descname"><span class="pre">NormalizeVCF</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.NormalizeVCF" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner" title="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner"><code class="xref py py-class docutils literal notranslate"><span class="pre">ParallelTaskRunner</span></code></a></p>
<p>A class for normalizing VCF files post-imputation in parallel.
This class provides functionality to process VCF files by normalizing them using
bcftools. It’s specifically designed to handle post-imputation VCF files and
split multiallelic variants into separate entries.
The class inherits from ParallelTaskRunner to enable parallel processing of
multiple VCF files, which improves performance for large-scale genomic datasets.</p>
<dl class="py attribute">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">Inherits</span> <span class="pre">all</span> <span class="pre">attributes</span> <span class="pre">from</span> <span class="pre">ParallelTaskRunner</span></span></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.NormalizeVCF.execute_task">
<span class="sig-name descname"><span class="pre">execute_task</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'uncompressed-'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.NormalizeVCF.execute_task" title="Link to this definition"></a></dt>
<dd><p>Execute the normalization task on VCF files in parallel.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.NormalizeVCF.normalize_vcf">
<span class="sig-name descname"><span class="pre">normalize_vcf</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'uncompressed-'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.NormalizeVCF.normalize_vcf" title="Link to this definition"></a></dt>
<dd><p>Normalize a single VCF file using bcftools norm with the -m -any option.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.NormalizeVCF.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_workers</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'uncompressed-'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.NormalizeVCF.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a ParallelTask processor.</p>
<p>This constructor sets up paths for processing multiple VCF in parallel and validates
that both input and output directories exist and are properly formatted.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_path</strong> (<em>Path</em>) – Path to the directory containing input files to process.</p></li>
<li><p><strong>output_path</strong> (<em>Path</em>) – Path to the directory where processed files will be saved.</p></li>
<li><p><strong>max_workers</strong> (<em>Optional</em><em>[</em><em>int</em><em>]</em>) – Maximum number of concurrent workers for parallel processing. If None, defaults to min(8, os.cpu_count()).</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If input_path or output_path is not a Path object.:</p></li>
<li><p><strong>FileNotFoundError</strong> – If input_path or output_path does not exist.:</p></li>
<li><p><strong>NotADirectoryError</strong> – If input_path or output_path is not a directory.:</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id6">
<span class="sig-name descname"><span class="pre">execute_task</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id6" title="Link to this definition"></a></dt>
<dd><p>Execute the post-imputation normalization task on VCF files.</p>
<p>This method collects filtered dose VCF files matching the pattern <code class="docutils literal notranslate"><span class="pre">filtered-*dose.vcf.gz</span></code>
and runs the normalization process on them. The normalized files will be prefixed with
the provided output_prefix.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>output_prefix</strong> (<em>str</em><em> (</em><em>optional</em><em>)</em>) – Prefix to add to the output files. Defaults to ‘uncompressed-‘.</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If output_prefix is not a string.:</p></li>
<li><p><strong>Returns:</strong> – </p></li>
<li><p><strong>--------</strong> – </p></li>
<li><p><strong>None</strong> – </p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id7">
<span class="sig-name descname"><span class="pre">normalize_vcf</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'uncompressed-'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id7" title="Link to this definition"></a></dt>
<dd><p>Normalizes a VCF file using bcftools norm with the -m -any option.
This method takes a VCF file, performs normalization using bcftools to split
multiallelic variants into separate entries, and outputs the normalized file
with the specified prefix.</p>
<section id="id8">
<h2>Parameters:<a class="headerlink" href="#id8" title="Link to this heading"></a></h2>
<blockquote>
<div><p>input_file (Path): Path to the input VCF file to be normalized
output_prefix (str, optional): Prefix for the output file name. Defaults to ‘uncompressed-’</p>
</div></blockquote>
</section>
<section id="id9">
<h2>Returns:<a class="headerlink" href="#id9" title="Link to this heading"></a></h2>
<blockquote>
<div><p>None</p>
</div></blockquote>
</section>
<section id="id10">
<h2>Raises:<a class="headerlink" href="#id10" title="Link to this heading"></a></h2>
<blockquote>
<div><p>FileExistsError: If the input file does not exist
IsADirectoryError: If the input file path points to a directory
TypeError: If output_prefix is not a string</p>
</div></blockquote>
</section>
<section id="note">
<h2>Note:<a class="headerlink" href="#note" title="Link to this heading"></a></h2>
<blockquote>
<div><p>The output file will be saved in the output_path directory with the naming
convention: output_prefix + base_name, where base_name is derived from the input file.</p>
</div></blockquote>
</section>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.preprocessing.PostImputation.</span></span><span class="sig-name descname"><span class="pre">ParallelTaskRunner</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A base class for running parallel tasks on files.</p>
<p>This class provides the basic infrastructure for parallel processing of files
using ThreadPoolExecutor. It handles file collection and parallel task execution
while providing progress monitoring and logging.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.input_path">
<span class="sig-name descname"><span class="pre">input_path</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.input_path" title="Link to this definition"></a></dt>
<dd><p>Directory path where input files are located.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.output_path">
<span class="sig-name descname"><span class="pre">output_path</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.output_path" title="Link to this definition"></a></dt>
<dd><p>Directory path where output files will be saved.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.max_workers">
<span class="sig-name descname"><span class="pre">max_workers</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.max_workers" title="Link to this definition"></a></dt>
<dd><p>Maximum number of worker threads to use. Defaults to min(8, CPU count).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>int</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.files">
<span class="sig-name descname"><span class="pre">files</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.files" title="Link to this definition"></a></dt>
<dd><p>List of files to be processed.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>List[Path]</p>
</dd>
</dl>
</dd></dl>

<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>TypeError</strong> – If input_path or output_path are not Path objects.:</p></li>
<li><p><strong>FileNotFoundError</strong> – If input_path or output_path don’t exist.:</p></li>
<li><p><strong>NotADirectoryError</strong> – If input_path or output_path are not directories.:</p></li>
</ul>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_workers</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a ParallelTask processor.</p>
<p>This constructor sets up paths for processing multiple VCF in parallel and validates
that both input and output directories exist and are properly formatted.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_path</strong> (<em>Path</em>) – Path to the directory containing input files to process.</p></li>
<li><p><strong>output_path</strong> (<em>Path</em>) – Path to the directory where processed files will be saved.</p></li>
<li><p><strong>max_workers</strong> (<em>Optional</em><em>[</em><em>int</em><em>]</em>) – Maximum number of concurrent workers for parallel processing. If None, defaults to min(8, os.cpu_count()).</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If input_path or output_path is not a Path object.:</p></li>
<li><p><strong>FileNotFoundError</strong> – If input_path or output_path does not exist.:</p></li>
<li><p><strong>NotADirectoryError</strong> – If input_path or output_path is not a directory.:</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.execute_task">
<span class="sig-name descname"><span class="pre">execute_task</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner.execute_task" title="Link to this definition"></a></dt>
<dd><p>Execute the specific post-imputation processing task.</p>
<p>This abstract method should be implemented by all subclasses to perform
their specific post-imputation processing operations. Implementations
should handle the execution logic for the particular task the subclass
is designed to perform.</p>
<section id="id11">
<h2>Returns:<a class="headerlink" href="#id11" title="Link to this heading"></a></h2>
<p>None</p>
</section>
<section id="id12">
<h2>Raises:<a class="headerlink" href="#id12" title="Link to this heading"></a></h2>
<p>NotImplementedError: If the subclass does not implement this method.</p>
</section>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ProcessVCF">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.preprocessing.PostImputation.</span></span><span class="sig-name descname"><span class="pre">ProcessVCF</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>ProcessVCF class for post-imputation processing of Variant Call Format (VCF) files.</p>
<p>This class provides a pipeline for processing VCF files through multiple sequential steps:</p>
<ol class="arabic simple">
<li><p>Unzipping VCF files (if compressed)</p></li>
<li><p>Filtering variants based on imputation quality (R²)</p></li>
<li><p>Normalizing variant representation</p></li>
<li><p>Normalizing against a reference genome</p></li>
<li><p>Indexing the normalized VCF files</p></li>
<li><p>Annotating variants with additional information</p></li>
<li><p>Concatenating multiple VCF files into a single output file</p></li>
</ol>
<dl class="field-list simple">
<dt class="field-odd">Variables<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_path</strong> (<em>pathlib.Path</em>) – Directory containing input VCF files.</p></li>
<li><p><strong>output_path</strong> (<em>pathlib.Path</em>) – Directory where processed files will be saved.</p></li>
<li><p><strong>process_vcf</strong> (<em>pathlib.Path</em>) – Subdirectory created for intermediate processing files.</p></li>
</ul>
</dd>
</dl>
<dl class="simple">
<dt><code class="xref py py-meth docutils literal notranslate"><span class="pre">execute_unzip(password)()</span></code></dt><dd><p>Unzips compressed VCF files.</p>
</dd>
<dt><code class="xref py py-meth docutils literal notranslate"><span class="pre">execute_filter(r2_threshold)()</span></code></dt><dd><p>Filters variants based on imputation quality.</p>
</dd>
<dt><a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_normalize" title="ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_normalize"><code class="xref py py-meth docutils literal notranslate"><span class="pre">execute_normalize()</span></code></a></dt><dd><p>Normalizes variant representation.</p>
</dd>
<dt><code class="xref py py-meth docutils literal notranslate"><span class="pre">execute_reference_normalize(build,</span> <span class="pre">reference_file)()</span></code></dt><dd><p>Normalizes variants against a reference genome.</p>
</dd>
<dt><code class="xref py py-meth docutils literal notranslate"><span class="pre">execute_index(pattern)()</span></code></dt><dd><p>Creates index files for the VCF files.</p>
</dd>
<dt><code class="xref py py-meth docutils literal notranslate"><span class="pre">execute_annotate(ref_annotation,</span> <span class="pre">output_prefix)()</span></code></dt><dd><p>Annotates variants with additional information.</p>
</dd>
<dt><code class="xref py py-meth docutils literal notranslate"><span class="pre">execute_concatenate(output_name,</span> <span class="pre">max_threads)()</span></code></dt><dd><p>Concatenates multiple VCF files into a single output file.</p>
</dd>
</dl>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ProcessVCF.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize the ProcessVCF class for post-imputation processing of VCF files.</p>
<p>This constructor sets up the input and output directories for processing VCF files
and creates a subdirectory for intermediate processing files.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_path</strong> (<em>Path</em>) – Path to the directory containing input VCF files.</p></li>
<li><p><strong>output_path</strong> (<em>Path</em>) – Path to the directory where processed files will be saved.</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If <cite>input_path</cite> or <cite>output_path</cite> is not of type <cite>Path</cite>.</p></li>
<li><p><strong>FileNotFoundError</strong> – If <cite>input_path</cite> or <cite>output_path</cite> does not exist.</p></li>
<li><p><strong>NotADirectoryError</strong> – If <cite>input_path</cite> or <cite>output_path</cite> is not a directory.</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>A subdirectory named <cite>process_vcf</cite> is created inside the <cite>input_path</cite> directory
for storing intermediate files during processing.</p></li>
<li><p>This class is designed to handle multiple sequential steps in VCF file processing,
such as unzipping, filtering, normalizing, and annotating.</p></li>
</ul>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_annotate">
<span class="sig-name descname"><span class="pre">execute_annotate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ref_annotation</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'annotated-'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_annotate" title="Link to this definition"></a></dt>
<dd><p>Annotates a VCF file using a reference annotation file.
This method initializes an AnnotateVCF object and executes the annotation
process on the current VCF file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ref_annotation</strong> (<em>Path</em>) – Path to the reference annotation file.</p></li>
<li><p><strong>output_prefix</strong> (<em>str</em><em>, </em><em>optional</em>) – Prefix to be added to the output file name. Default is ‘annotated-‘.</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_concatenate">
<span class="sig-name descname"><span class="pre">execute_concatenate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">output_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_threads</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_concatenate" title="Link to this definition"></a></dt>
<dd><p>Concatenates annotated VCF files using bcftools concat.
This method finds all annotated VCF files in the process_vcf directory,
sorts them, and concatenates them into a single compressed VCF file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>output_name</strong> (<em>str</em>) – Name of the output file.</p></li>
<li><p><strong>max_threads</strong> (<em>int</em><em> (</em><em>optional</em><em>)</em>) – Maximum number of threads to use for concatenation. If None, uses min(8, os.cpu_count()). Defaults to None.</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>TypeError</strong> – If output_name is not a string.:</p></li>
<li><p><strong>FileNotFoundError</strong> – If no annotated VCF files are found in the process_vcf directory.:</p></li>
<li><p><strong>ValueError</strong> – If max_threads is less than 1.:</p></li>
<li><p><strong>Notes:</strong> – </p></li>
<li><p><strong>------</strong> – </p></li>
<li><p><strong>The output file will be saved in the output_path directory.</strong> – </p></li>
<li><p><strong>The method uses the 'bcftools concat' command with Oz compression.</strong> – </p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_filter">
<span class="sig-name descname"><span class="pre">execute_filter</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">r2_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.3</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_filter" title="Link to this definition"></a></dt>
<dd><p>Execute a filtering operation on VCF data based on R² threshold.</p>
<p>This method filters variants in the processed VCF file by creating and
executing a FilterVariants object with the specified R² threshold.
Both input and output are set to the same process_vcf file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>r2_threshold</strong> (<em>float</em><em> (</em><em>optional</em><em>)</em>) – The R² threshold value for filtering variants. Variants with R² value below this threshold will be filtered out. Default is 0.3.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_index">
<span class="sig-name descname"><span class="pre">execute_index</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pattern</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'normalized-*dose.vcf.gz'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_index" title="Link to this definition"></a></dt>
<dd><p>Index VCF files matching a specific pattern.</p>
<p>This method creates an indexer for VCF files and executes the indexing task
on files that match the given pattern in the process_vcf directory.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>pattern</strong> (<em>str</em><em> (</em><em>optional</em><em>)</em><em>:</em>) – The glob pattern to match VCF files for indexing. Defaults to <code class="docutils literal notranslate"><span class="pre">normalized-*dose.vcf.gz</span></code>.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_normalize">
<span class="sig-name descname"><span class="pre">execute_normalize</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_normalize" title="Link to this definition"></a></dt>
<dd><p>Normalizes the VCF file using the NormalizeVCF class.
This method creates a NormalizeVCF object with the current processed VCF file
as both input and output, then executes the normalization task. The normalization
process updates the VCF file in place.</p>
<section id="id13">
<h2>Returns:<a class="headerlink" href="#id13" title="Link to this heading"></a></h2>
<blockquote>
<div><p>None: This method doesn’t return any value.</p>
</div></blockquote>
</section>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_reference_normalize">
<span class="sig-name descname"><span class="pre">execute_reference_normalize</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'38'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ref_genome</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_reference_normalize" title="Link to this definition"></a></dt>
<dd><p>Normalize the VCF file against a reference genome.
This method creates a ReferenceNormalizeVCF object and executes the normalization
task on the processed VCF file, using the specified genome build or reference file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>build</strong> (<em>str</em><em> (</em><em>optional</em><em>)</em>) – Genome build version to use. Defaults to ‘38’.</p></li>
<li><p><strong>reference_file</strong> (<em>Path</em><em> (</em><em>optional</em><em>)</em>) – Path to a custom reference file. If provided, this will be used instead of the default reference for the specified build. Defaults to None.</p></li>
<li><p><strong>Returns</strong></p></li>
<li><p><strong>--------</strong></p></li>
<li><p><strong>None</strong></p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_unzip">
<span class="sig-name descname"><span class="pre">execute_unzip</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">password</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ProcessVCF.execute_unzip" title="Link to this definition"></a></dt>
<dd><p>Unzips a VCF file using the UnzipVCF utility.
This method creates an instance of UnzipVCF with the input and process paths
from the current object, then executes the unzipping task. If the VCF file
is password-protected, a password can be provided.</p>
<section id="id14">
<h2>Parameters:<a class="headerlink" href="#id14" title="Link to this heading"></a></h2>
<blockquote>
<div><p>password (str, optional): Password for the protected zip file. Defaults to None.</p>
</div></blockquote>
</section>
<section id="id15">
<h2>Returns:<a class="headerlink" href="#id15" title="Link to this heading"></a></h2>
<blockquote>
<div><p>None</p>
</div></blockquote>
</section>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ReferenceNormalizeVCF">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.preprocessing.PostImputation.</span></span><span class="sig-name descname"><span class="pre">ReferenceNormalizeVCF</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ReferenceNormalizeVCF" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner" title="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner"><code class="xref py py-class docutils literal notranslate"><span class="pre">ParallelTaskRunner</span></code></a></p>
<p>A class for normalizing VCF files using a reference genome in parallel.
This class extends ParallelTaskRunner to process multiple VCF files concurrently,
normalizing them against a reference genome using bcftools. If a reference file
is not provided, it will automatically download the appropriate reference genome
based on the specified build.</p>
<section id="attributes">
<h2>Attributes:<a class="headerlink" href="#attributes" title="Link to this heading"></a></h2>
<blockquote>
<div><p>reference_file (str): Path to the reference genome file used for normalization.</p>
</div></blockquote>
</section>
<section id="methods">
<h2>Methods:<a class="headerlink" href="#methods" title="Link to this heading"></a></h2>
<blockquote>
<div><p>execute_task: Sets up normalization task and processes files in parallel.
normalize_with_reference: Normalizes a single VCF file using bcftools.</p>
</div></blockquote>
</section>
<section id="example">
<h2>Example:<a class="headerlink" href="#example" title="Link to this heading"></a></h2>
<blockquote>
<div><p>normalizer = ReferenceNormalizeVCF()
normalizer.execute_task(build=’38’, output_prefix=’normalized-‘)</p>
</div></blockquote>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ReferenceNormalizeVCF.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_workers</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'38'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'normalized-'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ReferenceNormalizeVCF.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a ParallelTask processor.</p>
<p>This constructor sets up paths for processing multiple VCF in parallel and validates
that both input and output directories exist and are properly formatted.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_path</strong> (<em>Path</em>) – Path to the directory containing input files to process.</p></li>
<li><p><strong>output_path</strong> (<em>Path</em>) – Path to the directory where processed files will be saved.</p></li>
<li><p><strong>max_workers</strong> (<em>Optional</em><em>[</em><em>int</em><em>]</em>) – Maximum number of concurrent workers for parallel processing. If None, defaults to min(8, os.cpu_count()).</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If input_path or output_path is not a Path object.:</p></li>
<li><p><strong>FileNotFoundError</strong> – If input_path or output_path does not exist.:</p></li>
<li><p><strong>NotADirectoryError</strong> – If input_path or output_path is not a directory.:</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ReferenceNormalizeVCF.execute_task">
<span class="sig-name descname"><span class="pre">execute_task</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ReferenceNormalizeVCF.execute_task" title="Link to this definition"></a></dt>
<dd><p>Execute the post-imputation normalization task with reference genome.
This method normalizes VCF files using a reference genome. If no reference file is provided,
it automatically downloads the appropriate reference genome based on the build parameter.</p>
<section id="id16">
<h3>Parameters:<a class="headerlink" href="#id16" title="Link to this heading"></a></h3>
<dl class="simple">
<dt>build<span class="classifier">str (optional)</span></dt><dd><p>Genome build version, either ‘37’ or ‘38’. Defaults to ‘38’.</p>
</dd>
<dt>output_prefix<span class="classifier">str (optional)</span></dt><dd><p>Prefix to add to output files. Defaults to ‘normalized-‘.</p>
</dd>
<dt>reference_file<span class="classifier">Path (optional)</span></dt><dd><p>Path to the reference genome file. If None or the file doesn’t exist, the reference will be downloaded. Defaults to None.</p>
</dd>
</dl>
</section>
<section id="id17">
<h3>Returns:<a class="headerlink" href="#id17" title="Link to this heading"></a></h3>
<p>None</p>
</section>
<section id="id18">
<h3>Raises:<a class="headerlink" href="#id18" title="Link to this heading"></a></h3>
<p>TypeError: If output_prefix is not a string.</p>
</section>
<section id="id19">
<h3>Notes:<a class="headerlink" href="#id19" title="Link to this heading"></a></h3>
<p>This method collects uncompressed dose VCF files using a pattern match and normalizes them against the reference genome. The downloaded reference genomes come from the 1000 Genomes Project.</p>
</section>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.ReferenceNormalizeVCF.normalize_with_reference">
<span class="sig-name descname"><span class="pre">normalize_with_reference</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_prefix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'normalized-'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.ReferenceNormalizeVCF.normalize_with_reference" title="Link to this definition"></a></dt>
<dd><p>Normalize a VCF file with a reference genome using bcftools.
This method takes an input VCF file and normalizes it against a reference genome
using bcftools norm. The normalized output is compressed with gzip (-Oz).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_file</strong> (<em>Path</em>) – Path to the input VCF file to be normalized.</p></li>
<li><p><strong>output_prefix</strong> (<em>str</em><em>, </em><em>default='normalized-'</em>) – Prefix to add to the output filename.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The method doesn’t return a value but creates a normalized VCF file
at the output_path location.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>None</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If output_prefix is not a string.</p></li>
<li><p><strong>subprocess.CalledProcessError</strong> – If the bcftools command fails.</p></li>
<li><p><strong>FileNotFoundError</strong> – If the input file cannot be found.</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>The output filename is constructed from the output_prefix and the
base name extracted from the input filename (after the first hyphen).</p>
</dd></dl>

</section>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.UnzipVCF">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.preprocessing.PostImputation.</span></span><span class="sig-name descname"><span class="pre">UnzipVCF</span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.UnzipVCF" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#ideal_genom.preprocessing.PostImputation.ParallelTaskRunner" title="ideal_genom.preprocessing.PostImputation.ParallelTaskRunner"><code class="xref py py-class docutils literal notranslate"><span class="pre">ParallelTaskRunner</span></code></a></p>
<p>A class for unzipping VCF (Variant Call Format) files after imputation, with support for parallel processing.
This class extends ParallelTaskRunner to efficiently extract VCF files from zip archives,
including password-protected ones. It collects all zip files in the working directory
and extracts their contents to the output directory.</p>
<dl class="py attribute">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">Inherits</span> <span class="pre">all</span> <span class="pre">attributes</span> <span class="pre">from</span> <span class="pre">ParallelTaskRunner</span></span></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.UnzipVCF.execute_task">
<span class="sig-name descname"><span class="pre">execute_task</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">password</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.UnzipVCF.execute_task" title="Link to this definition"></a></dt>
<dd><p>Execute the unzipping process for all zip files in the working directory</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.UnzipVCF.unzip_files">
<span class="sig-name descname"><span class="pre">unzip_files</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">zip_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">password</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.UnzipVCF.unzip_files" title="Link to this definition"></a></dt>
<dd><p>Extract files from a specific zip archive, handling password protection if needed</p>
</dd></dl>

<p class="rubric">Notes</p>
<ul class="simple">
<li><p>VCF files are commonly used in genomics for storing gene sequence variations</p></li>
<li><p>The class only extracts files (not directories) from the zip archives</p></li>
<li><p>All extracted files are placed directly in the output directory without preserving paths</p></li>
<li><p>This class is designed for post-imputation processing in genetic data pipelines</p></li>
</ul>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.PostImputation.UnzipVCF.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_workers</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">password</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.PostImputation.UnzipVCF.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a ParallelTask processor.</p>
<p>This constructor sets up paths for processing multiple VCF in parallel and validates
that both input and output directories exist and are properly formatted.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_path</strong> (<em>Path</em>) – Path to the directory containing input files to process.</p></li>
<li><p><strong>output_path</strong> (<em>Path</em>) – Path to the directory where processed files will be saved.</p></li>
<li><p><strong>max_workers</strong> (<em>Optional</em><em>[</em><em>int</em><em>]</em>) – Maximum number of concurrent workers for parallel processing. If None, defaults to min(8, os.cpu_count()).</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If input_path or output_path is not a Path object.:</p></li>
<li><p><strong>FileNotFoundError</strong> – If input_path or output_path does not exist.:</p></li>
<li><p><strong>NotADirectoryError</strong> – If input_path or output_path is not a directory.:</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id20">
<span class="sig-name descname"><span class="pre">execute_task</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id20" title="Link to this definition"></a></dt>
<dd><p>Execute the post-imputation unzipping task on VCF files.</p>
<p>This method performs the following steps:
1. Collects all zip files in the working directory
2. Unzips the VCF files, using the provided password if necessary</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>password</strong> (<em>Optional</em><em>[</em><em>str</em><em>]</em>) – Password to decrypt zip files if they are password-protected.
Default is None.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>This method doesn’t return any value.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id21">
<span class="sig-name descname"><span class="pre">unzip_files</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">zip_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">password</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id21" title="Link to this definition"></a></dt>
<dd><p>Extract files from a password-protected zip archive.
This method extracts all non-directory files from the specified zip archive
to the class’s output_path directory. If the zip file is password-protected,
provide the password as a parameter.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>zip_path</strong> (<em>Path</em>) – Path to the zip file to be extracted</p></li>
<li><p><strong>password</strong> (<em>str</em>) – Password for the zip file, None if the file is not password-protected</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>No exceptions are explicitly raised</strong><strong>, </strong><strong>but logging errors occur if the zip file is corrupted</strong> – </p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>Files are extracted to the output_path directory of the class instance.
Only files (not directories) are extracted from the archive.
File paths are not preserved - all files are placed directly in output_path.</p>
</dd></dl>

</dd></dl>

</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../gwas/gwas_glmm.html" class="btn btn-neutral float-left" title="GWAS with Generalized Linear Mixed Models (GLMM)" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="preparatory.html" class="btn btn-neutral float-right" title="Preparatory Module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Luis Giraldo Gonzalez Ricardo.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>