

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Preparatory Module &mdash; ideal-genom 0.1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=e59714d7" />

  
      <script src="../_static/jquery.js?v=5d32c60e"></script>
      <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../_static/documentation_options.js?v=01f34227"></script>
      <script src="../_static/doctools.js?v=9bcbadda"></script>
      <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Manhattan-type Plots Module" href="../visualization/manhattan_type.html" />
    <link rel="prev" title="PostImputation Module" href="PostImputation.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            ideal-genom
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gwas/gwas_glm.html">GWAS with Generalized Linear Models (GLM)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../gwas/gwas_glmm.html">GWAS with Generalized Linear Mixed Models (GLMM)</a></li>
<li class="toctree-l1"><a class="reference internal" href="PostImputation.html">PostImputation Module</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Preparatory Module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.preprocessing.preparatory.Preparatory"><code class="docutils literal notranslate"><span class="pre">Preparatory</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.preparatory.Preparatory.input_path"><code class="docutils literal notranslate"><span class="pre">Preparatory.input_path</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.preparatory.Preparatory.input_name"><code class="docutils literal notranslate"><span class="pre">Preparatory.input_name</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.preparatory.Preparatory.output_path"><code class="docutils literal notranslate"><span class="pre">Preparatory.output_path</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.preparatory.Preparatory.output_name"><code class="docutils literal notranslate"><span class="pre">Preparatory.output_name</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.preparatory.Preparatory.high_ld_file"><code class="docutils literal notranslate"><span class="pre">Preparatory.high_ld_file</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.preparatory.Preparatory.build"><code class="docutils literal notranslate"><span class="pre">Preparatory.build</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.preparatory.Preparatory.__init__"><code class="docutils literal notranslate"><span class="pre">Preparatory.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.preparatory.Preparatory.execute_ld_prunning"><code class="docutils literal notranslate"><span class="pre">Preparatory.execute_ld_prunning()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.preprocessing.preparatory.Preparatory.execute_pc_decomposition"><code class="docutils literal notranslate"><span class="pre">Preparatory.execute_pc_decomposition()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../visualization/manhattan_type.html">Manhattan-type Plots Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="../visualization/plots.html">Other Plots Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="../visualization/zoom_heatmap.html">Zoom-Heatmap Plot Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="../annotations.html">Annotations Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="../api_client.html">API Clients</a></li>
<li class="toctree-l1"><a class="reference internal" href="../get_examples.html">Examples Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="../get_references.html">References Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Helpers.html">ideal_genom.Helpers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../power_comp.html">Statistical Power Computation</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">ideal-genom</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Preparatory Module</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/preprocessing/preparatory.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-ideal_genom.preprocessing.preparatory">
<span id="preparatory-module"></span><h1>Preparatory Module<a class="headerlink" href="#module-ideal_genom.preprocessing.preparatory" title="Link to this heading"></a></h1>
<p>Class designed to run preparatory steps before conducting Genomic-Wide Association Studies (GWAS).</p>
<p>This class handles the pruning of high linkage disequilibrium (LD) regions and performs Principal Component Analysis (PCA) on the pruned data.
It uses PLINK software for the pruning and PCA operations, ensuring that the input data is in the correct format and that necessary files are present.
It also manages the fetching of high LD regions if they are not provided, using the FetcherLDRegions class from the ideal_genom package.
It is designed to be flexible with parameters such as missing rate, minor allele frequency, and number of principal components to compute.
It also allows for memory and thread management during the execution of PLINK</p>
<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.preparatory.Preparatory">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.preprocessing.preparatory.</span></span><span class="sig-name descname"><span class="pre">Preparatory</span></span><a class="headerlink" href="#ideal_genom.preprocessing.preparatory.Preparatory" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A class for preprocessing genomic data in preparation for analysis.</p>
<p>This class handles the preparatory steps needed for genomic data analysis,
including input validation, LD (Linkage Disequilibrium) pruning, and
PCA (Principal Component Analysis) decomposition.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.preparatory.Preparatory.input_path">
<span class="sig-name descname"><span class="pre">input_path</span></span><a class="headerlink" href="#ideal_genom.preprocessing.preparatory.Preparatory.input_path" title="Link to this definition"></a></dt>
<dd><p>Path to the directory containing input PLINK files (.bed, .bim, .fam)</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str or Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.preparatory.Preparatory.input_name">
<span class="sig-name descname"><span class="pre">input_name</span></span><a class="headerlink" href="#ideal_genom.preprocessing.preparatory.Preparatory.input_name" title="Link to this definition"></a></dt>
<dd><p>Base name of the input PLINK files (without extension)</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.preparatory.Preparatory.output_path">
<span class="sig-name descname"><span class="pre">output_path</span></span><a class="headerlink" href="#ideal_genom.preprocessing.preparatory.Preparatory.output_path" title="Link to this definition"></a></dt>
<dd><p>Path to the directory where output files will be saved</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str or Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.preparatory.Preparatory.output_name">
<span class="sig-name descname"><span class="pre">output_name</span></span><a class="headerlink" href="#ideal_genom.preprocessing.preparatory.Preparatory.output_name" title="Link to this definition"></a></dt>
<dd><p>Base name for the output files</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.preparatory.Preparatory.high_ld_file">
<span class="sig-name descname"><span class="pre">high_ld_file</span></span><a class="headerlink" href="#ideal_genom.preprocessing.preparatory.Preparatory.high_ld_file" title="Link to this definition"></a></dt>
<dd><p>Path to the high LD regions file. If not found, will be fetched automatically</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str or Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.preparatory.Preparatory.build">
<span class="sig-name descname"><span class="pre">build</span></span><a class="headerlink" href="#ideal_genom.preprocessing.preparatory.Preparatory.build" title="Link to this definition"></a></dt>
<dd><p>Genome build version, either ‘38’ or ‘37’</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str, default=’38’</p>
</dd>
</dl>
</dd></dl>

<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>ValueError</strong> – If input_path or output_path is None, or if input_name or output_name is None</p></li>
<li><p><strong>TypeError</strong> – If input_path or output_path is not of type str or Path, or if input_name or output_name is not of type str,
    or if build is not of type str</p></li>
<li><p><strong>FileNotFoundError</strong> – If the specified input_path or output_path does not exist, or if the required PLINK files (.bed, .bim, .fam) are not found,
    or if the high LD file is not found and cannot be fetched.</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>This class uses PLINK software for genomic data processing operations.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The class assumes that PLINK is installed and available in the system PATH.</p>
</div>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.preparatory.Preparatory.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">input_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">high_ld_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'38'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.preparatory.Preparatory.__init__" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.preparatory.Preparatory.execute_ld_prunning">
<span class="sig-name descname"><span class="pre">execute_ld_prunning</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mind</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maf</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.01</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">geno</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">hwe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">5e-06</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ind_pair</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">[50,</span> <span class="pre">5,</span> <span class="pre">0.2]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">memory</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.preparatory.Preparatory.execute_ld_prunning" title="Link to this definition"></a></dt>
<dd><p>Execute LD (Linkage Disequilibrium) pruning on genetic data using PLINK.</p>
<p>This method performs LD pruning in two steps:
1. Excludes high LD regions and identifies independent SNPs
2. Extracts the identified independent SNPs</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>mind</strong> (<em>float</em><em>, </em><em>optional</em><em> (</em><em>default=0.2</em><em>)</em>) – Missing rate per individual threshold. Excludes individuals with missing rate higher than threshold.</p></li>
<li><p><strong>maf</strong> (<em>float</em><em>, </em><em>optional</em><em> (</em><em>default=0.01</em><em>)</em>) – Minor allele frequency threshold. Must be between 0 and 0.5.</p></li>
<li><p><strong>geno</strong> (<em>float</em><em>, </em><em>optional</em><em> (</em><em>default=0.1</em><em>)</em>) – Missing rate per SNP threshold. Must be between 0 and 1.</p></li>
<li><p><strong>hwe</strong> (<em>float</em><em>, </em><em>optional</em><em> (</em><em>default=5e-6</em><em>)</em>) – Hardy-Weinberg equilibrium exact test p-value threshold. Must be between 0 and 1.</p></li>
<li><p><strong>ind_pair</strong> (<em>list</em><em>, </em><em>optional</em><em> (</em><em>default=</em><em>[</em><em>50</em><em>, </em><em>5</em><em>, </em><em>0.2</em><em>]</em><em>)</em>) – Parameters for pairwise pruning: [window size(variants), step size(variants), r^2 threshold]</p></li>
<li><p><strong>memory</strong> (<em>int</em><em>, </em><em>optional</em><em> (</em><em>default=None</em><em>)</em>) – Memory in MB to allocate. If None, uses 2/3 of available system memory.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The results are saved to disk and the pruned file path is stored in self.pruned_file</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>None</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If mind, maf, geno, or hwe are not float</p></li>
<li><p><strong>ValueError</strong> – If maf is not between 0 and 0.5
    If geno is not between 0 and 1
    If hwe is not between 0 and 1</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>Uses PLINK software for the pruning operations.
Operates on chromosomes 1-22 only.
Automatically determines optimal thread count based on system CPU cores.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.preprocessing.preparatory.Preparatory.execute_pc_decomposition">
<span class="sig-name descname"><span class="pre">execute_pc_decomposition</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pca</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.preprocessing.preparatory.Preparatory.execute_pc_decomposition" title="Link to this definition"></a></dt>
<dd><p>Execute PCA decomposition on pruned PLINK binary files.</p>
<p>This method performs Principal Component Analysis (PCA) on the pruned genotype data
using PLINK software. It requires the existence of pruned binary PLINK files
(.bed, .bim, .fam) and generates PCA eigenvectors and eigenvalues.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>pca</strong> (<em>int</em><em>, </em><em>default=10</em>) – Number of principal components to compute. Must be greater than 0.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>TypeError</strong> – If pca parameter is not an integer.</p></li>
<li><p><strong>ValueError</strong> – If pca parameter is less than 1.</p></li>
<li><p><strong>FileNotFoundError</strong> – If any of the required pruned PLINK files (.bed, .bim, .fam) are not found.</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>The method automatically determines the optimal number of threads to use based on
CPU count, reserving 2 cores for other processes. If CPU count cannot be determined,
it defaults to 10 threads.</p>
<p>The output files will be created in the same directory as the input files, using
the input name as prefix with extensions .eigenvec and .eigenval.</p>
</dd></dl>

</dd></dl>

</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="PostImputation.html" class="btn btn-neutral float-left" title="PostImputation Module" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../visualization/manhattan_type.html" class="btn btn-neutral float-right" title="Manhattan-type Plots Module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Luis Giraldo Gonzalez Ricardo.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>