{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import scipy.stats as stats\n",
    "\n",
    "import sys\n",
    "sys.path.append('/mnt/0A2AAC152AABFBB7/CGE/luxgiant-dstream')\n",
    "from luxgiant_dstream.gwas import GWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_gene = pd.read_csv(\"/home/luis/data/testDownStream/outputData/snps_top_hits.txt\", sep=\"\\t\")\n",
    "#df_gwas = pd.read_csv(\"/mnt/0A2AAC152AABFBB7/CGE/luxgiant-dstream/data/luxgiant_manhattan_input_final.txt\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"{'ïds': ['rs1462855', 'rs1775153', 'rs823116', 'rs12740609', 'rs74990530', 'rs61835654', 'rs6741007', 'rs4613239', 'rs13033661', 'rs6808178', 'rs6764058', 'rs34822404', 'rs3775146', 'rs6599389', 'rs6815138', 'rs34311866', 'rs4698412', 'rs1348207', 'rs7695720', 'rs72657707', 'rs367740068', 'rs356182', 'rs114685218', 'rs356213', 'rs10014396', 'rs1372518', 'rs79214011', 'rs57175349', 'rs114018162', 'rs1867598', 'rs41286192', 'rs858290', 'rs665980', 'rs1536072', 'rs4909940', 'rs329669', 'rs11223627', 'rs2708419', 'rs76904798', 'rs73109570', 'rs11060180', 'rs10847864', 'rs4341693', 'rs12934900', 'rs2285582', 'rs55938136', 'rs62061848', 'rs199452', 'rs199500', 'rs12456492', 'rs9975976', 'rs28638318']}\""
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snps_dict = {\"ïds\": df_gene['SNP'].to_list()}\n",
    "data_1 = str(snps_dict)\n",
    "data_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "\n",
    "params = json.load(open(\"//home/luis/data/testDownStream/configFiles/parameters.JSON\"))\n",
    "\n",
    "gwas = GWAS(\n",
    "    input_path='/home/luis/data/testDownStream/inputData',\n",
    "    input_name=\"subset_data_luxgiant\",\n",
    "    output_path='/home/luis/data/testDownStream/outputData',\n",
    "    output_name=\"results\",\n",
    "    config_dict=params,\n",
    "    dependables_path='/home/luis/data/testDownStream/dependables',\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "plink --bfile /home/luis/data/testDownStream/outputData/gwas_analysis/results --chr 1-22 --maf 0.05 --geno 0.1  --hwe 5e-08 --exclude /home/luis/data/testDownStream/dependables/high-LD-regions.txt --range --indep-pairwise 50 5 0.2 --threads 30 --make-bed --out /home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning\n",
      "PLINK v1.90b7.4 64-bit (18 Aug 2024)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning.log.\n",
      "Options in effect:\n",
      "  --bfile /home/luis/data/testDownStream/outputData/gwas_analysis/results\n",
      "  --chr 1-22\n",
      "  --exclude /home/luis/data/testDownStream/dependables/high-LD-regions.txt\n",
      "  --geno 0.1\n",
      "  --hwe 5e-08\n",
      "  --indep-pairwise 50 5 0.2\n",
      "  --maf 0.05\n",
      "  --make-bed\n",
      "  --out /home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning\n",
      "  --range\n",
      "  --threads 30\n",
      "\n",
      "Note: --range flag deprecated.  Use e.g. \"--extract range <filename>\".\n",
      "63927 MB RAM detected; reserving 31963 MB for main workspace.\n",
      "23169127 variants loaded from .bim file.\n",
      "466 people (322 males, 144 females) loaded from .fam.\n",
      "466 phenotype values loaded from .fam.\n",
      "--exclude range: 499022 variants excluded.\n",
      "--exclude range: 22670105 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 466 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is exactly 1.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "16865325 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "5804780 variants and 466 people pass filters and QC.\n",
      "Among remaining phenotypes, 198 are cases and 268 are controls.\n",
      "--make-bed to\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning.bed +\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning.bim +\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Pruned 430151 variants from chromosome 1, leaving 25018.\n",
      "Pruned 441787 variants from chromosome 2, leaving 24705.\n",
      "Pruned 395251 variants from chromosome 3, leaving 22420.\n",
      "Pruned 413631 variants from chromosome 4, leaving 22394.\n",
      "Pruned 349087 variants from chromosome 5, leaving 19566.\n",
      "Pruned 331786 variants from chromosome 6, leaving 18730.\n",
      "Pruned 331775 variants from chromosome 7, leaving 18835.\n",
      "Pruned 290244 variants from chromosome 8, leaving 15886.\n",
      "Pruned 242925 variants from chromosome 9, leaving 14408.\n",
      "Pruned 290505 variants from chromosome 10, leaving 16817.\n",
      "Pruned 271402 variants from chromosome 11, leaving 15157.\n",
      "Pruned 268082 variants from chromosome 12, leaving 16228.\n",
      "Pruned 207362 variants from chromosome 13, leaving 12020.\n",
      "Pruned 179107 variants from chromosome 14, leaving 10590.\n",
      "Pruned 160224 variants from chromosome 15, leaving 9873.\n",
      "Pruned 169512 variants from chromosome 16, leaving 10957.\n",
      "Pruned 143624 variants from chromosome 17, leaving 9827.\n",
      "Pruned 159419 variants from chromosome 18, leaving 9964.\n",
      "Pruned 126247 variants from chromosome 19, leaving 8312.\n",
      "Pruned 123823 variants from chromosome 20, leaving 8648.\n",
      "Pruned 80240 variants from chromosome 21, leaving 5244.\n",
      "Pruned 77417 variants from chromosome 22, leaving 5580.\n",
      "Pruning complete.  5483601 of 5804780 variants removed.\n",
      "Marker lists written to\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning.prune.in\n",
      "and\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning.prune.out\n",
      ".\n",
      "\n",
      "plink2 --bfile /home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning --extract /home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning.prune.in --make-bed --out /home/luis/data/testDownStream/outputData/gwas_analysis/results_LDpruned --threads 30\n",
      "PLINK v2.00a6LM 64-bit Intel (18 Aug 2024)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/luis/data/testDownStream/outputData/gwas_analysis/results_LDpruned.log.\n",
      "Options in effect:\n",
      "  --bfile /home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning\n",
      "  --extract /home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning.prune.in\n",
      "  --make-bed\n",
      "  --out /home/luis/data/testDownStream/outputData/gwas_analysis/results_LDpruned\n",
      "  --threads 30\n",
      "\n",
      "Start time: Thu Aug 29 15:07:10 2024\n",
      "63927 MiB RAM detected, ~60603 available; reserving 31963 MiB for main\n",
      "workspace.\n",
      "Using up to 30 threads (change this with --threads).\n",
      "466 samples (144 females, 322 males; 466 founders) loaded from\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning.fam.\n",
      "5804780 variants loaded from\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_prunning.bim.\n",
      "1 binary phenotype loaded (198 cases, 268 controls).\n",
      "--extract: 321179 variants remaining.\n",
      "321179 variants remaining after main filters.\n",
      "Writing\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_LDpruned.fam\n",
      "... done.\n",
      "Writing\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_LDpruned.bim\n",
      "... done.\n",
      "Writing\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_LDpruned.bed\n",
      "... 10111213141516171819202123242526272829303132333435373940414243444546474849505253545556575859606263646566676869717273747576777980818283858687899091929395969799done.\n",
      "End time: Thu Aug 29 15:07:11 2024\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'pass': True,\n",
       " 'step': 'ld_prune',\n",
       " 'output': {'plink_out': '/home/luis/data/testDownStream/outputData/gwas_analysis'}}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas.exclude_high_ld_hla()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.4 64-bit (18 Aug 2024)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/luis/data/testDownStream/outputData/gwas_analysis/results_pca.log.\n",
      "Options in effect:\n",
      "  --bfile /home/luis/data/testDownStream/outputData/gwas_analysis/results_LDpruned\n",
      "  --out /home/luis/data/testDownStream/outputData/gwas_analysis/results_pca\n",
      "  --pca 10\n",
      "  --threads 30\n",
      "\n",
      "63927 MB RAM detected; reserving 31963 MB for main workspace.\n",
      "321179 variants loaded from .bim file.\n",
      "466 people (322 males, 144 females) loaded from .fam.\n",
      "466 phenotype values loaded from .fam.\n",
      "Using up to 30 threads (change this with --threads).\n",
      "Before main variant filters, 466 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is exactly 1.\n",
      "321179 variants and 466 people pass filters and QC.\n",
      "Among remaining phenotypes, 198 are cases and 268 are controls.\n",
      "Relationship matrix calculation complete.\n",
      "--pca: Results saved to and eigenvectors]\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_pca.eigenval\n",
      "and\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_pca.eigenvec .\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'pass': True,\n",
       " 'step': 'pca_decomposition',\n",
       " 'output': {'plink_out': '/home/luis/data/testDownStream/outputData/gwas_analysis'}}"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas.pca_decomposition()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (18 Aug 2024)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/luis/data/testDownStream/outputData/gwas_analysis/results_glm.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --bfile /home/luis/data/testDownStream/inputData/subset_data_luxgiant\n",
      "  --ci 0.95\n",
      "  --covar /home/luis/data/testDownStream/outputData/gwas_analysis/results_pca.eigenvec\n",
      "  --glm hide-covar omit-ref sex cols=+a1freq,+beta\n",
      "  --hwe 5e-08\n",
      "  --maf 0.05\n",
      "  --mind 0.1\n",
      "  --out /home/luis/data/testDownStream/outputData/gwas_analysis/results_glm\n",
      "  --threads 30\n",
      "\n",
      "Start time: Thu Aug 29 15:07:48 2024\n",
      "63927 MiB RAM detected, ~60654 available; reserving 31963 MiB for main\n",
      "workspace.\n",
      "Using up to 30 threads (change this with --threads).\n",
      "466 samples (144 females, 322 males; 466 founders) loaded from\n",
      "/home/luis/data/testDownStream/inputData/subset_data_luxgiant.fam.\n",
      "23169127 variants loaded from\n",
      "/home/luis/data/testDownStream/inputData/subset_data_luxgiant.bim.\n",
      "1 binary phenotype loaded (198 cases, 268 controls).\n",
      "Calculating sample missingness rates... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "0 samples removed due to missing genotype data (--mind).\n",
      "10 covariates loaded from /home/luis/data/testDownStream/outputData/gwas_analysis/results_pca.eigenvec.\n",
      "466 samples (144 females, 322 males; 466 founders) remaining after main\n",
      "filters.\n",
      "198 cases and 268 controls remaining after main filters.\n",
      "Calculating allele frequencies... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test (founders only).\n",
      "17235930 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "5933197 variants remaining after main filters.\n",
      "--glm logistic-Firth hybrid regression on phenotype 'PHENO1': 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "Results written to /home/luis/data/testDownStream/outputData/gwas_analysis/results_glm.PHENO1.glm.logistic.hybrid .\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.04884.\n",
      "--adjust values (5933197 tests) written to\n",
      "/home/luis/data/testDownStream/outputData/gwas_analysis/results_glm.PHENO1.glm.logistic.hybrid.adjusted\n",
      ".\n",
      "End time: Thu Aug 29 15:08:22 2024\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'pass': True,\n",
       " 'step': 'association_analysis',\n",
       " 'output': {'plink_out': '/home/luis/data/testDownStream/outputData/gwas_analysis'}}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas.fixed_model_association_analysis()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 15:09:54 CEST on Thu Aug 29 2024.\n",
      "Hostname: IKEAB-TS-P3\n",
      "\n",
      "Accepted options:\n",
      "--bfile /home/luis/data/testDownStream/inputData/subset_data_luxgiant\n",
      "--maf 0.05\n",
      "--cojo-slct\n",
      "--cojo-file /home/luis/data/testDownStream/outputData/gwas_analysis/cojo_file.ma\n",
      "--out /home/luis/data/testDownStream/outputData/gwas_analysis/cojo_file\n",
      "--thread-num 30\n",
      "\n",
      "\n",
      "Reading PLINK FAM file from [/home/luis/data/testDownStream/inputData/subset_data_luxgiant.fam].\n",
      "466 individuals to be included from [/home/luis/data/testDownStream/inputData/subset_data_luxgiant.fam].\n",
      "Reading PLINK BIM file from [/home/luis/data/testDownStream/inputData/subset_data_luxgiant.bim].\n",
      "23169127 SNPs to be included from [/home/luis/data/testDownStream/inputData/subset_data_luxgiant.bim].\n",
      "Reading PLINK BED file from [/home/luis/data/testDownStream/inputData/subset_data_luxgiant.bed] in SNP-major format ...\n",
      "Genotype data for 466 individuals and 23169127 SNPs to be included from [/home/luis/data/testDownStream/inputData/subset_data_luxgiant.bed].\n",
      "Calculating allele frequencies ...\n",
      "Filtering SNPs with MAF > 0.05 ...\n",
      "After filtering SNPs with MAF > 0.05, there are 5933197 SNPs (17235930 SNPs with MAF < 0.05).\n",
      "\n",
      "Reading GWAS summary-level statistics from [/home/luis/data/testDownStream/outputData/gwas_analysis/cojo_file.ma] ...\n",
      "GWAS summary statistics of 5933197 SNPs read from [/home/luis/data/testDownStream/outputData/gwas_analysis/cojo_file.ma].\n",
      "Phenotypic variance estimated from summary statistics of all 5933197 SNPs: 4.4731 (variance of logit for case-control studies).\n",
      "Matching the GWAS meta-analysis results to the genotype data ...\n",
      "5933197 SNPs are matched to the genotype data.\n",
      "Calculating the variance of SNP genotypes ...\n",
      "\n",
      "Performing stepwise model selection on 5933197 SNPs to select association signals ... (p cutoff = 5e-08; collinearity cutoff = 0.9)\n",
      "(Assuming complete linkage equilibrium between SNPs which are more than 10Mb away from each other)\n",
      "No SNPs have been selected.\n",
      "\n",
      "Analysis finished at 15:13:18 CEST on Thu Aug 29 2024\n",
      "Overall computational time: 3 minutes 24 sec.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'pass': True,\n",
       " 'step': 'get_top_hits',\n",
       " 'output': {'plink_out': '/home/luis/data/testDownStream/outputData/gwas_analysis'}}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas.get_top_hits()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting context for 1:205708020\n",
      "Overlap no, distance\n",
      "Minimum reached for gene NUCKS1 at 4802\n",
      "4802\n",
      "Getting context for 1:226728477\n",
      "Overlap found\n",
      "Getting context for 4:958159\n",
      "Overlap found\n",
      "Getting context for 4:89704960\n",
      "Overlap found\n",
      "Getting context for 4:89835399\n",
      "Overlap found\n",
      "Getting context for 6:29964911\n",
      "Overlap no, distance\n",
      "Minimum reached for gene HLA-A at 15339\n",
      "15339\n",
      "Getting context for 6:32616036\n",
      "Overlap no, distance\n",
      "Minimum reached for gene HLA-DQA1 at 12143\n",
      "12143\n",
      "Getting context for 11:10521749\n",
      "Overlap found\n",
      "Getting context for 15:25662486\n",
      "Overlap no, distance\n",
      "Minimum reached for gene ATP10A at 14787\n",
      "14787\n",
      "Getting context for 17:45495053\n",
      "Overlap no, distance\n",
      "Minimum reached for gene PLEKHM1 at 4295\n",
      "4295\n",
      "Getting context for 17:62016181\n",
      "Overlap found\n",
      "Getting context for 22:41196673\n",
      "Overlap no, distance\n",
      "Minimum reached for gene L3MBTL2 at 8609\n",
      "8609\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>SNP</th>\n",
       "      <th>bp</th>\n",
       "      <th>context</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>rs3747973</td>\n",
       "      <td>205708020</td>\n",
       "      <td>NUCKS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>rs74990530</td>\n",
       "      <td>226728477</td>\n",
       "      <td>ITPKB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>rs34311866</td>\n",
       "      <td>958159</td>\n",
       "      <td>TMEM175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>rs356182</td>\n",
       "      <td>89704960</td>\n",
       "      <td>SNCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>rs7681440</td>\n",
       "      <td>89835399</td>\n",
       "      <td>SNCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>rs2517680</td>\n",
       "      <td>29964911</td>\n",
       "      <td>HLA-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>rs1846190</td>\n",
       "      <td>32616036</td>\n",
       "      <td>HLA-DQA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>11</td>\n",
       "      <td>rs4909940</td>\n",
       "      <td>10521749</td>\n",
       "      <td>RNF141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>15</td>\n",
       "      <td>rs528813377</td>\n",
       "      <td>25662486</td>\n",
       "      <td>ATP10A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>17</td>\n",
       "      <td>rs56328224</td>\n",
       "      <td>45495053</td>\n",
       "      <td>PLEKHM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>17</td>\n",
       "      <td>rs72843781</td>\n",
       "      <td>62016181</td>\n",
       "      <td>MED13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>22</td>\n",
       "      <td>rs2092563</td>\n",
       "      <td>41196673</td>\n",
       "      <td>L3MBTL2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Chr          SNP         bp   context\n",
       "0     1    rs3747973  205708020    NUCKS1\n",
       "1     1   rs74990530  226728477     ITPKB\n",
       "2     4   rs34311866     958159   TMEM175\n",
       "3     4     rs356182   89704960      SNCA\n",
       "4     4    rs7681440   89835399      SNCA\n",
       "5     6    rs2517680   29964911     HLA-A\n",
       "6     6    rs1846190   32616036  HLA-DQA1\n",
       "7    11    rs4909940   10521749    RNF141\n",
       "8    15  rs528813377   25662486    ATP10A\n",
       "9    17   rs56328224   45495053   PLEKHM1\n",
       "10   17   rs72843781   62016181     MED13\n",
       "11   22    rs2092563   41196673   L3MBTL2"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas.annotate_hits()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[]\n",
      "['NUCKS1']\n",
      "['NUCKS1', 'ITPKB']\n",
      "['NUCKS1', 'ITPKB', 'TMEM175']\n",
      "['NUCKS1', 'ITPKB', 'TMEM175', 'SNCA']\n",
      "['NUCKS1', 'ITPKB', 'TMEM175', 'SNCA', 'SNCA']\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'transcript_consequences'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[3], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m snps, genes \u001b[38;5;241m=\u001b[39m \u001b[43mgwas\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mannotate_hits\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      3\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mSNPs\u001b[39m\u001b[38;5;124m'\u001b[39m, snps)\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGenes\u001b[39m\u001b[38;5;124m'\u001b[39m, genes)\n",
      "File \u001b[0;32m~/CGE/luxgiant-dstream/luxgiant_dstream/gwas.py:296\u001b[0m, in \u001b[0;36mGWAS.annotate_hits\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    294\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m k \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;241m0\u001b[39m, \u001b[38;5;28mlen\u001b[39m(decoded)):\n\u001b[1;32m    295\u001b[0m     \u001b[38;5;28mprint\u001b[39m(genes)\n\u001b[0;32m--> 296\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m dictionary \u001b[38;5;129;01min\u001b[39;00m \u001b[43mdecoded\u001b[49m\u001b[43m[\u001b[49m\u001b[43mk\u001b[49m\u001b[43m]\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mtranscript_consequences\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m:\n\u001b[1;32m    297\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgene_symbol\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;129;01min\u001b[39;00m dictionary\u001b[38;5;241m.\u001b[39mkeys():\n\u001b[1;32m    298\u001b[0m             gen_symbol \u001b[38;5;241m=\u001b[39m dictionary[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgene_symbol\u001b[39m\u001b[38;5;124m'\u001b[39m]\n",
      "\u001b[0;31mKeyError\u001b[0m: 'transcript_consequences'"
     ]
    }
   ],
   "source": [
    "snps, genes = gwas.annotate_hits()\n",
    "\n",
    "print('SNPs', snps)\n",
    "print('Genes', genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'colocated_variants': [{'strand': 1,\n",
       "    'seq_region_name': '1',\n",
       "    'end': 205708020,\n",
       "    'id': 'rs3747973',\n",
       "    'start': 205708020,\n",
       "    'phenotype_or_disease': 1,\n",
       "    'allele_string': 'A/C/G/T',\n",
       "    'pubmed': [36914875, 33462484, 34887591, 30595370, 32203549],\n",
       "    'frequencies': {'T': {'gnomadg_amr': 0,\n",
       "      'gnomadg_nfe': 0,\n",
       "      'gnomadg_afr': 0,\n",
       "      'gnomadg_asj': 0,\n",
       "      'gnomadg_oth': 0,\n",
       "      'gnomadg_ami': 0,\n",
       "      'gnomadg': 5.299e-05,\n",
       "      'gnomadg_sas': 0,\n",
       "      'gnomadg_fin': 0,\n",
       "      'gnomadg_mid': 0,\n",
       "      'gnomadg_eas': 0.001552},\n",
       "     'G': {'gnomadg_nfe': 0.5992,\n",
       "      'afr': 0.1868,\n",
       "      'gnomadg': 0.4857,\n",
       "      'gnomadg_sas': 0.5686,\n",
       "      'gnomadg_ami': 0.6148,\n",
       "      'af': 0.4397,\n",
       "      'gnomadg_afr': 0.2357,\n",
       "      'gnomadg_amr': 0.4637,\n",
       "      'eas': 0.499,\n",
       "      'gnomadg_oth': 0.5293,\n",
       "      'gnomadg_asj': 0.6431,\n",
       "      'amr': 0.4107,\n",
       "      'gnomadg_fin': 0.6246,\n",
       "      'gnomadg_eas': 0.5314,\n",
       "      'gnomadg_mid': 0.6424,\n",
       "      'sas': 0.5869,\n",
       "      'eur': 0.5895}}}],\n",
       "  'transcript_consequences': [{'consequence_terms': ['downstream_gene_variant'],\n",
       "    'transcript_id': 'ENST00000367142',\n",
       "    'hgnc_id': 'HGNC:29923',\n",
       "    'gene_id': 'ENSG00000069275',\n",
       "    'impact': 'MODIFIER',\n",
       "    'gene_symbol_source': 'HGNC',\n",
       "    'biotype': 'protein_coding',\n",
       "    'strand': -1,\n",
       "    'distance': 4802,\n",
       "    'variant_allele': 'C',\n",
       "    'gene_symbol': 'NUCKS1'},\n",
       "   {'transcript_id': 'ENST00000367142',\n",
       "    'consequence_terms': ['downstream_gene_variant'],\n",
       "    'hgnc_id': 'HGNC:29923',\n",
       "    'gene_id': 'ENSG00000069275',\n",
       "    'biotype': 'protein_coding',\n",
       "    'gene_symbol_source': 'HGNC',\n",
       "    'impact': 'MODIFIER',\n",
       "    'gene_symbol': 'NUCKS1',\n",
       "    'variant_allele': 'G',\n",
       "    'distance': 4802,\n",
       "    'strand': -1},\n",
       "   {'biotype': 'protein_coding',\n",
       "    'impact': 'MODIFIER',\n",
       "    'gene_symbol_source': 'HGNC',\n",
       "    'strand': -1,\n",
       "    'variant_allele': 'T',\n",
       "    'gene_symbol': 'NUCKS1',\n",
       "    'distance': 4802,\n",
       "    'consequence_terms': ['downstream_gene_variant'],\n",
       "    'hgnc_id': 'HGNC:29923',\n",
       "    'transcript_id': 'ENST00000367142',\n",
       "    'gene_id': 'ENSG00000069275'}],\n",
       "  'allele_string': 'A/C/G/T',\n",
       "  'strand': 1,\n",
       "  'seq_region_name': '1',\n",
       "  'id': 'rs3747973',\n",
       "  'start': 205708020,\n",
       "  'most_severe_consequence': 'downstream_gene_variant',\n",
       "  'input': 'rs3747973',\n",
       "  'end': 205708020,\n",
       "  'assembly_name': 'GRCh38'}]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import requests, sys\n",
    " \n",
    "server = \"https://rest.ensembl.org\"\n",
    "ext = \"/vep/human/id/rs3747973?\"\n",
    " \n",
    "r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n",
    " \n",
    "if not r.ok:\n",
    "  r.raise_for_status()\n",
    "  sys.exit()\n",
    " \n",
    "decoded = r.json()\n",
    "decoded"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'colocated_variants': [{'strand': 1,\n",
       "    'end': 29964911,\n",
       "    'allele_string': 'G/A',\n",
       "    'frequencies': {'A': {'gnomadg_oth': 0.1155,\n",
       "      'gnomadg_mid': 0.1297,\n",
       "      'gnomadg_ami': 0.03956,\n",
       "      'afr': 0.121,\n",
       "      'gnomadg_nfe': 0.07318,\n",
       "      'af': 0.1174,\n",
       "      'gnomadg': 0.09384,\n",
       "      'gnomadg_amr': 0.1232,\n",
       "      'sas': 0.1299,\n",
       "      'gnomadg_asj': 0.1106,\n",
       "      'gnomadg_fin': 0.03458,\n",
       "      'amr': 0.1326,\n",
       "      'eur': 0.0696,\n",
       "      'gnomadg_afr': 0.1198,\n",
       "      'gnomadg_sas': 0.1526,\n",
       "      'eas': 0.1379,\n",
       "      'gnomadg_eas': 0.1306}},\n",
       "    'start': 29964911,\n",
       "    'seq_region_name': '6',\n",
       "    'id': 'rs2517680'}],\n",
       "  'id': 'rs2517680',\n",
       "  'intergenic_consequences': [{'impact': 'MODIFIER',\n",
       "    'consequence_terms': ['intergenic_variant'],\n",
       "    'variant_allele': 'A'}],\n",
       "  'input': 'rs2517680',\n",
       "  'seq_region_name': '6',\n",
       "  'most_severe_consequence': 'intergenic_variant',\n",
       "  'allele_string': 'G/A',\n",
       "  'start': 29964911,\n",
       "  'strand': 1,\n",
       "  'assembly_name': 'GRCh38',\n",
       "  'end': 29964911},\n",
       " {'strand': 1,\n",
       "  'start': 1222624,\n",
       "  'end': 1222624,\n",
       "  'assembly_name': 'GRCh38',\n",
       "  'seq_region_name': 'HSCHR6_MHC_APD_CTG1',\n",
       "  'intergenic_consequences': [{'impact': 'MODIFIER',\n",
       "    'variant_allele': 'A',\n",
       "    'consequence_terms': ['intergenic_variant']}],\n",
       "  'input': 'rs2517680',\n",
       "  'id': 'rs2517680',\n",
       "  'allele_string': 'G/A',\n",
       "  'most_severe_consequence': 'intergenic_variant'},\n",
       " {'assembly_name': 'GRCh38',\n",
       "  'end': 1444299,\n",
       "  'start': 1444299,\n",
       "  'strand': 1,\n",
       "  'most_severe_consequence': 'intergenic_variant',\n",
       "  'allele_string': 'G/A',\n",
       "  'id': 'rs2517680',\n",
       "  'input': 'rs2517680',\n",
       "  'intergenic_consequences': [{'impact': 'MODIFIER',\n",
       "    'variant_allele': 'A',\n",
       "    'consequence_terms': ['intergenic_variant']}],\n",
       "  'seq_region_name': 'HSCHR6_MHC_COX_CTG1'},\n",
       " {'seq_region_name': 'HSCHR6_MHC_DBB_CTG1',\n",
       "  'intergenic_consequences': [{'consequence_terms': ['intergenic_variant'],\n",
       "    'variant_allele': 'A',\n",
       "    'impact': 'MODIFIER'}],\n",
       "  'input': 'rs2517680',\n",
       "  'id': 'rs2517680',\n",
       "  'most_severe_consequence': 'intergenic_variant',\n",
       "  'allele_string': 'G/A',\n",
       "  'start': 1220451,\n",
       "  'strand': 1,\n",
       "  'end': 1220451,\n",
       "  'assembly_name': 'GRCh38'},\n",
       " {'most_severe_consequence': 'intergenic_variant',\n",
       "  'allele_string': 'A/G',\n",
       "  'id': 'rs2517680',\n",
       "  'intergenic_consequences': [{'impact': 'MODIFIER',\n",
       "    'variant_allele': 'G',\n",
       "    'consequence_terms': ['intergenic_variant']}],\n",
       "  'input': 'rs2517680',\n",
       "  'seq_region_name': 'HSCHR6_MHC_MANN_CTG1',\n",
       "  'assembly_name': 'GRCh38',\n",
       "  'end': 1225648,\n",
       "  'start': 1225648,\n",
       "  'strand': 1},\n",
       " {'start': 1308952,\n",
       "  'strand': 1,\n",
       "  'end': 1308952,\n",
       "  'assembly_name': 'GRCh38',\n",
       "  'input': 'rs2517680',\n",
       "  'intergenic_consequences': [{'impact': 'MODIFIER',\n",
       "    'variant_allele': 'A',\n",
       "    'consequence_terms': ['intergenic_variant']}],\n",
       "  'seq_region_name': 'HSCHR6_MHC_MCF_CTG1',\n",
       "  'id': 'rs2517680',\n",
       "  'most_severe_consequence': 'intergenic_variant',\n",
       "  'allele_string': 'G/A'},\n",
       " {'seq_region_name': 'HSCHR6_MHC_QBL_CTG1',\n",
       "  'intergenic_consequences': [{'impact': 'MODIFIER',\n",
       "    'variant_allele': 'A',\n",
       "    'consequence_terms': ['intergenic_variant']}],\n",
       "  'input': 'rs2517680',\n",
       "  'id': 'rs2517680',\n",
       "  'allele_string': 'G/A',\n",
       "  'most_severe_consequence': 'intergenic_variant',\n",
       "  'strand': 1,\n",
       "  'start': 1219894,\n",
       "  'end': 1219894,\n",
       "  'assembly_name': 'GRCh38'},\n",
       " {'input': 'rs2517680',\n",
       "  'intergenic_consequences': [{'consequence_terms': ['intergenic_variant'],\n",
       "    'variant_allele': 'A',\n",
       "    'impact': 'MODIFIER'}],\n",
       "  'seq_region_name': 'HSCHR6_MHC_SSTO_CTG1',\n",
       "  'id': 'rs2517680',\n",
       "  'most_severe_consequence': 'intergenic_variant',\n",
       "  'allele_string': 'G/A',\n",
       "  'start': 1262708,\n",
       "  'strand': 1,\n",
       "  'end': 1262708,\n",
       "  'assembly_name': 'GRCh38'}]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import requests, sys\n",
    " \n",
    "server = \"https://rest.ensembl.org\"\n",
    "ext = \"/vep/human/id/rs2517680?\"\n",
    " \n",
    "r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n",
    " \n",
    "if not r.ok:\n",
    "  r.raise_for_status()\n",
    "  sys.exit()\n",
    " \n",
    "decoded = r.json()\n",
    "decoded"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6\n",
      "29964911\n",
      "29964911\n"
     ]
    }
   ],
   "source": [
    "data_dict = decoded[0]\n",
    "print(data_dict.get('seq_region_name'))\n",
    "print(data_dict.get('start'))\n",
    "print(data_dict.get('end'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SNCA\n"
     ]
    }
   ],
   "source": [
    "for dictionary in data_dict['transcript_consequences']:\n",
    "    if 'gene_symbol' in dictionary.keys():\n",
    "        print(dictionary['gene_symbol'])\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EFNA3\n",
      "SLC45A3\n",
      "NUCKS1\n",
      "SLC41A1\n"
     ]
    }
   ],
   "source": [
    "def fetch_snp_annot(snp):\n",
    "\n",
    "    server = \"https://rest.ensembl.org\"\n",
    "    ext = f\"/vep/human/id/rs{snp}?\"\n",
    "\n",
    "    r = requests.get(server+ext, headers={ \"Content-Type\" : \"application/json\"})\n",
    " \n",
    "    if not r.ok:\n",
    "      r.raise_for_status()\n",
    "      sys.exit()\n",
    " \n",
    "    decoded = r.json()\n",
    "    return decoded[0]['transcript_consequences'][0]['gene_symbol']\n",
    "\n",
    "for snp in snps:\n",
    "    print(fetch_snp_annot(snp))\n",
    "    time.sleep(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import requests, sys\n",
    " \n",
    "server = \"https://rest.ensembl.org\"\n",
    "ext = \"/vep/human/id\"\n",
    "headers={ \"Content-Type\" : \"application/json\", \"Accept\" : \"application/json\"}\n",
    "r = requests.post(server+ext, headers=headers, data='{ \"ids\" : [\"rs1462855\", \"rs1775153\", \"rs823116\", \"rs12740609\"] }')\n",
    " \n",
    "if not r.ok:\n",
    "  r.raise_for_status()\n",
    "  sys.exit()\n",
    " \n",
    "decoded = r.json()\n",
    "len(decoded)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NUCKS1'"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "decoded[2]['transcript_consequences'][0]['gene_symbol']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No overlapping genes found.\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "\n",
    "# Replace with your SNP ID\n",
    "snp_id = \"rs2517680\"\n",
    "\n",
    "# VEP API URL for the given SNP ID\n",
    "vep_url = f\"https://rest.ensembl.org/vep/human/id/{snp_id}?content-type=application/json\"\n",
    "\n",
    "# Send the GET request to the VEP endpoint\n",
    "vep_response = requests.get(vep_url)\n",
    "\n",
    "# Check if the request was successful\n",
    "if vep_response.status_code == 200:\n",
    "    vep_data = vep_response.json()\n",
    "\n",
    "    # Loop through the VEP results to find transcript consequences\n",
    "    found_gene = False\n",
    "    for variant in vep_data:\n",
    "        for consequence in variant.get('transcript_consequences', []):\n",
    "            gene_id = consequence.get('gene_id')\n",
    "            gene_symbol = consequence.get('gene_symbol')\n",
    "\n",
    "            # If gene_symbol is found, print it\n",
    "            if gene_symbol:\n",
    "                print(f\"Gene ID: {gene_id}, Gene Symbol: {gene_symbol}\")\n",
    "                found_gene = True\n",
    "\n",
    "    # If no gene symbol was found, check for overlapping genes using coordinates\n",
    "    if not found_gene:\n",
    "        # Extract coordinates from VEP output\n",
    "        chrom = vep_data[0].get('seq_region_name')  # Chromosome\n",
    "        start = vep_data[0].get('start')  # Start position\n",
    "        end = vep_data[0].get('end')  # End position\n",
    "\n",
    "        # Use the Overlap API to find any genes overlapping this region\n",
    "        overlap_url = f\"https://rest.ensembl.org/overlap/region/human/{chrom}:{start}-{end}?feature=gene;content-type=application/json\"\n",
    "        overlap_response = requests.get(overlap_url)\n",
    "\n",
    "        # Check if the overlap request was successful\n",
    "        if overlap_response.status_code == 200:\n",
    "            overlap_data = overlap_response.json()\n",
    "\n",
    "            if overlap_data:\n",
    "                print(\"Overlapping genes found:\")\n",
    "                for gene in overlap_data:\n",
    "                    print(f\"Gene ID: {gene['id']}, Gene Symbol: {gene.get('external_name', 'N/A')}\")\n",
    "            else:\n",
    "                print(\"No overlapping genes found.\")\n",
    "        else:\n",
    "            print(f\"Error fetching overlapping genes: {overlap_response.status_code}\")\n",
    "else:\n",
    "    print(f\"Error fetching VEP data: {vep_response.status_code}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No overlapping genes found, fetching nearest genes...\n",
      "https://rest.ensembl.org/overlap/region/human/6:29964911-29964911?feature=gene;nearest=1;content-type=application/json\n",
      "No upstream gene found.\n",
      "No downstream gene found.\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "\n",
    "# Replace with your SNP ID\n",
    "snp_id = \"rs2517680\"\n",
    "\n",
    "# VEP API URL for the given SNP ID\n",
    "vep_url = f\"https://rest.ensembl.org/vep/human/id/{snp_id}?content-type=application/json\"\n",
    "\n",
    "# Send the GET request to the VEP endpoint\n",
    "vep_response = requests.get(vep_url)\n",
    "\n",
    "# Check if the request was successful\n",
    "if vep_response.status_code == 200:\n",
    "    vep_data = vep_response.json()\n",
    "\n",
    "    # Loop through the VEP results to find transcript consequences\n",
    "    found_gene = False\n",
    "    for variant in vep_data:\n",
    "        for consequence in variant.get('transcript_consequences', []):\n",
    "            gene_id = consequence.get('gene_id')\n",
    "            gene_symbol = consequence.get('gene_symbol')\n",
    "\n",
    "            # If gene_symbol is found, print it\n",
    "            if gene_symbol:\n",
    "                print(f\"Gene ID: {gene_id}, Gene Symbol: {gene_symbol}\")\n",
    "                found_gene = True\n",
    "\n",
    "    # If no gene symbol was found, check for overlapping genes using coordinates\n",
    "    if not found_gene:\n",
    "        # Extract coordinates from VEP output\n",
    "        chrom = vep_data[0].get('seq_region_name')  # Chromosome\n",
    "        start = vep_data[0].get('start')  # Start position\n",
    "        end = vep_data[0].get('end')  # End position\n",
    "\n",
    "        # Use the Overlap API to find overlapping genes in this region\n",
    "        overlap_url = f\"https://rest.ensembl.org/overlap/region/human/{chrom}:{start}-{end}?feature=gene;content-type=application/json\"\n",
    "        overlap_response = requests.get(overlap_url)\n",
    "\n",
    "        # Check if the overlap request was successful\n",
    "        if overlap_response.status_code == 200:\n",
    "            overlap_data = overlap_response.json()\n",
    "\n",
    "            if overlap_data:\n",
    "                print(\"Overlapping genes found:\")\n",
    "                for gene in overlap_data:\n",
    "                    print(f\"Gene ID: {gene['id']}, Gene Symbol: {gene.get('external_name', 'N/A')}\")\n",
    "            else:\n",
    "                print(\"No overlapping genes found, fetching nearest genes...\")\n",
    "\n",
    "                # Fetch nearest upstream and downstream genes using nearest=1\n",
    "                nearest_url = f\"https://rest.ensembl.org/overlap/region/human/{chrom}:{start}-{end}?feature=gene;nearest=1;content-type=application/json\"\n",
    "                print(nearest_url)\n",
    "                nearest_response = requests.get(nearest_url)\n",
    "\n",
    "                if nearest_response.status_code == 200:\n",
    "                    nearest_data = nearest_response.json()\n",
    "\n",
    "                    upstream_gene = None\n",
    "                    downstream_gene = None\n",
    "\n",
    "                    # Loop through the nearest gene results to identify upstream and downstream\n",
    "                    for gene in nearest_data:\n",
    "                        gene_start = gene['start']\n",
    "                        gene_end = gene['end']\n",
    "                        gene_id = gene['id']\n",
    "                        gene_symbol = gene.get('external_name', 'N/A')\n",
    "\n",
    "                        if gene_end < start:  # Gene is upstream\n",
    "                            upstream_gene = {\n",
    "                                'id': gene_id,\n",
    "                                'symbol': gene_symbol,\n",
    "                                'distance': start - gene_end\n",
    "                            }\n",
    "                        elif gene_start > end:  # Gene is downstream\n",
    "                            downstream_gene = {\n",
    "                                'id': gene_id,\n",
    "                                'symbol': gene_symbol,\n",
    "                                'distance': gene_start - end\n",
    "                            }\n",
    "\n",
    "                    # Print the nearest upstream and downstream genes with distances\n",
    "                    if upstream_gene:\n",
    "                        print(f\"Nearest upstream gene: {upstream_gene['symbol']} (Gene ID: {upstream_gene['id']})\")\n",
    "                        print(f\"Distance to upstream gene: {upstream_gene['distance']} bases\")\n",
    "                    else:\n",
    "                        print(\"No upstream gene found.\")\n",
    "\n",
    "                    if downstream_gene:\n",
    "                        print(f\"Nearest downstream gene: {downstream_gene['symbol']} (Gene ID: {downstream_gene['id']})\")\n",
    "                        print(f\"Distance to downstream gene: {downstream_gene['distance']} bases\")\n",
    "                    else:\n",
    "                        print(\"No downstream gene found.\")\n",
    "                else:\n",
    "                    print(f\"Error fetching nearest genes: {nearest_response.status_code}\")\n",
    "        else:\n",
    "            print(f\"Error fetching overlapping genes: {overlap_response.status_code}\")\n",
    "else:\n",
    "    print(f\"Error fetching VEP data: {vep_response.status_code}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No upstream gene found.\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "\n",
    "# Replace with your SNP ID\n",
    "snp_id = \"rs1333049\"\n",
    "\n",
    "# Step 1: Get SNP coordinates from the VEP endpoint\n",
    "vep_url = f\"https://rest.ensembl.org/vep/human/id/{snp_id}?content-type=application/json\"\n",
    "vep_response = requests.get(vep_url)\n",
    "\n",
    "# Check if the request was successful\n",
    "if vep_response.status_code == 200:\n",
    "    vep_data = vep_response.json()\n",
    "    \n",
    "    # Extract coordinates from VEP output\n",
    "    chrom = vep_data[0].get('seq_region_name')  # Chromosome\n",
    "    start = vep_data[0].get('start')  # Start position\n",
    "    end = vep_data[0].get('end')  # End position\n",
    "    \n",
    "    # Step 2: Use the Overlap API to find the nearest upstream gene\n",
    "    nearest_url = f\"https://rest.ensembl.org/overlap/region/human/{chrom}:{start}-{end}?feature=gene;nearest=1;content-type=application/json\"\n",
    "    nearest_response = requests.get(nearest_url)\n",
    "\n",
    "    if nearest_response.status_code == 200:\n",
    "        nearest_data = nearest_response.json()\n",
    "\n",
    "        upstream_gene = None\n",
    "\n",
    "        # Step 3: Filter for the closest upstream gene (gene_end < SNP_start)\n",
    "        for gene in nearest_data:\n",
    "            gene_end = gene['end']\n",
    "            gene_start = gene['start']\n",
    "            gene_id = gene['id']\n",
    "            gene_symbol = gene.get('external_name', 'N/A')\n",
    "\n",
    "            if gene_end < start:  # Gene is upstream\n",
    "                upstream_gene = {\n",
    "                    'id': gene_id,\n",
    "                    'symbol': gene_symbol,\n",
    "                    'distance': start - gene_end\n",
    "                }\n",
    "                break  # Stop after finding the closest upstream gene\n",
    "\n",
    "        if upstream_gene:\n",
    "            print(f\"Nearest upstream gene: {upstream_gene['symbol']} (Gene ID: {upstream_gene['id']})\")\n",
    "            print(f\"Distance to upstream gene: {upstream_gene['distance']} bases\")\n",
    "        else:\n",
    "            print(\"No upstream gene found.\")\n",
    "    else:\n",
    "        print(f\"Error fetching nearest genes: {nearest_response.status_code}\")\n",
    "else:\n",
    "    print(f\"Error fetching VEP data: {vep_response.status_code}\")\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
