

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>GWAS with Generalized Linear Mixed Models (GLMM) &mdash; ideal-genom 0.1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=e59714d7" />

  
      <script src="../_static/jquery.js?v=5d32c60e"></script>
      <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../_static/documentation_options.js?v=01f34227"></script>
      <script src="../_static/doctools.js?v=9bcbadda"></script>
      <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="PostImputation Module" href="../preprocessing/PostImputation.html" />
    <link rel="prev" title="GWAS with Generalized Linear Models (GLM)" href="gwas_glm.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            ideal-genom
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="gwas_glm.html">GWAS with Generalized Linear Models (GLM)</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">GWAS with Generalized Linear Mixed Models (GLMM)</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.gwas.gwas_glmm.GWASrandom"><code class="docutils literal notranslate"><span class="pre">GWASrandom</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.gwas.gwas_glmm.GWASrandom.__init__"><code class="docutils literal notranslate"><span class="pre">GWASrandom.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.gwas.gwas_glmm.GWASrandom.annotate_top_hits"><code class="docutils literal notranslate"><span class="pre">GWASrandom.annotate_top_hits()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.gwas.gwas_glmm.GWASrandom.compute_grm"><code class="docutils literal notranslate"><span class="pre">GWASrandom.compute_grm()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.gwas.gwas_glmm.GWASrandom.get_top_hits"><code class="docutils literal notranslate"><span class="pre">GWASrandom.get_top_hits()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.gwas.gwas_glmm.GWASrandom.prepare_aux_files"><code class="docutils literal notranslate"><span class="pre">GWASrandom.prepare_aux_files()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.gwas.gwas_glmm.GWASrandom.run_gwas_glmm"><code class="docutils literal notranslate"><span class="pre">GWASrandom.run_gwas_glmm()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../preprocessing/PostImputation.html">PostImputation Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="../preprocessing/preparatory.html">Preparatory Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="../annotations.html">ideal_genom.annotations</a></li>
<li class="toctree-l1"><a class="reference internal" href="../api_client.html">ideal_genom.api_client</a></li>
<li class="toctree-l1"><a class="reference internal" href="../get_examples.html">ideal_genom.get_examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="../get_references.html">ideal_genom.get_references</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Helpers.html">ideal_genom.Helpers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../power_comp.html">ideal_genom.power_comp</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">ideal-genom</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">GWAS with Generalized Linear Mixed Models (GLMM)</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/gwas/gwas_glmm.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-ideal_genom.gwas.gwas_glmm">
<span id="gwas-with-generalized-linear-mixed-models-glmm"></span><h1>GWAS with Generalized Linear Mixed Models (GLMM)<a class="headerlink" href="#module-ideal_genom.gwas.gwas_glmm" title="Link to this heading"></a></h1>
<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.gwas.gwas_glmm.GWASrandom">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.gwas.gwas_glmm.</span></span><span class="sig-name descname"><span class="pre">GWASrandom</span></span><a class="headerlink" href="#ideal_genom.gwas.gwas_glmm.GWASrandom" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.gwas.gwas_glmm.GWASrandom.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">input_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recompute</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.gwas.gwas_glmm.GWASrandom.__init__" title="Link to this definition"></a></dt>
<dd><p>Initializes the GWAS analysis with a Generalized Linear Mixed Model.</p>
<section id="parameters">
<h2>Parameters:<a class="headerlink" href="#parameters" title="Link to this heading"></a></h2>
<dl class="simple">
<dt>input_path (str):</dt><dd><p>Path to the input directory containing PLINK files.</p>
</dd>
<dt>input_name (str):</dt><dd><p>Base name of the input PLINK files (without extensions).</p>
</dd>
<dt>output_path (str):</dt><dd><p>Path to the output directory where results will be saved.</p>
</dd>
<dt>output_name (str):</dt><dd><p>Base name for the output files.</p>
</dd>
<dt>recompute (bool):</dt><dd><p>Flag indicating whether to recompute the analysis if results already exist. Default is True.</p>
</dd>
</dl>
</section>
<section id="raises">
<h2>Raises:<a class="headerlink" href="#raises" title="Link to this heading"></a></h2>
<p>ValueError: If input_path, output_path, input_name, or output_name are not provided.
FileNotFoundError: If the specified input_path or output_path does not exist.
FileNotFoundError: If the required PLINK files (.bed, .bim, .fam) are not found in the input_path.
TypeError: If input_name or output_name are not strings, or if recompute is not a boolean.</p>
</section>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.gwas.gwas_glmm.GWASrandom.annotate_top_hits">
<span class="sig-name descname"><span class="pre">annotate_top_hits</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gtf_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'38'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">anno_source</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'ensembl'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span></span></span><a class="headerlink" href="#ideal_genom.gwas.gwas_glmm.GWASrandom.annotate_top_hits" title="Link to this definition"></a></dt>
<dd><p>Annotate the top hits from the association analysis.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.gwas.gwas_glmm.GWASrandom.compute_grm">
<span class="sig-name descname"><span class="pre">compute_grm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">max_threads</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span></span></span><a class="headerlink" href="#ideal_genom.gwas.gwas_glmm.GWASrandom.compute_grm" title="Link to this definition"></a></dt>
<dd><p>Compute the Genetic Relationship Matrix (GRM) using GCTA software.</p>
<p>This method computes the GRM for the given input data using the GCTA software. It allows for multi-threaded execution and can optionally recompute the GRM if specified.</p>
<section id="id1">
<h2>Parameters:<a class="headerlink" href="#id1" title="Link to this heading"></a></h2>
<dl class="simple">
<dt>max_threads (int, optional):</dt><dd><p>The maximum number of threads to use for computation. If not specified, it defaults to the number of available CPU cores minus two. If the number of CPU cores cannot be determined, it defaults to 10.</p>
</dd>
</dl>
</section>
<section id="returns">
<h2>Returns:<a class="headerlink" href="#returns" title="Link to this heading"></a></h2>
<blockquote>
<div><p>dict: A dictionary containing the following keys:
- ‘pass’ (bool): Indicates whether the process completed successfully.
- ‘step’ (str): The name of the step performed (‘compute_grm’).
- ‘output’ (dict): A dictionary containing the output file paths with the key ‘gcta_out’.</p>
</div></blockquote>
</section>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.gwas.gwas_glmm.GWASrandom.get_top_hits">
<span class="sig-name descname"><span class="pre">get_top_hits</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">maf</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.01</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span></span></span><a class="headerlink" href="#ideal_genom.gwas.gwas_glmm.GWASrandom.get_top_hits" title="Link to this definition"></a></dt>
<dd><p>Get the top hits from the GWAS results.</p>
<p>This function processes the results of a genome-wide association study (GWAS) to identify the top hits based. It prepares the necessary files and optionally recomputes the results using GCTA.</p>
<section id="id2">
<h2>Parameters:<a class="headerlink" href="#id2" title="Link to this heading"></a></h2>
<dl class="simple">
<dt>maf (float, optional):</dt><dd><p>Minor allele frequency threshold. Default is 0.01. Must be between 0 and 1.</p>
</dd>
</dl>
</section>
<section id="id3">
<h2>Returns:<a class="headerlink" href="#id3" title="Link to this heading"></a></h2>
<dl class="simple">
<dt>dict:</dt><dd><p>A dictionary containing the status of the process, the step name, and the output directory.</p>
</dd>
</dl>
</section>
<section id="id4">
<h2>Raises:<a class="headerlink" href="#id4" title="Link to this heading"></a></h2>
<blockquote>
<div><p>TypeError: If <cite>maf</cite> is not of type float.
ValueError: If <cite>maf</cite> is not between 0 and 1.</p>
</div></blockquote>
</section>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.gwas.gwas_glmm.GWASrandom.prepare_aux_files">
<span class="sig-name descname"><span class="pre">prepare_aux_files</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span></span></span><a class="headerlink" href="#ideal_genom.gwas.gwas_glmm.GWASrandom.prepare_aux_files" title="Link to this definition"></a></dt>
<dd><p>Prepares auxiliary files for GWAS analysis by processing phenotype and sex data.</p>
<p>This function reads a .fam file, extracts and recodes phenotype and sex information, and writes the processed data to new files in the specified results directory.</p>
<section id="id5">
<h2>Returns:<a class="headerlink" href="#id5" title="Link to this heading"></a></h2>
<dl class="simple">
<dt>dict:</dt><dd><p>A dictionary containing the status of the process, the step name, and the output directory.</p>
</dd>
</dl>
</section>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.gwas.gwas_glmm.GWASrandom.run_gwas_glmm">
<span class="sig-name descname"><span class="pre">run_gwas_glmm</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">maf</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.01</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span></span></span><a class="headerlink" href="#ideal_genom.gwas.gwas_glmm.GWASrandom.run_gwas_glmm" title="Link to this definition"></a></dt>
<dd><p>Runs a Genome-Wide Association Study (GWAS) using a generalized linear mixed model (GLMMM).</p>
<section id="id6">
<h2>Parameters:<a class="headerlink" href="#id6" title="Link to this heading"></a></h2>
<dl class="simple">
<dt>maf (float):</dt><dd><p>Minor allele frequency threshold for filtering SNPs. Default is 0.01.</p>
</dd>
</dl>
</section>
<section id="id7">
<h2>Returns:<a class="headerlink" href="#id7" title="Link to this heading"></a></h2>
<dl class="simple">
<dt>dict:</dt><dd><p>A dictionary containing the status of the process, the step name, and the output directory.</p>
</dd>
</dl>
</section>
<section id="id8">
<h2>Raises:<a class="headerlink" href="#id8" title="Link to this heading"></a></h2>
<blockquote>
<div><p>TypeError: If <cite>maf</cite> is not of type float.
ValueError: If <cite>maf</cite> is not between 0 and 1.
FileExistsError: If required input files are not found in the results directory.</p>
</div></blockquote>
</section>
</dd></dl>

</dd></dl>

</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="gwas_glm.html" class="btn btn-neutral float-left" title="GWAS with Generalized Linear Models (GLM)" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../preprocessing/PostImputation.html" class="btn btn-neutral float-right" title="PostImputation Module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Luis Giraldo Gonzalez Ricardo.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>