

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>ideal_genom.get_references &mdash; ideal-genom 0.1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=01f34227"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="ideal_genom.Helpers" href="Helpers.html" />
    <link rel="prev" title="ideal_genom.get_examples" href="get_examples.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            ideal-genom
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="gwas/gwas_glm.html">GWAS with Generalized Linear Models (GLM)</a></li>
<li class="toctree-l1"><a class="reference internal" href="gwas/gwas_glmm.html">GWAS with Generalized Linear Mixed Models (GLMM)</a></li>
<li class="toctree-l1"><a class="reference internal" href="preprocessing/PostImputation.html">PostImputation Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="preprocessing/preparatory.html">Preparatory Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualization/manhattan_type.html">Manhattan-type Plots Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualization/plots.html">Other Plots Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="visualization/zoom_heatmap.html">Zoom-Heatmap Plot Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="annotations.html">Annotations Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="api_client.html">ideal_genom.api_client</a></li>
<li class="toctree-l1"><a class="reference internal" href="get_examples.html">ideal_genom.get_examples</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">ideal_genom.get_references</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher.__init__"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher.__init__()</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher.file_path"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher.file_path</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher.reference_url"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher.reference_url</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher.reference_file"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher.reference_file</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher._download_file"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher._download_file()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher.download_reference_file"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher.download_reference_file()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher.get_destination_folder"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher.get_destination_folder()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher.get_reference_url"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher.get_reference_url()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher.unzip_reference_file"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher.unzip_reference_file()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher.base_url"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.base_url</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher.build"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.build</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher.source"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.source</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher.latest_url"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.latest_url</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher.get_latest_release"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.get_latest_release()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher.__init__"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id0"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.get_latest_release()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher.base_url"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.base_url</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher.build"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.build</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher.source"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.source</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher.latest_url"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.latest_url</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher.get_latest_release"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.get_latest_release()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher.__init__"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id1"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.get_latest_release()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.get_references.FetcherLDRegions"><code class="docutils literal notranslate"><span class="pre">FetcherLDRegions</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.FetcherLDRegions.__init__"><code class="docutils literal notranslate"><span class="pre">FetcherLDRegions.__init__()</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.FetcherLDRegions.destination"><code class="docutils literal notranslate"><span class="pre">FetcherLDRegions.destination</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.FetcherLDRegions.built"><code class="docutils literal notranslate"><span class="pre">FetcherLDRegions.built</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.FetcherLDRegions.ld_regions"><code class="docutils literal notranslate"><span class="pre">FetcherLDRegions.ld_regions</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.FetcherLDRegions.get_ld_regions"><code class="docutils literal notranslate"><span class="pre">FetcherLDRegions.get_ld_regions()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher.base_url"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.base_url</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher.build"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.build</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher.source"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.source</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher.latest_url"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.latest_url</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher.get_latest_release"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.get_latest_release()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher.__init__"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id2"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.get_latest_release()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher</span></code></a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.build"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.build</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.source"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.source</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.base_url"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.base_url</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.destination_folder"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.destination_folder</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.latest_url"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.latest_url</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.gz_file"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.gz_file</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.gtf_file"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.gtf_file</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.get_latest_release"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.get_latest_release()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.download_latest"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.download_latest()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.get_destination_folder"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.get_destination_folder()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher._download_file"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher._download_file()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.unzip_latest"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.unzip_latest()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.get_all_genes"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.get_all_genes()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.__init__"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.__init__()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id3"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher._download_file()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id4"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.download_latest()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id6"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.get_all_genes()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id7"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.get_destination_folder()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id8"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.get_latest_release()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#id9"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.unzip_latest()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Helpers.html">ideal_genom.Helpers</a></li>
<li class="toctree-l1"><a class="reference internal" href="power_comp.html">ideal_genom.power_comp</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">ideal-genom</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">ideal_genom.get_references</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/get_references.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-ideal_genom.get_references">
<span id="ideal-genom-get-references"></span><h1>ideal_genom.get_references<a class="headerlink" href="#module-ideal_genom.get_references" title="Link to this heading"></a></h1>
<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.get_references.</span></span><span class="sig-name descname"><span class="pre">AssemblyReferenceFetcher</span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>AssemblyReferenceFetcher is responsible for retrieving genomic reference files.
This class provides functionality to locate, download, and process reference genome
assemblies from web servers. It handles URL construction, web scraping to find
appropriate files, downloading, and decompressing reference files.</p>
<section id="key-features">
<h2>Key Features<a class="headerlink" href="#key-features" title="Link to this heading"></a></h2>
<ul class="simple">
<li><p>Web scraping to identify correct reference files based on build version</p></li>
<li><p>Automatic download of reference genome files</p></li>
<li><p>Decompression of gzipped reference files</p></li>
<li><p>Management of file paths and destination directories</p></li>
</ul>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">base_url</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">extension</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">destination_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">avoid_substring</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'extra'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a reference retriever.
This method sets up the configuration for retrieving reference files from a
specified URL, typically for genomic data based on a particular build.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>base_url</strong> (<em>str</em>) – The base URL from which to retrieve reference files.</p></li>
<li><p><strong>build</strong> (<em>str</em>) – The genome build identifier (e.g., ‘hg38’, ‘GRCh37’).</p></li>
<li><p><strong>extension</strong> (<em>str</em>) – The file extension of the reference files to retrieve.</p></li>
<li><p><strong>destination_folder</strong> (<em>Optional</em><em>[</em><em>str</em><em>]</em><em>, </em><em>default=None</em>) – The folder where retrieved reference files will be saved.
If None, files might be saved to a default location or handled differently.</p></li>
<li><p><strong>avoid_substring</strong> (<em>str</em><em>, </em><em>default='extra'</em>) – A substring to avoid when processing reference files.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher.file_path">
<span class="sig-name descname"><span class="pre">file_path</span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher.file_path" title="Link to this definition"></a></dt>
<dd><p>Path to the downloaded reference file, initialized as None.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher.reference_url">
<span class="sig-name descname"><span class="pre">reference_url</span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher.reference_url" title="Link to this definition"></a></dt>
<dd><p>Complete URL to the reference file, initialized as None.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str or None</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher.reference_file">
<span class="sig-name descname"><span class="pre">reference_file</span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher.reference_file" title="Link to this definition"></a></dt>
<dd><p>Name or identifier of the reference file, initialized as None.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str or None</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher._download_file">
<span class="sig-name descname"><span class="pre">_download_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">url</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher._download_file" title="Link to this definition"></a></dt>
<dd><p>Download a file from the given URL and save it to <cite>file_path</cite>.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher.download_reference_file">
<span class="sig-name descname"><span class="pre">download_reference_file</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher.download_reference_file" title="Link to this definition"></a></dt>
<dd><p>Downloads the reference assembly sequence file from base_url to <cite>self.destination_folder</cite>.</p>
<section id="raises">
<h3>Raises:<a class="headerlink" href="#raises" title="Link to this heading"></a></h3>
<ul class="simple">
<li><p>AttributeError: If <cite>self.latest_url</cite> is not set.</p></li>
<li><p>requests.exceptions.RequestException: If the HTTP request fails.</p></li>
</ul>
</section>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher.get_destination_folder">
<span class="sig-name descname"><span class="pre">get_destination_folder</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher.get_destination_folder" title="Link to this definition"></a></dt>
<dd><p>Determine the destination folder for downloads.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher.get_reference_url">
<span class="sig-name descname"><span class="pre">get_reference_url</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher.get_reference_url" title="Link to this definition"></a></dt>
<dd><p>Retrieves the URL for the reference file from the base URL.
This method scrapes the base URL to find a link to the appropriate reference file.
It looks for links that contain the build version, have the correct file extension,
and do not contain the specified substring to avoid.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>str</strong></p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>The complete URL to the reference file.</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>Exception</strong> – If the base URL cannot be accessed.:</p></li>
<li><p><strong>FileNotFoundError</strong> – If no suitable reference file is found.:</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher.unzip_reference_file">
<span class="sig-name descname"><span class="pre">unzip_reference_file</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher.unzip_reference_file" title="Link to this definition"></a></dt>
<dd><p>Unzips the reference genome file if it’s compressed.
This method extracts the content of a gzipped reference genome file
and saves it as a FASTA file. After successful extraction, the original
compressed file is deleted.</p>
<section id="prerequisites">
<h3>Prerequisites<a class="headerlink" href="#prerequisites" title="Link to this heading"></a></h3>
<ul class="simple">
<li><p>self.reference_file must be set (via get_reference_url)</p></li>
<li><p>self.file_path must be set (via download_reference_file)</p></li>
</ul>
<dl class="field-list simple">
<dt class="field-odd">returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>str</strong></p>
</dd>
<dt class="field-even">rtype<span class="colon">:</span></dt>
<dd class="field-even"><p>Path to the unzipped FASTA file</p>
</dd>
<dt class="field-odd">raises AttributeError<span class="colon">:</span></dt>
<dd class="field-odd"><p>If self.reference_file or self.file_path are not set:</p>
</dd>
<dt class="field-even">raises OSError<span class="colon">:</span></dt>
<dd class="field-even"><p>If an error occurs during the unzipping process:</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>If the file is already unzipped (has .fa extension), returns its path</p></li>
<li><p>If the unzipped file already exists, returns its path without re-extracting</p></li>
<li><p>The original compressed file is deleted after successful extraction</p></li>
</ul>
</section>
</dd></dl>

</section>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.get_references.</span></span><span class="sig-name descname"><span class="pre">Ensembl37Fetcher</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher" title="ideal_genom.get_references.ReferenceDataFetcher"><code class="xref py py-class docutils literal notranslate"><span class="pre">ReferenceDataFetcher</span></code></a></p>
<p>A class for fetching reference genome data from Ensembl’s GRCh37 (hg19) repository.
This class specializes the ReferenceDataFetcher to work specifically with
Ensembl’s GRCh37 human genome build. It provides functionality to automatically
detect and download the latest available GTF file for Homo sapiens from the
Ensembl GRCh37 archive.
The fetcher connects to Ensembl’s FTP server, identifies the most recent release
available for GRCh37, and locates the chromosome GTF file for human genome data.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher.base_url">
<span class="sig-name descname"><span class="pre">base_url</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher.base_url" title="Link to this definition"></a></dt>
<dd><p>The base URL for Ensembl’s GRCh37 repository</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher.build">
<span class="sig-name descname"><span class="pre">build</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher.build" title="Link to this definition"></a></dt>
<dd><p>The genome build identifier (‘37’)</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher.source">
<span class="sig-name descname"><span class="pre">source</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher.source" title="Link to this definition"></a></dt>
<dd><p>The data source identifier (‘ensembl’)</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher.latest_url">
<span class="sig-name descname"><span class="pre">latest_url</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher.latest_url" title="Link to this definition"></a></dt>
<dd><p>The complete URL to the latest GTF file, populated after calling get_latest_release()</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher.get_latest_release">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher.get_latest_release" title="Link to this definition"></a></dt>
<dd><p>Identifies and sets the URL for the latest available GTF file</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">destination_folder</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a reference genome downloader for Ensembl GRCh37.
This constructor configures the downloader to retrieve data from Ensembl’s GRCh37
repository.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>destination_folder</strong> (<em>str</em><em>, </em><em>optional</em>) – The folder where downloaded files will be stored. If None, a default
location will be used based on the parent class implementation.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id0">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id0" title="Link to this definition"></a></dt>
<dd><p>Fetches the URL of the latest GTF file for Homo sapiens GRCh37 from Ensembl.
This method:
1. Connects to the base URL and identifies all available release folders
2. Determines the latest release by finding the highest release number
3. Navigates to the GTF directory for that release
4. Locates the Homo sapiens GRCh37 chromosome GTF file
5. Stores the complete download URL in self.latest_url</p>
<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>Exception</strong> – If the base URL cannot be accessed:</p></li>
<li><p><strong>Exception</strong> – If no release folders are found:</p></li>
<li><p><strong>Exception</strong> – If the latest release folder cannot be accessed:</p></li>
<li><p><strong>FileNotFoundError</strong> – If the GTF file is not found in the latest release:</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.get_references.</span></span><span class="sig-name descname"><span class="pre">Ensembl38Fetcher</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher" title="ideal_genom.get_references.ReferenceDataFetcher"><code class="xref py py-class docutils literal notranslate"><span class="pre">ReferenceDataFetcher</span></code></a></p>
<p>A fetcher class for downloading human genome reference data from Ensembl, specifically for GRCh38 (build 38).
This class provides functionality to download and access the latest GTF (Gene Transfer Format)
annotation files for the human genome from the Ensembl FTP server. It automatically
determines the most recent release available and handles the download process.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher.base_url">
<span class="sig-name descname"><span class="pre">base_url</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher.base_url" title="Link to this definition"></a></dt>
<dd><p>The base URL for Ensembl’s FTP server where human GTF files are stored</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher.build">
<span class="sig-name descname"><span class="pre">build</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher.build" title="Link to this definition"></a></dt>
<dd><p>The genome build identifier (38 for GRCh38)</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher.source">
<span class="sig-name descname"><span class="pre">source</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher.source" title="Link to this definition"></a></dt>
<dd><p>The data source identifier (‘ensembl’)</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher.latest_url">
<span class="sig-name descname"><span class="pre">latest_url</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher.latest_url" title="Link to this definition"></a></dt>
<dd><p>URL to the latest GTF file, populated after calling get_latest_release()</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher.get_latest_release">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher.get_latest_release" title="Link to this definition"></a></dt>
<dd><p>Identifies and sets the URL for the latest available GTF file</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">destination_folder</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize the Ensembl human genome reference downloader.
This class downloads human genome reference data from the Ensembl FTP server,
specifically for homo sapiens (human) GTF files from the current release.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>destination_folder</strong> (<em>str</em><em>, </em><em>optional</em>) – Directory where the downloaded files will be saved. If None, a default
location will be used (typically determined by the parent class).</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>This downloader is configured to use:
- GRCh38 (build 38) human genome assembly
- Ensembl as the data source
- The current GTF release from Ensembl</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id1">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id1" title="Link to this definition"></a></dt>
<dd><p>Fetches the latest GTF file dynamically from the specified base URL.</p>
<p>This method sends a GET request to the base URL, parses the HTML response to find the latest GTF file link, and sets the <cite>latest_url</cite> attribute to the full URL of the latest GTF file.</p>
<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>FileNotFoundError</strong> – If no GTF file is found in the HTML response.:</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.get_references.FetcherLDRegions">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.get_references.</span></span><span class="sig-name descname"><span class="pre">FetcherLDRegions</span></span><a class="headerlink" href="#ideal_genom.get_references.FetcherLDRegions" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.FetcherLDRegions.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">destination</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">built</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'38'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.FetcherLDRegions.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize LDRegions object.
This initializer sets up the destination path for LD regions files and the genome build version.
If no destination is provided, it defaults to a ‘data/ld_regions_files’ directory relative to
the parent directory of the current file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>destination</strong> (<em>Path</em><em>, </em><em>optional</em>) – Path where LD region files will be stored. If None, uses default path.</p></li>
<li><p><strong>built</strong> (<em>str</em><em>, </em><em>optional</em>) – Genome build version, defaults to ‘38’.</p></li>
</ul>
</dd>
</dl>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.FetcherLDRegions.destination">
<span class="sig-name descname"><span class="pre">destination</span></span><a class="headerlink" href="#ideal_genom.get_references.FetcherLDRegions.destination" title="Link to this definition"></a></dt>
<dd><p>Directory path where LD region files are stored</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.FetcherLDRegions.built">
<span class="sig-name descname"><span class="pre">built</span></span><a class="headerlink" href="#ideal_genom.get_references.FetcherLDRegions.built" title="Link to this definition"></a></dt>
<dd><p>Genome build version being used</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.FetcherLDRegions.ld_regions">
<span class="sig-name descname"><span class="pre">ld_regions</span></span><a class="headerlink" href="#ideal_genom.get_references.FetcherLDRegions.ld_regions" title="Link to this definition"></a></dt>
<dd><p>Placeholder for LD regions data, initially set to None</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>None</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.FetcherLDRegions.get_ld_regions">
<span class="sig-name descname"><span class="pre">get_ld_regions</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.get_references.FetcherLDRegions.get_ld_regions" title="Link to this definition"></a></dt>
<dd><p>Downloads or creates high LD regions file based on genome build version.
This method handles the retrieval of high Linkage Disequilibrium (LD) regions for
different genome builds (37 or 38). For build 37, it downloads the regions from a
GitHub repository. For build 38, it creates the file from predefined coordinates.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>Path</strong></p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Path to the created/downloaded LD regions file. Returns empty Path if download fails for build 37.</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>None explicitly</strong><strong>, </strong><strong>but may raise standard I/O related exceptions.</strong> – </p>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>For build 37: Downloads from genepi-freiburg/gwas repository</p></li>
<li><p>For build 38: Creates file from hardcoded coordinates from GWAS-pipeline</p></li>
<li><p>Files are named as ‘high-LD-regions_GRCh{build}.txt’</p></li>
<li><p>Creates destination directory if it doesn’t exist</p></li>
</ul>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.get_references.</span></span><span class="sig-name descname"><span class="pre">RefSeqFetcher</span></span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher" title="ideal_genom.get_references.ReferenceDataFetcher"><code class="xref py py-class docutils literal notranslate"><span class="pre">ReferenceDataFetcher</span></code></a></p>
<p>A class for fetching and downloading reference genome data from NCBI’s RefSeq repository.
This class extends ReferenceDataFetcher to specifically handle downloading human
genome reference files from the RefSeq database. It supports different genome
builds (e.g., ‘GRCh37’, ‘GRCh38’) and automatically identifies the latest version
available for the specified build.
The class handles navigating the NCBI FTP directory structure, finding the
appropriate GTF files for the requested genome build, and managing the
download process.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher.base_url">
<span class="sig-name descname"><span class="pre">base_url</span></span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher.base_url" title="Link to this definition"></a></dt>
<dd><p>The base URL for the NCBI RefSeq FTP server directory.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher.build">
<span class="sig-name descname"><span class="pre">build</span></span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher.build" title="Link to this definition"></a></dt>
<dd><p>The genome build version (‘37’ for GRCh37, ‘38’ for GRCh38).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher.source">
<span class="sig-name descname"><span class="pre">source</span></span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher.source" title="Link to this definition"></a></dt>
<dd><p>The source of the reference data (set to ‘refseq’).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher.latest_url">
<span class="sig-name descname"><span class="pre">latest_url</span></span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher.latest_url" title="Link to this definition"></a></dt>
<dd><p>URL to the latest GTF file, set after calling get_latest_release().</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher.get_latest_release">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher.get_latest_release" title="Link to this definition"></a></dt>
<dd><p>Identifies and sets the URL for the latest available GTF file</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">destination_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a RefSeq genome reference downloader.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>(</strong><strong>str</strong><strong>)</strong> (<em>build</em>)</p></li>
<li><p><strong>(</strong><strong>Optional</strong><strong>[</strong><strong>str</strong><strong>]</strong> (<em>destination_folder</em>) – reference files will be stored. If None, files will be stored in the default
location. Defaults to None.</p></li>
<li><p><strong>optional</strong><strong>)</strong> (<em>Directory path where the downloaded</em>) – reference files will be stored. If None, files will be stored in the default
location. Defaults to None.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id2">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id2" title="Link to this definition"></a></dt>
<dd><p>Fetches the latest GTF file dynamically from the specified base URL.</p>
<p>This method sends a GET request to the base URL, parses the HTML response to find the latest GTF file link, and sets the <cite>latest_url</cite> attribute to the full URL of the latest GTF file.</p>
<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>FileNotFoundError</strong> – If no GTF file is found in the HTML response.:</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.get_references.</span></span><span class="sig-name descname"><span class="pre">ReferenceDataFetcher</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A class for fetching, downloading, and processing reference genome data.
This class provides a framework for retrieving genomic reference data from various
sources. It handles downloading compressed files, unzipping them, and extracting
gene information from GTF files.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.build">
<span class="sig-name descname"><span class="pre">build</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.build" title="Link to this definition"></a></dt>
<dd><p>The genome build (e.g., ‘hg38’, ‘GRCh38’).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.source">
<span class="sig-name descname"><span class="pre">source</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.source" title="Link to this definition"></a></dt>
<dd><p>The data source (e.g., ‘ensembl’, ‘ucsc’).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.base_url">
<span class="sig-name descname"><span class="pre">base_url</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.base_url" title="Link to this definition"></a></dt>
<dd><p>The base URL to fetch data from.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.destination_folder">
<span class="sig-name descname"><span class="pre">destination_folder</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.destination_folder" title="Link to this definition"></a></dt>
<dd><p>The directory to save downloaded files. If None, defaults to project_root/data/{source}_latest.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Optional[str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.latest_url">
<span class="sig-name descname"><span class="pre">latest_url</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.latest_url" title="Link to this definition"></a></dt>
<dd><p>The URL of the latest release after calling get_latest_release().</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Optional[str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.gz_file">
<span class="sig-name descname"><span class="pre">gz_file</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.gz_file" title="Link to this definition"></a></dt>
<dd><p>Path to the downloaded compressed file.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Optional[str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.gtf_file">
<span class="sig-name descname"><span class="pre">gtf_file</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.gtf_file" title="Link to this definition"></a></dt>
<dd><p>Path to the uncompressed GTF file.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Optional[str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.get_latest_release">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.get_latest_release" title="Link to this definition"></a></dt>
<dd><p>Determine the specific URL for fetching data. Must be implemented by subclasses.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.download_latest">
<span class="sig-name descname"><span class="pre">download_latest</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.download_latest" title="Link to this definition"></a></dt>
<dd><p>Downloads the latest file and returns the file path.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.get_destination_folder">
<span class="sig-name descname"><span class="pre">get_destination_folder</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.get_destination_folder" title="Link to this definition"></a></dt>
<dd><p>Determines and creates the destination folder for downloads.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher._download_file">
<span class="sig-name descname"><span class="pre">_download_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">url</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher._download_file" title="Link to this definition"></a></dt>
<dd><p>Downloads a file from the given URL and saves it to file_path.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.unzip_latest">
<span class="sig-name descname"><span class="pre">unzip_latest</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.unzip_latest" title="Link to this definition"></a></dt>
<dd><p>Unzips the latest downloaded file and returns the path to the unzipped file.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.get_all_genes">
<span class="sig-name descname"><span class="pre">get_all_genes</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.get_all_genes" title="Link to this definition"></a></dt>
<dd><p>Processes the GTF file to extract all genes and saves them to a new file.
Returns the path to the new file.</p>
</dd></dl>

<p class="rubric">Notes</p>
<p>This is an abstract base class that requires subclasses to implement the
get_latest_release() method for specific data sources.</p>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">base_url</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">source</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">destination_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.__init__" title="Link to this definition"></a></dt>
<dd><p>Initialize a reference genome retrieval object.</p>
<p>This constructor sets up the necessary parameters for downloading reference genome files
from a specified source.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>(</strong><strong>str</strong><strong>)</strong> (<em>source</em>) – The base URL where reference genome files are hosted.</p></li>
<li><p><strong>(</strong><strong>str</strong><strong>)</strong> – The genome build/version (e.g., ‘GRCh38’, ‘hg19’).</p></li>
<li><p><strong>(</strong><strong>str</strong><strong>)</strong> – The source of the genome data (e.g., ‘ensembl’, ‘refseq’).</p></li>
<li><p><strong>(</strong><strong>Optional</strong><strong>[</strong><strong>str</strong><strong>]</strong> (<em>destination_folder</em>) – Directory where downloaded files will be saved. If None, files may be saved to a default location. Defaults to None.</p></li>
<li><p><strong>optional</strong><strong>)</strong> – Directory where downloaded files will be saved. If None, files may be saved to a default location. Defaults to None.</p></li>
<li><p><strong>Returns</strong> – None</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id3">
<span class="sig-name descname"><span class="pre">_download_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">url</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id3" title="Link to this definition"></a></dt>
<dd><p>Download a file from the given URL and save it to <cite>file_path</cite>.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id4">
<span class="sig-name descname"><span class="pre">download_latest</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#id4" title="Link to this definition"></a></dt>
<dd><p>Downloads the latest file from <cite>self.latest_url</cite> to <cite>self.destination_folder</cite>.</p>
<section id="id5">
<h2>Raises:<a class="headerlink" href="#id5" title="Link to this heading"></a></h2>
<blockquote>
<div><p>AttributeError: If <cite>self.latest_url</cite> is not set.
requests.exceptions.RequestException: If the HTTP request fails.</p>
</div></blockquote>
</section>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id6">
<span class="sig-name descname"><span class="pre">get_all_genes</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#id6" title="Link to this definition"></a></dt>
<dd><p>Extract all genes from the GTF file and save them to a new compressed file.
This method reads the GTF file specified in self.gtf_file, filters for gene features,
and creates a new GTF file containing only the gene entries. If the output file
already exists, it will return the path without reprocessing.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path to the compressed GTF file containing all genes</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>str</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>FileNotFoundError</strong> – If the reference GTF file (self.gtf_file) is not found:</p></li>
<li><p><strong>TypeError</strong> – If read_gtf does not return a pandas DataFrame:</p></li>
</ul>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The output file will be named based on the input GTF file with “-all_genes.gtf.gz” suffix</p>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id7">
<span class="sig-name descname"><span class="pre">get_destination_folder</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#id7" title="Link to this definition"></a></dt>
<dd><p>Determine the destination folder for downloads.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id8">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id8" title="Link to this definition"></a></dt>
<dd><p>Determine the specific URL for fetching data.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id9">
<span class="sig-name descname"><span class="pre">unzip_latest</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#id9" title="Link to this definition"></a></dt>
<dd><p>Unzips the latest downloaded file and stores it as a GTF file.</p>
</dd></dl>

</dd></dl>

</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="get_examples.html" class="btn btn-neutral float-left" title="ideal_genom.get_examples" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="Helpers.html" class="btn btn-neutral float-right" title="ideal_genom.Helpers" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Luis Giraldo Gonzalez Ricardo.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>