

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>ideal_genom package &mdash; ideal-genom 0.1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=01f34227"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="ideal_genom.annotations" href="annotations.html" />
    <link rel="prev" title="ideal-genom documentation" href="index.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            ideal-genom
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Package Modules</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">ideal_genom package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="#module-ideal_genom.Helpers">ideal_genom.Helpers module</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.Helpers.arg_parser"><code class="docutils literal notranslate"><span class="pre">arg_parser()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.Helpers.delete_temp_files"><code class="docutils literal notranslate"><span class="pre">delete_temp_files()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.Helpers.download_file"><code class="docutils literal notranslate"><span class="pre">download_file()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.Helpers.extract_gz_file"><code class="docutils literal notranslate"><span class="pre">extract_gz_file()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.Helpers.shell_do"><code class="docutils literal notranslate"><span class="pre">shell_do()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.Helpers.unzip_file_flat"><code class="docutils literal notranslate"><span class="pre">unzip_file_flat()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-ideal_genom.annotations">ideal_genom.annotations module</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.annotations.annotate_snp"><code class="docutils literal notranslate"><span class="pre">annotate_snp()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.annotations.annotate_variants"><code class="docutils literal notranslate"><span class="pre">annotate_variants()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.annotations.annotate_with_ensembl"><code class="docutils literal notranslate"><span class="pre">annotate_with_ensembl()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.annotations.annotate_with_refseq"><code class="docutils literal notranslate"><span class="pre">annotate_with_refseq()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.annotations.get_chr_to_NC"><code class="docutils literal notranslate"><span class="pre">get_chr_to_NC()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.annotations.get_closest_gene"><code class="docutils literal notranslate"><span class="pre">get_closest_gene()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.annotations.get_number_to_chr"><code class="docutils literal notranslate"><span class="pre">get_number_to_chr()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.annotations.gtf_to_all_genes"><code class="docutils literal notranslate"><span class="pre">gtf_to_all_genes()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.annotations.prepare_genome"><code class="docutils literal notranslate"><span class="pre">prepare_genome()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.annotations.prepare_gtf_path"><code class="docutils literal notranslate"><span class="pre">prepare_gtf_path()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-ideal_genom.api_client">ideal_genom.api_client module</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.api_client.GeneEnsemblRestClient"><code class="docutils literal notranslate"><span class="pre">GeneEnsemblRestClient</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.api_client.GeneEnsemblRestClient.get_gene_location"><code class="docutils literal notranslate"><span class="pre">GeneEnsemblRestClient.get_gene_location()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.api_client.GeneEnsemblRestClient.perform_rest_action"><code class="docutils literal notranslate"><span class="pre">GeneEnsemblRestClient.perform_rest_action()</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.api_client.VEPEnsemblRestClient"><code class="docutils literal notranslate"><span class="pre">VEPEnsemblRestClient</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.api_client.VEPEnsemblRestClient.ENSEMBL_API_URLS"><code class="docutils literal notranslate"><span class="pre">VEPEnsemblRestClient.ENSEMBL_API_URLS</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.api_client.VEPEnsemblRestClient.perform_rest_action"><code class="docutils literal notranslate"><span class="pre">VEPEnsemblRestClient.perform_rest_action()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.api_client.VEPEnsemblRestClient.post_vep_request"><code class="docutils literal notranslate"><span class="pre">VEPEnsemblRestClient.post_vep_request()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-ideal_genom.get_examples">ideal_genom.get_examples module</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_examples.get_bmi_japanese_gwas"><code class="docutils literal notranslate"><span class="pre">get_bmi_japanese_gwas()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_examples.get_bmi_japanese_gwas_sex_dependent"><code class="docutils literal notranslate"><span class="pre">get_bmi_japanese_gwas_sex_dependent()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_examples.get_top_cond_trumpet_quantitative"><code class="docutils literal notranslate"><span class="pre">get_top_cond_trumpet_quantitative()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_examples.get_top_loci_trumpet_quantitative"><code class="docutils literal notranslate"><span class="pre">get_top_loci_trumpet_quantitative()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_examples.get_trumpet_binary_example"><code class="docutils literal notranslate"><span class="pre">get_trumpet_binary_example()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_examples.get_trumpet_quantitative_example"><code class="docutils literal notranslate"><span class="pre">get_trumpet_quantitative_example()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-ideal_genom.get_references">ideal_genom.get_references module</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher.download_reference_file"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher.download_reference_file()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher.get_destination_folder"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher.get_destination_folder()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher.get_reference_url"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher.get_reference_url()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.AssemblyReferenceFetcher.unzip_reference_file"><code class="docutils literal notranslate"><span class="pre">AssemblyReferenceFetcher.unzip_reference_file()</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher.base_url"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.base_url</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher.build"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.build</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher.source"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.source</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher.latest_url"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.latest_url</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.Ensembl37Fetcher.get_latest_release"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.get_latest_release()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id0"><code class="docutils literal notranslate"><span class="pre">Ensembl37Fetcher.get_latest_release()</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher.base_url"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.base_url</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher.build"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.build</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher.source"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.source</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher.latest_url"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.latest_url</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.Ensembl38Fetcher.get_latest_release"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.get_latest_release()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id2"><code class="docutils literal notranslate"><span class="pre">Ensembl38Fetcher.get_latest_release()</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.FetcherLDRegions"><code class="docutils literal notranslate"><span class="pre">FetcherLDRegions</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.FetcherLDRegions.get_ld_regions"><code class="docutils literal notranslate"><span class="pre">FetcherLDRegions.get_ld_regions()</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher.base_url"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.base_url</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher.build"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.build</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher.source"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.source</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher.latest_url"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.latest_url</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.RefSeqFetcher.get_latest_release"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.get_latest_release()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id6"><code class="docutils literal notranslate"><span class="pre">RefSeqFetcher.get_latest_release()</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher</span></code></a><ul>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.build"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.build</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.source"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.source</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.base_url"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.base_url</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.destination_folder"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.destination_folder</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.latest_url"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.latest_url</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.gz_file"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.gz_file</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.gtf_file"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.gtf_file</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.get_latest_release"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.get_latest_release()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.download_latest"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.download_latest()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.get_destination_folder"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.get_destination_folder()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher._download_file"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher._download_file()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.unzip_latest"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.unzip_latest()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher.get_all_genes"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.get_all_genes()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id9"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.download_latest()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id11"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.get_all_genes()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id12"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.get_destination_folder()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id13"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.get_latest_release()</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="#id14"><code class="docutils literal notranslate"><span class="pre">ReferenceDataFetcher.unzip_latest()</span></code></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-ideal_genom.power_comp">ideal_genom.power_comp module</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.power_comp.calculate_power_binary"><code class="docutils literal notranslate"><span class="pre">calculate_power_binary()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.power_comp.calculate_power_quantitative"><code class="docutils literal notranslate"><span class="pre">calculate_power_quantitative()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.power_comp.get_beta_binary"><code class="docutils literal notranslate"><span class="pre">get_beta_binary()</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="#ideal_genom.power_comp.get_beta_quantitative"><code class="docutils literal notranslate"><span class="pre">get_beta_quantitative()</span></code></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#module-ideal_genom">Module contents</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Standalone Modules</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="annotations.html">ideal_genom.annotations</a></li>
<li class="toctree-l1"><a class="reference internal" href="api_client.html">ideal_genom.api_client</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">ideal-genom</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">ideal_genom package</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/ideal_genom.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="ideal-genom-package">
<h1>ideal_genom package<a class="headerlink" href="#ideal-genom-package" title="Link to this heading"></a></h1>
<section id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Link to this heading"></a></h2>
</section>
<section id="module-ideal_genom.Helpers">
<span id="ideal-genom-helpers-module"></span><h2>ideal_genom.Helpers module<a class="headerlink" href="#module-ideal_genom.Helpers" title="Link to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.Helpers.arg_parser">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.Helpers.</span></span><span class="sig-name descname"><span class="pre">arg_parser</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span></span></span><a class="headerlink" href="#ideal_genom.Helpers.arg_parser" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.Helpers.delete_temp_files">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.Helpers.</span></span><span class="sig-name descname"><span class="pre">delete_temp_files</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">files_to_keep</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path_to_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.Helpers.delete_temp_files" title="Link to this definition"></a></dt>
<dd><p>Function to delete temporary files that were created during the pipeline execution. Moreover, it creates a directory called ‘log_files’ to save al <cite>.log</cite> files originated from the pipeline execution.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>files_to_keep</strong> (<em>list</em>) – list of strings where its elements are the names of files and folders that should be kept.</p></li>
<li><p><strong>path_to_folder</strong> (<em>str</em>) – full path to the folder where the temporary files are located.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.Helpers.download_file">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.Helpers.</span></span><span class="sig-name descname"><span class="pre">download_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">url</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">local_filename</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.Helpers.download_file" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.Helpers.extract_gz_file">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.Helpers.</span></span><span class="sig-name descname"><span class="pre">extract_gz_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gz_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_gz</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.Helpers.extract_gz_file" title="Link to this definition"></a></dt>
<dd><p>Extracts a .gz file and saves the decompressed content in the same directory.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>gz_file</strong> (<em>str</em>) – Path to the .gz file.</p></li>
<li><p><strong>out_dir</strong> (<em>str</em>) – Directory where the decompressed file will be saved.</p></li>
<li><p><strong>remove_gz</strong> (<em>bool</em>) – If True, delete the .gz file after extraction.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Path to the extracted file.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.Helpers.shell_do">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.Helpers.</span></span><span class="sig-name descname"><span class="pre">shell_do</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">command</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">print_cmd</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">log</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_log</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">err</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.Helpers.shell_do" title="Link to this definition"></a></dt>
<dd><p>From GenoTools</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.Helpers.unzip_file_flat">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.Helpers.</span></span><span class="sig-name descname"><span class="pre">unzip_file_flat</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">in_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">target_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">remove_zip</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.Helpers.unzip_file_flat" title="Link to this definition"></a></dt>
<dd><p>Extracts a specific file from a ZIP archive, decompresses it if it’s a .gz file, and optionally deletes original files.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>in_file</strong> (<em>str</em>) – Path to the ZIP file.</p></li>
<li><p><strong>target_file</strong> (<em>str</em>) – The file inside the ZIP to extract.</p></li>
<li><p><strong>out_dir</strong> (<em>str</em>) – Directory where the extracted file will be saved.</p></li>
<li><p><strong>remove_zip</strong> (<em>bool</em>) – If True, delete the original ZIP file after extraction.</p></li>
<li><p><strong>remove_gz</strong> (<em>bool</em>) – If True, delete the .gz file after decompression.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Path to the final extracted file.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path</p>
</dd>
</dl>
</dd></dl>

</section>
<section id="module-ideal_genom.annotations">
<span id="ideal-genom-annotations-module"></span><h2>ideal_genom.annotations module<a class="headerlink" href="#module-ideal_genom.annotations" title="Link to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.annotations.annotate_snp">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.annotations.</span></span><span class="sig-name descname"><span class="pre">annotate_snp</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">insumstats</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gtf_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'CHR'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'POS'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'38'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">source</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'ensembl'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="headerlink" href="#ideal_genom.annotations.annotate_snp" title="Link to this definition"></a></dt>
<dd><p>Annotate SNPs with nearest gene name(s) using either Ensembl or RefSeq databases.
This function takes a DataFrame containing SNP data and annotates each variant
with information about the nearest gene(s) based on genomic coordinates.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>(</strong><strong>pd.DataFrame</strong><strong>)</strong> (<em>insumstats</em>)</p></li>
<li><p><strong>(</strong><strong>str</strong><strong>)</strong> (<em>gtf_path</em>)</p></li>
<li><p><strong>(</strong><strong>str</strong> (<em>source</em>)</p></li>
<li><p><strong>optional</strong><strong>)</strong> (<em>Source for gene annotation. Must be either &quot;ensembl&quot;</em><em> or </em><em>&quot;refseq&quot;. Defaults to &quot;ensembl&quot;.</em>)</p></li>
<li><p><strong>(</strong><strong>str</strong></p></li>
<li><p><strong>optional</strong><strong>)</strong></p></li>
<li><p><strong>(</strong><strong>str</strong></p></li>
<li><p><strong>optional</strong><strong>)</strong></p></li>
<li><p><strong>(</strong><strong>str</strong></p></li>
<li><p><strong>optional</strong><strong>)</strong></p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pd.DataFrame</strong></p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>A copy of the input DataFrame with additional gene annotation columns.</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If input is not a pandas DataFrame or if GTF path is not a string.:</p></li>
<li><p><strong>ValueError</strong> – If required columns are missing in the input DataFrame or if build/source parameters are invalid.:</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.annotations.annotate_variants">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.annotations.</span></span><span class="sig-name descname"><span class="pre">annotate_variants</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Genome</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">source</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'38'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="headerlink" href="#ideal_genom.annotations.annotate_variants" title="Link to this definition"></a></dt>
<dd><p>Annotate variants with their closest genes.</p>
<p>This function processes a DataFrame containing genomic variants and enriches it
with gene annotation information by finding the closest gene for each variant.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>output</strong> (<em>pd.DataFrame</em>) – DataFrame containing variant information to be annotated.</p></li>
<li><p><strong>output</strong> – DataFrame containing variant information to be annotated.</p></li>
<li><p><strong>data</strong> (<em>Genome</em>) – Genome object containing reference data for annotation.</p></li>
<li><p><strong>chrom</strong> (<em>str</em>) – Column name in the output DataFrame that contains chromosome information.</p></li>
<li><p><strong>pos</strong> (<em>str</em>) – Column name in the output DataFrame that contains position information.</p></li>
<li><p><strong>source</strong> (<em>str</em>) – Source of the gene annotation data.</p></li>
<li><p><strong>build</strong> (<em>str</em><em>, </em><em>default='38'</em>) – Genome build version (default is GRCh38).</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>DataFrame containing gene annotation information for each variant.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>This function applies the get_closest_gene function to each row in the input DataFrame
and returns the results as a DataFrame with the same index as the input.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.annotations.annotate_with_ensembl">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.annotations.</span></span><span class="sig-name descname"><span class="pre">annotate_with_ensembl</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gtf_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_gtf_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="headerlink" href="#ideal_genom.annotations.annotate_with_ensembl" title="Link to this definition"></a></dt>
<dd><p>Annotate variants with gene information from Ensembl database.
This function adds gene annotations to a DataFrame containing variant information
by looking up the genomic coordinates in Ensembl data. It adds ‘LOCATION’ and ‘GENE’
columns to the input DataFrame.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>output</strong> (<em>pd.DataFrame</em>) – DataFrame containing variant information with chromosome and position columns.</p></li>
<li><p><strong>chrom</strong> (<em>str</em>) – Name of the column in the DataFrame that contains chromosome information.</p></li>
<li><p><strong>pos</strong> (<em>str</em>) – Name of the column in the DataFrame that contains position information.</p></li>
<li><p><strong>build</strong> (<em>str</em>) – Genome build version to use. Must be one of ‘19’, ‘37’, or ‘38’.
Note that ‘19’ and ‘37’ are treated as equivalent (GRCh37).</p></li>
<li><p><strong>gtf_path</strong> (<em>str</em>) – Path to GTF file with gene annotations or None to use default paths.
If None, the appropriate GTF file will be downloaded or used from cache.</p></li>
<li><p><strong>is_gtf_path</strong> (<em>bool</em>) – If True, gtf_path is treated as a direct path to a GTF file.
If False, gtf_path is treated as a directory where the GTF file should be downloaded.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The input DataFrame with additional ‘LOCATION’ and ‘GENE’ columns containing
gene annotations from Ensembl.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If output is not a pandas DataFrame or if gtf_path is not a string (when provided).</p></li>
<li><p><strong>ValueError</strong> – If the required columns are not in the DataFrame or if the build is invalid.</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>The function supports both GRCh37 (build ‘19’ or ‘37’) and GRCh38 (build ‘38’)
and will download the appropriate annotation files if not already available.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.annotations.annotate_with_refseq">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.annotations.</span></span><span class="sig-name descname"><span class="pre">annotate_with_refseq</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">output</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gtf_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_gtf_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="headerlink" href="#ideal_genom.annotations.annotate_with_refseq" title="Link to this definition"></a></dt>
<dd><p>Annotate genomic variants with RefSeq gene information.
This function adds gene and location annotations to genomic variants using NCBI RefSeq data.
It processes the input DataFrame and adds two new columns: ‘LOCATION’ and ‘GENE’.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>(</strong><strong>pd.DataFrame</strong><strong>)</strong> (<em>output</em>)</p></li>
<li><p><strong>(</strong><strong>str</strong><strong>)</strong> (<em>gtf_path</em>)</p></li>
<li><p><strong>(</strong><strong>str</strong><strong>)</strong></p></li>
<li><p><strong>(</strong><strong>str</strong><strong>)</strong></p></li>
<li><p><strong>(</strong><strong>str</strong><strong>)</strong></p></li>
<li><p><strong>(</strong><strong>bool</strong><strong>)</strong> (<em>is_gtf_path</em>) – If False, gtf_path is treated as a directory.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pd.DataFrame</strong></p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>The input DataFrame with added ‘LOCATION’ and ‘GENE’ columns.</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If output is not a pandas DataFrame or if gtf_path is provided but not a string.:</p></li>
<li><p><strong>ValueError</strong> – If required columns are missing from output or if build is invalid.:</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>For builds ‘19’ and ‘37’, GRCh37 RefSeq annotations are used.</p></li>
<li><p>For build ‘38’, GRCh38 RefSeq annotations are used.</p></li>
<li><p>Only protein-coding genes are considered for annotation.</p></li>
</ul>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.annotations.get_chr_to_NC">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.annotations.</span></span><span class="sig-name descname"><span class="pre">get_chr_to_NC</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span></span></span><a class="headerlink" href="#ideal_genom.annotations.get_chr_to_NC" title="Link to this definition"></a></dt>
<dd><p>Returns a dictionary mapping between chromosome names and NCBI NC identifiers.</p>
<p>This function provides a mapping between chromosome names (like “1”, “X”, “MT”)
and their corresponding NCBI RefSeq accession numbers (like “NC_000001.10”)
for different human genome builds.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>(</strong><strong>str</strong><strong>)</strong> (<em>build</em>) – Note that builds “19” and “37” return the same mapping.</p></li>
<li><p><strong>(</strong><strong>bool</strong> (<em>inverse</em>) – identifiers are keys and chromosome names are values.
Defaults to False.</p></li>
<li><p><strong>optional</strong><strong>)</strong> (<em>If True</em><em>, </em><em>returns an inverted dictionary where NC</em>) – identifiers are keys and chromosome names are values.
Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>dict</strong> – or NC identifiers to chromosome names (if inverse=True).</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>A dictionary mapping chromosome names to NC identifiers (if inverse=False)</p>
</dd>
</dl>
<p class="rubric">References</p>
<p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.13">https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.13</a></p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.annotations.get_closest_gene">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.annotations.</span></span><span class="sig-name descname"><span class="pre">get_closest_gene</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Genome</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'CHR'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'POS'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_iter</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">20000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">step</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">50</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">source</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'ensembl'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'38'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">tuple</span></span></span><a class="headerlink" href="#ideal_genom.annotations.get_closest_gene" title="Link to this definition"></a></dt>
<dd><p>Find the closest gene to a given position in the genome.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>x</strong> – SNP information.</p></li>
<li><p><strong>(</strong><strong>pyensembl.Genome</strong><strong>)</strong> (<em>data</em>) – An instance of the Genome class containing gene annotations.</p></li>
<li><p><strong>(</strong><strong>str</strong> (<em>build</em>) – The key in the dictionary <cite>x</cite> that corresponds to the chromosome. Default is “CHR”.</p></li>
<li><p><strong>optional</strong><strong>)</strong> – The key in the dictionary <cite>x</cite> that corresponds to the chromosome. Default is “CHR”.</p></li>
<li><p><strong>(</strong><strong>str</strong> – The key in the dictionary <cite>x</cite> that corresponds to the position. Default is “POS”.</p></li>
<li><p><strong>optional</strong><strong>)</strong> – The key in the dictionary <cite>x</cite> that corresponds to the position. Default is “POS”.</p></li>
<li><p><strong>(</strong><strong>int</strong> (<em>step</em>) – The maximum number of iterations to search for a gene. Default is 20000.</p></li>
<li><p><strong>optional</strong><strong>)</strong> – The maximum number of iterations to search for a gene. Default is 20000.</p></li>
<li><p><strong>(</strong><strong>int</strong> – The step size for each iteration when searching for a gene. Default is 50.</p></li>
<li><p><strong>optional</strong><strong>)</strong> – The step size for each iteration when searching for a gene. Default is 50.</p></li>
<li><p><strong>(</strong><strong>str</strong> – The source of the gene annotations, either “ensembl” or “refseq”. Default is “ensembl”.</p></li>
<li><p><strong>optional</strong><strong>)</strong> – The source of the gene annotations, either “ensembl” or “refseq”. Default is “ensembl”.</p></li>
<li><p><strong>(</strong><strong>str</strong> – The genome build version, used when source is “refseq”. Default is “38”.</p></li>
<li><p><strong>optional</strong><strong>)</strong> – The genome build version, used when source is “refseq”. Default is “38”.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A tuple containing the distance to the closest gene and the gene name(s). If no gene is found, returns the distance and “intergenic”.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>tuple</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.annotations.get_number_to_chr">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.annotations.</span></span><span class="sig-name descname"><span class="pre">get_number_to_chr</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">in_chr</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xymt</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">['X',</span> <span class="pre">'Y',</span> <span class="pre">'MT']</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">xymt_num</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">[23,</span> <span class="pre">24,</span> <span class="pre">25]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">prefix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">''</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span></span></span><a class="headerlink" href="#ideal_genom.annotations.get_number_to_chr" title="Link to this definition"></a></dt>
<dd><p>Creates a dictionary mapping chromosome numbers to chromosome identifiers.</p>
<p>This function generates a mapping between chromosome numbers (as keys) and
chromosome identifiers (as values), with special handling for sex chromosomes
and mitochondrial chromosome.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>in_chr</strong> (<em>bool</em><em>, </em><em>default=False</em>) – If True, dictionary keys will be strings; if False, keys will be integers.</p></li>
<li><p><strong>xymt</strong> (<em>list</em><em>, </em><em>default=</em><em>[</em><em>&quot;X&quot;</em><em>,</em><em>&quot;Y&quot;</em><em>,</em><em>&quot;MT&quot;</em><em>]</em>) – List of string identifiers for the X, Y, and mitochondrial chromosomes.</p></li>
<li><p><strong>xymt_num</strong> (<em>list</em><em>, </em><em>default=</em><em>[</em><em>23</em><em>,</em><em>24</em><em>,</em><em>25</em><em>]</em>) – List of numeric identifiers corresponding to X, Y, and MT chromosomes.</p></li>
<li><p><strong>prefix</strong> (<em>str</em><em>, </em><em>default=&quot;&quot;</em>) – String prefix to add to all chromosome identifiers.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A dictionary mapping chromosome numbers to chromosome identifiers.
For autosomal chromosomes (1-199), maps to prefix+number.
For sex and mitochondrial chromosomes, maps to prefix+X/Y/MT.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">get_number_to_chr</span><span class="p">()</span>
<span class="go">{1: &#39;1&#39;, 2: &#39;2&#39;, ..., 23: &#39;X&#39;, 24: &#39;Y&#39;, 25: &#39;MT&#39;, ...}</span>
</pre></div>
</div>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">get_number_to_chr</span><span class="p">(</span><span class="n">in_chr</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">prefix</span><span class="o">=</span><span class="s2">&quot;chr&quot;</span><span class="p">)</span>
<span class="go">{&#39;1&#39;: &#39;chr1&#39;, &#39;2&#39;: &#39;chr2&#39;, ..., &#39;23&#39;: &#39;chrX&#39;, &#39;24&#39;: &#39;chrY&#39;, &#39;25&#39;: &#39;chrMT&#39;, ...}</span>
</pre></div>
</div>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.annotations.gtf_to_all_genes">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.annotations.</span></span><span class="sig-name descname"><span class="pre">gtf_to_all_genes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gtfpath</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.annotations.gtf_to_all_genes" title="Link to this definition"></a></dt>
<dd><p>Extract all gene records from a GTF file and save them to a new file.
This function reads a GTF file, extracts all gene records, and saves them to a new file
with the suffix ‘_all_genes.gtf.gz’. If the output file already exists, it will be returned
without regenerating it.</p>
<section id="parameter">
<h3>Parameter<a class="headerlink" href="#parameter" title="Link to this heading"></a></h3>
<blockquote>
<div><p>gtfpath (str): Path to the input GTF file.</p>
</div></blockquote>
<dl class="field-list simple">
<dt class="field-odd">returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>str</strong></p>
</dd>
<dt class="field-even">rtype<span class="colon">:</span></dt>
<dd class="field-even"><p>Path to the output file containing all gene records.</p>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The function uses the <cite>read_gtf</cite> function for initial parsing and pandas for extraction.
It logs information about the extraction process using the logger module.
The function assumes the GTF file has a standard format with gene_id attributes.</p>
</div>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.annotations.prepare_genome">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.annotations.</span></span><span class="sig-name descname"><span class="pre">prepare_genome</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gtf_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reference_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">annotation_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Genome</span></span></span><a class="headerlink" href="#ideal_genom.annotations.prepare_genome" title="Link to this definition"></a></dt>
<dd><p>Prepare a genome annotation by loading or creating a database from a GTF file.</p>
<p>This function creates a Genome object from a GTF file and ensures that the
corresponding database is indexed for efficient access.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>gtf_path</strong> (<em>str</em>) – Path to the GTF (Gene Transfer Format) file</p></li>
<li><p><strong>reference_name</strong> (<em>str</em>) – Name of the reference genome</p></li>
<li><p><strong>annotation_name</strong> (<em>str</em>) – Name of the annotation</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A Genome object initialized with the provided reference and annotation</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Genome</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>If the database file (with extension .db) doesn’t exist, this function
will create it by calling the index() method on the Genome object.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.annotations.prepare_gtf_path">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.annotations.</span></span><span class="sig-name descname"><span class="pre">prepare_gtf_path</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gtf_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">is_gtf_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">source</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.annotations.prepare_gtf_path" title="Link to this definition"></a></dt>
<dd><p>Prepares the path to a GTF (Gene Transfer Format) file for annotation purposes.
This function either uses a user-provided GTF file or downloads one from the
specified source (Ensembl or RefSeq) for the given genome build (GRCh37 or GRCh38).
If a download is required, it fetches the latest release, unzips it, and processes
it to extract all genes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>(</strong><strong>str</strong><strong>)</strong> (<em>build</em>)</p></li>
<li><p><strong>(</strong><strong>bool</strong><strong>)</strong> (<em>is_gtf_path</em>)</p></li>
<li><p><strong>(</strong><strong>str</strong><strong>)</strong></p></li>
<li><p><strong>(</strong><strong>str</strong><strong>)</strong></p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>str</strong></p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path to the prepared GTF file with all genes.</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>If gtf_path is None or is_gtf_path is False, a new GTF file will be downloaded.</p></li>
<li><p>If a valid gtf_path is provided, it will be processed using gtf_to_all_genes.</p></li>
<li><p>The function logs the actions being performed.</p></li>
</ul>
</dd></dl>

</section>
<section id="module-ideal_genom.api_client">
<span id="ideal-genom-api-client-module"></span><h2>ideal_genom.api_client module<a class="headerlink" href="#module-ideal_genom.api_client" title="Link to this heading"></a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.api_client.GeneEnsemblRestClient">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.api_client.</span></span><span class="sig-name descname"><span class="pre">GeneEnsemblRestClient</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">server</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'https://rest.ensembl.org'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reqs_per_sec</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">15</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.api_client.GeneEnsemblRestClient" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.api_client.GeneEnsemblRestClient.get_gene_location">
<span class="sig-name descname"><span class="pre">get_gene_location</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">gene_symbol</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.api_client.GeneEnsemblRestClient.get_gene_location" title="Link to this definition"></a></dt>
<dd><p>Fetch the start, end, strand, and chromosome for a given gene symbol using the Ensembl REST API.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>gene_symbol</strong> (<em>str</em>) – The symbol of the gene (e.g., “BRCA2”).</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A dictionary containing start, end, strand, and chromosome information, or an error message.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.api_client.GeneEnsemblRestClient.perform_rest_action">
<span class="sig-name descname"><span class="pre">perform_rest_action</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">method</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">endpoint</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">headers</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">params</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.api_client.GeneEnsemblRestClient.perform_rest_action" title="Link to this definition"></a></dt>
<dd><p>General method to perform REST actions with GET or POST</p>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.api_client.VEPEnsemblRestClient">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.api_client.</span></span><span class="sig-name descname"><span class="pre">VEPEnsemblRestClient</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">server</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reqs_per_sec</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">15</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.api_client.VEPEnsemblRestClient" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.api_client.VEPEnsemblRestClient.ENSEMBL_API_URLS">
<span class="sig-name descname"><span class="pre">ENSEMBL_API_URLS</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['https://rest.ensembl.org',</span> <span class="pre">'https://useast.ensembl.org',</span> <span class="pre">'https://asia.ensembl.org']</span></em><a class="headerlink" href="#ideal_genom.api_client.VEPEnsemblRestClient.ENSEMBL_API_URLS" title="Link to this definition"></a></dt>
<dd></dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.api_client.VEPEnsemblRestClient.perform_rest_action">
<span class="sig-name descname"><span class="pre">perform_rest_action</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">method</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">endpoint</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">headers</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">params</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">retry_count</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.api_client.VEPEnsemblRestClient.perform_rest_action" title="Link to this definition"></a></dt>
<dd><p>General method to perform REST actions with GET or POST</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.api_client.VEPEnsemblRestClient.post_vep_request">
<span class="sig-name descname"><span class="pre">post_vep_request</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ids</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.api_client.VEPEnsemblRestClient.post_vep_request" title="Link to this definition"></a></dt>
<dd><p>Specific method to perform the VEP POST request with InAct parameter set to false.</p>
</dd></dl>

</dd></dl>

</section>
<section id="module-ideal_genom.get_examples">
<span id="ideal-genom-get-examples-module"></span><h2>ideal_genom.get_examples module<a class="headerlink" href="#module-ideal_genom.get_examples" title="Link to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.get_examples.get_bmi_japanese_gwas">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.get_examples.</span></span><span class="sig-name descname"><span class="pre">get_bmi_japanese_gwas</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">tuple</span></span></span><a class="headerlink" href="#ideal_genom.get_examples.get_bmi_japanese_gwas" title="Link to this definition"></a></dt>
<dd><p>Downloads and prepares Japanese GWAS (Genome-Wide Association Studies) data for BMI (Body Mass Index).</p>
<p>This function retrieves data from the Japanese Biobank (BBJ) for both female and male subjects.
If the data files already exist locally, it returns their paths without downloading.
Otherwise, it downloads the zip archive, extracts the relevant files, and decompresses them.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><p><strong>tuple</strong> –</p>
<ul class="simple">
<li><p>Path to the female BMI GWAS data file</p></li>
<li><p>Path to the male BMI GWAS data file</p></li>
</ul>
</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>A tuple containing two Path objects:</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>The data is downloaded from the Japanese Database of Clinical and Genomic Data,
specifically the hum0014 dataset version 6 with 158k samples.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.get_examples.get_bmi_japanese_gwas_sex_dependent">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.get_examples.</span></span><span class="sig-name descname"><span class="pre">get_bmi_japanese_gwas_sex_dependent</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.get_examples.get_bmi_japanese_gwas_sex_dependent" title="Link to this definition"></a></dt>
<dd><p>Download and process sex-dependent BMI GWAS data from the Japanese Biobank (BBJ).</p>
<p>This function:
1. Downloads a supplementary Excel file from a 2017 Nature Genetics paper
2. Extracts relevant SNP information including sex-specific associations
3. Filters for SNPs with rs IDs
4. Converts the data to CSV format with columns: SNP, Sex, CHR, POS
5. Removes the original Excel file after processing</p>
<p>The study is the BioBank Japan GWAS which analyzed sex-dependent genetic
effects on BMI.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>Path</strong> – GWAS results</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Path to the processed CSV file containing the sex-dependent BMI</p>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If the CSV file already exists, it will not be re-downloaded or processed.</p>
</div>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.get_examples.get_top_cond_trumpet_quantitative">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.get_examples.</span></span><span class="sig-name descname"><span class="pre">get_top_cond_trumpet_quantitative</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.get_examples.get_top_cond_trumpet_quantitative" title="Link to this definition"></a></dt>
<dd><p>Downloads and processes conditional analysis data for height-associated variants from BBJ study.</p>
<p>This function retrieves conditional analysis data for height-associated variants from a BBJ
(BioBank Japan) 2019 study. It downloads an Excel file from the provided URL, extracts
relevant columns (rsID, candidate genes, chromosome, and position), filters for autosomal
variants, and saves the processed data as a tab-separated CSV file.</p>
<p>The function checks if the processed CSV file already exists before downloading to avoid
redundant operations. After processing, the original Excel file is deleted to save space.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>Path</strong> – for height-associated variants.</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Path to the processed CSV file containing the top conditional analysis hits</p>
</dd>
</dl>
<section id="side-effects">
<h3>Side effects<a class="headerlink" href="#side-effects" title="Link to this heading"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>Downloads an Excel file from an external URL if not already processed</p></li>
<li><p>Creates a CSV file in the data/sumstats directory</p></li>
<li><p>Removes the downloaded Excel file after processing</p></li>
</ul>
</div></blockquote>
</section>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.get_examples.get_top_loci_trumpet_quantitative">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.get_examples.</span></span><span class="sig-name descname"><span class="pre">get_top_loci_trumpet_quantitative</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.get_examples.get_top_loci_trumpet_quantitative" title="Link to this definition"></a></dt>
<dd><p>Downloads and processes top genetic loci data from a GWAS study on height.</p>
<p>This function retrieves a dataset from the supplementary materials of a
Biobank Japan (BBJ) height GWAS study published in Nature Communications
(2019). It downloads an Excel file containing significant loci, extracts
relevant columns, filters out X chromosome variants, and saves the
processed data as a CSV file.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>Path</strong></p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Path to the processed CSV file containing filtered top genetic loci.</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>The function checks if the processed CSV already exists before downloading</p></li>
<li><p>The original Excel file is deleted after processing</p></li>
<li><p>Only autosomal variants are kept (X chromosome variants are filtered out)</p></li>
<li><p>The processed file contains ‘Variants’ and ‘New_Locus’ columns</p></li>
</ul>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.get_examples.get_trumpet_binary_example">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.get_examples.</span></span><span class="sig-name descname"><span class="pre">get_trumpet_binary_example</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.get_examples.get_trumpet_binary_example" title="Link to this definition"></a></dt>
<dd><p>Downloads and prepares a GWAS summary statistics file for analysis.
This function retrieves a specific GWAS summary statistics file (GCST90278092)
from the EBI FTP server. It checks if the file already exists locally before
downloading. After download, it decompresses the gzipped file.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>Path</strong></p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Path to the uncompressed GWAS summary statistics file.</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>The file is stored in the ‘data/sumstats’ directory relative to the package root</p></li>
<li><p>If the file already exists locally, no download is performed</p></li>
<li><p>The original gzipped file is removed after decompression</p></li>
</ul>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.get_examples.get_trumpet_quantitative_example">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.get_examples.</span></span><span class="sig-name descname"><span class="pre">get_trumpet_quantitative_example</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.get_examples.get_trumpet_quantitative_example" title="Link to this definition"></a></dt>
<dd><p>Downloads and extracts a quantitative GWAS summary statistics example file for generating a trumpet plot.
This function checks if the BBJ Height GWAS summary statistics file already exists in the local data directory.
If not, it downloads the file from the Human Database (DBCLS), extracts it from the zip archive,
decompresses the gzipped file, and saves it to the data/sumstats directory.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>Path</strong> – (2019_BBJ_Height_autosomes_BOLT.txt)</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>Path to the extracted and decompressed GWAS summary statistics file</p>
</dd>
</dl>
<section id="id1">
<h3>Side effects<a class="headerlink" href="#id1" title="Link to this heading"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>Creates directories if they don’t exist</p></li>
<li><p>Downloads files from the internet if necessary</p></li>
<li><p>Extracts and decompresses files</p></li>
</ul>
</div></blockquote>
</section>
</dd></dl>

</section>
<section id="module-ideal_genom.get_references">
<span id="ideal-genom-get-references-module"></span><h2>ideal_genom.get_references module<a class="headerlink" href="#module-ideal_genom.get_references" title="Link to this heading"></a></h2>
<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.get_references.</span></span><span class="sig-name descname"><span class="pre">AssemblyReferenceFetcher</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">base_url</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">extension</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">destination_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">avoid_substring</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'extra'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>AssemblyReferenceFetcher is responsible for retrieving genomic reference files.
This class provides functionality to locate, download, and process reference genome
assemblies from web servers. It handles URL construction, web scraping to find
appropriate files, downloading, and decompressing reference files.</p>
<section id="key-features">
<h3>Key Features<a class="headerlink" href="#key-features" title="Link to this heading"></a></h3>
<ul class="simple">
<li><p>Web scraping to identify correct reference files based on build version</p></li>
<li><p>Automatic download of reference genome files</p></li>
<li><p>Decompression of gzipped reference files</p></li>
<li><p>Management of file paths and destination directories</p></li>
</ul>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher.download_reference_file">
<span class="sig-name descname"><span class="pre">download_reference_file</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher.download_reference_file" title="Link to this definition"></a></dt>
<dd><p>Downloads the reference assembly sequence file from base_url to <cite>self.destination_folder</cite>.</p>
<section id="raises">
<h4>Raises:<a class="headerlink" href="#raises" title="Link to this heading"></a></h4>
<blockquote>
<div><ul class="simple">
<li><p>AttributeError: If <cite>self.latest_url</cite> is not set.</p></li>
<li><p>requests.exceptions.RequestException: If the HTTP request fails.</p></li>
</ul>
</div></blockquote>
</section>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher.get_destination_folder">
<span class="sig-name descname"><span class="pre">get_destination_folder</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher.get_destination_folder" title="Link to this definition"></a></dt>
<dd><p>Determine the destination folder for downloads.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher.get_reference_url">
<span class="sig-name descname"><span class="pre">get_reference_url</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher.get_reference_url" title="Link to this definition"></a></dt>
<dd><p>Retrieves the URL for the reference file from the base URL.
This method scrapes the base URL to find a link to the appropriate reference file.
It looks for links that contain the build version, have the correct file extension,
and do not contain the specified substring to avoid.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>str</strong></p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>The complete URL to the reference file.</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>Exception</strong> – If the base URL cannot be accessed.:</p></li>
<li><p><strong>FileNotFoundError</strong> – If no suitable reference file is found.:</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.AssemblyReferenceFetcher.unzip_reference_file">
<span class="sig-name descname"><span class="pre">unzip_reference_file</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.get_references.AssemblyReferenceFetcher.unzip_reference_file" title="Link to this definition"></a></dt>
<dd><p>Unzips the reference genome file if it’s compressed.
This method extracts the content of a gzipped reference genome file
and saves it as a FASTA file. After successful extraction, the original
compressed file is deleted.</p>
<section id="prerequisites">
<h4>Prerequisites<a class="headerlink" href="#prerequisites" title="Link to this heading"></a></h4>
<blockquote>
<div><ul class="simple">
<li><p>self.reference_file must be set (via get_reference_url)</p></li>
<li><p>self.file_path must be set (via download_reference_file)</p></li>
</ul>
</div></blockquote>
<dl class="field-list simple">
<dt class="field-odd">returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>str</strong></p>
</dd>
<dt class="field-even">rtype<span class="colon">:</span></dt>
<dd class="field-even"><p>Path to the unzipped FASTA file</p>
</dd>
<dt class="field-odd">raises AttributeError<span class="colon">:</span></dt>
<dd class="field-odd"><p>If self.reference_file or self.file_path are not set:</p>
</dd>
<dt class="field-even">raises OSError<span class="colon">:</span></dt>
<dd class="field-even"><p>If an error occurs during the unzipping process:</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<ul class="simple">
<li><p>If the file is already unzipped (has .fa extension), returns its path</p></li>
<li><p>If the unzipped file already exists, returns its path without re-extracting</p></li>
<li><p>The original compressed file is deleted after successful extraction</p></li>
</ul>
</section>
</dd></dl>

</section>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.get_references.</span></span><span class="sig-name descname"><span class="pre">Ensembl37Fetcher</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">destination_folder</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher" title="ideal_genom.get_references.ReferenceDataFetcher"><code class="xref py py-class docutils literal notranslate"><span class="pre">ReferenceDataFetcher</span></code></a></p>
<p>A class for fetching reference genome data from Ensembl’s GRCh37 (hg19) repository.
This class specializes the ReferenceDataFetcher to work specifically with
Ensembl’s GRCh37 human genome build. It provides functionality to automatically
detect and download the latest available GTF file for Homo sapiens from the
Ensembl GRCh37 archive.
The fetcher connects to Ensembl’s FTP server, identifies the most recent release
available for GRCh37, and locates the chromosome GTF file for human genome data.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher.base_url">
<span class="sig-name descname"><span class="pre">base_url</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher.base_url" title="Link to this definition"></a></dt>
<dd><p>The base URL for Ensembl’s GRCh37 repository</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher.build">
<span class="sig-name descname"><span class="pre">build</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher.build" title="Link to this definition"></a></dt>
<dd><p>The genome build identifier (‘37’)</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher.source">
<span class="sig-name descname"><span class="pre">source</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher.source" title="Link to this definition"></a></dt>
<dd><p>The data source identifier (‘ensembl’)</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher.latest_url">
<span class="sig-name descname"><span class="pre">latest_url</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher.latest_url" title="Link to this definition"></a></dt>
<dd><p>The complete URL to the latest GTF file, populated after calling get_latest_release()</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl37Fetcher.get_latest_release">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.Ensembl37Fetcher.get_latest_release" title="Link to this definition"></a></dt>
<dd><p>Identifies and sets the URL for the latest available GTF file</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id0">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id0" title="Link to this definition"></a></dt>
<dd><p>Fetches the URL of the latest GTF file for Homo sapiens GRCh37 from Ensembl.
This method:
1. Connects to the base URL and identifies all available release folders
2. Determines the latest release by finding the highest release number
3. Navigates to the GTF directory for that release
4. Locates the Homo sapiens GRCh37 chromosome GTF file
5. Stores the complete download URL in self.latest_url</p>
<dl class="field-list simple">
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>Exception</strong> – If the base URL cannot be accessed:</p></li>
<li><p><strong>Exception</strong> – If no release folders are found:</p></li>
<li><p><strong>Exception</strong> – If the latest release folder cannot be accessed:</p></li>
<li><p><strong>FileNotFoundError</strong> – If the GTF file is not found in the latest release:</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.get_references.</span></span><span class="sig-name descname"><span class="pre">Ensembl38Fetcher</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">destination_folder</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher" title="ideal_genom.get_references.ReferenceDataFetcher"><code class="xref py py-class docutils literal notranslate"><span class="pre">ReferenceDataFetcher</span></code></a></p>
<p>A fetcher class for downloading human genome reference data from Ensembl, specifically for GRCh38 (build 38).
This class provides functionality to download and access the latest GTF (Gene Transfer Format)
annotation files for the human genome from the Ensembl FTP server. It automatically
determines the most recent release available and handles the download process.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher.base_url">
<span class="sig-name descname"><span class="pre">base_url</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher.base_url" title="Link to this definition"></a></dt>
<dd><p>The base URL for Ensembl’s FTP server where human GTF files are stored</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher.build">
<span class="sig-name descname"><span class="pre">build</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher.build" title="Link to this definition"></a></dt>
<dd><p>The genome build identifier (38 for GRCh38)</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher.source">
<span class="sig-name descname"><span class="pre">source</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher.source" title="Link to this definition"></a></dt>
<dd><p>The data source identifier (‘ensembl’)</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher.latest_url">
<span class="sig-name descname"><span class="pre">latest_url</span></span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher.latest_url" title="Link to this definition"></a></dt>
<dd><p>URL to the latest GTF file, populated after calling get_latest_release()</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.Ensembl38Fetcher.get_latest_release">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.Ensembl38Fetcher.get_latest_release" title="Link to this definition"></a></dt>
<dd><p>Identifies and sets the URL for the latest available GTF file</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id2">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id2" title="Link to this definition"></a></dt>
<dd><p>Fetches the latest GTF file dynamically from the specified base URL.</p>
<p>This method sends a GET request to the base URL, parses the HTML response to find the latest GTF file link, and sets the <cite>latest_url</cite> attribute to the full URL of the latest GTF file.</p>
<section id="id3">
<h3>Raises:<a class="headerlink" href="#id3" title="Link to this heading"></a></h3>
<p>FileNotFoundError: If no GTF file is found in the HTML response.</p>
</section>
<section id="returns">
<h3>Returns:<a class="headerlink" href="#returns" title="Link to this heading"></a></h3>
<p>None</p>
</section>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.get_references.FetcherLDRegions">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.get_references.</span></span><span class="sig-name descname"><span class="pre">FetcherLDRegions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">destination</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">built</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'38'</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.FetcherLDRegions" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.FetcherLDRegions.get_ld_regions">
<span class="sig-name descname"><span class="pre">get_ld_regions</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.get_references.FetcherLDRegions.get_ld_regions" title="Link to this definition"></a></dt>
<dd><p>Downloads or creates high LD regions file based on genome build version.
This method handles the retrieval of high Linkage Disequilibrium (LD) regions for
different genome builds (37 or 38). For build 37, it downloads the regions from a
GitHub repository. For build 38, it creates the file from predefined coordinates.</p>
<section id="id4">
<h3>Returns:<a class="headerlink" href="#id4" title="Link to this heading"></a></h3>
<dl class="simple">
<dt>Path: Path to the created/downloaded LD regions file. Returns empty Path if</dt><dd><p>download fails for build 37.</p>
</dd>
</dl>
</section>
<section id="id5">
<h3>Raises:<a class="headerlink" href="#id5" title="Link to this heading"></a></h3>
<blockquote>
<div><p>None explicitly, but may raise standard I/O related exceptions.</p>
</div></blockquote>
</section>
<section id="notes">
<h3>Notes:<a class="headerlink" href="#notes" title="Link to this heading"></a></h3>
<blockquote>
<div><ul class="simple">
<li><p>For build 37: Downloads from genepi-freiburg/gwas repository</p></li>
<li><p>For build 38: Creates file from hardcoded coordinates from GWAS-pipeline</p></li>
<li><p>Files are named as ‘high-LD-regions_GRCh{build}.txt’</p></li>
<li><p>Creates destination directory if it doesn’t exist</p></li>
</ul>
</div></blockquote>
</section>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.get_references.</span></span><span class="sig-name descname"><span class="pre">RefSeqFetcher</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">destination_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher" title="Link to this definition"></a></dt>
<dd><p>Bases: <a class="reference internal" href="#ideal_genom.get_references.ReferenceDataFetcher" title="ideal_genom.get_references.ReferenceDataFetcher"><code class="xref py py-class docutils literal notranslate"><span class="pre">ReferenceDataFetcher</span></code></a></p>
<p>A class for fetching and downloading reference genome data from NCBI’s RefSeq repository.
This class extends ReferenceDataFetcher to specifically handle downloading human
genome reference files from the RefSeq database. It supports different genome
builds (e.g., ‘GRCh37’, ‘GRCh38’) and automatically identifies the latest version
available for the specified build.
The class handles navigating the NCBI FTP directory structure, finding the
appropriate GTF files for the requested genome build, and managing the
download process.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher.base_url">
<span class="sig-name descname"><span class="pre">base_url</span></span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher.base_url" title="Link to this definition"></a></dt>
<dd><p>The base URL for the NCBI RefSeq FTP server directory.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher.build">
<span class="sig-name descname"><span class="pre">build</span></span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher.build" title="Link to this definition"></a></dt>
<dd><p>The genome build version (‘37’ for GRCh37, ‘38’ for GRCh38).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher.source">
<span class="sig-name descname"><span class="pre">source</span></span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher.source" title="Link to this definition"></a></dt>
<dd><p>The source of the reference data (set to ‘refseq’).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher.latest_url">
<span class="sig-name descname"><span class="pre">latest_url</span></span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher.latest_url" title="Link to this definition"></a></dt>
<dd><p>URL to the latest GTF file, set after calling get_latest_release().</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.RefSeqFetcher.get_latest_release">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.RefSeqFetcher.get_latest_release" title="Link to this definition"></a></dt>
<dd><p>Identifies and sets the URL for the latest available GTF file</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id6">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id6" title="Link to this definition"></a></dt>
<dd><p>Fetches the latest GTF file dynamically from the specified base URL.</p>
<p>This method sends a GET request to the base URL, parses the HTML response to find the latest GTF file link, and sets the <cite>latest_url</cite> attribute to the full URL of the latest GTF file.</p>
<section id="id7">
<h3>Raises:<a class="headerlink" href="#id7" title="Link to this heading"></a></h3>
<p>FileNotFoundError: If no GTF file is found in the HTML response.</p>
</section>
<section id="id8">
<h3>Returns:<a class="headerlink" href="#id8" title="Link to this heading"></a></h3>
<p>None</p>
</section>
</dd></dl>

</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">ideal_genom.get_references.</span></span><span class="sig-name descname"><span class="pre">ReferenceDataFetcher</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">base_url</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">source</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">destination_folder</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher" title="Link to this definition"></a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A class for fetching, downloading, and processing reference genome data.
This class provides a framework for retrieving genomic reference data from various
sources. It handles downloading compressed files, unzipping them, and extracting
gene information from GTF files.</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.build">
<span class="sig-name descname"><span class="pre">build</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.build" title="Link to this definition"></a></dt>
<dd><p>The genome build (e.g., ‘hg38’, ‘GRCh38’).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.source">
<span class="sig-name descname"><span class="pre">source</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.source" title="Link to this definition"></a></dt>
<dd><p>The data source (e.g., ‘ensembl’, ‘ucsc’).</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.base_url">
<span class="sig-name descname"><span class="pre">base_url</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.base_url" title="Link to this definition"></a></dt>
<dd><p>The base URL to fetch data from.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.destination_folder">
<span class="sig-name descname"><span class="pre">destination_folder</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.destination_folder" title="Link to this definition"></a></dt>
<dd><p>The directory to save downloaded files. If None, defaults to project_root/data/{source}_latest.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Optional[str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.latest_url">
<span class="sig-name descname"><span class="pre">latest_url</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.latest_url" title="Link to this definition"></a></dt>
<dd><p>The URL of the latest release after calling get_latest_release().</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Optional[str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.gz_file">
<span class="sig-name descname"><span class="pre">gz_file</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.gz_file" title="Link to this definition"></a></dt>
<dd><p>Path to the downloaded compressed file.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Optional[str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.gtf_file">
<span class="sig-name descname"><span class="pre">gtf_file</span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.gtf_file" title="Link to this definition"></a></dt>
<dd><p>Path to the uncompressed GTF file.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Optional[str]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.get_latest_release">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.get_latest_release" title="Link to this definition"></a></dt>
<dd><p>Determine the specific URL for fetching data. Must be implemented by subclasses.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.download_latest">
<span class="sig-name descname"><span class="pre">download_latest</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.download_latest" title="Link to this definition"></a></dt>
<dd><p>Downloads the latest file and returns the file path.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.get_destination_folder">
<span class="sig-name descname"><span class="pre">get_destination_folder</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.get_destination_folder" title="Link to this definition"></a></dt>
<dd><p>Determines and creates the destination folder for downloads.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher._download_file">
<span class="sig-name descname"><span class="pre">_download_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">url</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Path</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher._download_file" title="Link to this definition"></a></dt>
<dd><p>Downloads a file from the given URL and saves it to file_path.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.unzip_latest">
<span class="sig-name descname"><span class="pre">unzip_latest</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.unzip_latest" title="Link to this definition"></a></dt>
<dd><p>Unzips the latest downloaded file and returns the path to the unzipped file.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="ideal_genom.get_references.ReferenceDataFetcher.get_all_genes">
<span class="sig-name descname"><span class="pre">get_all_genes</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#ideal_genom.get_references.ReferenceDataFetcher.get_all_genes" title="Link to this definition"></a></dt>
<dd><p>Processes the GTF file to extract all genes and saves them to a new file.
Returns the path to the new file.</p>
</dd></dl>

<p class="rubric">Notes</p>
<p>This is an abstract base class that requires subclasses to implement the
get_latest_release() method for specific data sources.</p>
<dl class="py method">
<dt class="sig sig-object py" id="id9">
<span class="sig-name descname"><span class="pre">download_latest</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#id9" title="Link to this definition"></a></dt>
<dd><p>Downloads the latest file from <cite>self.latest_url</cite> to <cite>self.destination_folder</cite>.</p>
<section id="id10">
<h3>Raises:<a class="headerlink" href="#id10" title="Link to this heading"></a></h3>
<blockquote>
<div><p>AttributeError: If <cite>self.latest_url</cite> is not set.
requests.exceptions.RequestException: If the HTTP request fails.</p>
</div></blockquote>
</section>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id11">
<span class="sig-name descname"><span class="pre">get_all_genes</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#id11" title="Link to this definition"></a></dt>
<dd><p>Extract all genes from the GTF file and save them to a new compressed file.
This method reads the GTF file specified in self.gtf_file, filters for gene features,
and creates a new GTF file containing only the gene entries. If the output file
already exists, it will return the path without reprocessing.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p>Path to the compressed GTF file containing all genes</p>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>str</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>FileNotFoundError</strong> – If the reference GTF file (self.gtf_file) is not found:</p></li>
<li><p><strong>TypeError</strong> – If read_gtf does not return a pandas DataFrame:</p></li>
</ul>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The output file will be named based on the input GTF file with “-all_genes.gtf.gz” suffix</p>
</div>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id12">
<span class="sig-name descname"><span class="pre">get_destination_folder</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Path</span></span></span><a class="headerlink" href="#id12" title="Link to this definition"></a></dt>
<dd><p>Determine the destination folder for downloads.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id13">
<span class="sig-name descname"><span class="pre">get_latest_release</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#id13" title="Link to this definition"></a></dt>
<dd><p>Determine the specific URL for fetching data.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="id14">
<span class="sig-name descname"><span class="pre">unzip_latest</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="headerlink" href="#id14" title="Link to this definition"></a></dt>
<dd><p>Unzips the latest downloaded file and stores it as a GTF file.</p>
</dd></dl>

</dd></dl>

</section>
<section id="module-ideal_genom.power_comp">
<span id="ideal-genom-power-comp-module"></span><h2>ideal_genom.power_comp module<a class="headerlink" href="#module-ideal_genom.power_comp" title="Link to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.power_comp.calculate_power_binary">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.power_comp.</span></span><span class="sig-name descname"><span class="pre">calculate_power_binary</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">beta</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">ndarray</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">daf</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">ndarray</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">prevalence</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncase</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncontrol</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sig_level</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">5e-08</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">or_to_rr</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">ndarray</span></span></span><a class="headerlink" href="#ideal_genom.power_comp.calculate_power_binary" title="Link to this definition"></a></dt>
<dd><p>Calculate the statistical power of genetic association tests in a case-control study with binary traits.
This function computes the power to detect genetic associations in a case-control study design,
assuming an additive genetic model. It accounts for allele frequencies, effect sizes,
disease prevalence, and sample sizes to determine the probability of detecting true associations
at a specified significance level.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>beta</strong> (<em>np.ndarray</em>) – Log odds ratios representing effect sizes of variants.</p></li>
<li><p><strong>daf</strong> (<em>np.ndarray</em>) – Disease allele frequencies.</p></li>
<li><p><strong>prevalence</strong> (<em>float</em>) – Disease prevalence in the population.</p></li>
<li><p><strong>ncase</strong> (<em>int</em>) – Number of cases in the study.</p></li>
<li><p><strong>ncontrol</strong> (<em>int</em>) – Number of controls in the study.</p></li>
<li><p><strong>sig_level</strong> (<em>float</em><em>, </em><em>optional</em>) – Significance level threshold for declaring associations (default: 5e-8, standard for GWAS).</p></li>
<li><p><strong>or_to_rr</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True, converts odds ratios to risk ratios using the disease prevalence (default: False).</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Array of statistical power values for each variant, representing the probability
of detecting a true association at the specified significance level.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>np.ndarray</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>The function implements power calculation for an additive genetic model using normal
approximation. When or_to_rr is True, it converts odds ratio to relative risk using
the formula from Zhang and Yu (JAMA, 1998).</p>
<p class="rubric">References</p>
<p>Zhang J, Yu KF. What’s the Relative Risk? JAMA. 1998;280(19):1690-1691.
doi:10.1001/jama.280.19.1690</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.power_comp.calculate_power_quantitative">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.power_comp.</span></span><span class="sig-name descname"><span class="pre">calculate_power_quantitative</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">beta</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">ndarray</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">eaf</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">ndarray</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sig_level</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">5e-08</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variance</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">ndarray</span></span></span><a class="headerlink" href="#ideal_genom.power_comp.calculate_power_quantitative" title="Link to this definition"></a></dt>
<dd><p>Calculate statistical power for detecting genetic associations in quantitative traits.
This function computes the statistical power for quantitative trait association tests
based on effect sizes (beta), effect allele frequencies (EAF), sample size,
significance level, and trait variance.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>beta</strong> (<em>numpy.ndarray</em>) – Array of effect sizes (regression coefficients)</p></li>
<li><p><strong>eaf</strong> (<em>numpy.ndarray</em>) – Array of effect allele frequencies (between 0 and 1)</p></li>
<li><p><strong>sample_size</strong> (<em>int</em>) – Number of individuals in the study</p></li>
<li><p><strong>sig_level</strong> (<em>float</em><em>, </em><em>optional</em>) – Significance threshold (default: 5e-8, typical for GWAS)</p></li>
<li><p><strong>variance</strong> (<em>float</em><em>, </em><em>optional</em>) – Phenotypic variance of the trait (default: 1)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Array of statistical power values corresponding to each variant</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>numpy.ndarray</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>The calculation uses the non-central chi-square distribution to determine
the probability of detecting true associations at the specified significance level.
Power is calculated as 1 minus the cumulative distribution function of the
non-central chi-square distribution with 1 degree of freedom.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.power_comp.get_beta_binary">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.power_comp.</span></span><span class="sig-name descname"><span class="pre">get_beta_binary</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">prevalence</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncase</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ncontrol</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">eaf_range</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">tuple</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">(0.0001,</span> <span class="pre">0.5)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">beta_range</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">tuple</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">(0.0001,</span> <span class="pre">5)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sig_level</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">5e-08</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_matrix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">or_to_rr</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.power_comp.get_beta_binary" title="Link to this definition"></a></dt>
<dd><p>Find combinations of effect allele frequencies (EAF) and effect sizes (beta) that achieve
a specified statistical power for binary traits in genetic association studies.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>prevalence</strong> (<em>float</em>) – Disease prevalence in the population.</p></li>
<li><p><strong>ncase</strong> (<em>int</em>) – Number of cases in the study.</p></li>
<li><p><strong>ncontrol</strong> (<em>int</em>) – Number of controls in the study.</p></li>
<li><p><strong>eaf_range</strong> (<em>tuple</em><em>, </em><em>optional</em>) – Range of effect allele frequencies to consider (min, max). Default is (0.0001, 0.5).</p></li>
<li><p><strong>beta_range</strong> (<em>tuple</em><em>, </em><em>optional</em>) – Range of effect sizes to consider (min, max). Default is (0.0001, 5).</p></li>
<li><p><strong>t</strong> (<em>float</em><em>, </em><em>optional</em>) – Target power threshold. Combinations with power &gt;= t will be returned. Default is 0.</p></li>
<li><p><strong>sig_level</strong> (<em>float</em><em>, </em><em>optional</em>) – Significance level (alpha) for power calculation. Default is 5e-8 (genome-wide significance).</p></li>
<li><p><strong>n_matrix</strong> (<em>int</em><em>, </em><em>optional</em>) – Resolution of the EAF-beta matrix. Default is 500.</p></li>
<li><p><strong>or_to_rr</strong> (<em>bool</em><em>, </em><em>optional</em>) – If True, converts odds ratios to relative risks using the provided prevalence.
If False, approximates genetic relative risk (GRR) using odds ratios. Default is False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>DataFrame containing EAF and beta combinations that achieve the target power.
The DataFrame has columns ‘eaf’ and ‘beta’ and is sorted by ‘eaf’ in descending order.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas.DataFrame</p>
</dd>
</dl>
<p class="rubric">Notes</p>
<p>When prevalence is less than 10%, GRR is very similar to OR if or_to_rr is set to False.
The function uses a greedy algorithm to find EAF-beta combinations with power &gt;= t.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.power_comp.get_beta_quantitative">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.power_comp.</span></span><span class="sig-name descname"><span class="pre">get_beta_quantitative</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">eaf_range</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">float</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">(1e-05,</span> <span class="pre">0.5)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">beta_range</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">float</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">(1e-05,</span> <span class="pre">5)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sig_level</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">5e-08</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variance</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_matrix</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">500</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="headerlink" href="#ideal_genom.power_comp.get_beta_quantitative" title="Link to this definition"></a></dt>
<dd><p>Calculate beta values for quantitative traits based on power threshold.</p>
<section id="parameters">
<h3>Parameters:<a class="headerlink" href="#parameters" title="Link to this heading"></a></h3>
<dl class="simple">
<dt>eaf_range<span class="classifier">Tuple[float, float]</span></dt><dd><p>Range of effect allele frequencies (min, max)</p>
</dd>
<dt>beta_range<span class="classifier">Tuple[float, float]</span></dt><dd><p>Range of beta values (min, max)</p>
</dd>
<dt>t<span class="classifier">float</span></dt><dd><p>Power threshold (0 to 1)</p>
</dd>
<dt>sample_size<span class="classifier">int</span></dt><dd><p>Sample size</p>
</dd>
<dt>sig_level<span class="classifier">float</span></dt><dd><p>Significance level</p>
</dd>
<dt>variance<span class="classifier">float</span></dt><dd><p>Variance</p>
</dd>
<dt>n_matrix<span class="classifier">int</span></dt><dd><p>Size of the grid for calculations</p>
</dd>
</dl>
</section>
<section id="id15">
<h3>Returns:<a class="headerlink" href="#id15" title="Link to this heading"></a></h3>
<dl class="simple">
<dt>pd.DataFrame</dt><dd><p>DataFrame containing eaf and beta values meeting the threshold</p>
</dd>
</dl>
</section>
</dd></dl>

</section>
<section id="module-ideal_genom">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-ideal_genom" title="Link to this heading"></a></h2>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="index.html" class="btn btn-neutral float-left" title="ideal-genom documentation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="annotations.html" class="btn btn-neutral float-right" title="ideal_genom.annotations" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Luis Giraldo Gonzalez Ricardo.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>