

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Manhattan-type Plots Module &mdash; ideal-genom 0.1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=80d5e7a1" />
      <link rel="stylesheet" type="text/css" href="../_static/css/theme.css?v=e59714d7" />

  
      <script src="../_static/jquery.js?v=5d32c60e"></script>
      <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="../_static/documentation_options.js?v=01f34227"></script>
      <script src="../_static/doctools.js?v=9bcbadda"></script>
      <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Other Plots Module" href="plots.html" />
    <link rel="prev" title="Preparatory Module" href="../preprocessing/preparatory.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            ideal-genom
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gwas/gwas_glm.html">GWAS with Generalized Linear Models (GLM)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../gwas/gwas_glmm.html">GWAS with Generalized Linear Mixed Models (GLMM)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../preprocessing/PostImputation.html">PostImputation Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="../preprocessing/preparatory.html">Preparatory Module</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Manhattan-type Plots Module</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.visualization.manhattan_type.compute_relative_pos"><code class="docutils literal notranslate"><span class="pre">compute_relative_pos()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.visualization.manhattan_type.find_chromosomes_center"><code class="docutils literal notranslate"><span class="pre">find_chromosomes_center()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.visualization.manhattan_type.manhattan_draw"><code class="docutils literal notranslate"><span class="pre">manhattan_draw()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.visualization.manhattan_type.manhattan_process_data"><code class="docutils literal notranslate"><span class="pre">manhattan_process_data()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.visualization.manhattan_type.manhattan_type_annotate"><code class="docutils literal notranslate"><span class="pre">manhattan_type_annotate()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.visualization.manhattan_type.miami_draw"><code class="docutils literal notranslate"><span class="pre">miami_draw()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.visualization.manhattan_type.miami_draw_anno_lines"><code class="docutils literal notranslate"><span class="pre">miami_draw_anno_lines()</span></code></a></li>
<li class="toctree-l2"><a class="reference internal" href="#ideal_genom.visualization.manhattan_type.miami_process_data"><code class="docutils literal notranslate"><span class="pre">miami_process_data()</span></code></a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="plots.html">Other Plots Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="zoom_heatmap.html">Zoom-Heatmap Plot Module</a></li>
<li class="toctree-l1"><a class="reference internal" href="../annotations.html">ideal_genom.annotations</a></li>
<li class="toctree-l1"><a class="reference internal" href="../api_client.html">ideal_genom.api_client</a></li>
<li class="toctree-l1"><a class="reference internal" href="../get_examples.html">ideal_genom.get_examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="../get_references.html">ideal_genom.get_references</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Helpers.html">ideal_genom.Helpers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../power_comp.html">ideal_genom.power_comp</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">ideal-genom</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Manhattan-type Plots Module</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/visualization/manhattan_type.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-ideal_genom.visualization.manhattan_type">
<span id="manhattan-type-plots-module"></span><h1>Manhattan-type Plots Module<a class="headerlink" href="#module-ideal_genom.visualization.manhattan_type" title="Link to this heading"></a></h1>
<p>Generate Manhattan and Miami plots for genome-wide association studies (GWAS).</p>
<p>This module provides functions to create Manhattan and Miami plots to visualize
the significance of genetic variants across the genome.</p>
<p>Features:
- Data processing and visualization of GWAS summary statistics.
- Annotation of SNPs with gene information from various sources.
- Highlighting and labeling of specific SNPs of interest.</p>
<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.visualization.manhattan_type.compute_relative_pos">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.visualization.manhattan_type.</span></span><span class="sig-name descname"><span class="pre">compute_relative_pos</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chr_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'CHR'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'POS'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'p'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="headerlink" href="#ideal_genom.visualization.manhattan_type.compute_relative_pos" title="Link to this definition"></a></dt>
<dd><p>Compute the relative position of probes/SNPs across chromosomes and add a -log10(p-value) column.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>data</strong> (<em>pd.DataFrame</em>) – Input DataFrame containing genomic data.</p></li>
<li><p><strong>chr_col</strong> (<em>str</em>) – Column name for chromosome identifiers. Default is ‘CHR’.</p></li>
<li><p><strong>pos_col</strong> (<em>str</em>) – Column name for base pair positions. Default is ‘POS’.</p></li>
<li><p><strong>p_col</strong> (<em>str</em>) – Column name for p-values. Default is ‘p’.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>pd.DataFrame</strong></p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>DataFrame with additional columns for relative positions and -log10(p-values).</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If input data is not a pandas DataFrame.</p></li>
<li><p><strong>ValueError</strong> – If <cite>chr_col</cite>, <cite>pos_col</cite> or <cite>p_col</cite> columns are not found in the DataFrame.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.visualization.manhattan_type.find_chromosomes_center">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.visualization.manhattan_type.</span></span><span class="sig-name descname"><span class="pre">find_chromosomes_center</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chr_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'CHR'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chr_pos_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'rel_pos'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">DataFrame</span></span></span><a class="headerlink" href="#ideal_genom.visualization.manhattan_type.find_chromosomes_center" title="Link to this definition"></a></dt>
<dd><p>Calculate the center positions of chromosomes in a given DataFrame.</p>
<p>This function takes a DataFrame containing chromosome data and calculates the center position
for each chromosome based on the specified chromosome column and chromosome position column.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>data</strong> (<em>pd.DataFrame</em>) – The input DataFrame containing chromosome data.</p></li>
<li><p><strong>chr_col</strong> (<em>str</em><em>, </em><em>optional</em>) – The name of the column representing chromosome identifiers (default is ‘CHR’).</p></li>
<li><p><strong>chr_pos_col</strong> (<em>str</em><em>, </em><em>optional</em>) – The name of the column representing relative positions within chromosomes (default is ‘rel_pos’).</p></li>
<li><p><strong>Returns</strong></p></li>
<li><p><strong>--------</strong></p></li>
<li><p><strong>pd.DataFrame</strong> – A DataFrame with columns ‘CHR’ and ‘center’, where ‘CHR’ contains chromosome identifiers
and ‘center’ contains the calculated center positions for each chromosome.</p></li>
</ul>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If the input data is not a pandas DataFrame.</p></li>
<li><p><strong>ValueError</strong> – If the specified chromosome or chromosome position columns are not found in the DataFrame.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.visualization.manhattan_type.manhattan_draw">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.visualization.manhattan_type.</span></span><span class="sig-name descname"><span class="pre">manhattan_draw</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data_df:</span> <span class="pre">~pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">snp_col:</span> <span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chr_col:</span> <span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_col:</span> <span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p_col:</span> <span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_dir:</span> <span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">to_highlight:</span> <span class="pre">~pandas.core.frame.DataFrame</span> <span class="pre">=</span> <span class="pre">Empty</span> <span class="pre">DataFrame</span> <span class="pre">Columns:</span> <span class="pre">[]</span> <span class="pre">Index:</span> <span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">highlight_hue:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'hue'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">to_annotate:</span> <span class="pre">~pandas.core.frame.DataFrame</span> <span class="pre">=</span> <span class="pre">Empty</span> <span class="pre">DataFrame</span> <span class="pre">Columns:</span> <span class="pre">[]</span> <span class="pre">Index:</span> <span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gen_col:</span> <span class="pre">str</span> <span class="pre">|</span> <span class="pre">None</span> <span class="pre">=</span> <span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'38'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">anno_source='ensembl'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gtf_path:</span> <span class="pre">str</span> <span class="pre">|</span> <span class="pre">None</span> <span class="pre">=</span> <span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_name:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'manhattan_plot.png'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">upper_cap:</span> <span class="pre">float</span> <span class="pre">|</span> <span class="pre">None</span> <span class="pre">=</span> <span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome_line:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">5e-08</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">suggestive_line:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">1e-05</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">yaxis_margin:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dpi:</span> <span class="pre">int</span> <span class="pre">=</span> <span class="pre">400</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">bool</span></span></span><a class="headerlink" href="#ideal_genom.visualization.manhattan_type.manhattan_draw" title="Link to this definition"></a></dt>
<dd><p>Generate a Manhattan plot for genomic data.</p>
<p>This function creates a Manhattan plot for visualizing the statistical significance of genetic
variants across the genome, typically used in Genome-Wide Association Studies (GWAS).
The plot shows -log10(p-values) against genomic positions, organized by chromosome.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>data_df</strong> (<em>pd.DataFrame</em>) – DataFrame containing the SNP data to be plotted.</p></li>
<li><p><strong>snp_col</strong> (<em>str</em>) – Column name in data_df that contains the SNP identifiers.</p></li>
<li><p><strong>chr_col</strong> (<em>str</em>) – Column name in data_df that contains the chromosome information.</p></li>
<li><p><strong>pos_col</strong> (<em>str</em>) – Column name in data_df that contains the position information.</p></li>
<li><p><strong>p_col</strong> (<em>str</em>) – Column name in data_df that contains the p-values.</p></li>
<li><p><strong>plot_dir</strong> (<em>str</em>) – Directory path where the plot will be saved.</p></li>
<li><p><strong>to_highlight</strong> (<em>pd.DataFrame</em><em>, </em><em>optional</em>) – DataFrame containing SNPs to be highlighted in the plot.</p></li>
<li><p><strong>highlight_hue</strong> (<em>str</em><em>, </em><em>default='hue'</em>) – Column name in to_highlight to be used for color-coding highlighted SNPs.</p></li>
<li><p><strong>to_annotate</strong> (<em>pd.DataFrame</em><em>, </em><em>optional</em>) – DataFrame containing SNPs to be annotated in the plot.</p></li>
<li><p><strong>gen_col</strong> (<em>str</em><em>, </em><em>optional</em>) – Column name for gene information to be used in annotation.</p></li>
<li><p><strong>build</strong> (<em>str</em><em>, </em><em>default='38'</em>) – Genome build version (‘38’ for GRCh38, etc.).</p></li>
<li><p><strong>anno_source</strong> (<em>str</em><em>, </em><em>default='ensembl'</em>) – Source for gene annotation information.</p></li>
<li><p><strong>gtf_path</strong> (<em>str</em><em>, </em><em>optional</em>) – Path to a GTF file for gene annotation if not using online sources.</p></li>
<li><p><strong>save_name</strong> (<em>str</em><em>, </em><em>default='manhattan_plot.png'</em>) – Filename for the saved plot.</p></li>
<li><p><strong>upper_cap</strong> (<em>float</em><em>, </em><em>optional</em>) – Upper limit for log10(p-values) to be capped at.</p></li>
<li><p><strong>genome_line</strong> (<em>float</em><em>, </em><em>default=5e-8</em>) – P-value threshold for genome-wide significance line.</p></li>
<li><p><strong>suggestive_line</strong> (<em>float</em><em>, </em><em>default=1e-5</em>) – P-value threshold for suggestive significance line.</p></li>
<li><p><strong>yaxis_margin</strong> (<em>float</em><em>, </em><em>default=10</em>) – Additional margin to add to the y-axis maximum.</p></li>
<li><p><strong>dpi</strong> (<em>int</em><em>, </em><em>default=400</em>) – Resolution of the saved image in dots per inch.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><em>bool</em> – True if the plot was successfully created and saved.</p></li>
<li><p><em>Side Effects</em></p></li>
<li><p><em>————</em></p></li>
<li><p><em>Saves a Manhattan plot image to the specified directory.</em></p></li>
</ul>
</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>TypeError</strong> – If input data is not a pandas DataFrame or to_highlight/to_annotate are not of correct type.</p></li>
<li><p><strong>ValueError</strong> – If required columns are not found in the input DataFrame.</p></li>
<li><p><strong>FileNotFoundError</strong> – If the specified plot directory does not exist.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.visualization.manhattan_type.manhattan_process_data">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.visualization.manhattan_type.</span></span><span class="sig-name descname"><span class="pre">manhattan_process_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chr_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'CHR'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'POS'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'p'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span></span></span><a class="headerlink" href="#ideal_genom.visualization.manhattan_type.manhattan_process_data" title="Link to this definition"></a></dt>
<dd><p>Processes the input DataFrame to prepare data for a Manhattan plot.</p>
<p>The function computes the relative positions of SNPs across chromosomes,
calculates the -log10(p-values), and finds the center positions of each chromosome.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>data_df</strong> (<em>pd.DataFrame</em>) – The input DataFrame containing genomic data.</p></li>
<li><p><strong>chr_col</strong> (<em>str</em><em> (</em><em>optional</em><em>)</em>) – The column name for chromosome data. Defaults to ‘CHR’.</p></li>
<li><p><strong>pos_col</strong> (<em>str</em><em> (</em><em>optional</em><em>)</em>) – The column name for position data. Defaults to ‘POS’.</p></li>
<li><p><strong>p_col</strong> (<em>str</em><em> (</em><em>optional</em><em>)</em>) – The column name for p-value data. Defaults to ‘p’.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><p><strong>dict</strong> –</p>
<ul class="simple">
<li><p>‘data’ (pd.DataFrame): The processed DataFrame with relative positions and log-transformed p-values.</p></li>
<li><p>’axis’ (dict): The center positions of each chromosome for plotting.</p></li>
<li><p>’maxp’ (float): The maximum log-transformed p-value.</p></li>
</ul>
</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>A dictionary containing processed data for the Manhattan plot with the following keys:</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.visualization.manhattan_type.manhattan_type_annotate">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.visualization.manhattan_type.</span></span><span class="sig-name descname"><span class="pre">manhattan_type_annotate</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">axes</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Axes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variants_toanno</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_x_axis</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">suggestive_line</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">genome_line</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Axes</span></span></span><a class="headerlink" href="#ideal_genom.visualization.manhattan_type.manhattan_type_annotate" title="Link to this definition"></a></dt>
<dd><p>Annotates a Manhattan plot with gene names.</p>
<p>This function uses the textalloc library to place gene names on a Manhattan plot.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>axes</strong> (<em>Axes</em><em> (</em><em>matplotlib.axes.Axes</em><em>)</em>) – The matplotlib axes object where the Manhattan plot is drawn.</p></li>
<li><p><strong>data</strong> (<em>pd.DataFrame</em>) – DataFrame containing the scatter plot data with columns ‘rel_pos’ and ‘log10p’.</p></li>
<li><p><strong>variants_toanno</strong> (<em>pd.DataFrame</em>) – DataFrame containing the variants to annotate with columns ‘rel_pos’, ‘log10p’, and ‘GENENAME’.</p></li>
<li><p><strong>max_x_axis</strong> (<em>float</em>) – The maximum value for the x-axis.</p></li>
<li><p><strong>suggestive_line</strong> (<em>float</em>) – The y-value for the suggestive significance line.</p></li>
<li><p><strong>genome_line</strong> (<em>float</em>) – The y-value for the genome-wide significance line.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The matplotlib axes object with annotations.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Axes</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>TypeError</strong> – If the input parameters are not of the expected types.</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.visualization.manhattan_type.miami_draw">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.visualization.manhattan_type.</span></span><span class="sig-name descname"><span class="pre">miami_draw</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df_top:</span> <span class="pre">~pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_bottom:</span> <span class="pre">~pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">snp_col:</span> <span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chr_col:</span> <span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_col:</span> <span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p_col:</span> <span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plots_dir:</span> <span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">top_highlights:</span> <span class="pre">list</span> <span class="pre">=</span> <span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">top_annotations:</span> <span class="pre">~pandas.core.frame.DataFrame</span> <span class="pre">=</span> <span class="pre">Empty</span> <span class="pre">DataFrame</span> <span class="pre">Columns:</span> <span class="pre">[]</span> <span class="pre">Index:</span> <span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bottom_highlights:</span> <span class="pre">list</span> <span class="pre">=</span> <span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bottom_annotations:</span> <span class="pre">~pandas.core.frame.DataFrame</span> <span class="pre">=</span> <span class="pre">Empty</span> <span class="pre">DataFrame</span> <span class="pre">Columns:</span> <span class="pre">[]</span> <span class="pre">Index:</span> <span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">top_gen_col:</span> <span class="pre">str</span> <span class="pre">|</span> <span class="pre">None</span> <span class="pre">=</span> <span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bottom_gen_col:</span> <span class="pre">str</span> <span class="pre">|</span> <span class="pre">None</span> <span class="pre">=</span> <span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gtf_path:</span> <span class="pre">str</span> <span class="pre">|</span> <span class="pre">None</span> <span class="pre">=</span> <span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">source:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'ensemble'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">build:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'38'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">save_name:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'miami_plot.jpeg'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">legend_top:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'top</span> <span class="pre">GWAS'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">legend_bottom:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'bottom</span> <span class="pre">GWAS'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dpi:</span> <span class="pre">int</span> <span class="pre">=</span> <span class="pre">400</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">bool</span></span></span><a class="headerlink" href="#ideal_genom.visualization.manhattan_type.miami_draw" title="Link to this definition"></a></dt>
<dd><p>Draws a Miami plot (a combination of two Manhattan plots) for visualizing GWAS results.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>df_top</strong> (<em>pd.DataFrame</em>) – DataFrame containing the top plot data.</p></li>
<li><p><strong>df_bottom</strong> (<em>pd.DataFrame</em>) – DataFrame containing the bottom plot data.</p></li>
<li><p><strong>snp_col</strong> (<em>str</em>) – Column name for SNP identifiers.</p></li>
<li><p><strong>chr_col</strong> (<em>str</em>) – Column name for chromosome identifiers.</p></li>
<li><p><strong>pos_col</strong> (<em>str</em>) – Column name for base pair positions.</p></li>
<li><p><strong>p_col</strong> (<em>str</em>) – Column name for p-values.</p></li>
<li><p><strong>plots_dir</strong> (<em>str</em>) – Directory where the plot will be saved.</p></li>
<li><p><strong>top_highlights</strong> (<em>list</em><em>, </em><em>optional</em>) – List of SNPs to highlight in the top plot.</p></li>
<li><p><strong>top_annotations</strong> (<em>list</em><em>, </em><em>optional</em>) – List of SNPs to annotate in the top plot.</p></li>
<li><p><strong>bottom_highlights</strong> (<em>list</em><em>, </em><em>optional</em>) – List of SNPs to highlight in the bottom plot.</p></li>
<li><p><strong>bottom_annotations</strong> (<em>list</em><em>, </em><em>optional</em>) – List of SNPs to annotate in the bottom plot.</p></li>
<li><p><strong>gtf_path</strong> (<em>str</em><em>, </em><em>optional</em>) – Path to the GTF file for gene annotation. If None, the file will be downloaded.</p></li>
<li><p><strong>save_name</strong> (<em>str</em><em>, </em><em>optional</em>) – Name of the file to save the plot as. Default is ‘miami_plot.jpeg’.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><em>bool</em> – True if the plot is successfully created and saved, False otherwise.</p></li>
<li><p><em>Side Effects</em></p></li>
<li><p><em>————</em></p></li>
<li><p><em>Saves a Miami plot image to the specified directory.</em></p></li>
</ul>
</p>
</dd>
<dt class="field-odd">Raises<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>TypeError</strong> – If <cite>df_top</cite>, <cite>df_bottom</cite>, <cite>top_annotations</cite>, or <cite>bottom_annotations</cite> are not pandas DataFrames.</p></li>
<li><p><strong>TypeError</strong> – If <cite>top_highlights</cite> or <cite>bottom_highlights</cite> are not lists of SNP identifiers.</p></li>
<li><p><strong>TypeError</strong> – If <cite>save_name</cite>, <cite>legend_top</cite>, or <cite>legend_bottom</cite> are not strings.</p></li>
<li><p><strong>ValueError</strong> – If required columns (<cite>chr_col</cite>, <cite>pos_col</cite>, <cite>p_col</cite>) are not found in the DataFrames.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.visualization.manhattan_type.miami_draw_anno_lines">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.visualization.manhattan_type.</span></span><span class="sig-name descname"><span class="pre">miami_draw_anno_lines</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">renderer</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">RendererBase</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">axes</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Axes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">texts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variants_toanno</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#ideal_genom.visualization.manhattan_type.miami_draw_anno_lines" title="Link to this definition"></a></dt>
<dd><p>Draws annotation lines from text labels to their corresponding data points on a plot.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>renderer</strong> (<em>RendererBase</em>) – The renderer used to draw the plot.</p></li>
<li><p><strong>axes</strong> (<em>Axes</em><em> (</em><em>matplotlib.axes.Axes</em><em>)</em>) – The axes on which the plot is drawn.</p></li>
<li><p><strong>texts</strong> (<em>list</em>) – A list of text objects to annotate.</p></li>
<li><p><strong>variants_toanno</strong> (<em>pd.DataFrame</em>) – A DataFrame containing the data points to annotate, with columns ‘GENENAME’, ‘rel_pos’, and ‘log10p’.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The axes with the annotation lines drawn.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>Axes</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="ideal_genom.visualization.manhattan_type.miami_process_data">
<span class="sig-prename descclassname"><span class="pre">ideal_genom.visualization.manhattan_type.</span></span><span class="sig-name descname"><span class="pre">miami_process_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">data_top</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">data_bottom</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chr_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pos_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p_col</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">dict</span></span></span><a class="headerlink" href="#ideal_genom.visualization.manhattan_type.miami_process_data" title="Link to this definition"></a></dt>
<dd><p>Processes Miami plot data by preparing, computing relative positions, and splitting the data.</p>
<p>For each part of the Miami plot (top and bottom), this function computes the relative positions of SNPs,
calculates the -log10(p-values), and finds the center positions of chromosomes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>data_top</strong> (<em>pd.DataFrame</em>) – The top part of the data to be processed.</p></li>
<li><p><strong>data_bottom</strong> (<em>pd.DataFrame</em>) – The bottom part of the data to be processed.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><p><strong>dict</strong> –</p>
<ul class="simple">
<li><p>‘upper’: DataFrame containing the top part of the processed data.</p></li>
<li><p>’lower’: DataFrame containing the bottom part of the processed data.</p></li>
<li><p>’axis’: The center positions of the chromosomes.</p></li>
<li><p>’maxp’: The maximum -log10(p-value) in the data.</p></li>
</ul>
</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>A dictionary containing the processed data with the following keys:</p>
</dd>
<dt class="field-even">Raises<span class="colon">:</span></dt>
<dd class="field-even"><ul class="simple">
<li><p><strong>TypeError</strong> – If <cite>data_top</cite> or <cite>data_bottom</cite> is not a pandas DataFrame.</p></li>
<li><p><strong>ValueError</strong> – If the specified columns (<cite>chr_col</cite>, <cite>pos_col</cite>, <cite>p_col</cite>) are not found in the DataFrames.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../preprocessing/preparatory.html" class="btn btn-neutral float-left" title="Preparatory Module" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="plots.html" class="btn btn-neutral float-right" title="Other Plots Module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Luis Giraldo Gonzalez Ricardo.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>